// @vercel/sdk/esm/models/addprojectmemberop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The project role of the member that will be added.
 */
export const AddProjectMemberRequestBodyProjectMembersRole = {
    Admin: "ADMIN",
    ProjectDeveloper: "PROJECT_DEVELOPER",
    ProjectViewer: "PROJECT_VIEWER",
};
/**
 * The project role of the member that will be added.
 */
export const AddProjectMemberRequestBodyRole = {
    Admin: "ADMIN",
    ProjectDeveloper: "PROJECT_DEVELOPER",
    ProjectViewer: "PROJECT_VIEWER",
};
/**
 * The project role of the member that will be added.
 */
export const RequestBodyRole = {
    Admin: "ADMIN",
    ProjectDeveloper: "PROJECT_DEVELOPER",
    ProjectViewer: "PROJECT_VIEWER",
};
/** @internal */
export const AddProjectMemberRequestBodyProjectMembersRole$inboundSchema = z
    .nativeEnum(AddProjectMemberRequestBodyProjectMembersRole);
/** @internal */
export const AddProjectMemberRequestBodyProjectMembersRole$outboundSchema = AddProjectMemberRequestBodyProjectMembersRole$inboundSchema;
/** @internal */
export const AddProjectMemberRequestBody3$inboundSchema = z.object({
    uid: z.string().optional(),
    username: z.string().optional(),
    email: z.string(),
    role: AddProjectMemberRequestBodyProjectMembersRole$inboundSchema,
});
/** @internal */
export const AddProjectMemberRequestBody3$outboundSchema = z.object({
    uid: z.string().optional(),
    username: z.string().optional(),
    email: z.string(),
    role: AddProjectMemberRequestBodyProjectMembersRole$outboundSchema,
});
export function addProjectMemberRequestBody3ToJSON(addProjectMemberRequestBody3) {
    return JSON.stringify(AddProjectMemberRequestBody3$outboundSchema.parse(addProjectMemberRequestBody3));
}
export function addProjectMemberRequestBody3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => AddProjectMemberRequestBody3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddProjectMemberRequestBody3' from JSON`);
}
/** @internal */
export const AddProjectMemberRequestBodyRole$inboundSchema = z.nativeEnum(AddProjectMemberRequestBodyRole);
/** @internal */
export const AddProjectMemberRequestBodyRole$outboundSchema = AddProjectMemberRequestBodyRole$inboundSchema;
/** @internal */
export const AddProjectMemberRequestBody2$inboundSchema = z.object({
    uid: z.string().optional(),
    username: z.string(),
    email: z.string().optional(),
    role: AddProjectMemberRequestBodyRole$inboundSchema,
});
/** @internal */
export const AddProjectMemberRequestBody2$outboundSchema = z.object({
    uid: z.string().optional(),
    username: z.string(),
    email: z.string().optional(),
    role: AddProjectMemberRequestBodyRole$outboundSchema,
});
export function addProjectMemberRequestBody2ToJSON(addProjectMemberRequestBody2) {
    return JSON.stringify(AddProjectMemberRequestBody2$outboundSchema.parse(addProjectMemberRequestBody2));
}
export function addProjectMemberRequestBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => AddProjectMemberRequestBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddProjectMemberRequestBody2' from JSON`);
}
/** @internal */
export const RequestBodyRole$inboundSchema = z.nativeEnum(RequestBodyRole);
/** @internal */
export const RequestBodyRole$outboundSchema = RequestBodyRole$inboundSchema;
/** @internal */
export const AddProjectMemberRequestBody1$inboundSchema = z.object({
    uid: z.string(),
    username: z.string().optional(),
    email: z.string().optional(),
    role: RequestBodyRole$inboundSchema,
});
/** @internal */
export const AddProjectMemberRequestBody1$outboundSchema = z.object({
    uid: z.string(),
    username: z.string().optional(),
    email: z.string().optional(),
    role: RequestBodyRole$outboundSchema,
});
export function addProjectMemberRequestBody1ToJSON(addProjectMemberRequestBody1) {
    return JSON.stringify(AddProjectMemberRequestBody1$outboundSchema.parse(addProjectMemberRequestBody1));
}
export function addProjectMemberRequestBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => AddProjectMemberRequestBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddProjectMemberRequestBody1' from JSON`);
}
/** @internal */
export const AddProjectMemberRequestBody$inboundSchema = z.union([
    z.lazy(() => AddProjectMemberRequestBody1$inboundSchema),
    z.lazy(() => AddProjectMemberRequestBody2$inboundSchema),
    z.lazy(() => AddProjectMemberRequestBody3$inboundSchema),
]);
/** @internal */
export const AddProjectMemberRequestBody$outboundSchema = z.union([
    z.lazy(() => AddProjectMemberRequestBody1$outboundSchema),
    z.lazy(() => AddProjectMemberRequestBody2$outboundSchema),
    z.lazy(() => AddProjectMemberRequestBody3$outboundSchema),
]);
export function addProjectMemberRequestBodyToJSON(addProjectMemberRequestBody) {
    return JSON.stringify(AddProjectMemberRequestBody$outboundSchema.parse(addProjectMemberRequestBody));
}
export function addProjectMemberRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AddProjectMemberRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddProjectMemberRequestBody' from JSON`);
}
/** @internal */
export const AddProjectMemberRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.union([
        z.lazy(() => AddProjectMemberRequestBody1$inboundSchema),
        z.lazy(() => AddProjectMemberRequestBody2$inboundSchema),
        z.lazy(() => AddProjectMemberRequestBody3$inboundSchema),
    ]),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const AddProjectMemberRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.union([
        z.lazy(() => AddProjectMemberRequestBody1$outboundSchema),
        z.lazy(() => AddProjectMemberRequestBody2$outboundSchema),
        z.lazy(() => AddProjectMemberRequestBody3$outboundSchema),
    ]),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function addProjectMemberRequestToJSON(addProjectMemberRequest) {
    return JSON.stringify(AddProjectMemberRequest$outboundSchema.parse(addProjectMemberRequest));
}
export function addProjectMemberRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AddProjectMemberRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddProjectMemberRequest' from JSON`);
}
/** @internal */
export const AddProjectMemberResponseBody$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const AddProjectMemberResponseBody$outboundSchema = z.object({
    id: z.string(),
});
export function addProjectMemberResponseBodyToJSON(addProjectMemberResponseBody) {
    return JSON.stringify(AddProjectMemberResponseBody$outboundSchema.parse(addProjectMemberResponseBody));
}
export function addProjectMemberResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AddProjectMemberResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddProjectMemberResponseBody' from JSON`);
}
