// @vercel/sdk/esm/models/approverollingreleasestageop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The current state of the rolling release
 */
export const ApproveRollingReleaseStageState = {
    Active: "ACTIVE",
    Complete: "COMPLETE",
    Aborted: "ABORTED",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const ApproveRollingReleaseStageTarget = {
    Staging: "staging",
    Production: "production",
};
/**
 * Where was the deployment created from
 */
export const ApproveRollingReleaseStageSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const ApproveRollingReleaseStageReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const ApproveRollingReleaseStageRollingReleaseTarget = {
    Staging: "staging",
    Production: "production",
};
/**
 * Where was the deployment created from
 */
export const ApproveRollingReleaseStageRollingReleaseSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const ApproveRollingReleaseStageRollingReleaseReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * The advancement type of the rolling release
 */
export const ApproveRollingReleaseStageAdvancementType = {
    Automatic: "automatic",
    ManualApproval: "manual-approval",
};
/** @internal */
export const ApproveRollingReleaseStageRequestBody$inboundSchema = z.object({
    nextStageIndex: z.number(),
    canaryDeploymentId: z.string(),
});
/** @internal */
export const ApproveRollingReleaseStageRequestBody$outboundSchema = z.object({
    nextStageIndex: z.number(),
    canaryDeploymentId: z.string(),
});
export function approveRollingReleaseStageRequestBodyToJSON(approveRollingReleaseStageRequestBody) {
    return JSON.stringify(ApproveRollingReleaseStageRequestBody$outboundSchema.parse(approveRollingReleaseStageRequestBody));
}
export function approveRollingReleaseStageRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageRequestBody' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => ApproveRollingReleaseStageRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const ApproveRollingReleaseStageRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => ApproveRollingReleaseStageRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function approveRollingReleaseStageRequestToJSON(approveRollingReleaseStageRequest) {
    return JSON.stringify(ApproveRollingReleaseStageRequest$outboundSchema.parse(approveRollingReleaseStageRequest));
}
export function approveRollingReleaseStageRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageRequest' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageState$inboundSchema = z.nativeEnum(ApproveRollingReleaseStageState);
/** @internal */
export const ApproveRollingReleaseStageState$outboundSchema = ApproveRollingReleaseStageState$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageTarget$inboundSchema = z.nativeEnum(ApproveRollingReleaseStageTarget);
/** @internal */
export const ApproveRollingReleaseStageTarget$outboundSchema = ApproveRollingReleaseStageTarget$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageSource$inboundSchema = z.nativeEnum(ApproveRollingReleaseStageSource);
/** @internal */
export const ApproveRollingReleaseStageSource$outboundSchema = ApproveRollingReleaseStageSource$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageReadyState$inboundSchema = z.nativeEnum(ApproveRollingReleaseStageReadyState);
/** @internal */
export const ApproveRollingReleaseStageReadyState$outboundSchema = ApproveRollingReleaseStageReadyState$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageCurrentDeployment$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(ApproveRollingReleaseStageTarget$inboundSchema)
        .optional(),
    source: ApproveRollingReleaseStageSource$inboundSchema.optional(),
    createdAt: z.number(),
    readyState: ApproveRollingReleaseStageReadyState$inboundSchema,
    readyStateAt: z.number().optional(),
});
/** @internal */
export const ApproveRollingReleaseStageCurrentDeployment$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(ApproveRollingReleaseStageTarget$outboundSchema)
        .optional(),
    source: ApproveRollingReleaseStageSource$outboundSchema.optional(),
    createdAt: z.number(),
    readyState: ApproveRollingReleaseStageReadyState$outboundSchema,
    readyStateAt: z.number().optional(),
});
export function approveRollingReleaseStageCurrentDeploymentToJSON(approveRollingReleaseStageCurrentDeployment) {
    return JSON.stringify(ApproveRollingReleaseStageCurrentDeployment$outboundSchema.parse(approveRollingReleaseStageCurrentDeployment));
}
export function approveRollingReleaseStageCurrentDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageCurrentDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageCurrentDeployment' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageRollingReleaseTarget$inboundSchema = z
    .nativeEnum(ApproveRollingReleaseStageRollingReleaseTarget);
/** @internal */
export const ApproveRollingReleaseStageRollingReleaseTarget$outboundSchema = ApproveRollingReleaseStageRollingReleaseTarget$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageRollingReleaseSource$inboundSchema = z
    .nativeEnum(ApproveRollingReleaseStageRollingReleaseSource);
/** @internal */
export const ApproveRollingReleaseStageRollingReleaseSource$outboundSchema = ApproveRollingReleaseStageRollingReleaseSource$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageRollingReleaseReadyState$inboundSchema = z
    .nativeEnum(ApproveRollingReleaseStageRollingReleaseReadyState);
/** @internal */
export const ApproveRollingReleaseStageRollingReleaseReadyState$outboundSchema = ApproveRollingReleaseStageRollingReleaseReadyState$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageCanaryDeployment$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(ApproveRollingReleaseStageRollingReleaseTarget$inboundSchema).optional(),
    source: ApproveRollingReleaseStageRollingReleaseSource$inboundSchema
        .optional(),
    createdAt: z.number(),
    readyState: ApproveRollingReleaseStageRollingReleaseReadyState$inboundSchema,
    readyStateAt: z.number().optional(),
});
/** @internal */
export const ApproveRollingReleaseStageCanaryDeployment$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(ApproveRollingReleaseStageRollingReleaseTarget$outboundSchema).optional(),
    source: ApproveRollingReleaseStageRollingReleaseSource$outboundSchema
        .optional(),
    createdAt: z.number(),
    readyState: ApproveRollingReleaseStageRollingReleaseReadyState$outboundSchema,
    readyStateAt: z.number().optional(),
});
export function approveRollingReleaseStageCanaryDeploymentToJSON(approveRollingReleaseStageCanaryDeployment) {
    return JSON.stringify(ApproveRollingReleaseStageCanaryDeployment$outboundSchema.parse(approveRollingReleaseStageCanaryDeployment));
}
export function approveRollingReleaseStageCanaryDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageCanaryDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageCanaryDeployment' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageAdvancementType$inboundSchema = z
    .nativeEnum(ApproveRollingReleaseStageAdvancementType);
/** @internal */
export const ApproveRollingReleaseStageAdvancementType$outboundSchema = ApproveRollingReleaseStageAdvancementType$inboundSchema;
/** @internal */
export const ApproveRollingReleaseStageStages$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const ApproveRollingReleaseStageStages$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function approveRollingReleaseStageStagesToJSON(approveRollingReleaseStageStages) {
    return JSON.stringify(ApproveRollingReleaseStageStages$outboundSchema.parse(approveRollingReleaseStageStages));
}
export function approveRollingReleaseStageStagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageStages$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageStages' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageActiveStage$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const ApproveRollingReleaseStageActiveStage$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function approveRollingReleaseStageActiveStageToJSON(approveRollingReleaseStageActiveStage) {
    return JSON.stringify(ApproveRollingReleaseStageActiveStage$outboundSchema.parse(approveRollingReleaseStageActiveStage));
}
export function approveRollingReleaseStageActiveStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageActiveStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageActiveStage' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageNextStage$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const ApproveRollingReleaseStageNextStage$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function approveRollingReleaseStageNextStageToJSON(approveRollingReleaseStageNextStage) {
    return JSON.stringify(ApproveRollingReleaseStageNextStage$outboundSchema.parse(approveRollingReleaseStageNextStage));
}
export function approveRollingReleaseStageNextStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageNextStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageNextStage' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageRollingRelease$inboundSchema = z.object({
    state: ApproveRollingReleaseStageState$inboundSchema,
    currentDeployment: z.nullable(z.lazy(() => ApproveRollingReleaseStageCurrentDeployment$inboundSchema)),
    canaryDeployment: z.nullable(z.lazy(() => ApproveRollingReleaseStageCanaryDeployment$inboundSchema)),
    queuedDeploymentId: z.nullable(z.string()),
    advancementType: ApproveRollingReleaseStageAdvancementType$inboundSchema,
    stages: z.array(z.lazy(() => ApproveRollingReleaseStageStages$inboundSchema)),
    activeStage: z.nullable(z.lazy(() => ApproveRollingReleaseStageActiveStage$inboundSchema)),
    nextStage: z.nullable(z.lazy(() => ApproveRollingReleaseStageNextStage$inboundSchema)),
    startedAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const ApproveRollingReleaseStageRollingRelease$outboundSchema = z.object({
    state: ApproveRollingReleaseStageState$outboundSchema,
    currentDeployment: z.nullable(z.lazy(() => ApproveRollingReleaseStageCurrentDeployment$outboundSchema)),
    canaryDeployment: z.nullable(z.lazy(() => ApproveRollingReleaseStageCanaryDeployment$outboundSchema)),
    queuedDeploymentId: z.nullable(z.string()),
    advancementType: ApproveRollingReleaseStageAdvancementType$outboundSchema,
    stages: z.array(z.lazy(() => ApproveRollingReleaseStageStages$outboundSchema)),
    activeStage: z.nullable(z.lazy(() => ApproveRollingReleaseStageActiveStage$outboundSchema)),
    nextStage: z.nullable(z.lazy(() => ApproveRollingReleaseStageNextStage$outboundSchema)),
    startedAt: z.number(),
    updatedAt: z.number(),
});
export function approveRollingReleaseStageRollingReleaseToJSON(approveRollingReleaseStageRollingRelease) {
    return JSON.stringify(ApproveRollingReleaseStageRollingRelease$outboundSchema.parse(approveRollingReleaseStageRollingRelease));
}
export function approveRollingReleaseStageRollingReleaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageRollingRelease$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageRollingRelease' from JSON`);
}
/** @internal */
export const ApproveRollingReleaseStageResponseBody$inboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => ApproveRollingReleaseStageRollingRelease$inboundSchema)),
});
/** @internal */
export const ApproveRollingReleaseStageResponseBody$outboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => ApproveRollingReleaseStageRollingRelease$outboundSchema)),
});
export function approveRollingReleaseStageResponseBodyToJSON(approveRollingReleaseStageResponseBody) {
    return JSON.stringify(ApproveRollingReleaseStageResponseBody$outboundSchema.parse(approveRollingReleaseStageResponseBody));
}
export function approveRollingReleaseStageResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ApproveRollingReleaseStageResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ApproveRollingReleaseStageResponseBody' from JSON`);
}
