// @vercel/sdk/esm/models/authuserlimited.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const AuthUserLimited$inboundSchema = z.object({
    limited: z.boolean(),
    id: z.string(),
    email: z.string(),
    name: z.nullable(z.string()),
    username: z.string(),
    avatar: z.nullable(z.string()),
    defaultTeamId: z.nullable(z.string()),
});
/** @internal */
export const AuthUserLimited$outboundSchema = z.object({
    limited: z.boolean(),
    id: z.string(),
    email: z.string(),
    name: z.nullable(z.string()),
    username: z.string(),
    avatar: z.nullable(z.string()),
    defaultTeamId: z.nullable(z.string()),
});
export function authUserLimitedToJSON(authUserLimited) {
    return JSON.stringify(AuthUserLimited$outboundSchema.parse(authUserLimited));
}
export function authUserLimitedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AuthUserLimited$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AuthUserLimited' from JSON`);
}
