// @vercel/sdk/esm/models/authtoken.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const AuthTokenScopesType = {
    Team: "team",
};
export const AuthTokenScopesOrigin = {
    Saml: "saml",
    Github: "github",
    Gitlab: "gitlab",
    Bitbucket: "bitbucket",
    Email: "email",
    Manual: "manual",
    Passkey: "passkey",
    Otp: "otp",
    Sms: "sms",
    Invite: "invite",
    Google: "google",
    Apple: "apple",
    App: "app",
};
export const ScopesType = {
    User: "user",
};
/**
 * Possible multi-factor origins
 */
export const AuthTokenScopes1Origin = {
    Totp: "totp",
    Webauthn: "webauthn",
    RecoveryCode: "recovery-code",
};
export const ScopesOrigin = {
    Saml: "saml",
    Github: "github",
    Gitlab: "gitlab",
    Bitbucket: "bitbucket",
    Email: "email",
    Manual: "manual",
    Passkey: "passkey",
    Otp: "otp",
    Sms: "sms",
    Invite: "invite",
    Google: "google",
    Apple: "apple",
    App: "app",
};
/** @internal */
export const AuthTokenScopesType$inboundSchema = z.nativeEnum(AuthTokenScopesType);
/** @internal */
export const AuthTokenScopesType$outboundSchema = AuthTokenScopesType$inboundSchema;
/** @internal */
export const AuthTokenScopesOrigin$inboundSchema = z.nativeEnum(AuthTokenScopesOrigin);
/** @internal */
export const AuthTokenScopesOrigin$outboundSchema = AuthTokenScopesOrigin$inboundSchema;
/** @internal */
export const Scopes2$inboundSchema = z.object({
    type: AuthTokenScopesType$inboundSchema,
    teamId: z.string(),
    origin: AuthTokenScopesOrigin$inboundSchema.optional(),
    createdAt: z.number(),
    expiresAt: z.number().optional(),
});
/** @internal */
export const Scopes2$outboundSchema = z.object({
    type: AuthTokenScopesType$outboundSchema,
    teamId: z.string(),
    origin: AuthTokenScopesOrigin$outboundSchema.optional(),
    createdAt: z.number(),
    expiresAt: z.number().optional(),
});
export function scopes2ToJSON(scopes2) {
    return JSON.stringify(Scopes2$outboundSchema.parse(scopes2));
}
export function scopes2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Scopes2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Scopes2' from JSON`);
}
/** @internal */
export const ScopesType$inboundSchema = z
    .nativeEnum(ScopesType);
/** @internal */
export const ScopesType$outboundSchema = ScopesType$inboundSchema;
/** @internal */
export const AuthTokenScopes1Origin$inboundSchema = z.nativeEnum(AuthTokenScopes1Origin);
/** @internal */
export const AuthTokenScopes1Origin$outboundSchema = AuthTokenScopes1Origin$inboundSchema;
/** @internal */
export const Sudo$inboundSchema = z
    .object({
    origin: AuthTokenScopes1Origin$inboundSchema,
    expiresAt: z.number(),
});
/** @internal */
export const Sudo$outboundSchema = z.object({
    origin: AuthTokenScopes1Origin$outboundSchema,
    expiresAt: z.number(),
});
export function sudoToJSON(sudo) {
    return JSON.stringify(Sudo$outboundSchema.parse(sudo));
}
export function sudoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sudo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sudo' from JSON`);
}
/** @internal */
export const ScopesOrigin$inboundSchema = z.nativeEnum(ScopesOrigin);
/** @internal */
export const ScopesOrigin$outboundSchema = ScopesOrigin$inboundSchema;
/** @internal */
export const Scopes1$inboundSchema = z.object({
    type: ScopesType$inboundSchema,
    sudo: z.lazy(() => Sudo$inboundSchema).optional(),
    origin: ScopesOrigin$inboundSchema.optional(),
    createdAt: z.number(),
    expiresAt: z.number().optional(),
});
/** @internal */
export const Scopes1$outboundSchema = z.object({
    type: ScopesType$outboundSchema,
    sudo: z.lazy(() => Sudo$outboundSchema).optional(),
    origin: ScopesOrigin$outboundSchema.optional(),
    createdAt: z.number(),
    expiresAt: z.number().optional(),
});
export function scopes1ToJSON(scopes1) {
    return JSON.stringify(Scopes1$outboundSchema.parse(scopes1));
}
export function scopes1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Scopes1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Scopes1' from JSON`);
}
/** @internal */
export const Scopes$inboundSchema = z
    .union([
    z.lazy(() => Scopes2$inboundSchema),
    z.lazy(() => Scopes1$inboundSchema),
]);
/** @internal */
export const Scopes$outboundSchema = z.union([
    z.lazy(() => Scopes2$outboundSchema),
    z.lazy(() => Scopes1$outboundSchema),
]);
export function scopesToJSON(scopes) {
    return JSON.stringify(Scopes$outboundSchema.parse(scopes));
}
export function scopesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Scopes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Scopes' from JSON`);
}
/** @internal */
export const AuthToken$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    type: z.string(),
    origin: z.string().optional(),
    scopes: z.array(z.union([
        z.lazy(() => Scopes2$inboundSchema),
        z.lazy(() => Scopes1$inboundSchema),
    ])).optional(),
    expiresAt: z.number().optional(),
    activeAt: z.number(),
    createdAt: z.number(),
});
/** @internal */
export const AuthToken$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    type: z.string(),
    origin: z.string().optional(),
    scopes: z.array(z.union([
        z.lazy(() => Scopes2$outboundSchema),
        z.lazy(() => Scopes1$outboundSchema),
    ])).optional(),
    expiresAt: z.number().optional(),
    activeAt: z.number(),
    createdAt: z.number(),
});
export function authTokenToJSON(authToken) {
    return JSON.stringify(AuthToken$outboundSchema.parse(authToken));
}
export function authTokenFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AuthToken$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AuthToken' from JSON`);
}
