// @vercel/sdk/esm/models/batchremoveprojectenvop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const BatchRemoveProjectEnvRequestBody$inboundSchema = z.object({
    ids: z.array(z.string()),
});
/** @internal */
export const BatchRemoveProjectEnvRequestBody$outboundSchema = z.object({
    ids: z.array(z.string()),
});
export function batchRemoveProjectEnvRequestBodyToJSON(batchRemoveProjectEnvRequestBody) {
    return JSON.stringify(BatchRemoveProjectEnvRequestBody$outboundSchema.parse(batchRemoveProjectEnvRequestBody));
}
export function batchRemoveProjectEnvRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BatchRemoveProjectEnvRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BatchRemoveProjectEnvRequestBody' from JSON`);
}
/** @internal */
export const BatchRemoveProjectEnvRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => BatchRemoveProjectEnvRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const BatchRemoveProjectEnvRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => BatchRemoveProjectEnvRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function batchRemoveProjectEnvRequestToJSON(batchRemoveProjectEnvRequest) {
    return JSON.stringify(BatchRemoveProjectEnvRequest$outboundSchema.parse(batchRemoveProjectEnvRequest));
}
export function batchRemoveProjectEnvRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BatchRemoveProjectEnvRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BatchRemoveProjectEnvRequest' from JSON`);
}
/** @internal */
export const BatchRemoveProjectEnvResponseBody$inboundSchema = z.object({
    deleted: z.number(),
    ids: z.array(z.string()),
});
/** @internal */
export const BatchRemoveProjectEnvResponseBody$outboundSchema = z.object({
    deleted: z.number(),
    ids: z.array(z.string()),
});
export function batchRemoveProjectEnvResponseBodyToJSON(batchRemoveProjectEnvResponseBody) {
    return JSON.stringify(BatchRemoveProjectEnvResponseBody$outboundSchema.parse(batchRemoveProjectEnvResponseBody));
}
export function batchRemoveProjectEnvResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BatchRemoveProjectEnvResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BatchRemoveProjectEnvResponseBody' from JSON`);
}
