// @vercel/sdk/esm/models/canceldeploymentop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { FlagJSONValue$inboundSchema, FlagJSONValue$outboundSchema, } from "./flagjsonvalue.js";
export const CancelDeploymentFramework = {
    Blitzjs: "blitzjs",
    Nextjs: "nextjs",
    Gatsby: "gatsby",
    Remix: "remix",
    ReactRouter: "react-router",
    Astro: "astro",
    Hexo: "hexo",
    Eleventy: "eleventy",
    Docusaurus2: "docusaurus-2",
    Docusaurus: "docusaurus",
    Preact: "preact",
    Solidstart1: "solidstart-1",
    Solidstart: "solidstart",
    Dojo: "dojo",
    Ember: "ember",
    Vue: "vue",
    Scully: "scully",
    IonicAngular: "ionic-angular",
    Angular: "angular",
    Polymer: "polymer",
    Svelte: "svelte",
    Sveltekit: "sveltekit",
    Sveltekit1: "sveltekit-1",
    IonicReact: "ionic-react",
    CreateReactApp: "create-react-app",
    Gridsome: "gridsome",
    Umijs: "umijs",
    Sapper: "sapper",
    Saber: "saber",
    Stencil: "stencil",
    Nuxtjs: "nuxtjs",
    Redwoodjs: "redwoodjs",
    Hugo: "hugo",
    Jekyll: "jekyll",
    Brunch: "brunch",
    Middleman: "middleman",
    Zola: "zola",
    Hydrogen: "hydrogen",
    Vite: "vite",
    TanstackStart: "tanstack-start",
    Vitepress: "vitepress",
    Vuepress: "vuepress",
    Parcel: "parcel",
    Fastapi: "fastapi",
    Flask: "flask",
    Fasthtml: "fasthtml",
    SanityV3: "sanity-v3",
    Sanity: "sanity",
    Storybook: "storybook",
    Nitro: "nitro",
    Hono: "hono",
    Express: "express",
    H3: "h3",
    Nestjs: "nestjs",
    Elysia: "elysia",
    Fastify: "fastify",
    Xmcp: "xmcp",
};
export const CancelDeploymentDeploymentsStatus = {
    Skipped: "skipped",
    Pending: "pending",
    Ready: "ready",
    Error: "error",
    Timeout: "timeout",
};
/**
 * Must be `http` or `https`.
 */
export const CancelDeploymentProtocol = {
    Http: "http",
    Https: "https",
};
export const CancelDeploymentFormats = {
    ImageAvif: "image/avif",
    ImageWebp: "image/webp",
};
export const CancelDeploymentContentDispositionType = {
    Inline: "inline",
    Attachment: "attachment",
};
export const CancelDeploymentDeploymentsReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Ready: "READY",
};
export const CancelDeploymentStatus = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * The type of environment (production, preview, or development)
 */
export const CancelDeploymentCustomEnvironmentType = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of matching to perform
 */
export const CancelDeploymentCustomEnvironmentDeploymentsType = {
    EndsWith: "endsWith",
    StartsWith: "startsWith",
    Equals: "equals",
};
export const CancelDeploymentOomReport = {
    OutOfMemory: "out-of-memory",
};
export const CancelDeploymentChecksState = {
    Registered: "registered",
    Running: "running",
    Completed: "completed",
};
export const CancelDeploymentChecksConclusion = {
    Skipped: "skipped",
    Succeeded: "succeeded",
    Failed: "failed",
    Canceled: "canceled",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type = {
    Bitbucket: "bitbucket",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type = {
    Gitlab: "gitlab",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type = {
    GithubLimited: "github-limited",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type = {
    GithubCustomHost: "github-custom-host",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type = {
    Github: "github",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type = {
    Custom: "custom",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type = {
    Bitbucket: "bitbucket",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type = {
    Bitbucket: "bitbucket",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type = {
    Gitlab: "gitlab",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType = {
    GithubLimited: "github-limited",
};
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType = {
    GithubLimited: "github-limited",
};
export const CancelDeploymentGitSourceDeploymentsResponse200Type = {
    GithubCustomHost: "github-custom-host",
};
export const CancelDeploymentGitSourceDeploymentsResponseType = {
    GithubCustomHost: "github-custom-host",
};
export const CancelDeploymentGitSourceDeploymentsType = {
    Github: "github",
};
export const CancelDeploymentGitSourceType = {
    Github: "github",
};
/**
 * If set it overrides the `projectSettings.nodeVersion` for this deployment.
 */
export const CancelDeploymentNodeVersion = {
    TwentyFourDotX: "24.x",
    TwentyTwoDotX: "22.x",
    TwentyDotX: "20.x",
    EighteenDotX: "18.x",
    SixteenDotX: "16.x",
    FourteenDotX: "14.x",
    TwelveDotX: "12.x",
    TenDotX: "10.x",
    EightDot10DotX: "8.10.x",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const CancelDeploymentReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * Substate of deployment when readyState is 'READY' Tracks whether or not deployment has seen production traffic: - STAGED: never seen production traffic - ROLLING: in the process of having production traffic gradually transitioned. - PROMOTED: has seen production traffic
 */
export const CancelDeploymentReadySubstate = {
    Staged: "STAGED",
    Rolling: "ROLLING",
    Promoted: "PROMOTED",
};
/**
 * Where was the deployment created from
 */
export const CancelDeploymentSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const CancelDeploymentTarget = {
    Staging: "staging",
    Production: "production",
};
export const CancelDeploymentType = {
    Lambdas: "LAMBDAS",
};
export const CancelDeploymentArchitecture = {
    X8664: "x86_64",
    Arm64: "arm64",
};
/**
 * Event type - must be "queue/v1beta" (REQUIRED)
 */
export const CancelDeploymentDeploymentsType = {
    QueueV1beta: "queue/v1beta",
};
export const CancelDeploymentPlan = {
    Pro: "pro",
    Enterprise: "enterprise",
    Hobby: "hobby",
};
export const RoutesHandle = {
    Error: "error",
    Filesystem: "filesystem",
    Hit: "hit",
    Miss: "miss",
    Rewrite: "rewrite",
    Resource: "resource",
};
export const CancelDeploymentHasDeploymentsType = {
    Header: "header",
    Cookie: "cookie",
    Query: "query",
};
export const CancelDeploymentHasType = {
    Host: "host",
};
export const CancelDeploymentMissingDeploymentsType = {
    Header: "header",
    Cookie: "cookie",
    Query: "query",
};
export const CancelDeploymentMissingType = {
    Host: "host",
};
export const CancelDeploymentRoutesAction = {
    Challenge: "challenge",
    Deny: "deny",
};
export const CancelDeploymentRoutesType = {
    RequestHeaders: "request.headers",
    RequestQuery: "request.query",
    ResponseHeaders: "response.headers",
};
export const RoutesOp = {
    Append: "append",
    Set: "set",
    Delete: "delete",
};
export const CancelDeploymentGitRepoDeploymentsResponseType = {
    Bitbucket: "bitbucket",
};
export const CancelDeploymentGitRepoDeploymentsResponseOwnerType = {
    Team: "team",
    User: "user",
};
export const CancelDeploymentGitRepoDeploymentsType = {
    Github: "github",
};
export const CancelDeploymentGitRepoDeploymentsOwnerType = {
    Team: "team",
    User: "user",
};
export const CancelDeploymentGitRepoType = {
    Gitlab: "gitlab",
};
export const CancelDeploymentGitRepoOwnerType = {
    Team: "team",
    User: "user",
};
/**
 * The result of the microfrontends config upload during deployment creation / build. Only set for default app deployments. The config upload is attempted during deployment create, and then again during the build. If the config is not in the root directory, or the deployment is prebuilt, the config cannot be uploaded during deployment create. The upload during deployment build finds the config even if it's not in the root directory, as it has access to all files. Uploading the config during create is ideal, as then all child deployments are guaranteed to have access to the default app deployment config even if the default app has not yet started building. If the config is not uploaded, the child app will show as building until the config has been uploaded during the default app build. - `success` - The config was uploaded successfully, either when the deployment was created or during the build. - `waiting_on_build` - The config could not be uploaded during deployment create, will be attempted again during the build. - `no_config` - No config was found. Only set once the build has not found the config in any of the deployment's files. - `undefined` - Legacy deployments, or there was an error uploading the config during deployment create.
 */
export const MicrofrontendsMfeConfigUploadState = {
    Success: "success",
    WaitingOnBuild: "waiting_on_build",
    NoConfig: "no_config",
};
export const CancelDeploymentFunctionType = {
    Fluid: "fluid",
    Standard: "standard",
};
export const CancelDeploymentFunctionMemoryType = {
    Standard: "standard",
    StandardLegacy: "standard_legacy",
    Performance: "performance",
};
export const CancelDeploymentState = {
    Succeeded: "succeeded",
    Failed: "failed",
    Pending: "pending",
};
/** @internal */
export const CancelDeploymentRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const CancelDeploymentRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function cancelDeploymentRequestToJSON(cancelDeploymentRequest) {
    return JSON.stringify(CancelDeploymentRequest$outboundSchema.parse(cancelDeploymentRequest));
}
export function cancelDeploymentRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRequest' from JSON`);
}
/** @internal */
export const CancelDeploymentAliasAssignedAt$inboundSchema = z.union([z.number(), z.boolean()]);
/** @internal */
export const CancelDeploymentAliasAssignedAt$outboundSchema = z.union([z.number(), z.boolean()]);
export function cancelDeploymentAliasAssignedAtToJSON(cancelDeploymentAliasAssignedAt) {
    return JSON.stringify(CancelDeploymentAliasAssignedAt$outboundSchema.parse(cancelDeploymentAliasAssignedAt));
}
export function cancelDeploymentAliasAssignedAtFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentAliasAssignedAt$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentAliasAssignedAt' from JSON`);
}
/** @internal */
export const CancelDeploymentBuild$inboundSchema = z.object({
    env: z.array(z.string()),
});
/** @internal */
export const CancelDeploymentBuild$outboundSchema = z.object({
    env: z.array(z.string()),
});
export function cancelDeploymentBuildToJSON(cancelDeploymentBuild) {
    return JSON.stringify(CancelDeploymentBuild$outboundSchema.parse(cancelDeploymentBuild));
}
export function cancelDeploymentBuildFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentBuild$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentBuild' from JSON`);
}
/** @internal */
export const CancelDeploymentBuilds$inboundSchema = z.object({
    use: z.string(),
    src: z.string().optional(),
    config: z.record(z.any()).optional(),
});
/** @internal */
export const CancelDeploymentBuilds$outboundSchema = z.object({
    use: z.string(),
    src: z.string().optional(),
    config: z.record(z.any()).optional(),
});
export function cancelDeploymentBuildsToJSON(cancelDeploymentBuilds) {
    return JSON.stringify(CancelDeploymentBuilds$outboundSchema.parse(cancelDeploymentBuilds));
}
export function cancelDeploymentBuildsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentBuilds$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentBuilds' from JSON`);
}
/** @internal */
export const CancelDeploymentFramework$inboundSchema = z.nativeEnum(CancelDeploymentFramework);
/** @internal */
export const CancelDeploymentFramework$outboundSchema = CancelDeploymentFramework$inboundSchema;
/** @internal */
export const CancelDeploymentSpeedInsights$inboundSchema = z.object({
    id: z.string(),
    enabledAt: z.number().optional(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    hasData: z.boolean().optional(),
    paidAt: z.number().optional(),
});
/** @internal */
export const CancelDeploymentSpeedInsights$outboundSchema = z.object({
    id: z.string(),
    enabledAt: z.number().optional(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    hasData: z.boolean().optional(),
    paidAt: z.number().optional(),
});
export function cancelDeploymentSpeedInsightsToJSON(cancelDeploymentSpeedInsights) {
    return JSON.stringify(CancelDeploymentSpeedInsights$outboundSchema.parse(cancelDeploymentSpeedInsights));
}
export function cancelDeploymentSpeedInsightsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentSpeedInsights$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentSpeedInsights' from JSON`);
}
/** @internal */
export const CancelDeploymentWebAnalytics$inboundSchema = z.object({
    id: z.string(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    enabledAt: z.number().optional(),
    hasData: z.boolean().optional(),
});
/** @internal */
export const CancelDeploymentWebAnalytics$outboundSchema = z.object({
    id: z.string(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    enabledAt: z.number().optional(),
    hasData: z.boolean().optional(),
});
export function cancelDeploymentWebAnalyticsToJSON(cancelDeploymentWebAnalytics) {
    return JSON.stringify(CancelDeploymentWebAnalytics$outboundSchema.parse(cancelDeploymentWebAnalytics));
}
export function cancelDeploymentWebAnalyticsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentWebAnalytics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentWebAnalytics' from JSON`);
}
/** @internal */
export const CancelDeploymentProjectSettings$inboundSchema = z.object({
    buildCommand: z.nullable(z.string()).optional(),
    devCommand: z.nullable(z.string()).optional(),
    framework: z.nullable(CancelDeploymentFramework$inboundSchema).optional(),
    commandForIgnoringBuildStep: z.nullable(z.string()).optional(),
    installCommand: z.nullable(z.string()).optional(),
    outputDirectory: z.nullable(z.string()).optional(),
    speedInsights: z.lazy(() => CancelDeploymentSpeedInsights$inboundSchema)
        .optional(),
    webAnalytics: z.lazy(() => CancelDeploymentWebAnalytics$inboundSchema)
        .optional(),
});
/** @internal */
export const CancelDeploymentProjectSettings$outboundSchema = z.object({
    buildCommand: z.nullable(z.string()).optional(),
    devCommand: z.nullable(z.string()).optional(),
    framework: z.nullable(CancelDeploymentFramework$outboundSchema).optional(),
    commandForIgnoringBuildStep: z.nullable(z.string()).optional(),
    installCommand: z.nullable(z.string()).optional(),
    outputDirectory: z.nullable(z.string()).optional(),
    speedInsights: z.lazy(() => CancelDeploymentSpeedInsights$outboundSchema)
        .optional(),
    webAnalytics: z.lazy(() => CancelDeploymentWebAnalytics$outboundSchema)
        .optional(),
});
export function cancelDeploymentProjectSettingsToJSON(cancelDeploymentProjectSettings) {
    return JSON.stringify(CancelDeploymentProjectSettings$outboundSchema.parse(cancelDeploymentProjectSettings));
}
export function cancelDeploymentProjectSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentProjectSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentProjectSettings' from JSON`);
}
/** @internal */
export const CancelDeploymentDeploymentsStatus$inboundSchema = z.nativeEnum(CancelDeploymentDeploymentsStatus);
/** @internal */
export const CancelDeploymentDeploymentsStatus$outboundSchema = CancelDeploymentDeploymentsStatus$inboundSchema;
/** @internal */
export const CancelDeploymentIntegrations$inboundSchema = z.object({
    status: CancelDeploymentDeploymentsStatus$inboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
    skippedAt: z.number().optional(),
    skippedBy: z.string().optional(),
});
/** @internal */
export const CancelDeploymentIntegrations$outboundSchema = z.object({
    status: CancelDeploymentDeploymentsStatus$outboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
    skippedAt: z.number().optional(),
    skippedBy: z.string().optional(),
});
export function cancelDeploymentIntegrationsToJSON(cancelDeploymentIntegrations) {
    return JSON.stringify(CancelDeploymentIntegrations$outboundSchema.parse(cancelDeploymentIntegrations));
}
export function cancelDeploymentIntegrationsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentIntegrations$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentIntegrations' from JSON`);
}
/** @internal */
export const CancelDeploymentProtocol$inboundSchema = z.nativeEnum(CancelDeploymentProtocol);
/** @internal */
export const CancelDeploymentProtocol$outboundSchema = CancelDeploymentProtocol$inboundSchema;
/** @internal */
export const CancelDeploymentRemotePatterns$inboundSchema = z.object({
    protocol: CancelDeploymentProtocol$inboundSchema.optional(),
    hostname: z.string(),
    port: z.string().optional(),
    pathname: z.string().optional(),
    search: z.string().optional(),
});
/** @internal */
export const CancelDeploymentRemotePatterns$outboundSchema = z.object({
    protocol: CancelDeploymentProtocol$outboundSchema.optional(),
    hostname: z.string(),
    port: z.string().optional(),
    pathname: z.string().optional(),
    search: z.string().optional(),
});
export function cancelDeploymentRemotePatternsToJSON(cancelDeploymentRemotePatterns) {
    return JSON.stringify(CancelDeploymentRemotePatterns$outboundSchema.parse(cancelDeploymentRemotePatterns));
}
export function cancelDeploymentRemotePatternsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRemotePatterns$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRemotePatterns' from JSON`);
}
/** @internal */
export const CancelDeploymentLocalPatterns$inboundSchema = z.object({
    pathname: z.string().optional(),
    search: z.string().optional(),
});
/** @internal */
export const CancelDeploymentLocalPatterns$outboundSchema = z.object({
    pathname: z.string().optional(),
    search: z.string().optional(),
});
export function cancelDeploymentLocalPatternsToJSON(cancelDeploymentLocalPatterns) {
    return JSON.stringify(CancelDeploymentLocalPatterns$outboundSchema.parse(cancelDeploymentLocalPatterns));
}
export function cancelDeploymentLocalPatternsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentLocalPatterns$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentLocalPatterns' from JSON`);
}
/** @internal */
export const CancelDeploymentFormats$inboundSchema = z.nativeEnum(CancelDeploymentFormats);
/** @internal */
export const CancelDeploymentFormats$outboundSchema = CancelDeploymentFormats$inboundSchema;
/** @internal */
export const CancelDeploymentContentDispositionType$inboundSchema = z.nativeEnum(CancelDeploymentContentDispositionType);
/** @internal */
export const CancelDeploymentContentDispositionType$outboundSchema = CancelDeploymentContentDispositionType$inboundSchema;
/** @internal */
export const CancelDeploymentImages$inboundSchema = z.object({
    sizes: z.array(z.number()).optional(),
    qualities: z.array(z.number()).optional(),
    domains: z.array(z.string()).optional(),
    remotePatterns: z.array(z.lazy(() => CancelDeploymentRemotePatterns$inboundSchema)).optional(),
    localPatterns: z.array(z.lazy(() => CancelDeploymentLocalPatterns$inboundSchema)).optional(),
    minimumCacheTTL: z.number().optional(),
    formats: z.array(CancelDeploymentFormats$inboundSchema).optional(),
    dangerouslyAllowSVG: z.boolean().optional(),
    contentSecurityPolicy: z.string().optional(),
    contentDispositionType: CancelDeploymentContentDispositionType$inboundSchema
        .optional(),
});
/** @internal */
export const CancelDeploymentImages$outboundSchema = z.object({
    sizes: z.array(z.number()).optional(),
    qualities: z.array(z.number()).optional(),
    domains: z.array(z.string()).optional(),
    remotePatterns: z.array(z.lazy(() => CancelDeploymentRemotePatterns$outboundSchema)).optional(),
    localPatterns: z.array(z.lazy(() => CancelDeploymentLocalPatterns$outboundSchema)).optional(),
    minimumCacheTTL: z.number().optional(),
    formats: z.array(CancelDeploymentFormats$outboundSchema).optional(),
    dangerouslyAllowSVG: z.boolean().optional(),
    contentSecurityPolicy: z.string().optional(),
    contentDispositionType: CancelDeploymentContentDispositionType$outboundSchema
        .optional(),
});
export function cancelDeploymentImagesToJSON(cancelDeploymentImages) {
    return JSON.stringify(CancelDeploymentImages$outboundSchema.parse(cancelDeploymentImages));
}
export function cancelDeploymentImagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentImages$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentImages' from JSON`);
}
/** @internal */
export const CancelDeploymentCreator$inboundSchema = z.object({
    uid: z.string(),
    username: z.string().optional(),
    avatar: z.string().optional(),
});
/** @internal */
export const CancelDeploymentCreator$outboundSchema = z.object({
    uid: z.string(),
    username: z.string().optional(),
    avatar: z.string().optional(),
});
export function cancelDeploymentCreatorToJSON(cancelDeploymentCreator) {
    return JSON.stringify(CancelDeploymentCreator$outboundSchema.parse(cancelDeploymentCreator));
}
export function cancelDeploymentCreatorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCreator$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCreator' from JSON`);
}
/** @internal */
export const CancelDeploymentDeploymentsReadyState$inboundSchema = z.nativeEnum(CancelDeploymentDeploymentsReadyState);
/** @internal */
export const CancelDeploymentDeploymentsReadyState$outboundSchema = CancelDeploymentDeploymentsReadyState$inboundSchema;
/** @internal */
export const CancelDeploymentOutput$inboundSchema = z.object({
    path: z.string(),
    functionName: z.string(),
});
/** @internal */
export const CancelDeploymentOutput$outboundSchema = z.object({
    path: z.string(),
    functionName: z.string(),
});
export function cancelDeploymentOutputToJSON(cancelDeploymentOutput) {
    return JSON.stringify(CancelDeploymentOutput$outboundSchema.parse(cancelDeploymentOutput));
}
export function cancelDeploymentOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentOutput' from JSON`);
}
/** @internal */
export const CancelDeploymentLambdas$inboundSchema = z.object({
    id: z.string(),
    createdAt: z.number().optional(),
    readyState: CancelDeploymentDeploymentsReadyState$inboundSchema.optional(),
    entrypoint: z.nullable(z.string()).optional(),
    readyStateAt: z.number().optional(),
    output: z.array(z.lazy(() => CancelDeploymentOutput$inboundSchema)),
});
/** @internal */
export const CancelDeploymentLambdas$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.number().optional(),
    readyState: CancelDeploymentDeploymentsReadyState$outboundSchema.optional(),
    entrypoint: z.nullable(z.string()).optional(),
    readyStateAt: z.number().optional(),
    output: z.array(z.lazy(() => CancelDeploymentOutput$outboundSchema)),
});
export function cancelDeploymentLambdasToJSON(cancelDeploymentLambdas) {
    return JSON.stringify(CancelDeploymentLambdas$outboundSchema.parse(cancelDeploymentLambdas));
}
export function cancelDeploymentLambdasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentLambdas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentLambdas' from JSON`);
}
/** @internal */
export const CancelDeploymentStatus$inboundSchema = z.nativeEnum(CancelDeploymentStatus);
/** @internal */
export const CancelDeploymentStatus$outboundSchema = CancelDeploymentStatus$inboundSchema;
/** @internal */
export const CancelDeploymentTeam$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    avatar: z.string().optional(),
    slug: z.string(),
});
/** @internal */
export const CancelDeploymentTeam$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    avatar: z.string().optional(),
    slug: z.string(),
});
export function cancelDeploymentTeamToJSON(cancelDeploymentTeam) {
    return JSON.stringify(CancelDeploymentTeam$outboundSchema.parse(cancelDeploymentTeam));
}
export function cancelDeploymentTeamFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentTeam$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentTeam' from JSON`);
}
/** @internal */
export const CancelDeploymentCustomEnvironment2$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CancelDeploymentCustomEnvironment2$outboundSchema = z.object({
    id: z.string(),
});
export function cancelDeploymentCustomEnvironment2ToJSON(cancelDeploymentCustomEnvironment2) {
    return JSON.stringify(CancelDeploymentCustomEnvironment2$outboundSchema.parse(cancelDeploymentCustomEnvironment2));
}
export function cancelDeploymentCustomEnvironment2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCustomEnvironment2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCustomEnvironment2' from JSON`);
}
/** @internal */
export const CancelDeploymentCustomEnvironmentType$inboundSchema = z.nativeEnum(CancelDeploymentCustomEnvironmentType);
/** @internal */
export const CancelDeploymentCustomEnvironmentType$outboundSchema = CancelDeploymentCustomEnvironmentType$inboundSchema;
/** @internal */
export const CancelDeploymentCustomEnvironmentDeploymentsType$inboundSchema = z
    .nativeEnum(CancelDeploymentCustomEnvironmentDeploymentsType);
/** @internal */
export const CancelDeploymentCustomEnvironmentDeploymentsType$outboundSchema = CancelDeploymentCustomEnvironmentDeploymentsType$inboundSchema;
/** @internal */
export const CancelDeploymentCustomEnvironmentBranchMatcher$inboundSchema = z.object({
    type: CancelDeploymentCustomEnvironmentDeploymentsType$inboundSchema,
    pattern: z.string(),
});
/** @internal */
export const CancelDeploymentCustomEnvironmentBranchMatcher$outboundSchema = z.object({
    type: CancelDeploymentCustomEnvironmentDeploymentsType$outboundSchema,
    pattern: z.string(),
});
export function cancelDeploymentCustomEnvironmentBranchMatcherToJSON(cancelDeploymentCustomEnvironmentBranchMatcher) {
    return JSON.stringify(CancelDeploymentCustomEnvironmentBranchMatcher$outboundSchema.parse(cancelDeploymentCustomEnvironmentBranchMatcher));
}
export function cancelDeploymentCustomEnvironmentBranchMatcherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCustomEnvironmentBranchMatcher$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCustomEnvironmentBranchMatcher' from JSON`);
}
/** @internal */
export const CancelDeploymentCustomEnvironmentVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const CancelDeploymentCustomEnvironmentVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
export function cancelDeploymentCustomEnvironmentVerificationToJSON(cancelDeploymentCustomEnvironmentVerification) {
    return JSON.stringify(CancelDeploymentCustomEnvironmentVerification$outboundSchema.parse(cancelDeploymentCustomEnvironmentVerification));
}
export function cancelDeploymentCustomEnvironmentVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCustomEnvironmentVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCustomEnvironmentVerification' from JSON`);
}
/** @internal */
export const CancelDeploymentCustomEnvironmentDomains$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => CancelDeploymentCustomEnvironmentVerification$inboundSchema)).optional(),
});
/** @internal */
export const CancelDeploymentCustomEnvironmentDomains$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => CancelDeploymentCustomEnvironmentVerification$outboundSchema)).optional(),
});
export function cancelDeploymentCustomEnvironmentDomainsToJSON(cancelDeploymentCustomEnvironmentDomains) {
    return JSON.stringify(CancelDeploymentCustomEnvironmentDomains$outboundSchema.parse(cancelDeploymentCustomEnvironmentDomains));
}
export function cancelDeploymentCustomEnvironmentDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCustomEnvironmentDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCustomEnvironmentDomains' from JSON`);
}
/** @internal */
export const CancelDeploymentCustomEnvironment1$inboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: CancelDeploymentCustomEnvironmentType$inboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => CancelDeploymentCustomEnvironmentBranchMatcher$inboundSchema).optional(),
    domains: z.array(z.lazy(() => CancelDeploymentCustomEnvironmentDomains$inboundSchema)).optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const CancelDeploymentCustomEnvironment1$outboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: CancelDeploymentCustomEnvironmentType$outboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => CancelDeploymentCustomEnvironmentBranchMatcher$outboundSchema).optional(),
    domains: z.array(z.lazy(() => CancelDeploymentCustomEnvironmentDomains$outboundSchema)).optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
export function cancelDeploymentCustomEnvironment1ToJSON(cancelDeploymentCustomEnvironment1) {
    return JSON.stringify(CancelDeploymentCustomEnvironment1$outboundSchema.parse(cancelDeploymentCustomEnvironment1));
}
export function cancelDeploymentCustomEnvironment1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCustomEnvironment1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCustomEnvironment1' from JSON`);
}
/** @internal */
export const CancelDeploymentCustomEnvironment$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentCustomEnvironment1$inboundSchema),
    z.lazy(() => CancelDeploymentCustomEnvironment2$inboundSchema),
]);
/** @internal */
export const CancelDeploymentCustomEnvironment$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentCustomEnvironment1$outboundSchema),
    z.lazy(() => CancelDeploymentCustomEnvironment2$outboundSchema),
]);
export function cancelDeploymentCustomEnvironmentToJSON(cancelDeploymentCustomEnvironment) {
    return JSON.stringify(CancelDeploymentCustomEnvironment$outboundSchema.parse(cancelDeploymentCustomEnvironment));
}
export function cancelDeploymentCustomEnvironmentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCustomEnvironment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCustomEnvironment' from JSON`);
}
/** @internal */
export const CancelDeploymentOomReport$inboundSchema = z.nativeEnum(CancelDeploymentOomReport);
/** @internal */
export const CancelDeploymentOomReport$outboundSchema = CancelDeploymentOomReport$inboundSchema;
/** @internal */
export const CancelDeploymentAliasError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
/** @internal */
export const CancelDeploymentAliasError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
export function cancelDeploymentAliasErrorToJSON(cancelDeploymentAliasError) {
    return JSON.stringify(CancelDeploymentAliasError$outboundSchema.parse(cancelDeploymentAliasError));
}
export function cancelDeploymentAliasErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentAliasError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentAliasError' from JSON`);
}
/** @internal */
export const CancelDeploymentAliasWarning$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    link: z.string().optional(),
    action: z.string().optional(),
});
/** @internal */
export const CancelDeploymentAliasWarning$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    link: z.string().optional(),
    action: z.string().optional(),
});
export function cancelDeploymentAliasWarningToJSON(cancelDeploymentAliasWarning) {
    return JSON.stringify(CancelDeploymentAliasWarning$outboundSchema.parse(cancelDeploymentAliasWarning));
}
export function cancelDeploymentAliasWarningFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentAliasWarning$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentAliasWarning' from JSON`);
}
/** @internal */
export const CancelDeploymentChecksState$inboundSchema = z.nativeEnum(CancelDeploymentChecksState);
/** @internal */
export const CancelDeploymentChecksState$outboundSchema = CancelDeploymentChecksState$inboundSchema;
/** @internal */
export const CancelDeploymentChecksConclusion$inboundSchema = z.nativeEnum(CancelDeploymentChecksConclusion);
/** @internal */
export const CancelDeploymentChecksConclusion$outboundSchema = CancelDeploymentChecksConclusion$inboundSchema;
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$inboundSchema;
/** @internal */
export const GitSource15$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    owner: z.string().optional(),
    slug: z.string().optional(),
    workspaceUuid: z.string(),
    repoUuid: z.string(),
});
/** @internal */
export const GitSource15$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    owner: z.string().optional(),
    slug: z.string().optional(),
    workspaceUuid: z.string(),
    repoUuid: z.string(),
});
export function gitSource15ToJSON(gitSource15) {
    return JSON.stringify(GitSource15$outboundSchema.parse(gitSource15));
}
export function gitSource15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource15' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$inboundSchema;
/** @internal */
export const GitSource14$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    projectId: z.number(),
});
/** @internal */
export const GitSource14$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    projectId: z.number(),
});
export function gitSource14ToJSON(gitSource14) {
    return JSON.stringify(GitSource14$outboundSchema.parse(gitSource14));
}
export function gitSource14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource14' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$inboundSchema;
/** @internal */
export const GitSource13$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
/** @internal */
export const GitSource13$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
export function gitSource13ToJSON(gitSource13) {
    return JSON.stringify(GitSource13$outboundSchema.parse(gitSource13));
}
export function gitSource13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource13' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$inboundSchema;
/** @internal */
export const GitSource12$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$inboundSchema,
    host: z.string(),
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
/** @internal */
export const GitSource12$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$outboundSchema,
    host: z.string(),
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
export function gitSource12ToJSON(gitSource12) {
    return JSON.stringify(GitSource12$outboundSchema.parse(gitSource12));
}
export function gitSource12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource12' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$inboundSchema;
/** @internal */
export const GitSource11$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
/** @internal */
export const GitSource11$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
export function gitSource11ToJSON(gitSource11) {
    return JSON.stringify(GitSource11$outboundSchema.parse(gitSource11));
}
export function gitSource11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource11' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$inboundSchema;
/** @internal */
export const GitSource10$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    gitUrl: z.string(),
});
/** @internal */
export const GitSource10$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    gitUrl: z.string(),
});
export function gitSource10ToJSON(gitSource10) {
    return JSON.stringify(GitSource10$outboundSchema.parse(gitSource10));
}
export function gitSource10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource10' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$inboundSchema;
/** @internal */
export const GitSource9$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$inboundSchema,
    owner: z.string(),
    slug: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const GitSource9$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$outboundSchema,
    owner: z.string(),
    slug: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function gitSource9ToJSON(gitSource9) {
    return JSON.stringify(GitSource9$outboundSchema.parse(gitSource9));
}
export function gitSource9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource9' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$inboundSchema;
/** @internal */
export const GitSource8$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$inboundSchema,
    workspaceUuid: z.string().optional(),
    repoUuid: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const GitSource8$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$outboundSchema,
    workspaceUuid: z.string().optional(),
    repoUuid: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function gitSource8ToJSON(gitSource8) {
    return JSON.stringify(GitSource8$outboundSchema.parse(gitSource8));
}
export function gitSource8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource8' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$inboundSchema;
/** @internal */
export const CancelDeploymentGitSourceProjectId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentGitSourceProjectId$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentGitSourceProjectIdToJSON(cancelDeploymentGitSourceProjectId) {
    return JSON.stringify(CancelDeploymentGitSourceProjectId$outboundSchema.parse(cancelDeploymentGitSourceProjectId));
}
export function cancelDeploymentGitSourceProjectIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSourceProjectId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSourceProjectId' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSource7$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$inboundSchema,
    projectId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource7$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$outboundSchema,
    projectId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource7ToJSON(cancelDeploymentGitSource7) {
    return JSON.stringify(CancelDeploymentGitSource7$outboundSchema.parse(cancelDeploymentGitSource7));
}
export function cancelDeploymentGitSource7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource7' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const CancelDeploymentGitSource6$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$inboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource6$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$outboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource6ToJSON(cancelDeploymentGitSource6) {
    return JSON.stringify(CancelDeploymentGitSource6$outboundSchema.parse(cancelDeploymentGitSource6));
}
export function cancelDeploymentGitSource6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource6' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponseRepoId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponseRepoId$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentGitSourceDeploymentsResponseRepoIdToJSON(cancelDeploymentGitSourceDeploymentsResponseRepoId) {
    return JSON.stringify(CancelDeploymentGitSourceDeploymentsResponseRepoId$outboundSchema.parse(cancelDeploymentGitSourceDeploymentsResponseRepoId));
}
export function cancelDeploymentGitSourceDeploymentsResponseRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSourceDeploymentsResponseRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSourceDeploymentsResponseRepoId' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSource5$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$inboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource5$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$outboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource5ToJSON(cancelDeploymentGitSource5) {
    return JSON.stringify(CancelDeploymentGitSource5$outboundSchema.parse(cancelDeploymentGitSource5));
}
export function cancelDeploymentGitSource5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource5' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200Type$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceDeploymentsResponse200Type);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponse200Type$outboundSchema = CancelDeploymentGitSourceDeploymentsResponse200Type$inboundSchema;
/** @internal */
export const CancelDeploymentGitSource4$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200Type$inboundSchema,
    host: z.string(),
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource4$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponse200Type$outboundSchema,
    host: z.string(),
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource4ToJSON(cancelDeploymentGitSource4) {
    return JSON.stringify(CancelDeploymentGitSource4$outboundSchema.parse(cancelDeploymentGitSource4));
}
export function cancelDeploymentGitSource4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource4' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponseType$inboundSchema = z
    .nativeEnum(CancelDeploymentGitSourceDeploymentsResponseType);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsResponseType$outboundSchema = CancelDeploymentGitSourceDeploymentsResponseType$inboundSchema;
/** @internal */
export const CancelDeploymentGitSourceDeploymentsRepoId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsRepoId$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentGitSourceDeploymentsRepoIdToJSON(cancelDeploymentGitSourceDeploymentsRepoId) {
    return JSON.stringify(CancelDeploymentGitSourceDeploymentsRepoId$outboundSchema.parse(cancelDeploymentGitSourceDeploymentsRepoId));
}
export function cancelDeploymentGitSourceDeploymentsRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSourceDeploymentsRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSourceDeploymentsRepoId' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSource3$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponseType$inboundSchema,
    host: z.string(),
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource3$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsResponseType$outboundSchema,
    host: z.string(),
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource3ToJSON(cancelDeploymentGitSource3) {
    return JSON.stringify(CancelDeploymentGitSource3$outboundSchema.parse(cancelDeploymentGitSource3));
}
export function cancelDeploymentGitSource3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource3' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceDeploymentsType$inboundSchema = z
    .nativeEnum(CancelDeploymentGitSourceDeploymentsType);
/** @internal */
export const CancelDeploymentGitSourceDeploymentsType$outboundSchema = CancelDeploymentGitSourceDeploymentsType$inboundSchema;
/** @internal */
export const CancelDeploymentGitSource2$inboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsType$inboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource2$outboundSchema = z.object({
    type: CancelDeploymentGitSourceDeploymentsType$outboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource2ToJSON(cancelDeploymentGitSource2) {
    return JSON.stringify(CancelDeploymentGitSource2$outboundSchema.parse(cancelDeploymentGitSource2));
}
export function cancelDeploymentGitSource2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource2' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSourceType$inboundSchema = z.nativeEnum(CancelDeploymentGitSourceType);
/** @internal */
export const CancelDeploymentGitSourceType$outboundSchema = CancelDeploymentGitSourceType$inboundSchema;
/** @internal */
export const CancelDeploymentGitSourceRepoId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentGitSourceRepoId$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentGitSourceRepoIdToJSON(cancelDeploymentGitSourceRepoId) {
    return JSON.stringify(CancelDeploymentGitSourceRepoId$outboundSchema.parse(cancelDeploymentGitSourceRepoId));
}
export function cancelDeploymentGitSourceRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSourceRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSourceRepoId' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSource1$inboundSchema = z.object({
    type: CancelDeploymentGitSourceType$inboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CancelDeploymentGitSource1$outboundSchema = z.object({
    type: CancelDeploymentGitSourceType$outboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function cancelDeploymentGitSource1ToJSON(cancelDeploymentGitSource1) {
    return JSON.stringify(CancelDeploymentGitSource1$outboundSchema.parse(cancelDeploymentGitSource1));
}
export function cancelDeploymentGitSource1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource1' from JSON`);
}
/** @internal */
export const CancelDeploymentGitSource$inboundSchema = z.union([
    z.lazy(() => GitSource12$inboundSchema),
    z.lazy(() => GitSource15$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource4$inboundSchema),
    z.lazy(() => GitSource10$inboundSchema),
    z.lazy(() => GitSource11$inboundSchema),
    z.lazy(() => GitSource13$inboundSchema),
    z.lazy(() => GitSource14$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource2$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource3$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource6$inboundSchema),
    z.lazy(() => GitSource9$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource1$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource5$inboundSchema),
    z.lazy(() => CancelDeploymentGitSource7$inboundSchema),
    z.lazy(() => GitSource8$inboundSchema),
]);
/** @internal */
export const CancelDeploymentGitSource$outboundSchema = z.union([
    z.lazy(() => GitSource12$outboundSchema),
    z.lazy(() => GitSource15$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource4$outboundSchema),
    z.lazy(() => GitSource10$outboundSchema),
    z.lazy(() => GitSource11$outboundSchema),
    z.lazy(() => GitSource13$outboundSchema),
    z.lazy(() => GitSource14$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource2$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource3$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource6$outboundSchema),
    z.lazy(() => GitSource9$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource1$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource5$outboundSchema),
    z.lazy(() => CancelDeploymentGitSource7$outboundSchema),
    z.lazy(() => GitSource8$outboundSchema),
]);
export function cancelDeploymentGitSourceToJSON(cancelDeploymentGitSource) {
    return JSON.stringify(CancelDeploymentGitSource$outboundSchema.parse(cancelDeploymentGitSource));
}
export function cancelDeploymentGitSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitSource' from JSON`);
}
/** @internal */
export const CancelDeploymentNodeVersion$inboundSchema = z.nativeEnum(CancelDeploymentNodeVersion);
/** @internal */
export const CancelDeploymentNodeVersion$outboundSchema = CancelDeploymentNodeVersion$inboundSchema;
/** @internal */
export const CancelDeploymentProject$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    framework: z.nullable(z.string()).optional(),
});
/** @internal */
export const CancelDeploymentProject$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    framework: z.nullable(z.string()).optional(),
});
export function cancelDeploymentProjectToJSON(cancelDeploymentProject) {
    return JSON.stringify(CancelDeploymentProject$outboundSchema.parse(cancelDeploymentProject));
}
export function cancelDeploymentProjectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentProject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentProject' from JSON`);
}
/** @internal */
export const CancelDeploymentReadyState$inboundSchema = z.nativeEnum(CancelDeploymentReadyState);
/** @internal */
export const CancelDeploymentReadyState$outboundSchema = CancelDeploymentReadyState$inboundSchema;
/** @internal */
export const CancelDeploymentReadySubstate$inboundSchema = z.nativeEnum(CancelDeploymentReadySubstate);
/** @internal */
export const CancelDeploymentReadySubstate$outboundSchema = CancelDeploymentReadySubstate$inboundSchema;
/** @internal */
export const CancelDeploymentSource$inboundSchema = z.nativeEnum(CancelDeploymentSource);
/** @internal */
export const CancelDeploymentSource$outboundSchema = CancelDeploymentSource$inboundSchema;
/** @internal */
export const CancelDeploymentTarget$inboundSchema = z.nativeEnum(CancelDeploymentTarget);
/** @internal */
export const CancelDeploymentTarget$outboundSchema = CancelDeploymentTarget$inboundSchema;
/** @internal */
export const CancelDeploymentType$inboundSchema = z.nativeEnum(CancelDeploymentType);
/** @internal */
export const CancelDeploymentType$outboundSchema = CancelDeploymentType$inboundSchema;
/** @internal */
export const CancelDeploymentOidcTokenClaims$inboundSchema = z.object({
    iss: z.string(),
    sub: z.string(),
    scope: z.string(),
    aud: z.string(),
    owner: z.string(),
    owner_id: z.string(),
    project: z.string(),
    project_id: z.string(),
    environment: z.string(),
    plan: z.string(),
}).transform((v) => {
    return remap$(v, {
        "owner_id": "ownerId",
        "project_id": "projectId",
    });
});
/** @internal */
export const CancelDeploymentOidcTokenClaims$outboundSchema = z.object({
    iss: z.string(),
    sub: z.string(),
    scope: z.string(),
    aud: z.string(),
    owner: z.string(),
    ownerId: z.string(),
    project: z.string(),
    projectId: z.string(),
    environment: z.string(),
    plan: z.string(),
}).transform((v) => {
    return remap$(v, {
        ownerId: "owner_id",
        projectId: "project_id",
    });
});
export function cancelDeploymentOidcTokenClaimsToJSON(cancelDeploymentOidcTokenClaims) {
    return JSON.stringify(CancelDeploymentOidcTokenClaims$outboundSchema.parse(cancelDeploymentOidcTokenClaims));
}
export function cancelDeploymentOidcTokenClaimsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentOidcTokenClaims$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentOidcTokenClaims' from JSON`);
}
/** @internal */
export const CancelDeploymentCrons$inboundSchema = z.object({
    schedule: z.string(),
    path: z.string(),
});
/** @internal */
export const CancelDeploymentCrons$outboundSchema = z.object({
    schedule: z.string(),
    path: z.string(),
});
export function cancelDeploymentCronsToJSON(cancelDeploymentCrons) {
    return JSON.stringify(CancelDeploymentCrons$outboundSchema.parse(cancelDeploymentCrons));
}
export function cancelDeploymentCronsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentCrons$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentCrons' from JSON`);
}
/** @internal */
export const CancelDeploymentArchitecture$inboundSchema = z.nativeEnum(CancelDeploymentArchitecture);
/** @internal */
export const CancelDeploymentArchitecture$outboundSchema = CancelDeploymentArchitecture$inboundSchema;
/** @internal */
export const CancelDeploymentDeploymentsType$inboundSchema = z.nativeEnum(CancelDeploymentDeploymentsType);
/** @internal */
export const CancelDeploymentDeploymentsType$outboundSchema = CancelDeploymentDeploymentsType$inboundSchema;
/** @internal */
export const CancelDeploymentExperimentalTriggers$inboundSchema = z.object({
    type: CancelDeploymentDeploymentsType$inboundSchema,
    topic: z.string(),
    consumer: z.string(),
    maxDeliveries: z.number().optional(),
    retryAfterSeconds: z.number().optional(),
    initialDelaySeconds: z.number().optional(),
});
/** @internal */
export const CancelDeploymentExperimentalTriggers$outboundSchema = z.object({
    type: CancelDeploymentDeploymentsType$outboundSchema,
    topic: z.string(),
    consumer: z.string(),
    maxDeliveries: z.number().optional(),
    retryAfterSeconds: z.number().optional(),
    initialDelaySeconds: z.number().optional(),
});
export function cancelDeploymentExperimentalTriggersToJSON(cancelDeploymentExperimentalTriggers) {
    return JSON.stringify(CancelDeploymentExperimentalTriggers$outboundSchema.parse(cancelDeploymentExperimentalTriggers));
}
export function cancelDeploymentExperimentalTriggersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentExperimentalTriggers$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentExperimentalTriggers' from JSON`);
}
/** @internal */
export const CancelDeploymentFunctions$inboundSchema = z.object({
    architecture: CancelDeploymentArchitecture$inboundSchema.optional(),
    memory: z.number().optional(),
    maxDuration: z.number().optional(),
    runtime: z.string().optional(),
    includeFiles: z.string().optional(),
    excludeFiles: z.string().optional(),
    experimentalTriggers: z.array(z.lazy(() => CancelDeploymentExperimentalTriggers$inboundSchema)).optional(),
    supportsCancellation: z.boolean().optional(),
});
/** @internal */
export const CancelDeploymentFunctions$outboundSchema = z.object({
    architecture: CancelDeploymentArchitecture$outboundSchema.optional(),
    memory: z.number().optional(),
    maxDuration: z.number().optional(),
    runtime: z.string().optional(),
    includeFiles: z.string().optional(),
    excludeFiles: z.string().optional(),
    experimentalTriggers: z.array(z.lazy(() => CancelDeploymentExperimentalTriggers$outboundSchema)).optional(),
    supportsCancellation: z.boolean().optional(),
});
export function cancelDeploymentFunctionsToJSON(cancelDeploymentFunctions) {
    return JSON.stringify(CancelDeploymentFunctions$outboundSchema.parse(cancelDeploymentFunctions));
}
export function cancelDeploymentFunctionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentFunctions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentFunctions' from JSON`);
}
/** @internal */
export const CancelDeploymentPlan$inboundSchema = z.nativeEnum(CancelDeploymentPlan);
/** @internal */
export const CancelDeploymentPlan$outboundSchema = CancelDeploymentPlan$inboundSchema;
/** @internal */
export const CancelDeploymentRoutes3$inboundSchema = z.object({
    src: z.string(),
    continue: z.boolean(),
    middleware: z.number(),
});
/** @internal */
export const CancelDeploymentRoutes3$outboundSchema = z.object({
    src: z.string(),
    continue: z.boolean(),
    middleware: z.number(),
});
export function cancelDeploymentRoutes3ToJSON(cancelDeploymentRoutes3) {
    return JSON.stringify(CancelDeploymentRoutes3$outboundSchema.parse(cancelDeploymentRoutes3));
}
export function cancelDeploymentRoutes3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutes3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutes3' from JSON`);
}
/** @internal */
export const RoutesHandle$inboundSchema = z.nativeEnum(RoutesHandle);
/** @internal */
export const RoutesHandle$outboundSchema = RoutesHandle$inboundSchema;
/** @internal */
export const CancelDeploymentRoutes2$inboundSchema = z.object({
    handle: RoutesHandle$inboundSchema,
    src: z.string().optional(),
    dest: z.string().optional(),
    status: z.number().optional(),
});
/** @internal */
export const CancelDeploymentRoutes2$outboundSchema = z.object({
    handle: RoutesHandle$outboundSchema,
    src: z.string().optional(),
    dest: z.string().optional(),
    status: z.number().optional(),
});
export function cancelDeploymentRoutes2ToJSON(cancelDeploymentRoutes2) {
    return JSON.stringify(CancelDeploymentRoutes2$outboundSchema.parse(cancelDeploymentRoutes2));
}
export function cancelDeploymentRoutes2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutes2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutes2' from JSON`);
}
/** @internal */
export const CancelDeploymentHasDeploymentsType$inboundSchema = z.nativeEnum(CancelDeploymentHasDeploymentsType);
/** @internal */
export const CancelDeploymentHasDeploymentsType$outboundSchema = CancelDeploymentHasDeploymentsType$inboundSchema;
/** @internal */
export const CancelDeploymentValueDeploymentsEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentValueDeploymentsEq$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentValueDeploymentsEqToJSON(cancelDeploymentValueDeploymentsEq) {
    return JSON.stringify(CancelDeploymentValueDeploymentsEq$outboundSchema.parse(cancelDeploymentValueDeploymentsEq));
}
export function cancelDeploymentValueDeploymentsEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueDeploymentsEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueDeploymentsEq' from JSON`);
}
/** @internal */
export const CancelDeploymentValueDeployments2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CancelDeploymentValueDeployments2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function cancelDeploymentValueDeployments2ToJSON(cancelDeploymentValueDeployments2) {
    return JSON.stringify(CancelDeploymentValueDeployments2$outboundSchema.parse(cancelDeploymentValueDeployments2));
}
export function cancelDeploymentValueDeployments2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueDeployments2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueDeployments2' from JSON`);
}
/** @internal */
export const CancelDeploymentHasDeploymentsValue$inboundSchema = z.union([
    z.string(),
    z.lazy(() => CancelDeploymentValueDeployments2$inboundSchema),
]);
/** @internal */
export const CancelDeploymentHasDeploymentsValue$outboundSchema = z.union([
    z.string(),
    z.lazy(() => CancelDeploymentValueDeployments2$outboundSchema),
]);
export function cancelDeploymentHasDeploymentsValueToJSON(cancelDeploymentHasDeploymentsValue) {
    return JSON.stringify(CancelDeploymentHasDeploymentsValue$outboundSchema.parse(cancelDeploymentHasDeploymentsValue));
}
export function cancelDeploymentHasDeploymentsValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentHasDeploymentsValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentHasDeploymentsValue' from JSON`);
}
/** @internal */
export const CancelDeploymentHas2$inboundSchema = z.object({
    type: CancelDeploymentHasDeploymentsType$inboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValueDeployments2$inboundSchema),
    ]).optional(),
});
/** @internal */
export const CancelDeploymentHas2$outboundSchema = z.object({
    type: CancelDeploymentHasDeploymentsType$outboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValueDeployments2$outboundSchema),
    ]).optional(),
});
export function cancelDeploymentHas2ToJSON(cancelDeploymentHas2) {
    return JSON.stringify(CancelDeploymentHas2$outboundSchema.parse(cancelDeploymentHas2));
}
export function cancelDeploymentHas2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentHas2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentHas2' from JSON`);
}
/** @internal */
export const CancelDeploymentHasType$inboundSchema = z.nativeEnum(CancelDeploymentHasType);
/** @internal */
export const CancelDeploymentHasType$outboundSchema = CancelDeploymentHasType$inboundSchema;
/** @internal */
export const CancelDeploymentValueEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentValueEq$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentValueEqToJSON(cancelDeploymentValueEq) {
    return JSON.stringify(CancelDeploymentValueEq$outboundSchema.parse(cancelDeploymentValueEq));
}
export function cancelDeploymentValueEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueEq' from JSON`);
}
/** @internal */
export const CancelDeploymentValue2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CancelDeploymentValue2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function cancelDeploymentValue2ToJSON(cancelDeploymentValue2) {
    return JSON.stringify(CancelDeploymentValue2$outboundSchema.parse(cancelDeploymentValue2));
}
export function cancelDeploymentValue2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValue2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValue2' from JSON`);
}
/** @internal */
export const CancelDeploymentHasValue$inboundSchema = z.union([z.string(), z.lazy(() => CancelDeploymentValue2$inboundSchema)]);
/** @internal */
export const CancelDeploymentHasValue$outboundSchema = z.union([z.string(), z.lazy(() => CancelDeploymentValue2$outboundSchema)]);
export function cancelDeploymentHasValueToJSON(cancelDeploymentHasValue) {
    return JSON.stringify(CancelDeploymentHasValue$outboundSchema.parse(cancelDeploymentHasValue));
}
export function cancelDeploymentHasValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentHasValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentHasValue' from JSON`);
}
/** @internal */
export const CancelDeploymentHas1$inboundSchema = z.object({
    type: CancelDeploymentHasType$inboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValue2$inboundSchema),
    ]),
});
/** @internal */
export const CancelDeploymentHas1$outboundSchema = z.object({
    type: CancelDeploymentHasType$outboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValue2$outboundSchema),
    ]),
});
export function cancelDeploymentHas1ToJSON(cancelDeploymentHas1) {
    return JSON.stringify(CancelDeploymentHas1$outboundSchema.parse(cancelDeploymentHas1));
}
export function cancelDeploymentHas1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentHas1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentHas1' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutesHas$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentHas1$inboundSchema),
    z.lazy(() => CancelDeploymentHas2$inboundSchema),
]);
/** @internal */
export const CancelDeploymentRoutesHas$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentHas1$outboundSchema),
    z.lazy(() => CancelDeploymentHas2$outboundSchema),
]);
export function cancelDeploymentRoutesHasToJSON(cancelDeploymentRoutesHas) {
    return JSON.stringify(CancelDeploymentRoutesHas$outboundSchema.parse(cancelDeploymentRoutesHas));
}
export function cancelDeploymentRoutesHasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutesHas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutesHas' from JSON`);
}
/** @internal */
export const CancelDeploymentMissingDeploymentsType$inboundSchema = z.nativeEnum(CancelDeploymentMissingDeploymentsType);
/** @internal */
export const CancelDeploymentMissingDeploymentsType$outboundSchema = CancelDeploymentMissingDeploymentsType$inboundSchema;
/** @internal */
export const CancelDeploymentValueDeploymentsResponse200Eq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentValueDeploymentsResponse200Eq$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentValueDeploymentsResponse200EqToJSON(cancelDeploymentValueDeploymentsResponse200Eq) {
    return JSON.stringify(CancelDeploymentValueDeploymentsResponse200Eq$outboundSchema.parse(cancelDeploymentValueDeploymentsResponse200Eq));
}
export function cancelDeploymentValueDeploymentsResponse200EqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueDeploymentsResponse200Eq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueDeploymentsResponse200Eq' from JSON`);
}
/** @internal */
export const CancelDeploymentValueDeploymentsResponse2002$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CancelDeploymentValueDeploymentsResponse2002$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function cancelDeploymentValueDeploymentsResponse2002ToJSON(cancelDeploymentValueDeploymentsResponse2002) {
    return JSON.stringify(CancelDeploymentValueDeploymentsResponse2002$outboundSchema.parse(cancelDeploymentValueDeploymentsResponse2002));
}
export function cancelDeploymentValueDeploymentsResponse2002FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueDeploymentsResponse2002$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueDeploymentsResponse2002' from JSON`);
}
/** @internal */
export const CancelDeploymentMissingDeploymentsValue$inboundSchema = z.union([
    z.string(),
    z.lazy(() => CancelDeploymentValueDeploymentsResponse2002$inboundSchema),
]);
/** @internal */
export const CancelDeploymentMissingDeploymentsValue$outboundSchema = z.union([
    z.string(),
    z.lazy(() => CancelDeploymentValueDeploymentsResponse2002$outboundSchema),
]);
export function cancelDeploymentMissingDeploymentsValueToJSON(cancelDeploymentMissingDeploymentsValue) {
    return JSON.stringify(CancelDeploymentMissingDeploymentsValue$outboundSchema.parse(cancelDeploymentMissingDeploymentsValue));
}
export function cancelDeploymentMissingDeploymentsValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMissingDeploymentsValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMissingDeploymentsValue' from JSON`);
}
/** @internal */
export const CancelDeploymentMissing2$inboundSchema = z.object({
    type: CancelDeploymentMissingDeploymentsType$inboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValueDeploymentsResponse2002$inboundSchema),
    ]).optional(),
});
/** @internal */
export const CancelDeploymentMissing2$outboundSchema = z.object({
    type: CancelDeploymentMissingDeploymentsType$outboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValueDeploymentsResponse2002$outboundSchema),
    ]).optional(),
});
export function cancelDeploymentMissing2ToJSON(cancelDeploymentMissing2) {
    return JSON.stringify(CancelDeploymentMissing2$outboundSchema.parse(cancelDeploymentMissing2));
}
export function cancelDeploymentMissing2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMissing2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMissing2' from JSON`);
}
/** @internal */
export const CancelDeploymentMissingType$inboundSchema = z.nativeEnum(CancelDeploymentMissingType);
/** @internal */
export const CancelDeploymentMissingType$outboundSchema = CancelDeploymentMissingType$inboundSchema;
/** @internal */
export const CancelDeploymentValueDeploymentsResponseEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentValueDeploymentsResponseEq$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentValueDeploymentsResponseEqToJSON(cancelDeploymentValueDeploymentsResponseEq) {
    return JSON.stringify(CancelDeploymentValueDeploymentsResponseEq$outboundSchema.parse(cancelDeploymentValueDeploymentsResponseEq));
}
export function cancelDeploymentValueDeploymentsResponseEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueDeploymentsResponseEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueDeploymentsResponseEq' from JSON`);
}
/** @internal */
export const CancelDeploymentValueDeploymentsResponse2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CancelDeploymentValueDeploymentsResponse2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function cancelDeploymentValueDeploymentsResponse2ToJSON(cancelDeploymentValueDeploymentsResponse2) {
    return JSON.stringify(CancelDeploymentValueDeploymentsResponse2$outboundSchema.parse(cancelDeploymentValueDeploymentsResponse2));
}
export function cancelDeploymentValueDeploymentsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentValueDeploymentsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentValueDeploymentsResponse2' from JSON`);
}
/** @internal */
export const CancelDeploymentMissingValue$inboundSchema = z.union([
    z.string(),
    z.lazy(() => CancelDeploymentValueDeploymentsResponse2$inboundSchema),
]);
/** @internal */
export const CancelDeploymentMissingValue$outboundSchema = z.union([
    z.string(),
    z.lazy(() => CancelDeploymentValueDeploymentsResponse2$outboundSchema),
]);
export function cancelDeploymentMissingValueToJSON(cancelDeploymentMissingValue) {
    return JSON.stringify(CancelDeploymentMissingValue$outboundSchema.parse(cancelDeploymentMissingValue));
}
export function cancelDeploymentMissingValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMissingValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMissingValue' from JSON`);
}
/** @internal */
export const CancelDeploymentMissing1$inboundSchema = z.object({
    type: CancelDeploymentMissingType$inboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValueDeploymentsResponse2$inboundSchema),
    ]),
});
/** @internal */
export const CancelDeploymentMissing1$outboundSchema = z.object({
    type: CancelDeploymentMissingType$outboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CancelDeploymentValueDeploymentsResponse2$outboundSchema),
    ]),
});
export function cancelDeploymentMissing1ToJSON(cancelDeploymentMissing1) {
    return JSON.stringify(CancelDeploymentMissing1$outboundSchema.parse(cancelDeploymentMissing1));
}
export function cancelDeploymentMissing1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMissing1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMissing1' from JSON`);
}
/** @internal */
export const RoutesMissing$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentMissing1$inboundSchema),
    z.lazy(() => CancelDeploymentMissing2$inboundSchema),
]);
/** @internal */
export const RoutesMissing$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentMissing1$outboundSchema),
    z.lazy(() => CancelDeploymentMissing2$outboundSchema),
]);
export function routesMissingToJSON(routesMissing) {
    return JSON.stringify(RoutesMissing$outboundSchema.parse(routesMissing));
}
export function routesMissingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesMissing$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesMissing' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutesAction$inboundSchema = z.nativeEnum(CancelDeploymentRoutesAction);
/** @internal */
export const CancelDeploymentRoutesAction$outboundSchema = CancelDeploymentRoutesAction$inboundSchema;
/** @internal */
export const CancelDeploymentRoutesMitigate$inboundSchema = z.object({
    action: CancelDeploymentRoutesAction$inboundSchema,
});
/** @internal */
export const CancelDeploymentRoutesMitigate$outboundSchema = z.object({
    action: CancelDeploymentRoutesAction$outboundSchema,
});
export function cancelDeploymentRoutesMitigateToJSON(cancelDeploymentRoutesMitigate) {
    return JSON.stringify(CancelDeploymentRoutesMitigate$outboundSchema.parse(cancelDeploymentRoutesMitigate));
}
export function cancelDeploymentRoutesMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutesMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutesMitigate' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutesType$inboundSchema = z.nativeEnum(CancelDeploymentRoutesType);
/** @internal */
export const CancelDeploymentRoutesType$outboundSchema = CancelDeploymentRoutesType$inboundSchema;
/** @internal */
export const RoutesOp$inboundSchema = z
    .nativeEnum(RoutesOp);
/** @internal */
export const RoutesOp$outboundSchema = RoutesOp$inboundSchema;
/** @internal */
export const CancelDeploymentKeyEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CancelDeploymentKeyEq$outboundSchema = z.union([z.string(), z.number()]);
export function cancelDeploymentKeyEqToJSON(cancelDeploymentKeyEq) {
    return JSON.stringify(CancelDeploymentKeyEq$outboundSchema.parse(cancelDeploymentKeyEq));
}
export function cancelDeploymentKeyEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentKeyEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentKeyEq' from JSON`);
}
/** @internal */
export const CancelDeploymentKey2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CancelDeploymentKey2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function cancelDeploymentKey2ToJSON(cancelDeploymentKey2) {
    return JSON.stringify(CancelDeploymentKey2$outboundSchema.parse(cancelDeploymentKey2));
}
export function cancelDeploymentKey2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentKey2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentKey2' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutesKey$inboundSchema = z.union([z.string(), z.lazy(() => CancelDeploymentKey2$inboundSchema)]);
/** @internal */
export const CancelDeploymentRoutesKey$outboundSchema = z.union([z.string(), z.lazy(() => CancelDeploymentKey2$outboundSchema)]);
export function cancelDeploymentRoutesKeyToJSON(cancelDeploymentRoutesKey) {
    return JSON.stringify(CancelDeploymentRoutesKey$outboundSchema.parse(cancelDeploymentRoutesKey));
}
export function cancelDeploymentRoutesKeyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutesKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutesKey' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutesTarget$inboundSchema = z.object({
    key: z.union([z.string(), z.lazy(() => CancelDeploymentKey2$inboundSchema)]),
});
/** @internal */
export const CancelDeploymentRoutesTarget$outboundSchema = z.object({
    key: z.union([z.string(), z.lazy(() => CancelDeploymentKey2$outboundSchema)]),
});
export function cancelDeploymentRoutesTargetToJSON(cancelDeploymentRoutesTarget) {
    return JSON.stringify(CancelDeploymentRoutesTarget$outboundSchema.parse(cancelDeploymentRoutesTarget));
}
export function cancelDeploymentRoutesTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutesTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutesTarget' from JSON`);
}
/** @internal */
export const RoutesArgs$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const RoutesArgs$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function routesArgsToJSON(routesArgs) {
    return JSON.stringify(RoutesArgs$outboundSchema.parse(routesArgs));
}
export function routesArgsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesArgs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesArgs' from JSON`);
}
/** @internal */
export const RoutesTransforms$inboundSchema = z.object({
    type: CancelDeploymentRoutesType$inboundSchema,
    op: RoutesOp$inboundSchema,
    target: z.lazy(() => CancelDeploymentRoutesTarget$inboundSchema),
    args: z.union([z.string(), z.array(z.string())]).optional(),
});
/** @internal */
export const RoutesTransforms$outboundSchema = z.object({
    type: CancelDeploymentRoutesType$outboundSchema,
    op: RoutesOp$outboundSchema,
    target: z.lazy(() => CancelDeploymentRoutesTarget$outboundSchema),
    args: z.union([z.string(), z.array(z.string())]).optional(),
});
export function routesTransformsToJSON(routesTransforms) {
    return JSON.stringify(RoutesTransforms$outboundSchema.parse(routesTransforms));
}
export function routesTransformsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesTransforms$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesTransforms' from JSON`);
}
/** @internal */
export const RoutesLocale$inboundSchema = z.object({
    redirect: z.record(z.string()).optional(),
    cookie: z.string().optional(),
});
/** @internal */
export const RoutesLocale$outboundSchema = z.object({
    redirect: z.record(z.string()).optional(),
    cookie: z.string().optional(),
});
export function routesLocaleToJSON(routesLocale) {
    return JSON.stringify(RoutesLocale$outboundSchema.parse(routesLocale));
}
export function routesLocaleFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesLocale$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesLocale' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutes1$inboundSchema = z.object({
    src: z.string(),
    dest: z.string().optional(),
    headers: z.record(z.string()).optional(),
    methods: z.array(z.string()).optional(),
    continue: z.boolean().optional(),
    override: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    check: z.boolean().optional(),
    important: z.boolean().optional(),
    status: z.number().optional(),
    has: z.array(z.union([
        z.lazy(() => CancelDeploymentHas1$inboundSchema),
        z.lazy(() => CancelDeploymentHas2$inboundSchema),
    ])).optional(),
    missing: z.array(z.union([
        z.lazy(() => CancelDeploymentMissing1$inboundSchema),
        z.lazy(() => CancelDeploymentMissing2$inboundSchema),
    ])).optional(),
    mitigate: z.lazy(() => CancelDeploymentRoutesMitigate$inboundSchema)
        .optional(),
    transforms: z.array(z.lazy(() => RoutesTransforms$inboundSchema)).optional(),
    locale: z.lazy(() => RoutesLocale$inboundSchema).optional(),
    middlewarePath: z.string().optional(),
    middlewareRawSrc: z.array(z.string()).optional(),
    middleware: z.number().optional(),
});
/** @internal */
export const CancelDeploymentRoutes1$outboundSchema = z.object({
    src: z.string(),
    dest: z.string().optional(),
    headers: z.record(z.string()).optional(),
    methods: z.array(z.string()).optional(),
    continue: z.boolean().optional(),
    override: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    check: z.boolean().optional(),
    important: z.boolean().optional(),
    status: z.number().optional(),
    has: z.array(z.union([
        z.lazy(() => CancelDeploymentHas1$outboundSchema),
        z.lazy(() => CancelDeploymentHas2$outboundSchema),
    ])).optional(),
    missing: z.array(z.union([
        z.lazy(() => CancelDeploymentMissing1$outboundSchema),
        z.lazy(() => CancelDeploymentMissing2$outboundSchema),
    ])).optional(),
    mitigate: z.lazy(() => CancelDeploymentRoutesMitigate$outboundSchema)
        .optional(),
    transforms: z.array(z.lazy(() => RoutesTransforms$outboundSchema)).optional(),
    locale: z.lazy(() => RoutesLocale$outboundSchema).optional(),
    middlewarePath: z.string().optional(),
    middlewareRawSrc: z.array(z.string()).optional(),
    middleware: z.number().optional(),
});
export function cancelDeploymentRoutes1ToJSON(cancelDeploymentRoutes1) {
    return JSON.stringify(CancelDeploymentRoutes1$outboundSchema.parse(cancelDeploymentRoutes1));
}
export function cancelDeploymentRoutes1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutes1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutes1' from JSON`);
}
/** @internal */
export const CancelDeploymentRoutes$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentRoutes3$inboundSchema),
    z.lazy(() => CancelDeploymentRoutes1$inboundSchema),
    z.lazy(() => CancelDeploymentRoutes2$inboundSchema),
]);
/** @internal */
export const CancelDeploymentRoutes$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentRoutes3$outboundSchema),
    z.lazy(() => CancelDeploymentRoutes1$outboundSchema),
    z.lazy(() => CancelDeploymentRoutes2$outboundSchema),
]);
export function cancelDeploymentRoutesToJSON(cancelDeploymentRoutes) {
    return JSON.stringify(CancelDeploymentRoutes$outboundSchema.parse(cancelDeploymentRoutes));
}
export function cancelDeploymentRoutesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentRoutes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentRoutes' from JSON`);
}
/** @internal */
export const CancelDeploymentGitRepoDeploymentsResponseType$inboundSchema = z
    .nativeEnum(CancelDeploymentGitRepoDeploymentsResponseType);
/** @internal */
export const CancelDeploymentGitRepoDeploymentsResponseType$outboundSchema = CancelDeploymentGitRepoDeploymentsResponseType$inboundSchema;
/** @internal */
export const CancelDeploymentGitRepoDeploymentsResponseOwnerType$inboundSchema = z.nativeEnum(CancelDeploymentGitRepoDeploymentsResponseOwnerType);
/** @internal */
export const CancelDeploymentGitRepoDeploymentsResponseOwnerType$outboundSchema = CancelDeploymentGitRepoDeploymentsResponseOwnerType$inboundSchema;
/** @internal */
export const CancelDeploymentGitRepo3$inboundSchema = z.object({
    owner: z.string(),
    repoUuid: z.string(),
    slug: z.string(),
    type: CancelDeploymentGitRepoDeploymentsResponseType$inboundSchema,
    workspaceUuid: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CancelDeploymentGitRepoDeploymentsResponseOwnerType$inboundSchema,
});
/** @internal */
export const CancelDeploymentGitRepo3$outboundSchema = z.object({
    owner: z.string(),
    repoUuid: z.string(),
    slug: z.string(),
    type: CancelDeploymentGitRepoDeploymentsResponseType$outboundSchema,
    workspaceUuid: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CancelDeploymentGitRepoDeploymentsResponseOwnerType$outboundSchema,
});
export function cancelDeploymentGitRepo3ToJSON(cancelDeploymentGitRepo3) {
    return JSON.stringify(CancelDeploymentGitRepo3$outboundSchema.parse(cancelDeploymentGitRepo3));
}
export function cancelDeploymentGitRepo3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitRepo3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitRepo3' from JSON`);
}
/** @internal */
export const CancelDeploymentGitRepoDeploymentsType$inboundSchema = z.nativeEnum(CancelDeploymentGitRepoDeploymentsType);
/** @internal */
export const CancelDeploymentGitRepoDeploymentsType$outboundSchema = CancelDeploymentGitRepoDeploymentsType$inboundSchema;
/** @internal */
export const CancelDeploymentGitRepoDeploymentsOwnerType$inboundSchema = z
    .nativeEnum(CancelDeploymentGitRepoDeploymentsOwnerType);
/** @internal */
export const CancelDeploymentGitRepoDeploymentsOwnerType$outboundSchema = CancelDeploymentGitRepoDeploymentsOwnerType$inboundSchema;
/** @internal */
export const CancelDeploymentGitRepo2$inboundSchema = z.object({
    org: z.string(),
    repo: z.string(),
    repoId: z.number(),
    type: CancelDeploymentGitRepoDeploymentsType$inboundSchema,
    repoOwnerId: z.number(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CancelDeploymentGitRepoDeploymentsOwnerType$inboundSchema,
});
/** @internal */
export const CancelDeploymentGitRepo2$outboundSchema = z.object({
    org: z.string(),
    repo: z.string(),
    repoId: z.number(),
    type: CancelDeploymentGitRepoDeploymentsType$outboundSchema,
    repoOwnerId: z.number(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CancelDeploymentGitRepoDeploymentsOwnerType$outboundSchema,
});
export function cancelDeploymentGitRepo2ToJSON(cancelDeploymentGitRepo2) {
    return JSON.stringify(CancelDeploymentGitRepo2$outboundSchema.parse(cancelDeploymentGitRepo2));
}
export function cancelDeploymentGitRepo2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitRepo2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitRepo2' from JSON`);
}
/** @internal */
export const CancelDeploymentGitRepoType$inboundSchema = z.nativeEnum(CancelDeploymentGitRepoType);
/** @internal */
export const CancelDeploymentGitRepoType$outboundSchema = CancelDeploymentGitRepoType$inboundSchema;
/** @internal */
export const CancelDeploymentGitRepoOwnerType$inboundSchema = z.nativeEnum(CancelDeploymentGitRepoOwnerType);
/** @internal */
export const CancelDeploymentGitRepoOwnerType$outboundSchema = CancelDeploymentGitRepoOwnerType$inboundSchema;
/** @internal */
export const CancelDeploymentGitRepo1$inboundSchema = z.object({
    namespace: z.string(),
    projectId: z.number(),
    type: CancelDeploymentGitRepoType$inboundSchema,
    url: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CancelDeploymentGitRepoOwnerType$inboundSchema,
});
/** @internal */
export const CancelDeploymentGitRepo1$outboundSchema = z.object({
    namespace: z.string(),
    projectId: z.number(),
    type: CancelDeploymentGitRepoType$outboundSchema,
    url: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CancelDeploymentGitRepoOwnerType$outboundSchema,
});
export function cancelDeploymentGitRepo1ToJSON(cancelDeploymentGitRepo1) {
    return JSON.stringify(CancelDeploymentGitRepo1$outboundSchema.parse(cancelDeploymentGitRepo1));
}
export function cancelDeploymentGitRepo1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitRepo1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitRepo1' from JSON`);
}
/** @internal */
export const CancelDeploymentGitRepo$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentGitRepo2$inboundSchema),
    z.lazy(() => CancelDeploymentGitRepo3$inboundSchema),
    z.lazy(() => CancelDeploymentGitRepo1$inboundSchema),
]);
/** @internal */
export const CancelDeploymentGitRepo$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentGitRepo2$outboundSchema),
    z.lazy(() => CancelDeploymentGitRepo3$outboundSchema),
    z.lazy(() => CancelDeploymentGitRepo1$outboundSchema),
]);
export function cancelDeploymentGitRepoToJSON(cancelDeploymentGitRepo) {
    return JSON.stringify(CancelDeploymentGitRepo$outboundSchema.parse(cancelDeploymentGitRepo));
}
export function cancelDeploymentGitRepoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentGitRepo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentGitRepo' from JSON`);
}
/** @internal */
export const CancelDeploymentFlags2$inboundSchema = z.object({});
/** @internal */
export const CancelDeploymentFlags2$outboundSchema = z.object({});
export function cancelDeploymentFlags2ToJSON(cancelDeploymentFlags2) {
    return JSON.stringify(CancelDeploymentFlags2$outboundSchema.parse(cancelDeploymentFlags2));
}
export function cancelDeploymentFlags2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentFlags2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentFlags2' from JSON`);
}
/** @internal */
export const FlagsOptions$inboundSchema = z.object({
    value: z.nullable(FlagJSONValue$inboundSchema),
    label: z.string().optional(),
});
/** @internal */
export const FlagsOptions$outboundSchema = z.object({
    value: z.nullable(FlagJSONValue$outboundSchema),
    label: z.string().optional(),
});
export function flagsOptionsToJSON(flagsOptions) {
    return JSON.stringify(FlagsOptions$outboundSchema.parse(flagsOptions));
}
export function flagsOptionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FlagsOptions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FlagsOptions' from JSON`);
}
/** @internal */
export const CancelDeploymentFlagsDefinitions$inboundSchema = z.object({
    options: z.array(z.lazy(() => FlagsOptions$inboundSchema)).optional(),
    url: z.string().optional(),
    description: z.string().optional(),
});
/** @internal */
export const CancelDeploymentFlagsDefinitions$outboundSchema = z.object({
    options: z.array(z.lazy(() => FlagsOptions$outboundSchema)).optional(),
    url: z.string().optional(),
    description: z.string().optional(),
});
export function cancelDeploymentFlagsDefinitionsToJSON(cancelDeploymentFlagsDefinitions) {
    return JSON.stringify(CancelDeploymentFlagsDefinitions$outboundSchema.parse(cancelDeploymentFlagsDefinitions));
}
export function cancelDeploymentFlagsDefinitionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentFlagsDefinitions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentFlagsDefinitions' from JSON`);
}
/** @internal */
export const CancelDeploymentFlags1$inboundSchema = z.object({
    definitions: z.record(z.lazy(() => CancelDeploymentFlagsDefinitions$inboundSchema)),
});
/** @internal */
export const CancelDeploymentFlags1$outboundSchema = z.object({
    definitions: z.record(z.lazy(() => CancelDeploymentFlagsDefinitions$outboundSchema)),
});
export function cancelDeploymentFlags1ToJSON(cancelDeploymentFlags1) {
    return JSON.stringify(CancelDeploymentFlags1$outboundSchema.parse(cancelDeploymentFlags1));
}
export function cancelDeploymentFlags1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentFlags1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentFlags1' from JSON`);
}
/** @internal */
export const CancelDeploymentFlags$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentFlags1$inboundSchema),
    z.array(z.lazy(() => CancelDeploymentFlags2$inboundSchema)),
]);
/** @internal */
export const CancelDeploymentFlags$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentFlags1$outboundSchema),
    z.array(z.lazy(() => CancelDeploymentFlags2$outboundSchema)),
]);
export function cancelDeploymentFlagsToJSON(cancelDeploymentFlags) {
    return JSON.stringify(CancelDeploymentFlags$outboundSchema.parse(cancelDeploymentFlags));
}
export function cancelDeploymentFlagsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentFlags$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentFlags' from JSON`);
}
/** @internal */
export const MicrofrontendsApplications$inboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    productionHost: z.string(),
    deploymentAlias: z.string().optional(),
    deploymentHost: z.string().optional(),
});
/** @internal */
export const MicrofrontendsApplications$outboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    productionHost: z.string(),
    deploymentAlias: z.string().optional(),
    deploymentHost: z.string().optional(),
});
export function microfrontendsApplicationsToJSON(microfrontendsApplications) {
    return JSON.stringify(MicrofrontendsApplications$outboundSchema.parse(microfrontendsApplications));
}
export function microfrontendsApplicationsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MicrofrontendsApplications$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MicrofrontendsApplications' from JSON`);
}
/** @internal */
export const MicrofrontendsMfeConfigUploadState$inboundSchema = z.nativeEnum(MicrofrontendsMfeConfigUploadState);
/** @internal */
export const MicrofrontendsMfeConfigUploadState$outboundSchema = MicrofrontendsMfeConfigUploadState$inboundSchema;
/** @internal */
export const CancelDeploymentMicrofrontends2$inboundSchema = z.object({
    isDefaultApp: z.boolean(),
    applications: z.record(z.lazy(() => MicrofrontendsApplications$inboundSchema))
        .optional(),
    mfeConfigUploadState: MicrofrontendsMfeConfigUploadState$inboundSchema
        .optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
/** @internal */
export const CancelDeploymentMicrofrontends2$outboundSchema = z.object({
    isDefaultApp: z.boolean(),
    applications: z.record(z.lazy(() => MicrofrontendsApplications$outboundSchema)).optional(),
    mfeConfigUploadState: MicrofrontendsMfeConfigUploadState$outboundSchema
        .optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
export function cancelDeploymentMicrofrontends2ToJSON(cancelDeploymentMicrofrontends2) {
    return JSON.stringify(CancelDeploymentMicrofrontends2$outboundSchema.parse(cancelDeploymentMicrofrontends2));
}
export function cancelDeploymentMicrofrontends2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMicrofrontends2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMicrofrontends2' from JSON`);
}
/** @internal */
export const CancelDeploymentMicrofrontends1$inboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
/** @internal */
export const CancelDeploymentMicrofrontends1$outboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
export function cancelDeploymentMicrofrontends1ToJSON(cancelDeploymentMicrofrontends1) {
    return JSON.stringify(CancelDeploymentMicrofrontends1$outboundSchema.parse(cancelDeploymentMicrofrontends1));
}
export function cancelDeploymentMicrofrontends1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMicrofrontends1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMicrofrontends1' from JSON`);
}
/** @internal */
export const CancelDeploymentMicrofrontends$inboundSchema = z.union([
    z.lazy(() => CancelDeploymentMicrofrontends2$inboundSchema),
    z.lazy(() => CancelDeploymentMicrofrontends1$inboundSchema),
]);
/** @internal */
export const CancelDeploymentMicrofrontends$outboundSchema = z.union([
    z.lazy(() => CancelDeploymentMicrofrontends2$outboundSchema),
    z.lazy(() => CancelDeploymentMicrofrontends1$outboundSchema),
]);
export function cancelDeploymentMicrofrontendsToJSON(cancelDeploymentMicrofrontends) {
    return JSON.stringify(CancelDeploymentMicrofrontends$outboundSchema.parse(cancelDeploymentMicrofrontends));
}
export function cancelDeploymentMicrofrontendsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentMicrofrontends$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentMicrofrontends' from JSON`);
}
/** @internal */
export const CancelDeploymentFunctionType$inboundSchema = z.nativeEnum(CancelDeploymentFunctionType);
/** @internal */
export const CancelDeploymentFunctionType$outboundSchema = CancelDeploymentFunctionType$inboundSchema;
/** @internal */
export const CancelDeploymentFunctionMemoryType$inboundSchema = z.nativeEnum(CancelDeploymentFunctionMemoryType);
/** @internal */
export const CancelDeploymentFunctionMemoryType$outboundSchema = CancelDeploymentFunctionMemoryType$inboundSchema;
/** @internal */
export const CancelDeploymentConfig$inboundSchema = z.object({
    version: z.number().optional(),
    functionType: CancelDeploymentFunctionType$inboundSchema,
    functionMemoryType: CancelDeploymentFunctionMemoryType$inboundSchema,
    functionTimeout: z.nullable(z.number()),
    secureComputePrimaryRegion: z.nullable(z.string()),
    secureComputeFallbackRegion: z.nullable(z.string()),
    isUsingActiveCPU: z.boolean().optional(),
});
/** @internal */
export const CancelDeploymentConfig$outboundSchema = z.object({
    version: z.number().optional(),
    functionType: CancelDeploymentFunctionType$outboundSchema,
    functionMemoryType: CancelDeploymentFunctionMemoryType$outboundSchema,
    functionTimeout: z.nullable(z.number()),
    secureComputePrimaryRegion: z.nullable(z.string()),
    secureComputeFallbackRegion: z.nullable(z.string()),
    isUsingActiveCPU: z.boolean().optional(),
});
export function cancelDeploymentConfigToJSON(cancelDeploymentConfig) {
    return JSON.stringify(CancelDeploymentConfig$outboundSchema.parse(cancelDeploymentConfig));
}
export function cancelDeploymentConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentConfig' from JSON`);
}
/** @internal */
export const CancelDeploymentState$inboundSchema = z.nativeEnum(CancelDeploymentState);
/** @internal */
export const CancelDeploymentState$outboundSchema = CancelDeploymentState$inboundSchema;
/** @internal */
export const CancelDeploymentDeploymentAlias$inboundSchema = z.object({
    state: CancelDeploymentState$inboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
});
/** @internal */
export const CancelDeploymentDeploymentAlias$outboundSchema = z.object({
    state: CancelDeploymentState$outboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
});
export function cancelDeploymentDeploymentAliasToJSON(cancelDeploymentDeploymentAlias) {
    return JSON.stringify(CancelDeploymentDeploymentAlias$outboundSchema.parse(cancelDeploymentDeploymentAlias));
}
export function cancelDeploymentDeploymentAliasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentDeploymentAlias$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentDeploymentAlias' from JSON`);
}
/** @internal */
export const CancelDeploymentChecks$inboundSchema = z.object({
    "deployment-alias": z.lazy(() => CancelDeploymentDeploymentAlias$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "deployment-alias": "deploymentAlias",
    });
});
/** @internal */
export const CancelDeploymentChecks$outboundSchema = z.object({
    deploymentAlias: z.lazy(() => CancelDeploymentDeploymentAlias$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        deploymentAlias: "deployment-alias",
    });
});
export function cancelDeploymentChecksToJSON(cancelDeploymentChecks) {
    return JSON.stringify(CancelDeploymentChecks$outboundSchema.parse(cancelDeploymentChecks));
}
export function cancelDeploymentChecksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentChecks$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentChecks' from JSON`);
}
/** @internal */
export const CancelDeploymentResponseBody$inboundSchema = z.object({
    aliasAssignedAt: z.nullable(z.union([z.number(), z.boolean()])).optional(),
    alwaysRefuseToBuild: z.boolean().optional(),
    build: z.lazy(() => CancelDeploymentBuild$inboundSchema),
    buildArtifactUrls: z.array(z.string()).optional(),
    builds: z.array(z.lazy(() => CancelDeploymentBuilds$inboundSchema))
        .optional(),
    env: z.array(z.string()),
    inspectorUrl: z.nullable(z.string()),
    isInConcurrentBuildsQueue: z.boolean(),
    isInSystemBuildsQueue: z.boolean(),
    projectSettings: z.lazy(() => CancelDeploymentProjectSettings$inboundSchema),
    readyStateReason: z.string().optional(),
    integrations: z.lazy(() => CancelDeploymentIntegrations$inboundSchema)
        .optional(),
    images: z.lazy(() => CancelDeploymentImages$inboundSchema).optional(),
    alias: z.array(z.string()).optional(),
    aliasAssigned: z.boolean(),
    bootedAt: z.number(),
    buildingAt: z.number(),
    buildContainerFinishedAt: z.number().optional(),
    buildSkipped: z.boolean(),
    creator: z.lazy(() => CancelDeploymentCreator$inboundSchema),
    initReadyAt: z.number().optional(),
    isFirstBranchDeployment: z.boolean().optional(),
    lambdas: z.array(z.lazy(() => CancelDeploymentLambdas$inboundSchema))
        .optional(),
    public: z.boolean(),
    ready: z.number().optional(),
    status: CancelDeploymentStatus$inboundSchema,
    team: z.lazy(() => CancelDeploymentTeam$inboundSchema).optional(),
    userAliases: z.array(z.string()).optional(),
    previewCommentsEnabled: z.boolean().optional(),
    ttyBuildLogs: z.boolean().optional(),
    customEnvironment: z.union([
        z.lazy(() => CancelDeploymentCustomEnvironment1$inboundSchema),
        z.lazy(() => CancelDeploymentCustomEnvironment2$inboundSchema),
    ]).optional(),
    oomReport: CancelDeploymentOomReport$inboundSchema.optional(),
    id: z.string(),
    aliasError: z.nullable(z.lazy(() => CancelDeploymentAliasError$inboundSchema))
        .optional(),
    aliasFinal: z.nullable(z.string()).optional(),
    aliasWarning: z.nullable(z.lazy(() => CancelDeploymentAliasWarning$inboundSchema)).optional(),
    autoAssignCustomDomains: z.boolean().optional(),
    automaticAliases: z.array(z.string()).optional(),
    buildErrorAt: z.number().optional(),
    checksState: CancelDeploymentChecksState$inboundSchema.optional(),
    checksConclusion: CancelDeploymentChecksConclusion$inboundSchema.optional(),
    createdAt: z.number(),
    deletedAt: z.nullable(z.number()).optional(),
    defaultRoute: z.string().optional(),
    canceledAt: z.number().optional(),
    errorCode: z.string().optional(),
    errorLink: z.string().optional(),
    errorMessage: z.nullable(z.string()).optional(),
    errorStep: z.string().optional(),
    passiveRegions: z.array(z.string()).optional(),
    gitSource: z.union([
        z.lazy(() => GitSource12$inboundSchema),
        z.lazy(() => GitSource15$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource4$inboundSchema),
        z.lazy(() => GitSource10$inboundSchema),
        z.lazy(() => GitSource11$inboundSchema),
        z.lazy(() => GitSource13$inboundSchema),
        z.lazy(() => GitSource14$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource2$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource3$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource6$inboundSchema),
        z.lazy(() => GitSource9$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource1$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource5$inboundSchema),
        z.lazy(() => CancelDeploymentGitSource7$inboundSchema),
        z.lazy(() => GitSource8$inboundSchema),
    ]).optional(),
    name: z.string(),
    meta: z.record(z.string()),
    originCacheRegion: z.string().optional(),
    nodeVersion: CancelDeploymentNodeVersion$inboundSchema.optional(),
    project: z.lazy(() => CancelDeploymentProject$inboundSchema).optional(),
    readyState: CancelDeploymentReadyState$inboundSchema,
    readySubstate: CancelDeploymentReadySubstate$inboundSchema.optional(),
    regions: z.array(z.string()),
    softDeletedByRetention: z.boolean().optional(),
    source: CancelDeploymentSource$inboundSchema.optional(),
    target: z.nullable(CancelDeploymentTarget$inboundSchema).optional(),
    type: CancelDeploymentType$inboundSchema,
    undeletedAt: z.number().optional(),
    url: z.string(),
    version: z.number(),
    oidcTokenClaims: z.lazy(() => CancelDeploymentOidcTokenClaims$inboundSchema)
        .optional(),
    connectBuildsEnabled: z.boolean().optional(),
    connectConfigurationId: z.string().optional(),
    createdIn: z.string(),
    crons: z.array(z.lazy(() => CancelDeploymentCrons$inboundSchema)).optional(),
    functions: z.nullable(z.record(z.lazy(() => CancelDeploymentFunctions$inboundSchema))).optional(),
    monorepoManager: z.nullable(z.string()).optional(),
    ownerId: z.string(),
    passiveConnectConfigurationId: z.string().optional(),
    plan: CancelDeploymentPlan$inboundSchema,
    projectId: z.string(),
    routes: z.nullable(z.array(z.union([
        z.lazy(() => CancelDeploymentRoutes3$inboundSchema),
        z.lazy(() => CancelDeploymentRoutes1$inboundSchema),
        z.lazy(() => CancelDeploymentRoutes2$inboundSchema),
    ]))),
    gitRepo: z.nullable(z.union([
        z.lazy(() => CancelDeploymentGitRepo2$inboundSchema),
        z.lazy(() => CancelDeploymentGitRepo3$inboundSchema),
        z.lazy(() => CancelDeploymentGitRepo1$inboundSchema),
    ])).optional(),
    flags: z.union([
        z.lazy(() => CancelDeploymentFlags1$inboundSchema),
        z.array(z.lazy(() => CancelDeploymentFlags2$inboundSchema)),
    ]).optional(),
    microfrontends: z.union([
        z.lazy(() => CancelDeploymentMicrofrontends2$inboundSchema),
        z.lazy(() => CancelDeploymentMicrofrontends1$inboundSchema),
    ]).optional(),
    config: z.lazy(() => CancelDeploymentConfig$inboundSchema).optional(),
    checks: z.lazy(() => CancelDeploymentChecks$inboundSchema).optional(),
});
/** @internal */
export const CancelDeploymentResponseBody$outboundSchema = z.object({
    aliasAssignedAt: z.nullable(z.union([z.number(), z.boolean()])).optional(),
    alwaysRefuseToBuild: z.boolean().optional(),
    build: z.lazy(() => CancelDeploymentBuild$outboundSchema),
    buildArtifactUrls: z.array(z.string()).optional(),
    builds: z.array(z.lazy(() => CancelDeploymentBuilds$outboundSchema))
        .optional(),
    env: z.array(z.string()),
    inspectorUrl: z.nullable(z.string()),
    isInConcurrentBuildsQueue: z.boolean(),
    isInSystemBuildsQueue: z.boolean(),
    projectSettings: z.lazy(() => CancelDeploymentProjectSettings$outboundSchema),
    readyStateReason: z.string().optional(),
    integrations: z.lazy(() => CancelDeploymentIntegrations$outboundSchema)
        .optional(),
    images: z.lazy(() => CancelDeploymentImages$outboundSchema).optional(),
    alias: z.array(z.string()).optional(),
    aliasAssigned: z.boolean(),
    bootedAt: z.number(),
    buildingAt: z.number(),
    buildContainerFinishedAt: z.number().optional(),
    buildSkipped: z.boolean(),
    creator: z.lazy(() => CancelDeploymentCreator$outboundSchema),
    initReadyAt: z.number().optional(),
    isFirstBranchDeployment: z.boolean().optional(),
    lambdas: z.array(z.lazy(() => CancelDeploymentLambdas$outboundSchema))
        .optional(),
    public: z.boolean(),
    ready: z.number().optional(),
    status: CancelDeploymentStatus$outboundSchema,
    team: z.lazy(() => CancelDeploymentTeam$outboundSchema).optional(),
    userAliases: z.array(z.string()).optional(),
    previewCommentsEnabled: z.boolean().optional(),
    ttyBuildLogs: z.boolean().optional(),
    customEnvironment: z.union([
        z.lazy(() => CancelDeploymentCustomEnvironment1$outboundSchema),
        z.lazy(() => CancelDeploymentCustomEnvironment2$outboundSchema),
    ]).optional(),
    oomReport: CancelDeploymentOomReport$outboundSchema.optional(),
    id: z.string(),
    aliasError: z.nullable(z.lazy(() => CancelDeploymentAliasError$outboundSchema)).optional(),
    aliasFinal: z.nullable(z.string()).optional(),
    aliasWarning: z.nullable(z.lazy(() => CancelDeploymentAliasWarning$outboundSchema)).optional(),
    autoAssignCustomDomains: z.boolean().optional(),
    automaticAliases: z.array(z.string()).optional(),
    buildErrorAt: z.number().optional(),
    checksState: CancelDeploymentChecksState$outboundSchema.optional(),
    checksConclusion: CancelDeploymentChecksConclusion$outboundSchema.optional(),
    createdAt: z.number(),
    deletedAt: z.nullable(z.number()).optional(),
    defaultRoute: z.string().optional(),
    canceledAt: z.number().optional(),
    errorCode: z.string().optional(),
    errorLink: z.string().optional(),
    errorMessage: z.nullable(z.string()).optional(),
    errorStep: z.string().optional(),
    passiveRegions: z.array(z.string()).optional(),
    gitSource: z.union([
        z.lazy(() => GitSource12$outboundSchema),
        z.lazy(() => GitSource15$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource4$outboundSchema),
        z.lazy(() => GitSource10$outboundSchema),
        z.lazy(() => GitSource11$outboundSchema),
        z.lazy(() => GitSource13$outboundSchema),
        z.lazy(() => GitSource14$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource2$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource3$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource6$outboundSchema),
        z.lazy(() => GitSource9$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource1$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource5$outboundSchema),
        z.lazy(() => CancelDeploymentGitSource7$outboundSchema),
        z.lazy(() => GitSource8$outboundSchema),
    ]).optional(),
    name: z.string(),
    meta: z.record(z.string()),
    originCacheRegion: z.string().optional(),
    nodeVersion: CancelDeploymentNodeVersion$outboundSchema.optional(),
    project: z.lazy(() => CancelDeploymentProject$outboundSchema).optional(),
    readyState: CancelDeploymentReadyState$outboundSchema,
    readySubstate: CancelDeploymentReadySubstate$outboundSchema.optional(),
    regions: z.array(z.string()),
    softDeletedByRetention: z.boolean().optional(),
    source: CancelDeploymentSource$outboundSchema.optional(),
    target: z.nullable(CancelDeploymentTarget$outboundSchema).optional(),
    type: CancelDeploymentType$outboundSchema,
    undeletedAt: z.number().optional(),
    url: z.string(),
    version: z.number(),
    oidcTokenClaims: z.lazy(() => CancelDeploymentOidcTokenClaims$outboundSchema)
        .optional(),
    connectBuildsEnabled: z.boolean().optional(),
    connectConfigurationId: z.string().optional(),
    createdIn: z.string(),
    crons: z.array(z.lazy(() => CancelDeploymentCrons$outboundSchema)).optional(),
    functions: z.nullable(z.record(z.lazy(() => CancelDeploymentFunctions$outboundSchema))).optional(),
    monorepoManager: z.nullable(z.string()).optional(),
    ownerId: z.string(),
    passiveConnectConfigurationId: z.string().optional(),
    plan: CancelDeploymentPlan$outboundSchema,
    projectId: z.string(),
    routes: z.nullable(z.array(z.union([
        z.lazy(() => CancelDeploymentRoutes3$outboundSchema),
        z.lazy(() => CancelDeploymentRoutes1$outboundSchema),
        z.lazy(() => CancelDeploymentRoutes2$outboundSchema),
    ]))),
    gitRepo: z.nullable(z.union([
        z.lazy(() => CancelDeploymentGitRepo2$outboundSchema),
        z.lazy(() => CancelDeploymentGitRepo3$outboundSchema),
        z.lazy(() => CancelDeploymentGitRepo1$outboundSchema),
    ])).optional(),
    flags: z.union([
        z.lazy(() => CancelDeploymentFlags1$outboundSchema),
        z.array(z.lazy(() => CancelDeploymentFlags2$outboundSchema)),
    ]).optional(),
    microfrontends: z.union([
        z.lazy(() => CancelDeploymentMicrofrontends2$outboundSchema),
        z.lazy(() => CancelDeploymentMicrofrontends1$outboundSchema),
    ]).optional(),
    config: z.lazy(() => CancelDeploymentConfig$outboundSchema).optional(),
    checks: z.lazy(() => CancelDeploymentChecks$outboundSchema).optional(),
});
export function cancelDeploymentResponseBodyToJSON(cancelDeploymentResponseBody) {
    return JSON.stringify(CancelDeploymentResponseBody$outboundSchema.parse(cancelDeploymentResponseBody));
}
export function cancelDeploymentResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CancelDeploymentResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CancelDeploymentResponseBody' from JSON`);
}
