// @vercel/sdk/esm/models/completerollingreleaseop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The current state of the rolling release
 */
export const CompleteRollingReleaseState = {
    Active: "ACTIVE",
    Complete: "COMPLETE",
    Aborted: "ABORTED",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const CompleteRollingReleaseTarget = {
    Staging: "staging",
    Production: "production",
};
/**
 * Where was the deployment created from
 */
export const CompleteRollingReleaseSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const CompleteRollingReleaseReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const CompleteRollingReleaseRollingReleaseTarget = {
    Staging: "staging",
    Production: "production",
};
/**
 * Where was the deployment created from
 */
export const CompleteRollingReleaseRollingReleaseSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const CompleteRollingReleaseRollingReleaseReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * The advancement type of the rolling release
 */
export const CompleteRollingReleaseAdvancementType = {
    Automatic: "automatic",
    ManualApproval: "manual-approval",
};
/** @internal */
export const CompleteRollingReleaseRequestBody$inboundSchema = z.object({
    canaryDeploymentId: z.string(),
});
/** @internal */
export const CompleteRollingReleaseRequestBody$outboundSchema = z.object({
    canaryDeploymentId: z.string(),
});
export function completeRollingReleaseRequestBodyToJSON(completeRollingReleaseRequestBody) {
    return JSON.stringify(CompleteRollingReleaseRequestBody$outboundSchema.parse(completeRollingReleaseRequestBody));
}
export function completeRollingReleaseRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseRequestBody' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => CompleteRollingReleaseRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const CompleteRollingReleaseRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => CompleteRollingReleaseRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function completeRollingReleaseRequestToJSON(completeRollingReleaseRequest) {
    return JSON.stringify(CompleteRollingReleaseRequest$outboundSchema.parse(completeRollingReleaseRequest));
}
export function completeRollingReleaseRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseRequest' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseState$inboundSchema = z.nativeEnum(CompleteRollingReleaseState);
/** @internal */
export const CompleteRollingReleaseState$outboundSchema = CompleteRollingReleaseState$inboundSchema;
/** @internal */
export const CompleteRollingReleaseTarget$inboundSchema = z.nativeEnum(CompleteRollingReleaseTarget);
/** @internal */
export const CompleteRollingReleaseTarget$outboundSchema = CompleteRollingReleaseTarget$inboundSchema;
/** @internal */
export const CompleteRollingReleaseSource$inboundSchema = z.nativeEnum(CompleteRollingReleaseSource);
/** @internal */
export const CompleteRollingReleaseSource$outboundSchema = CompleteRollingReleaseSource$inboundSchema;
/** @internal */
export const CompleteRollingReleaseReadyState$inboundSchema = z.nativeEnum(CompleteRollingReleaseReadyState);
/** @internal */
export const CompleteRollingReleaseReadyState$outboundSchema = CompleteRollingReleaseReadyState$inboundSchema;
/** @internal */
export const CompleteRollingReleaseCurrentDeployment$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(CompleteRollingReleaseTarget$inboundSchema).optional(),
    source: CompleteRollingReleaseSource$inboundSchema.optional(),
    createdAt: z.number(),
    readyState: CompleteRollingReleaseReadyState$inboundSchema,
    readyStateAt: z.number().optional(),
});
/** @internal */
export const CompleteRollingReleaseCurrentDeployment$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(CompleteRollingReleaseTarget$outboundSchema).optional(),
    source: CompleteRollingReleaseSource$outboundSchema.optional(),
    createdAt: z.number(),
    readyState: CompleteRollingReleaseReadyState$outboundSchema,
    readyStateAt: z.number().optional(),
});
export function completeRollingReleaseCurrentDeploymentToJSON(completeRollingReleaseCurrentDeployment) {
    return JSON.stringify(CompleteRollingReleaseCurrentDeployment$outboundSchema.parse(completeRollingReleaseCurrentDeployment));
}
export function completeRollingReleaseCurrentDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseCurrentDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseCurrentDeployment' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseRollingReleaseTarget$inboundSchema = z
    .nativeEnum(CompleteRollingReleaseRollingReleaseTarget);
/** @internal */
export const CompleteRollingReleaseRollingReleaseTarget$outboundSchema = CompleteRollingReleaseRollingReleaseTarget$inboundSchema;
/** @internal */
export const CompleteRollingReleaseRollingReleaseSource$inboundSchema = z
    .nativeEnum(CompleteRollingReleaseRollingReleaseSource);
/** @internal */
export const CompleteRollingReleaseRollingReleaseSource$outboundSchema = CompleteRollingReleaseRollingReleaseSource$inboundSchema;
/** @internal */
export const CompleteRollingReleaseRollingReleaseReadyState$inboundSchema = z
    .nativeEnum(CompleteRollingReleaseRollingReleaseReadyState);
/** @internal */
export const CompleteRollingReleaseRollingReleaseReadyState$outboundSchema = CompleteRollingReleaseRollingReleaseReadyState$inboundSchema;
/** @internal */
export const CompleteRollingReleaseCanaryDeployment$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(CompleteRollingReleaseRollingReleaseTarget$inboundSchema)
        .optional(),
    source: CompleteRollingReleaseRollingReleaseSource$inboundSchema.optional(),
    createdAt: z.number(),
    readyState: CompleteRollingReleaseRollingReleaseReadyState$inboundSchema,
    readyStateAt: z.number().optional(),
});
/** @internal */
export const CompleteRollingReleaseCanaryDeployment$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(CompleteRollingReleaseRollingReleaseTarget$outboundSchema)
        .optional(),
    source: CompleteRollingReleaseRollingReleaseSource$outboundSchema.optional(),
    createdAt: z.number(),
    readyState: CompleteRollingReleaseRollingReleaseReadyState$outboundSchema,
    readyStateAt: z.number().optional(),
});
export function completeRollingReleaseCanaryDeploymentToJSON(completeRollingReleaseCanaryDeployment) {
    return JSON.stringify(CompleteRollingReleaseCanaryDeployment$outboundSchema.parse(completeRollingReleaseCanaryDeployment));
}
export function completeRollingReleaseCanaryDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseCanaryDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseCanaryDeployment' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseAdvancementType$inboundSchema = z.nativeEnum(CompleteRollingReleaseAdvancementType);
/** @internal */
export const CompleteRollingReleaseAdvancementType$outboundSchema = CompleteRollingReleaseAdvancementType$inboundSchema;
/** @internal */
export const CompleteRollingReleaseStages$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const CompleteRollingReleaseStages$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function completeRollingReleaseStagesToJSON(completeRollingReleaseStages) {
    return JSON.stringify(CompleteRollingReleaseStages$outboundSchema.parse(completeRollingReleaseStages));
}
export function completeRollingReleaseStagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseStages$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseStages' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseActiveStage$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const CompleteRollingReleaseActiveStage$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function completeRollingReleaseActiveStageToJSON(completeRollingReleaseActiveStage) {
    return JSON.stringify(CompleteRollingReleaseActiveStage$outboundSchema.parse(completeRollingReleaseActiveStage));
}
export function completeRollingReleaseActiveStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseActiveStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseActiveStage' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseNextStage$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const CompleteRollingReleaseNextStage$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function completeRollingReleaseNextStageToJSON(completeRollingReleaseNextStage) {
    return JSON.stringify(CompleteRollingReleaseNextStage$outboundSchema.parse(completeRollingReleaseNextStage));
}
export function completeRollingReleaseNextStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseNextStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseNextStage' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseRollingRelease$inboundSchema = z.object({
    state: CompleteRollingReleaseState$inboundSchema,
    currentDeployment: z.nullable(z.lazy(() => CompleteRollingReleaseCurrentDeployment$inboundSchema)),
    canaryDeployment: z.nullable(z.lazy(() => CompleteRollingReleaseCanaryDeployment$inboundSchema)),
    queuedDeploymentId: z.nullable(z.string()),
    advancementType: CompleteRollingReleaseAdvancementType$inboundSchema,
    stages: z.array(z.lazy(() => CompleteRollingReleaseStages$inboundSchema)),
    activeStage: z.nullable(z.lazy(() => CompleteRollingReleaseActiveStage$inboundSchema)),
    nextStage: z.nullable(z.lazy(() => CompleteRollingReleaseNextStage$inboundSchema)),
    startedAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const CompleteRollingReleaseRollingRelease$outboundSchema = z.object({
    state: CompleteRollingReleaseState$outboundSchema,
    currentDeployment: z.nullable(z.lazy(() => CompleteRollingReleaseCurrentDeployment$outboundSchema)),
    canaryDeployment: z.nullable(z.lazy(() => CompleteRollingReleaseCanaryDeployment$outboundSchema)),
    queuedDeploymentId: z.nullable(z.string()),
    advancementType: CompleteRollingReleaseAdvancementType$outboundSchema,
    stages: z.array(z.lazy(() => CompleteRollingReleaseStages$outboundSchema)),
    activeStage: z.nullable(z.lazy(() => CompleteRollingReleaseActiveStage$outboundSchema)),
    nextStage: z.nullable(z.lazy(() => CompleteRollingReleaseNextStage$outboundSchema)),
    startedAt: z.number(),
    updatedAt: z.number(),
});
export function completeRollingReleaseRollingReleaseToJSON(completeRollingReleaseRollingRelease) {
    return JSON.stringify(CompleteRollingReleaseRollingRelease$outboundSchema.parse(completeRollingReleaseRollingRelease));
}
export function completeRollingReleaseRollingReleaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseRollingRelease$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseRollingRelease' from JSON`);
}
/** @internal */
export const CompleteRollingReleaseResponseBody$inboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => CompleteRollingReleaseRollingRelease$inboundSchema)),
});
/** @internal */
export const CompleteRollingReleaseResponseBody$outboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => CompleteRollingReleaseRollingRelease$outboundSchema)),
});
export function completeRollingReleaseResponseBodyToJSON(completeRollingReleaseResponseBody) {
    return JSON.stringify(CompleteRollingReleaseResponseBody$outboundSchema.parse(completeRollingReleaseResponseBody));
}
export function completeRollingReleaseResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CompleteRollingReleaseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CompleteRollingReleaseResponseBody' from JSON`);
}
