// @vercel/sdk/esm/models/createprojectenvop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The type of environment variable
 */
export const CreateProjectEnv2Type = {
    System: "system",
    Secret: "secret",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
};
export const CreateProjectEnv2Target = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of environment variable
 */
export const TwoType = {
    System: "system",
    Secret: "secret",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
};
export const TwoTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of environment variable
 */
export const CreateProjectEnv1Type = {
    System: "system",
    Secret: "secret",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
};
export const CreateProjectEnv1Target = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of environment variable
 */
export const OneType = {
    System: "system",
    Secret: "secret",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
};
export const OneTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateProjectEnvTargetProjects2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateProjectEnvCreatedType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type = {
    PostgresDatabase: "postgres-database",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type = {
    PostgresPassword: "postgres-password",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type = {
    PostgresHost: "postgres-host",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type = {
    PostgresUser: "postgres-user",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type = {
    PostgresUrl: "postgres-url",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type = {
    RedisUrl: "redis-url",
};
export const CreateProjectEnvCreatedProjectsResponseType = {
    FlagsSecret: "flags-secret",
};
export const CreateProjectEnvTarget2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateProjectEnvTarget1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreatedType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type = {
    PostgresDatabase: "postgres-database",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type = {
    PostgresPassword: "postgres-password",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type = {
    PostgresHost: "postgres-host",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type = {
    PostgresUser: "postgres-user",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType = {
    PostgresUrl: "postgres-url",
};
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const CreateProjectEnvContentHintProjectsResponse201Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const CreateProjectEnvContentHintProjectsResponseType = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const CreateProjectEnvContentHintProjectsType = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const CreateProjectEnvContentHintType = {
    RedisUrl: "redis-url",
};
export const CreateProjectEnvCreatedProjectsType = {
    FlagsSecret: "flags-secret",
};
export const CreateProjectEnvValue2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateProjectEnvTargetProjectsResponse2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateProjectEnvTargetProjects1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/** @internal */
export const CreateProjectEnv2Type$inboundSchema = z.nativeEnum(CreateProjectEnv2Type);
/** @internal */
export const CreateProjectEnv2Type$outboundSchema = CreateProjectEnv2Type$inboundSchema;
/** @internal */
export const CreateProjectEnv2Target$inboundSchema = z.nativeEnum(CreateProjectEnv2Target);
/** @internal */
export const CreateProjectEnv2Target$outboundSchema = CreateProjectEnv2Target$inboundSchema;
/** @internal */
export const Two2$inboundSchema = z
    .object({
    key: z.string(),
    value: z.string(),
    type: CreateProjectEnv2Type$inboundSchema,
    target: z.array(CreateProjectEnv2Target$inboundSchema).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()),
});
/** @internal */
export const Two2$outboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    type: CreateProjectEnv2Type$outboundSchema,
    target: z.array(CreateProjectEnv2Target$outboundSchema).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()),
});
export function two2ToJSON(two2) {
    return JSON.stringify(Two2$outboundSchema.parse(two2));
}
export function two2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Two2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Two2' from JSON`);
}
/** @internal */
export const TwoType$inboundSchema = z
    .nativeEnum(TwoType);
/** @internal */
export const TwoType$outboundSchema = TwoType$inboundSchema;
/** @internal */
export const TwoTarget$inboundSchema = z
    .nativeEnum(TwoTarget);
/** @internal */
export const TwoTarget$outboundSchema = TwoTarget$inboundSchema;
/** @internal */
export const Two1$inboundSchema = z
    .object({
    key: z.string(),
    value: z.string(),
    type: TwoType$inboundSchema,
    target: z.array(TwoTarget$inboundSchema),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const Two1$outboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    type: TwoType$outboundSchema,
    target: z.array(TwoTarget$outboundSchema),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function two1ToJSON(two1) {
    return JSON.stringify(Two1$outboundSchema.parse(two1));
}
export function two1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Two1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Two1' from JSON`);
}
/** @internal */
export const CreateProjectEnvRequestBody2$inboundSchema = z.union([
    z.lazy(() => Two1$inboundSchema),
    z.lazy(() => Two2$inboundSchema),
]);
/** @internal */
export const CreateProjectEnvRequestBody2$outboundSchema = z.union([
    z.lazy(() => Two1$outboundSchema),
    z.lazy(() => Two2$outboundSchema),
]);
export function createProjectEnvRequestBody2ToJSON(createProjectEnvRequestBody2) {
    return JSON.stringify(CreateProjectEnvRequestBody2$outboundSchema.parse(createProjectEnvRequestBody2));
}
export function createProjectEnvRequestBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvRequestBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvRequestBody2' from JSON`);
}
/** @internal */
export const CreateProjectEnv1Type$inboundSchema = z.nativeEnum(CreateProjectEnv1Type);
/** @internal */
export const CreateProjectEnv1Type$outboundSchema = CreateProjectEnv1Type$inboundSchema;
/** @internal */
export const CreateProjectEnv1Target$inboundSchema = z.nativeEnum(CreateProjectEnv1Target);
/** @internal */
export const CreateProjectEnv1Target$outboundSchema = CreateProjectEnv1Target$inboundSchema;
/** @internal */
export const CreateProjectEnv12$inboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    type: CreateProjectEnv1Type$inboundSchema,
    target: z.array(CreateProjectEnv1Target$inboundSchema).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()),
});
/** @internal */
export const CreateProjectEnv12$outboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    type: CreateProjectEnv1Type$outboundSchema,
    target: z.array(CreateProjectEnv1Target$outboundSchema).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()),
});
export function createProjectEnv12ToJSON(createProjectEnv12) {
    return JSON.stringify(CreateProjectEnv12$outboundSchema.parse(createProjectEnv12));
}
export function createProjectEnv12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnv12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnv12' from JSON`);
}
/** @internal */
export const OneType$inboundSchema = z
    .nativeEnum(OneType);
/** @internal */
export const OneType$outboundSchema = OneType$inboundSchema;
/** @internal */
export const OneTarget$inboundSchema = z
    .nativeEnum(OneTarget);
/** @internal */
export const OneTarget$outboundSchema = OneTarget$inboundSchema;
/** @internal */
export const CreateProjectEnv11$inboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    type: OneType$inboundSchema,
    target: z.array(OneTarget$inboundSchema),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const CreateProjectEnv11$outboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    type: OneType$outboundSchema,
    target: z.array(OneTarget$outboundSchema),
    gitBranch: z.nullable(z.string()).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function createProjectEnv11ToJSON(createProjectEnv11) {
    return JSON.stringify(CreateProjectEnv11$outboundSchema.parse(createProjectEnv11));
}
export function createProjectEnv11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnv11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnv11' from JSON`);
}
/** @internal */
export const CreateProjectEnvRequestBody1$inboundSchema = z.union([
    z.lazy(() => CreateProjectEnv11$inboundSchema),
    z.lazy(() => CreateProjectEnv12$inboundSchema),
]);
/** @internal */
export const CreateProjectEnvRequestBody1$outboundSchema = z.union([
    z.lazy(() => CreateProjectEnv11$outboundSchema),
    z.lazy(() => CreateProjectEnv12$outboundSchema),
]);
export function createProjectEnvRequestBody1ToJSON(createProjectEnvRequestBody1) {
    return JSON.stringify(CreateProjectEnvRequestBody1$outboundSchema.parse(createProjectEnvRequestBody1));
}
export function createProjectEnvRequestBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvRequestBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvRequestBody1' from JSON`);
}
/** @internal */
export const CreateProjectEnvRequestBody$inboundSchema = z.union([
    z.union([
        z.lazy(() => CreateProjectEnv11$inboundSchema),
        z.lazy(() => CreateProjectEnv12$inboundSchema),
    ]),
    z.array(z.union([
        z.lazy(() => Two1$inboundSchema),
        z.lazy(() => Two2$inboundSchema),
    ])),
]);
/** @internal */
export const CreateProjectEnvRequestBody$outboundSchema = z.union([
    z.union([
        z.lazy(() => CreateProjectEnv11$outboundSchema),
        z.lazy(() => CreateProjectEnv12$outboundSchema),
    ]),
    z.array(z.union([
        z.lazy(() => Two1$outboundSchema),
        z.lazy(() => Two2$outboundSchema),
    ])),
]);
export function createProjectEnvRequestBodyToJSON(createProjectEnvRequestBody) {
    return JSON.stringify(CreateProjectEnvRequestBody$outboundSchema.parse(createProjectEnvRequestBody));
}
export function createProjectEnvRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvRequestBody' from JSON`);
}
/** @internal */
export const CreateProjectEnvRequest$inboundSchema = z.object({
    idOrName: z.string(),
    upsert: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.union([
        z.union([
            z.lazy(() => CreateProjectEnv11$inboundSchema),
            z.lazy(() => CreateProjectEnv12$inboundSchema),
        ]),
        z.array(z.union([
            z.lazy(() => Two1$inboundSchema),
            z.lazy(() => Two2$inboundSchema),
        ])),
    ]),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const CreateProjectEnvRequest$outboundSchema = z.object({
    idOrName: z.string(),
    upsert: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.union([
        z.union([
            z.lazy(() => CreateProjectEnv11$outboundSchema),
            z.lazy(() => CreateProjectEnv12$outboundSchema),
        ]),
        z.array(z.union([
            z.lazy(() => Two1$outboundSchema),
            z.lazy(() => Two2$outboundSchema),
        ])),
    ]),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function createProjectEnvRequestToJSON(createProjectEnvRequest) {
    return JSON.stringify(CreateProjectEnvRequest$outboundSchema.parse(createProjectEnvRequest));
}
export function createProjectEnvRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvRequest' from JSON`);
}
/** @internal */
export const CreateProjectEnvTargetProjects2$inboundSchema = z.nativeEnum(CreateProjectEnvTargetProjects2);
/** @internal */
export const CreateProjectEnvTargetProjects2$outboundSchema = CreateProjectEnvTargetProjects2$inboundSchema;
/** @internal */
export const CreateProjectEnvCreatedTarget$inboundSchema = z.union([
    z.array(z.string()),
    CreateProjectEnvTargetProjects2$inboundSchema,
]);
/** @internal */
export const CreateProjectEnvCreatedTarget$outboundSchema = z.union([
    z.array(z.string()),
    CreateProjectEnvTargetProjects2$outboundSchema,
]);
export function createProjectEnvCreatedTargetToJSON(createProjectEnvCreatedTarget) {
    return JSON.stringify(CreateProjectEnvCreatedTarget$outboundSchema.parse(createProjectEnvCreatedTarget));
}
export function createProjectEnvCreatedTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvCreatedTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvCreatedTarget' from JSON`);
}
/** @internal */
export const CreateProjectEnvCreatedType$inboundSchema = z.nativeEnum(CreateProjectEnvCreatedType);
/** @internal */
export const CreateProjectEnvCreatedType$outboundSchema = CreateProjectEnvCreatedType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects15$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects15$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated215Type$outboundSchema,
    projectId: z.string(),
});
export function createProjectEnvContentHintProjects15ToJSON(createProjectEnvContentHintProjects15) {
    return JSON.stringify(CreateProjectEnvContentHintProjects15$outboundSchema.parse(createProjectEnvContentHintProjects15));
}
export function createProjectEnvContentHintProjects15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects15' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects14$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects14$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated214Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function createProjectEnvContentHintProjects14ToJSON(createProjectEnvContentHintProjects14) {
    return JSON.stringify(CreateProjectEnvContentHintProjects14$outboundSchema.parse(createProjectEnvContentHintProjects14));
}
export function createProjectEnvContentHintProjects14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects14' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects13$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects13$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated213Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects13ToJSON(createProjectEnvContentHintProjects13) {
    return JSON.stringify(CreateProjectEnvContentHintProjects13$outboundSchema.parse(createProjectEnvContentHintProjects13));
}
export function createProjectEnvContentHintProjects13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects13' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects12$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects12$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated212Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects12ToJSON(createProjectEnvContentHintProjects12) {
    return JSON.stringify(CreateProjectEnvContentHintProjects12$outboundSchema.parse(createProjectEnvContentHintProjects12));
}
export function createProjectEnvContentHintProjects12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects12' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects11$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects11$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated211Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects11ToJSON(createProjectEnvContentHintProjects11) {
    return JSON.stringify(CreateProjectEnvContentHintProjects11$outboundSchema.parse(createProjectEnvContentHintProjects11));
}
export function createProjectEnvContentHintProjects11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects11' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects10$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects10$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated210Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects10ToJSON(createProjectEnvContentHintProjects10) {
    return JSON.stringify(CreateProjectEnvContentHintProjects10$outboundSchema.parse(createProjectEnvContentHintProjects10));
}
export function createProjectEnvContentHintProjects10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects10' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects9$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects9$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated29Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects9ToJSON(createProjectEnvContentHintProjects9) {
    return JSON.stringify(CreateProjectEnvContentHintProjects9$outboundSchema.parse(createProjectEnvContentHintProjects9));
}
export function createProjectEnvContentHintProjects9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects9' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects8$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects8$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated28Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects8ToJSON(createProjectEnvContentHintProjects8) {
    return JSON.stringify(CreateProjectEnvContentHintProjects8$outboundSchema.parse(createProjectEnvContentHintProjects8));
}
export function createProjectEnvContentHintProjects8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects8' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects7$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects7$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated27Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects7ToJSON(createProjectEnvContentHintProjects7) {
    return JSON.stringify(CreateProjectEnvContentHintProjects7$outboundSchema.parse(createProjectEnvContentHintProjects7));
}
export function createProjectEnvContentHintProjects7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects7' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects6$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects6$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated26Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects6ToJSON(createProjectEnvContentHintProjects6) {
    return JSON.stringify(CreateProjectEnvContentHintProjects6$outboundSchema.parse(createProjectEnvContentHintProjects6));
}
export function createProjectEnvContentHintProjects6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects6' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects5$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects5$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated25Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects5ToJSON(createProjectEnvContentHintProjects5) {
    return JSON.stringify(CreateProjectEnvContentHintProjects5$outboundSchema.parse(createProjectEnvContentHintProjects5));
}
export function createProjectEnvContentHintProjects5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects5' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects4$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects4$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated24Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects4ToJSON(createProjectEnvContentHintProjects4) {
    return JSON.stringify(CreateProjectEnvContentHintProjects4$outboundSchema.parse(createProjectEnvContentHintProjects4));
}
export function createProjectEnvContentHintProjects4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects4' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects3$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects3$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated23Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects3ToJSON(createProjectEnvContentHintProjects3) {
    return JSON.stringify(CreateProjectEnvContentHintProjects3$outboundSchema.parse(createProjectEnvContentHintProjects3));
}
export function createProjectEnvContentHintProjects3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects3' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects2$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects2$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated22Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects2ToJSON(createProjectEnvContentHintProjects2) {
    return JSON.stringify(CreateProjectEnvContentHintProjects2$outboundSchema.parse(createProjectEnvContentHintProjects2));
}
export function createProjectEnvContentHintProjects2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects2' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjects1$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHintProjects1$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated2Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHintProjects1ToJSON(createProjectEnvContentHintProjects1) {
    return JSON.stringify(CreateProjectEnvContentHintProjects1$outboundSchema.parse(createProjectEnvContentHintProjects1));
}
export function createProjectEnvContentHintProjects1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHintProjects1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHintProjects1' from JSON`);
}
/** @internal */
export const CreateProjectEnvCreatedContentHint$inboundSchema = z.union([
    z.lazy(() => CreateProjectEnvContentHintProjects14$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects1$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects2$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects3$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects4$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects5$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects6$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects7$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects8$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects9$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects10$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects11$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects12$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects13$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects15$inboundSchema),
]);
/** @internal */
export const CreateProjectEnvCreatedContentHint$outboundSchema = z.union([
    z.lazy(() => CreateProjectEnvContentHintProjects14$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects1$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects2$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects3$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects4$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects5$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects6$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects7$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects8$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects9$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects10$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects11$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects12$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects13$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHintProjects15$outboundSchema),
]);
export function createProjectEnvCreatedContentHintToJSON(createProjectEnvCreatedContentHint) {
    return JSON.stringify(CreateProjectEnvCreatedContentHint$outboundSchema.parse(createProjectEnvCreatedContentHint));
}
export function createProjectEnvCreatedContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvCreatedContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvCreatedContentHint' from JSON`);
}
/** @internal */
export const CreateProjectEnvCreatedProjectsResponseType$inboundSchema = z
    .nativeEnum(CreateProjectEnvCreatedProjectsResponseType);
/** @internal */
export const CreateProjectEnvCreatedProjectsResponseType$outboundSchema = CreateProjectEnvCreatedProjectsResponseType$inboundSchema;
/** @internal */
export const CreateProjectEnvCreatedInternalContentHint$inboundSchema = z.object({
    type: CreateProjectEnvCreatedProjectsResponseType$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const CreateProjectEnvCreatedInternalContentHint$outboundSchema = z.object({
    type: CreateProjectEnvCreatedProjectsResponseType$outboundSchema,
    encryptedValue: z.string(),
});
export function createProjectEnvCreatedInternalContentHintToJSON(createProjectEnvCreatedInternalContentHint) {
    return JSON.stringify(CreateProjectEnvCreatedInternalContentHint$outboundSchema.parse(createProjectEnvCreatedInternalContentHint));
}
export function createProjectEnvCreatedInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvCreatedInternalContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvCreatedInternalContentHint' from JSON`);
}
/** @internal */
export const Created2$inboundSchema = z.object({
    target: z.union([
        z.array(z.string()),
        CreateProjectEnvTargetProjects2$inboundSchema,
    ]).optional(),
    type: CreateProjectEnvCreatedType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => CreateProjectEnvContentHintProjects14$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects1$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects2$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects3$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects4$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects5$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects6$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects7$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects8$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects9$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects10$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects11$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects12$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects13$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => CreateProjectEnvCreatedInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
    system: z.boolean().optional(),
});
/** @internal */
export const Created2$outboundSchema = z.object({
    target: z.union([
        z.array(z.string()),
        CreateProjectEnvTargetProjects2$outboundSchema,
    ]).optional(),
    type: CreateProjectEnvCreatedType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => CreateProjectEnvContentHintProjects14$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects1$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects2$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects3$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects4$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects5$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects6$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects7$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects8$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects9$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects10$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects11$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects12$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects13$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHintProjects15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => CreateProjectEnvCreatedInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
    system: z.boolean().optional(),
});
export function created2ToJSON(created2) {
    return JSON.stringify(Created2$outboundSchema.parse(created2));
}
export function created2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Created2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Created2' from JSON`);
}
/** @internal */
export const CreateProjectEnvTarget2$inboundSchema = z.nativeEnum(CreateProjectEnvTarget2);
/** @internal */
export const CreateProjectEnvTarget2$outboundSchema = CreateProjectEnvTarget2$inboundSchema;
/** @internal */
export const CreateProjectEnvTarget1$inboundSchema = z.nativeEnum(CreateProjectEnvTarget1);
/** @internal */
export const CreateProjectEnvTarget1$outboundSchema = CreateProjectEnvTarget1$inboundSchema;
/** @internal */
export const CreatedTarget$inboundSchema = z.union([
    z.array(CreateProjectEnvTarget1$inboundSchema),
    CreateProjectEnvTarget2$inboundSchema,
]);
/** @internal */
export const CreatedTarget$outboundSchema = z.union([
    z.array(CreateProjectEnvTarget1$outboundSchema),
    CreateProjectEnvTarget2$outboundSchema,
]);
export function createdTargetToJSON(createdTarget) {
    return JSON.stringify(CreatedTarget$outboundSchema.parse(createdTarget));
}
export function createdTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreatedTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreatedTarget' from JSON`);
}
/** @internal */
export const CreatedType$inboundSchema = z
    .nativeEnum(CreatedType);
/** @internal */
export const CreatedType$outboundSchema = CreatedType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint15$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint15$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated115Type$outboundSchema,
    projectId: z.string(),
});
export function createProjectEnvContentHint15ToJSON(createProjectEnvContentHint15) {
    return JSON.stringify(CreateProjectEnvContentHint15$outboundSchema.parse(createProjectEnvContentHint15));
}
export function createProjectEnvContentHint15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint15' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint14$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint14$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated114Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function createProjectEnvContentHint14ToJSON(createProjectEnvContentHint14) {
    return JSON.stringify(CreateProjectEnvContentHint14$outboundSchema.parse(createProjectEnvContentHint14));
}
export function createProjectEnvContentHint14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint14' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint13$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint13$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated113Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint13ToJSON(createProjectEnvContentHint13) {
    return JSON.stringify(CreateProjectEnvContentHint13$outboundSchema.parse(createProjectEnvContentHint13));
}
export function createProjectEnvContentHint13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint13' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint12$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint12$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated112Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint12ToJSON(createProjectEnvContentHint12) {
    return JSON.stringify(CreateProjectEnvContentHint12$outboundSchema.parse(createProjectEnvContentHint12));
}
export function createProjectEnvContentHint12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint12' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint11$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint11$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated111Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint11ToJSON(createProjectEnvContentHint11) {
    return JSON.stringify(CreateProjectEnvContentHint11$outboundSchema.parse(createProjectEnvContentHint11));
}
export function createProjectEnvContentHint11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint11' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint10$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint10$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated110Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint10ToJSON(createProjectEnvContentHint10) {
    return JSON.stringify(CreateProjectEnvContentHint10$outboundSchema.parse(createProjectEnvContentHint10));
}
export function createProjectEnvContentHint10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint10' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint9$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint9$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated19Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint9ToJSON(createProjectEnvContentHint9) {
    return JSON.stringify(CreateProjectEnvContentHint9$outboundSchema.parse(createProjectEnvContentHint9));
}
export function createProjectEnvContentHint9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint9' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint8$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint8$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreated1Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint8ToJSON(createProjectEnvContentHint8) {
    return JSON.stringify(CreateProjectEnvContentHint8$outboundSchema.parse(createProjectEnvContentHint8));
}
export function createProjectEnvContentHint8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint8' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint7$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint7$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyCreatedType$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint7ToJSON(createProjectEnvContentHint7) {
    return JSON.stringify(CreateProjectEnvContentHint7$outboundSchema.parse(createProjectEnvContentHint7));
}
export function createProjectEnvContentHint7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint7' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint6$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint6$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONResponseBodyType$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint6ToJSON(createProjectEnvContentHint6) {
    return JSON.stringify(CreateProjectEnvContentHint6$outboundSchema.parse(createProjectEnvContentHint6));
}
export function createProjectEnvContentHint6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint6' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType$outboundSchema = CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint5$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint5$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201ApplicationJSONType$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint5ToJSON(createProjectEnvContentHint5) {
    return JSON.stringify(CreateProjectEnvContentHint5$outboundSchema.parse(createProjectEnvContentHint5));
}
export function createProjectEnvContentHint5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint5' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201Type$inboundSchema = z
    .nativeEnum(CreateProjectEnvContentHintProjectsResponse201Type);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponse201Type$outboundSchema = CreateProjectEnvContentHintProjectsResponse201Type$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint4$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint4$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponse201Type$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint4ToJSON(createProjectEnvContentHint4) {
    return JSON.stringify(CreateProjectEnvContentHint4$outboundSchema.parse(createProjectEnvContentHint4));
}
export function createProjectEnvContentHint4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint4' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsResponseType$inboundSchema = z
    .nativeEnum(CreateProjectEnvContentHintProjectsResponseType);
/** @internal */
export const CreateProjectEnvContentHintProjectsResponseType$outboundSchema = CreateProjectEnvContentHintProjectsResponseType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint3$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponseType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint3$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsResponseType$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint3ToJSON(createProjectEnvContentHint3) {
    return JSON.stringify(CreateProjectEnvContentHint3$outboundSchema.parse(createProjectEnvContentHint3));
}
export function createProjectEnvContentHint3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint3' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintProjectsType$inboundSchema = z
    .nativeEnum(CreateProjectEnvContentHintProjectsType);
/** @internal */
export const CreateProjectEnvContentHintProjectsType$outboundSchema = CreateProjectEnvContentHintProjectsType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint2$inboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint2$outboundSchema = z.object({
    type: CreateProjectEnvContentHintProjectsType$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint2ToJSON(createProjectEnvContentHint2) {
    return JSON.stringify(CreateProjectEnvContentHint2$outboundSchema.parse(createProjectEnvContentHint2));
}
export function createProjectEnvContentHint2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint2' from JSON`);
}
/** @internal */
export const CreateProjectEnvContentHintType$inboundSchema = z.nativeEnum(CreateProjectEnvContentHintType);
/** @internal */
export const CreateProjectEnvContentHintType$outboundSchema = CreateProjectEnvContentHintType$inboundSchema;
/** @internal */
export const CreateProjectEnvContentHint1$inboundSchema = z.object({
    type: CreateProjectEnvContentHintType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const CreateProjectEnvContentHint1$outboundSchema = z.object({
    type: CreateProjectEnvContentHintType$outboundSchema,
    storeId: z.string(),
});
export function createProjectEnvContentHint1ToJSON(createProjectEnvContentHint1) {
    return JSON.stringify(CreateProjectEnvContentHint1$outboundSchema.parse(createProjectEnvContentHint1));
}
export function createProjectEnvContentHint1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvContentHint1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvContentHint1' from JSON`);
}
/** @internal */
export const CreatedContentHint$inboundSchema = z.union([
    z.lazy(() => CreateProjectEnvContentHint14$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint1$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint2$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint3$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint4$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint5$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint6$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint7$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint8$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint9$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint10$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint11$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint12$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint13$inboundSchema),
    z.lazy(() => CreateProjectEnvContentHint15$inboundSchema),
]);
/** @internal */
export const CreatedContentHint$outboundSchema = z.union([
    z.lazy(() => CreateProjectEnvContentHint14$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint1$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint2$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint3$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint4$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint5$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint6$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint7$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint8$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint9$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint10$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint11$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint12$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint13$outboundSchema),
    z.lazy(() => CreateProjectEnvContentHint15$outboundSchema),
]);
export function createdContentHintToJSON(createdContentHint) {
    return JSON.stringify(CreatedContentHint$outboundSchema.parse(createdContentHint));
}
export function createdContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreatedContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreatedContentHint' from JSON`);
}
/** @internal */
export const CreateProjectEnvCreatedProjectsType$inboundSchema = z.nativeEnum(CreateProjectEnvCreatedProjectsType);
/** @internal */
export const CreateProjectEnvCreatedProjectsType$outboundSchema = CreateProjectEnvCreatedProjectsType$inboundSchema;
/** @internal */
export const CreatedInternalContentHint$inboundSchema = z.object({
    type: CreateProjectEnvCreatedProjectsType$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const CreatedInternalContentHint$outboundSchema = z.object({
    type: CreateProjectEnvCreatedProjectsType$outboundSchema,
    encryptedValue: z.string(),
});
export function createdInternalContentHintToJSON(createdInternalContentHint) {
    return JSON.stringify(CreatedInternalContentHint$outboundSchema.parse(createdInternalContentHint));
}
export function createdInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreatedInternalContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreatedInternalContentHint' from JSON`);
}
/** @internal */
export const Created1$inboundSchema = z.object({
    target: z.union([
        z.array(CreateProjectEnvTarget1$inboundSchema),
        CreateProjectEnvTarget2$inboundSchema,
    ]).optional(),
    type: CreatedType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => CreateProjectEnvContentHint14$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint1$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint2$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint3$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint4$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint5$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint6$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint7$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint8$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint9$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint10$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint11$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint12$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint13$inboundSchema),
        z.lazy(() => CreateProjectEnvContentHint15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => CreatedInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
    system: z.boolean().optional(),
});
/** @internal */
export const Created1$outboundSchema = z.object({
    target: z.union([
        z.array(CreateProjectEnvTarget1$outboundSchema),
        CreateProjectEnvTarget2$outboundSchema,
    ]).optional(),
    type: CreatedType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => CreateProjectEnvContentHint14$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint1$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint2$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint3$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint4$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint5$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint6$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint7$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint8$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint9$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint10$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint11$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint12$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint13$outboundSchema),
        z.lazy(() => CreateProjectEnvContentHint15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => CreatedInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
    system: z.boolean().optional(),
});
export function created1ToJSON(created1) {
    return JSON.stringify(Created1$outboundSchema.parse(created1));
}
export function created1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Created1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Created1' from JSON`);
}
/** @internal */
export const CreateProjectEnvCreated$inboundSchema = z.union([
    z.lazy(() => Created1$inboundSchema),
    z.array(z.lazy(() => Created2$inboundSchema)),
]);
/** @internal */
export const CreateProjectEnvCreated$outboundSchema = z.union([
    z.lazy(() => Created1$outboundSchema),
    z.array(z.lazy(() => Created2$outboundSchema)),
]);
export function createProjectEnvCreatedToJSON(createProjectEnvCreated) {
    return JSON.stringify(CreateProjectEnvCreated$outboundSchema.parse(createProjectEnvCreated));
}
export function createProjectEnvCreatedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvCreated$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvCreated' from JSON`);
}
/** @internal */
export const CreateProjectEnvValue2$inboundSchema = z.nativeEnum(CreateProjectEnvValue2);
/** @internal */
export const CreateProjectEnvValue2$outboundSchema = CreateProjectEnvValue2$inboundSchema;
/** @internal */
export const CreateProjectEnvValue$inboundSchema = z.union([z.string(), z.array(CreateProjectEnvValue2$inboundSchema)]);
/** @internal */
export const CreateProjectEnvValue$outboundSchema = z.union([z.string(), z.array(CreateProjectEnvValue2$outboundSchema)]);
export function createProjectEnvValueToJSON(createProjectEnvValue) {
    return JSON.stringify(CreateProjectEnvValue$outboundSchema.parse(createProjectEnvValue));
}
export function createProjectEnvValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvValue' from JSON`);
}
/** @internal */
export const CreateProjectEnvTargetProjectsResponse2$inboundSchema = z
    .nativeEnum(CreateProjectEnvTargetProjectsResponse2);
/** @internal */
export const CreateProjectEnvTargetProjectsResponse2$outboundSchema = CreateProjectEnvTargetProjectsResponse2$inboundSchema;
/** @internal */
export const CreateProjectEnvTargetProjects1$inboundSchema = z.nativeEnum(CreateProjectEnvTargetProjects1);
/** @internal */
export const CreateProjectEnvTargetProjects1$outboundSchema = CreateProjectEnvTargetProjects1$inboundSchema;
/** @internal */
export const CreateProjectEnvTarget$inboundSchema = z.union([
    z.array(CreateProjectEnvTargetProjects1$inboundSchema),
    CreateProjectEnvTargetProjectsResponse2$inboundSchema,
]);
/** @internal */
export const CreateProjectEnvTarget$outboundSchema = z.union([
    z.array(CreateProjectEnvTargetProjects1$outboundSchema),
    CreateProjectEnvTargetProjectsResponse2$outboundSchema,
]);
export function createProjectEnvTargetToJSON(createProjectEnvTarget) {
    return JSON.stringify(CreateProjectEnvTarget$outboundSchema.parse(createProjectEnvTarget));
}
export function createProjectEnvTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvTarget' from JSON`);
}
/** @internal */
export const CreateProjectEnvError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    key: z.string().optional(),
    envVarId: z.string().optional(),
    envVarKey: z.string().optional(),
    action: z.string().optional(),
    link: z.string().optional(),
    value: z.union([z.string(), z.array(CreateProjectEnvValue2$inboundSchema)])
        .optional(),
    gitBranch: z.string().optional(),
    target: z.union([
        z.array(CreateProjectEnvTargetProjects1$inboundSchema),
        CreateProjectEnvTargetProjectsResponse2$inboundSchema,
    ]).optional(),
    project: z.string().optional(),
});
/** @internal */
export const CreateProjectEnvError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    key: z.string().optional(),
    envVarId: z.string().optional(),
    envVarKey: z.string().optional(),
    action: z.string().optional(),
    link: z.string().optional(),
    value: z.union([z.string(), z.array(CreateProjectEnvValue2$outboundSchema)])
        .optional(),
    gitBranch: z.string().optional(),
    target: z.union([
        z.array(CreateProjectEnvTargetProjects1$outboundSchema),
        CreateProjectEnvTargetProjectsResponse2$outboundSchema,
    ]).optional(),
    project: z.string().optional(),
});
export function createProjectEnvErrorToJSON(createProjectEnvError) {
    return JSON.stringify(CreateProjectEnvError$outboundSchema.parse(createProjectEnvError));
}
export function createProjectEnvErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvError' from JSON`);
}
/** @internal */
export const CreateProjectEnvFailed$inboundSchema = z.object({
    error: z.lazy(() => CreateProjectEnvError$inboundSchema),
});
/** @internal */
export const CreateProjectEnvFailed$outboundSchema = z.object({
    error: z.lazy(() => CreateProjectEnvError$outboundSchema),
});
export function createProjectEnvFailedToJSON(createProjectEnvFailed) {
    return JSON.stringify(CreateProjectEnvFailed$outboundSchema.parse(createProjectEnvFailed));
}
export function createProjectEnvFailedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvFailed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvFailed' from JSON`);
}
/** @internal */
export const CreateProjectEnvResponseBody$inboundSchema = z.object({
    created: z.union([
        z.lazy(() => Created1$inboundSchema),
        z.array(z.lazy(() => Created2$inboundSchema)),
    ]),
    failed: z.array(z.lazy(() => CreateProjectEnvFailed$inboundSchema)),
});
/** @internal */
export const CreateProjectEnvResponseBody$outboundSchema = z.object({
    created: z.union([
        z.lazy(() => Created1$outboundSchema),
        z.array(z.lazy(() => Created2$outboundSchema)),
    ]),
    failed: z.array(z.lazy(() => CreateProjectEnvFailed$outboundSchema)),
});
export function createProjectEnvResponseBodyToJSON(createProjectEnvResponseBody) {
    return JSON.stringify(CreateProjectEnvResponseBody$outboundSchema.parse(createProjectEnvResponseBody));
}
export function createProjectEnvResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateProjectEnvResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateProjectEnvResponseBody' from JSON`);
}
