// @vercel/sdk/esm/models/createdeploymentop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { FlagJSONValue$inboundSchema, FlagJSONValue$outboundSchema, } from "./flagjsonvalue.js";
/**
 * Forces a new deployment even if there is a previous similar deployment
 */
export const ForceNew = {
    Zero: "0",
    One: "1",
};
/**
 * Allows to skip framework detection so the API would not fail to ask for confirmation
 */
export const SkipAutoDetectionConfirmation = {
    Zero: "0",
    One: "1",
};
/**
 * The file content encoding, it could be either a base64 (useful for images, etc.) of the files or the plain text for source code.
 */
export const Encoding = {
    Base64: "base64",
    Utf8: "utf-8",
};
export const CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type = {
    Bitbucket: "bitbucket",
};
export const CreateDeploymentGitSourceDeploymentsRequestRequestBodyType = {
    Bitbucket: "bitbucket",
};
export const CreateDeploymentGitSourceDeploymentsRequestType = {
    Gitlab: "gitlab",
};
export const CreateDeploymentGitSourceDeploymentsType = {
    GithubLimited: "github-limited",
};
export const CreateDeploymentGitSourceType = {
    GithubLimited: "github-limited",
};
export const GitSourceType = {
    Github: "github",
};
export const CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type = {
    Github: "github",
};
/**
 * The framework that is being used for this project. When `null` is used no framework is selected
 */
export const Framework = {
    Blitzjs: "blitzjs",
    Nextjs: "nextjs",
    Gatsby: "gatsby",
    Remix: "remix",
    ReactRouter: "react-router",
    Astro: "astro",
    Hexo: "hexo",
    Eleventy: "eleventy",
    Docusaurus2: "docusaurus-2",
    Docusaurus: "docusaurus",
    Preact: "preact",
    Solidstart1: "solidstart-1",
    Solidstart: "solidstart",
    Dojo: "dojo",
    Ember: "ember",
    Vue: "vue",
    Scully: "scully",
    IonicAngular: "ionic-angular",
    Angular: "angular",
    Polymer: "polymer",
    Svelte: "svelte",
    Sveltekit: "sveltekit",
    Sveltekit1: "sveltekit-1",
    IonicReact: "ionic-react",
    CreateReactApp: "create-react-app",
    Gridsome: "gridsome",
    Umijs: "umijs",
    Sapper: "sapper",
    Saber: "saber",
    Stencil: "stencil",
    Nuxtjs: "nuxtjs",
    Redwoodjs: "redwoodjs",
    Hugo: "hugo",
    Jekyll: "jekyll",
    Brunch: "brunch",
    Middleman: "middleman",
    Zola: "zola",
    Hydrogen: "hydrogen",
    Vite: "vite",
    TanstackStart: "tanstack-start",
    Vitepress: "vitepress",
    Vuepress: "vuepress",
    Parcel: "parcel",
    Fastapi: "fastapi",
    Flask: "flask",
    Fasthtml: "fasthtml",
    SanityV3: "sanity-v3",
    Sanity: "sanity",
    Storybook: "storybook",
    Nitro: "nitro",
    Hono: "hono",
    Express: "express",
    H3: "h3",
    Nestjs: "nestjs",
    Elysia: "elysia",
    Fastify: "fastify",
    Xmcp: "xmcp",
};
/**
 * Override the Node.js version that should be used for this deployment
 */
export const NodeVersion = {
    TwentyFourDotX: "24.x",
    TwentyTwoDotX: "22.x",
    TwentyDotX: "20.x",
    EighteenDotX: "18.x",
    SixteenDotX: "16.x",
    FourteenDotX: "14.x",
    TwelveDotX: "12.x",
    TenDotX: "10.x",
    EightDot10DotX: "8.10.x",
};
export const CreateDeploymentFramework = {
    Blitzjs: "blitzjs",
    Nextjs: "nextjs",
    Gatsby: "gatsby",
    Remix: "remix",
    ReactRouter: "react-router",
    Astro: "astro",
    Hexo: "hexo",
    Eleventy: "eleventy",
    Docusaurus2: "docusaurus-2",
    Docusaurus: "docusaurus",
    Preact: "preact",
    Solidstart1: "solidstart-1",
    Solidstart: "solidstart",
    Dojo: "dojo",
    Ember: "ember",
    Vue: "vue",
    Scully: "scully",
    IonicAngular: "ionic-angular",
    Angular: "angular",
    Polymer: "polymer",
    Svelte: "svelte",
    Sveltekit: "sveltekit",
    Sveltekit1: "sveltekit-1",
    IonicReact: "ionic-react",
    CreateReactApp: "create-react-app",
    Gridsome: "gridsome",
    Umijs: "umijs",
    Sapper: "sapper",
    Saber: "saber",
    Stencil: "stencil",
    Nuxtjs: "nuxtjs",
    Redwoodjs: "redwoodjs",
    Hugo: "hugo",
    Jekyll: "jekyll",
    Brunch: "brunch",
    Middleman: "middleman",
    Zola: "zola",
    Hydrogen: "hydrogen",
    Vite: "vite",
    TanstackStart: "tanstack-start",
    Vitepress: "vitepress",
    Vuepress: "vuepress",
    Parcel: "parcel",
    Fastapi: "fastapi",
    Flask: "flask",
    Fasthtml: "fasthtml",
    SanityV3: "sanity-v3",
    Sanity: "sanity",
    Storybook: "storybook",
    Nitro: "nitro",
    Hono: "hono",
    Express: "express",
    H3: "h3",
    Nestjs: "nestjs",
    Elysia: "elysia",
    Fastify: "fastify",
    Xmcp: "xmcp",
};
export const CreateDeploymentDeploymentsStatus = {
    Skipped: "skipped",
    Pending: "pending",
    Ready: "ready",
    Error: "error",
    Timeout: "timeout",
};
/**
 * Must be `http` or `https`.
 */
export const Protocol = {
    Http: "http",
    Https: "https",
};
export const Formats = {
    ImageAvif: "image/avif",
    ImageWebp: "image/webp",
};
export const ContentDispositionType = {
    Inline: "inline",
    Attachment: "attachment",
};
export const CreateDeploymentReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Ready: "READY",
};
export const CreateDeploymentStatus = {
    Queued: "QUEUED",
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * The type of environment (production, preview, or development)
 */
export const CustomEnvironmentType = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of matching to perform
 */
export const CreateDeploymentCustomEnvironmentType = {
    EndsWith: "endsWith",
    StartsWith: "startsWith",
    Equals: "equals",
};
export const OomReport = {
    OutOfMemory: "out-of-memory",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const ReadyState = {
    Queued: "QUEUED",
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Ready: "READY",
    Canceled: "CANCELED",
};
export const CreateDeploymentType = {
    Lambdas: "LAMBDAS",
};
export const ChecksState = {
    Registered: "registered",
    Running: "running",
    Completed: "completed",
};
export const ChecksConclusion = {
    Succeeded: "succeeded",
    Failed: "failed",
    Skipped: "skipped",
    Canceled: "canceled",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type = {
    Bitbucket: "bitbucket",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type = {
    Gitlab: "gitlab",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type = {
    GithubLimited: "github-limited",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type = {
    GithubCustomHost: "github-custom-host",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type = {
    Github: "github",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type = {
    Custom: "custom",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type = {
    Bitbucket: "bitbucket",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type = {
    Bitbucket: "bitbucket",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type = {
    Gitlab: "gitlab",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type = {
    GithubLimited: "github-limited",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type = {
    GithubLimited: "github-limited",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType = {
    GithubCustomHost: "github-custom-host",
};
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType = {
    GithubCustomHost: "github-custom-host",
};
export const CreateDeploymentGitSourceDeploymentsResponse200Type = {
    Github: "github",
};
export const CreateDeploymentGitSourceDeploymentsResponseType = {
    Github: "github",
};
/**
 * If set it overrides the `projectSettings.nodeVersion` for this deployment.
 */
export const CreateDeploymentNodeVersion = {
    TwentyFourDotX: "24.x",
    TwentyTwoDotX: "22.x",
    TwentyDotX: "20.x",
    EighteenDotX: "18.x",
    SixteenDotX: "16.x",
    FourteenDotX: "14.x",
    TwelveDotX: "12.x",
    TenDotX: "10.x",
    EightDot10DotX: "8.10.x",
};
/**
 * Substate of deployment when readyState is 'READY' Tracks whether or not deployment has seen production traffic: - STAGED: never seen production traffic - ROLLING: in the process of having production traffic gradually transitioned. - PROMOTED: has seen production traffic
 */
export const ReadySubstate = {
    Staged: "STAGED",
    Rolling: "ROLLING",
    Promoted: "PROMOTED",
};
/**
 * Where was the deployment created from
 */
export const CreateDeploymentSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const CreateDeploymentTarget = {
    Staging: "staging",
    Production: "production",
};
export const CreateDeploymentPlan = {
    Pro: "pro",
    Enterprise: "enterprise",
    Hobby: "hobby",
};
export const Architecture = {
    X8664: "x86_64",
    Arm64: "arm64",
};
/**
 * Event type - must be "queue/v1beta" (REQUIRED)
 */
export const CreateDeploymentDeploymentsType = {
    QueueV1beta: "queue/v1beta",
};
export const Handle = {
    Error: "error",
    Filesystem: "filesystem",
    Hit: "hit",
    Miss: "miss",
    Rewrite: "rewrite",
    Resource: "resource",
};
export const CreateDeploymentHasDeploymentsType = {
    Header: "header",
    Cookie: "cookie",
    Query: "query",
};
export const CreateDeploymentHasType = {
    Host: "host",
};
export const CreateDeploymentMissingType = {
    Header: "header",
    Cookie: "cookie",
    Query: "query",
};
export const MissingType = {
    Host: "host",
};
export const RoutesAction = {
    Challenge: "challenge",
    Deny: "deny",
};
export const RoutesType = {
    RequestHeaders: "request.headers",
    RequestQuery: "request.query",
    ResponseHeaders: "response.headers",
};
export const CreateDeploymentRoutesOp = {
    Append: "append",
    Set: "set",
    Delete: "delete",
};
export const CreateDeploymentGitRepoDeploymentsType = {
    Bitbucket: "bitbucket",
};
export const CreateDeploymentGitRepoOwnerType = {
    Team: "team",
    User: "user",
};
export const CreateDeploymentGitRepoType = {
    Github: "github",
};
export const GitRepoOwnerType = {
    Team: "team",
    User: "user",
};
export const GitRepoType = {
    Gitlab: "gitlab",
};
export const OwnerType = {
    Team: "team",
    User: "user",
};
/**
 * The result of the microfrontends config upload during deployment creation / build. Only set for default app deployments. The config upload is attempted during deployment create, and then again during the build. If the config is not in the root directory, or the deployment is prebuilt, the config cannot be uploaded during deployment create. The upload during deployment build finds the config even if it's not in the root directory, as it has access to all files. Uploading the config during create is ideal, as then all child deployments are guaranteed to have access to the default app deployment config even if the default app has not yet started building. If the config is not uploaded, the child app will show as building until the config has been uploaded during the default app build. - `success` - The config was uploaded successfully, either when the deployment was created or during the build. - `waiting_on_build` - The config could not be uploaded during deployment create, will be attempted again during the build. - `no_config` - No config was found. Only set once the build has not found the config in any of the deployment's files. - `undefined` - Legacy deployments, or there was an error uploading the config during deployment create.
 */
export const MfeConfigUploadState = {
    Success: "success",
    WaitingOnBuild: "waiting_on_build",
    NoConfig: "no_config",
};
export const FunctionType = {
    Fluid: "fluid",
    Standard: "standard",
};
export const FunctionMemoryType = {
    Standard: "standard",
    StandardLegacy: "standard_legacy",
    Performance: "performance",
};
export const CreateDeploymentState = {
    Succeeded: "succeeded",
    Failed: "failed",
    Pending: "pending",
};
/** @internal */
export const ForceNew$inboundSchema = z
    .nativeEnum(ForceNew);
/** @internal */
export const ForceNew$outboundSchema = ForceNew$inboundSchema;
/** @internal */
export const SkipAutoDetectionConfirmation$inboundSchema = z.nativeEnum(SkipAutoDetectionConfirmation);
/** @internal */
export const SkipAutoDetectionConfirmation$outboundSchema = SkipAutoDetectionConfirmation$inboundSchema;
/** @internal */
export const UploadedFile$inboundSchema = z.object({
    file: z.string(),
    sha: z.string().optional(),
    size: z.number().int().optional(),
});
/** @internal */
export const UploadedFile$outboundSchema = z.object({
    file: z.string(),
    sha: z.string().optional(),
    size: z.number().int().optional(),
});
export function uploadedFileToJSON(uploadedFile) {
    return JSON.stringify(UploadedFile$outboundSchema.parse(uploadedFile));
}
export function uploadedFileFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UploadedFile$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UploadedFile' from JSON`);
}
/** @internal */
export const Encoding$inboundSchema = z
    .nativeEnum(Encoding);
/** @internal */
export const Encoding$outboundSchema = Encoding$inboundSchema;
/** @internal */
export const InlinedFile$inboundSchema = z.object({
    data: z.string(),
    encoding: Encoding$inboundSchema.optional(),
    file: z.string(),
});
/** @internal */
export const InlinedFile$outboundSchema = z.object({
    data: z.string(),
    encoding: Encoding$outboundSchema.optional(),
    file: z.string(),
});
export function inlinedFileToJSON(inlinedFile) {
    return JSON.stringify(InlinedFile$outboundSchema.parse(inlinedFile));
}
export function inlinedFileFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InlinedFile$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InlinedFile' from JSON`);
}
/** @internal */
export const Files$inboundSchema = z
    .union([
    z.lazy(() => InlinedFile$inboundSchema),
    z.lazy(() => UploadedFile$inboundSchema),
]);
/** @internal */
export const Files$outboundSchema = z.union([
    z.lazy(() => InlinedFile$outboundSchema),
    z.lazy(() => UploadedFile$outboundSchema),
]);
export function filesToJSON(files) {
    return JSON.stringify(Files$outboundSchema.parse(files));
}
export function filesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Files$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Files' from JSON`);
}
/** @internal */
export const GitMetadata$inboundSchema = z.object({
    remoteUrl: z.string().optional(),
    commitAuthorName: z.string().optional(),
    commitAuthorEmail: z.string().optional(),
    commitMessage: z.string().optional(),
    commitRef: z.string().optional(),
    commitSha: z.string().optional(),
    dirty: z.boolean().optional(),
    ci: z.boolean().optional(),
    ciType: z.string().optional(),
    ciGitProviderUsername: z.string().optional(),
    ciGitRepoVisibility: z.string().optional(),
});
/** @internal */
export const GitMetadata$outboundSchema = z.object({
    remoteUrl: z.string().optional(),
    commitAuthorName: z.string().optional(),
    commitAuthorEmail: z.string().optional(),
    commitMessage: z.string().optional(),
    commitRef: z.string().optional(),
    commitSha: z.string().optional(),
    dirty: z.boolean().optional(),
    ci: z.boolean().optional(),
    ciType: z.string().optional(),
    ciGitProviderUsername: z.string().optional(),
    ciGitRepoVisibility: z.string().optional(),
});
export function gitMetadataToJSON(gitMetadata) {
    return JSON.stringify(GitMetadata$outboundSchema.parse(gitMetadata));
}
export function gitMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitMetadata' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type$outboundSchema = CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type$inboundSchema;
/** @internal */
export const GitSource7$inboundSchema = z.object({
    owner: z.string(),
    ref: z.string(),
    sha: z.string().optional(),
    slug: z.string(),
    type: CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type$inboundSchema,
});
/** @internal */
export const GitSource7$outboundSchema = z.object({
    owner: z.string(),
    ref: z.string(),
    sha: z.string().optional(),
    slug: z.string(),
    type: CreateDeploymentGitSourceDeploymentsRequestRequestBody7Type$outboundSchema,
});
export function gitSource7ToJSON(gitSource7) {
    return JSON.stringify(GitSource7$outboundSchema.parse(gitSource7));
}
export function gitSource7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource7' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestRequestBodyType$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsRequestRequestBodyType);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestRequestBodyType$outboundSchema = CreateDeploymentGitSourceDeploymentsRequestRequestBodyType$inboundSchema;
/** @internal */
export const GitSource6$inboundSchema = z.object({
    ref: z.string(),
    repoUuid: z.string(),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsRequestRequestBodyType$inboundSchema,
    workspaceUuid: z.string().optional(),
});
/** @internal */
export const GitSource6$outboundSchema = z.object({
    ref: z.string(),
    repoUuid: z.string(),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsRequestRequestBodyType$outboundSchema,
    workspaceUuid: z.string().optional(),
});
export function gitSource6ToJSON(gitSource6) {
    return JSON.stringify(GitSource6$outboundSchema.parse(gitSource6));
}
export function gitSource6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource6' from JSON`);
}
/** @internal */
export const ProjectId$inboundSchema = z.union([z.number(), z.string()]);
/** @internal */
export const ProjectId$outboundSchema = z.union([z.number(), z.string()]);
export function projectIdToJSON(projectId) {
    return JSON.stringify(ProjectId$outboundSchema.parse(projectId));
}
export function projectIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProjectId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProjectId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestType$inboundSchema = z
    .nativeEnum(CreateDeploymentGitSourceDeploymentsRequestType);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestType$outboundSchema = CreateDeploymentGitSourceDeploymentsRequestType$inboundSchema;
/** @internal */
export const GitSource5$inboundSchema = z.object({
    projectId: z.union([z.number(), z.string()]),
    ref: z.string(),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsRequestType$inboundSchema,
});
/** @internal */
export const GitSource5$outboundSchema = z.object({
    projectId: z.union([z.number(), z.string()]),
    ref: z.string(),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsRequestType$outboundSchema,
});
export function gitSource5ToJSON(gitSource5) {
    return JSON.stringify(GitSource5$outboundSchema.parse(gitSource5));
}
export function gitSource5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource5' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsType$inboundSchema = z
    .nativeEnum(CreateDeploymentGitSourceDeploymentsType);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsType$outboundSchema = CreateDeploymentGitSourceDeploymentsType$inboundSchema;
/** @internal */
export const GitSource4$inboundSchema = z.object({
    org: z.string(),
    ref: z.string(),
    repo: z.string(),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsType$inboundSchema,
});
/** @internal */
export const GitSource4$outboundSchema = z.object({
    org: z.string(),
    ref: z.string(),
    repo: z.string(),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsType$outboundSchema,
});
export function gitSource4ToJSON(gitSource4) {
    return JSON.stringify(GitSource4$outboundSchema.parse(gitSource4));
}
export function gitSource4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource4' from JSON`);
}
/** @internal */
export const GitSourceRepoId$inboundSchema = z.union([z.number(), z.string()]);
/** @internal */
export const GitSourceRepoId$outboundSchema = z.union([z.number(), z.string()]);
export function gitSourceRepoIdToJSON(gitSourceRepoId) {
    return JSON.stringify(GitSourceRepoId$outboundSchema.parse(gitSourceRepoId));
}
export function gitSourceRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSourceRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSourceRepoId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceType$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceType);
/** @internal */
export const CreateDeploymentGitSourceType$outboundSchema = CreateDeploymentGitSourceType$inboundSchema;
/** @internal */
export const GitSource3$inboundSchema = z.object({
    ref: z.string(),
    repoId: z.union([z.number(), z.string()]),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceType$inboundSchema,
});
/** @internal */
export const GitSource3$outboundSchema = z.object({
    ref: z.string(),
    repoId: z.union([z.number(), z.string()]),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceType$outboundSchema,
});
export function gitSource3ToJSON(gitSource3) {
    return JSON.stringify(GitSource3$outboundSchema.parse(gitSource3));
}
export function gitSource3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource3' from JSON`);
}
/** @internal */
export const GitSourceType$inboundSchema = z.nativeEnum(GitSourceType);
/** @internal */
export const GitSourceType$outboundSchema = GitSourceType$inboundSchema;
/** @internal */
export const GitSource2$inboundSchema = z.object({
    org: z.string(),
    ref: z.string(),
    repo: z.string(),
    sha: z.string().optional(),
    type: GitSourceType$inboundSchema,
});
/** @internal */
export const GitSource2$outboundSchema = z.object({
    org: z.string(),
    ref: z.string(),
    repo: z.string(),
    sha: z.string().optional(),
    type: GitSourceType$outboundSchema,
});
export function gitSource2ToJSON(gitSource2) {
    return JSON.stringify(GitSource2$outboundSchema.parse(gitSource2));
}
export function gitSource2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource2' from JSON`);
}
/** @internal */
export const RepoId$inboundSchema = z
    .union([z.number(), z.string()]);
/** @internal */
export const RepoId$outboundSchema = z.union([z.number(), z.string()]);
export function repoIdToJSON(repoId) {
    return JSON.stringify(RepoId$outboundSchema.parse(repoId));
}
export function repoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RepoId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type$outboundSchema = CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type$inboundSchema;
/** @internal */
export const GitSource1$inboundSchema = z.object({
    ref: z.string(),
    repoId: z.union([z.number(), z.string()]),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type$inboundSchema,
});
/** @internal */
export const GitSource1$outboundSchema = z.object({
    ref: z.string(),
    repoId: z.union([z.number(), z.string()]),
    sha: z.string().optional(),
    type: CreateDeploymentGitSourceDeploymentsRequestRequestBody1Type$outboundSchema,
});
export function gitSource1ToJSON(gitSource1) {
    return JSON.stringify(GitSource1$outboundSchema.parse(gitSource1));
}
export function gitSource1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource1' from JSON`);
}
/** @internal */
export const GitSource$inboundSchema = z.union([
    z.lazy(() => GitSource2$inboundSchema),
    z.lazy(() => GitSource4$inboundSchema),
    z.lazy(() => GitSource7$inboundSchema),
    z.lazy(() => GitSource1$inboundSchema),
    z.lazy(() => GitSource3$inboundSchema),
    z.lazy(() => GitSource5$inboundSchema),
    z.lazy(() => GitSource6$inboundSchema),
]);
/** @internal */
export const GitSource$outboundSchema = z.union([
    z.lazy(() => GitSource2$outboundSchema),
    z.lazy(() => GitSource4$outboundSchema),
    z.lazy(() => GitSource7$outboundSchema),
    z.lazy(() => GitSource1$outboundSchema),
    z.lazy(() => GitSource3$outboundSchema),
    z.lazy(() => GitSource5$outboundSchema),
    z.lazy(() => GitSource6$outboundSchema),
]);
export function gitSourceToJSON(gitSource) {
    return JSON.stringify(GitSource$outboundSchema.parse(gitSource));
}
export function gitSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSource' from JSON`);
}
/** @internal */
export const Framework$inboundSchema = z
    .nativeEnum(Framework);
/** @internal */
export const Framework$outboundSchema = Framework$inboundSchema;
/** @internal */
export const NodeVersion$inboundSchema = z
    .nativeEnum(NodeVersion);
/** @internal */
export const NodeVersion$outboundSchema = NodeVersion$inboundSchema;
/** @internal */
export const ProjectSettings$inboundSchema = z.object({
    buildCommand: z.nullable(z.string()).optional(),
    commandForIgnoringBuildStep: z.nullable(z.string()).optional(),
    devCommand: z.nullable(z.string()).optional(),
    framework: z.nullable(Framework$inboundSchema).optional(),
    installCommand: z.nullable(z.string()).optional(),
    nodeVersion: NodeVersion$inboundSchema.optional(),
    outputDirectory: z.nullable(z.string()).optional(),
    rootDirectory: z.nullable(z.string()).optional(),
    serverlessFunctionRegion: z.nullable(z.string()).optional(),
    skipGitConnectDuringLink: z.boolean().optional(),
    sourceFilesOutsideRootDirectory: z.boolean().optional(),
});
/** @internal */
export const ProjectSettings$outboundSchema = z.object({
    buildCommand: z.nullable(z.string()).optional(),
    commandForIgnoringBuildStep: z.nullable(z.string()).optional(),
    devCommand: z.nullable(z.string()).optional(),
    framework: z.nullable(Framework$outboundSchema).optional(),
    installCommand: z.nullable(z.string()).optional(),
    nodeVersion: NodeVersion$outboundSchema.optional(),
    outputDirectory: z.nullable(z.string()).optional(),
    rootDirectory: z.nullable(z.string()).optional(),
    serverlessFunctionRegion: z.nullable(z.string()).optional(),
    skipGitConnectDuringLink: z.boolean().optional(),
    sourceFilesOutsideRootDirectory: z.boolean().optional(),
});
export function projectSettingsToJSON(projectSettings) {
    return JSON.stringify(ProjectSettings$outboundSchema.parse(projectSettings));
}
export function projectSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProjectSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProjectSettings' from JSON`);
}
/** @internal */
export const CreateDeploymentRequestBody$inboundSchema = z.object({
    customEnvironmentSlugOrId: z.string().optional(),
    deploymentId: z.string().optional(),
    files: z.array(z.union([
        z.lazy(() => InlinedFile$inboundSchema),
        z.lazy(() => UploadedFile$inboundSchema),
    ])).optional(),
    gitMetadata: z.lazy(() => GitMetadata$inboundSchema).optional(),
    gitSource: z.union([
        z.lazy(() => GitSource2$inboundSchema),
        z.lazy(() => GitSource4$inboundSchema),
        z.lazy(() => GitSource7$inboundSchema),
        z.lazy(() => GitSource1$inboundSchema),
        z.lazy(() => GitSource3$inboundSchema),
        z.lazy(() => GitSource5$inboundSchema),
        z.lazy(() => GitSource6$inboundSchema),
    ]).optional(),
    meta: z.record(z.string()).optional(),
    monorepoManager: z.nullable(z.string()).optional(),
    name: z.string(),
    project: z.string().optional(),
    projectSettings: z.lazy(() => ProjectSettings$inboundSchema).optional(),
    target: z.string().optional(),
    withLatestCommit: z.boolean().optional(),
});
/** @internal */
export const CreateDeploymentRequestBody$outboundSchema = z.object({
    customEnvironmentSlugOrId: z.string().optional(),
    deploymentId: z.string().optional(),
    files: z.array(z.union([
        z.lazy(() => InlinedFile$outboundSchema),
        z.lazy(() => UploadedFile$outboundSchema),
    ])).optional(),
    gitMetadata: z.lazy(() => GitMetadata$outboundSchema).optional(),
    gitSource: z.union([
        z.lazy(() => GitSource2$outboundSchema),
        z.lazy(() => GitSource4$outboundSchema),
        z.lazy(() => GitSource7$outboundSchema),
        z.lazy(() => GitSource1$outboundSchema),
        z.lazy(() => GitSource3$outboundSchema),
        z.lazy(() => GitSource5$outboundSchema),
        z.lazy(() => GitSource6$outboundSchema),
    ]).optional(),
    meta: z.record(z.string()).optional(),
    monorepoManager: z.nullable(z.string()).optional(),
    name: z.string(),
    project: z.string().optional(),
    projectSettings: z.lazy(() => ProjectSettings$outboundSchema).optional(),
    target: z.string().optional(),
    withLatestCommit: z.boolean().optional(),
});
export function createDeploymentRequestBodyToJSON(createDeploymentRequestBody) {
    return JSON.stringify(CreateDeploymentRequestBody$outboundSchema.parse(createDeploymentRequestBody));
}
export function createDeploymentRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentRequestBody' from JSON`);
}
/** @internal */
export const CreateDeploymentRequest$inboundSchema = z.object({
    forceNew: ForceNew$inboundSchema.optional(),
    skipAutoDetectionConfirmation: SkipAutoDetectionConfirmation$inboundSchema
        .optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => CreateDeploymentRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const CreateDeploymentRequest$outboundSchema = z.object({
    forceNew: ForceNew$outboundSchema.optional(),
    skipAutoDetectionConfirmation: SkipAutoDetectionConfirmation$outboundSchema
        .optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => CreateDeploymentRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function createDeploymentRequestToJSON(createDeploymentRequest) {
    return JSON.stringify(CreateDeploymentRequest$outboundSchema.parse(createDeploymentRequest));
}
export function createDeploymentRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentRequest' from JSON`);
}
/** @internal */
export const AliasAssignedAt$inboundSchema = z.union([z.number(), z.boolean()]);
/** @internal */
export const AliasAssignedAt$outboundSchema = z.union([z.number(), z.boolean()]);
export function aliasAssignedAtToJSON(aliasAssignedAt) {
    return JSON.stringify(AliasAssignedAt$outboundSchema.parse(aliasAssignedAt));
}
export function aliasAssignedAtFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AliasAssignedAt$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AliasAssignedAt' from JSON`);
}
/** @internal */
export const Build$inboundSchema = z
    .object({
    env: z.array(z.string()),
});
/** @internal */
export const Build$outboundSchema = z.object({
    env: z.array(z.string()),
});
export function buildToJSON(build) {
    return JSON.stringify(Build$outboundSchema.parse(build));
}
export function buildFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Build$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Build' from JSON`);
}
/** @internal */
export const Builds$inboundSchema = z
    .object({
    use: z.string(),
    src: z.string().optional(),
    config: z.record(z.any()).optional(),
});
/** @internal */
export const Builds$outboundSchema = z.object({
    use: z.string(),
    src: z.string().optional(),
    config: z.record(z.any()).optional(),
});
export function buildsToJSON(builds) {
    return JSON.stringify(Builds$outboundSchema.parse(builds));
}
export function buildsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Builds$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Builds' from JSON`);
}
/** @internal */
export const CreateDeploymentFramework$inboundSchema = z.nativeEnum(CreateDeploymentFramework);
/** @internal */
export const CreateDeploymentFramework$outboundSchema = CreateDeploymentFramework$inboundSchema;
/** @internal */
export const CreateDeploymentSpeedInsights$inboundSchema = z.object({
    id: z.string(),
    enabledAt: z.number().optional(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    hasData: z.boolean().optional(),
    paidAt: z.number().optional(),
});
/** @internal */
export const CreateDeploymentSpeedInsights$outboundSchema = z.object({
    id: z.string(),
    enabledAt: z.number().optional(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    hasData: z.boolean().optional(),
    paidAt: z.number().optional(),
});
export function createDeploymentSpeedInsightsToJSON(createDeploymentSpeedInsights) {
    return JSON.stringify(CreateDeploymentSpeedInsights$outboundSchema.parse(createDeploymentSpeedInsights));
}
export function createDeploymentSpeedInsightsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentSpeedInsights$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentSpeedInsights' from JSON`);
}
/** @internal */
export const CreateDeploymentWebAnalytics$inboundSchema = z.object({
    id: z.string(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    enabledAt: z.number().optional(),
    hasData: z.boolean().optional(),
});
/** @internal */
export const CreateDeploymentWebAnalytics$outboundSchema = z.object({
    id: z.string(),
    disabledAt: z.number().optional(),
    canceledAt: z.number().optional(),
    enabledAt: z.number().optional(),
    hasData: z.boolean().optional(),
});
export function createDeploymentWebAnalyticsToJSON(createDeploymentWebAnalytics) {
    return JSON.stringify(CreateDeploymentWebAnalytics$outboundSchema.parse(createDeploymentWebAnalytics));
}
export function createDeploymentWebAnalyticsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentWebAnalytics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentWebAnalytics' from JSON`);
}
/** @internal */
export const CreateDeploymentProjectSettings$inboundSchema = z.object({
    buildCommand: z.nullable(z.string()).optional(),
    devCommand: z.nullable(z.string()).optional(),
    framework: z.nullable(CreateDeploymentFramework$inboundSchema).optional(),
    commandForIgnoringBuildStep: z.nullable(z.string()).optional(),
    installCommand: z.nullable(z.string()).optional(),
    outputDirectory: z.nullable(z.string()).optional(),
    speedInsights: z.lazy(() => CreateDeploymentSpeedInsights$inboundSchema)
        .optional(),
    webAnalytics: z.lazy(() => CreateDeploymentWebAnalytics$inboundSchema)
        .optional(),
});
/** @internal */
export const CreateDeploymentProjectSettings$outboundSchema = z.object({
    buildCommand: z.nullable(z.string()).optional(),
    devCommand: z.nullable(z.string()).optional(),
    framework: z.nullable(CreateDeploymentFramework$outboundSchema).optional(),
    commandForIgnoringBuildStep: z.nullable(z.string()).optional(),
    installCommand: z.nullable(z.string()).optional(),
    outputDirectory: z.nullable(z.string()).optional(),
    speedInsights: z.lazy(() => CreateDeploymentSpeedInsights$outboundSchema)
        .optional(),
    webAnalytics: z.lazy(() => CreateDeploymentWebAnalytics$outboundSchema)
        .optional(),
});
export function createDeploymentProjectSettingsToJSON(createDeploymentProjectSettings) {
    return JSON.stringify(CreateDeploymentProjectSettings$outboundSchema.parse(createDeploymentProjectSettings));
}
export function createDeploymentProjectSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentProjectSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentProjectSettings' from JSON`);
}
/** @internal */
export const CreateDeploymentDeploymentsStatus$inboundSchema = z.nativeEnum(CreateDeploymentDeploymentsStatus);
/** @internal */
export const CreateDeploymentDeploymentsStatus$outboundSchema = CreateDeploymentDeploymentsStatus$inboundSchema;
/** @internal */
export const Integrations$inboundSchema = z.object({
    status: CreateDeploymentDeploymentsStatus$inboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
    skippedAt: z.number().optional(),
    skippedBy: z.string().optional(),
});
/** @internal */
export const Integrations$outboundSchema = z.object({
    status: CreateDeploymentDeploymentsStatus$outboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
    skippedAt: z.number().optional(),
    skippedBy: z.string().optional(),
});
export function integrationsToJSON(integrations) {
    return JSON.stringify(Integrations$outboundSchema.parse(integrations));
}
export function integrationsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Integrations$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Integrations' from JSON`);
}
/** @internal */
export const Protocol$inboundSchema = z
    .nativeEnum(Protocol);
/** @internal */
export const Protocol$outboundSchema = Protocol$inboundSchema;
/** @internal */
export const RemotePatterns$inboundSchema = z.object({
    protocol: Protocol$inboundSchema.optional(),
    hostname: z.string(),
    port: z.string().optional(),
    pathname: z.string().optional(),
    search: z.string().optional(),
});
/** @internal */
export const RemotePatterns$outboundSchema = z.object({
    protocol: Protocol$outboundSchema.optional(),
    hostname: z.string(),
    port: z.string().optional(),
    pathname: z.string().optional(),
    search: z.string().optional(),
});
export function remotePatternsToJSON(remotePatterns) {
    return JSON.stringify(RemotePatterns$outboundSchema.parse(remotePatterns));
}
export function remotePatternsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemotePatterns$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemotePatterns' from JSON`);
}
/** @internal */
export const LocalPatterns$inboundSchema = z.object({
    pathname: z.string().optional(),
    search: z.string().optional(),
});
/** @internal */
export const LocalPatterns$outboundSchema = z.object({
    pathname: z.string().optional(),
    search: z.string().optional(),
});
export function localPatternsToJSON(localPatterns) {
    return JSON.stringify(LocalPatterns$outboundSchema.parse(localPatterns));
}
export function localPatternsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => LocalPatterns$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LocalPatterns' from JSON`);
}
/** @internal */
export const Formats$inboundSchema = z
    .nativeEnum(Formats);
/** @internal */
export const Formats$outboundSchema = Formats$inboundSchema;
/** @internal */
export const ContentDispositionType$inboundSchema = z.nativeEnum(ContentDispositionType);
/** @internal */
export const ContentDispositionType$outboundSchema = ContentDispositionType$inboundSchema;
/** @internal */
export const Images$inboundSchema = z
    .object({
    sizes: z.array(z.number()).optional(),
    qualities: z.array(z.number()).optional(),
    domains: z.array(z.string()).optional(),
    remotePatterns: z.array(z.lazy(() => RemotePatterns$inboundSchema))
        .optional(),
    localPatterns: z.array(z.lazy(() => LocalPatterns$inboundSchema))
        .optional(),
    minimumCacheTTL: z.number().optional(),
    formats: z.array(Formats$inboundSchema).optional(),
    dangerouslyAllowSVG: z.boolean().optional(),
    contentSecurityPolicy: z.string().optional(),
    contentDispositionType: ContentDispositionType$inboundSchema.optional(),
});
/** @internal */
export const Images$outboundSchema = z.object({
    sizes: z.array(z.number()).optional(),
    qualities: z.array(z.number()).optional(),
    domains: z.array(z.string()).optional(),
    remotePatterns: z.array(z.lazy(() => RemotePatterns$outboundSchema))
        .optional(),
    localPatterns: z.array(z.lazy(() => LocalPatterns$outboundSchema)).optional(),
    minimumCacheTTL: z.number().optional(),
    formats: z.array(Formats$outboundSchema).optional(),
    dangerouslyAllowSVG: z.boolean().optional(),
    contentSecurityPolicy: z.string().optional(),
    contentDispositionType: ContentDispositionType$outboundSchema.optional(),
});
export function imagesToJSON(images) {
    return JSON.stringify(Images$outboundSchema.parse(images));
}
export function imagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Images$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Images' from JSON`);
}
/** @internal */
export const Creator$inboundSchema = z.object({
    uid: z.string(),
    username: z.string().optional(),
    avatar: z.string().optional(),
});
/** @internal */
export const Creator$outboundSchema = z.object({
    uid: z.string(),
    username: z.string().optional(),
    avatar: z.string().optional(),
});
export function creatorToJSON(creator) {
    return JSON.stringify(Creator$outboundSchema.parse(creator));
}
export function creatorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Creator$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Creator' from JSON`);
}
/** @internal */
export const CreateDeploymentReadyState$inboundSchema = z.nativeEnum(CreateDeploymentReadyState);
/** @internal */
export const CreateDeploymentReadyState$outboundSchema = CreateDeploymentReadyState$inboundSchema;
/** @internal */
export const CreateDeploymentOutput$inboundSchema = z.object({
    path: z.string(),
    functionName: z.string(),
});
/** @internal */
export const CreateDeploymentOutput$outboundSchema = z.object({
    path: z.string(),
    functionName: z.string(),
});
export function createDeploymentOutputToJSON(createDeploymentOutput) {
    return JSON.stringify(CreateDeploymentOutput$outboundSchema.parse(createDeploymentOutput));
}
export function createDeploymentOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentOutput' from JSON`);
}
/** @internal */
export const Lambdas$inboundSchema = z.object({
    id: z.string(),
    createdAt: z.number().optional(),
    readyState: CreateDeploymentReadyState$inboundSchema.optional(),
    entrypoint: z.nullable(z.string()).optional(),
    readyStateAt: z.number().optional(),
    output: z.array(z.lazy(() => CreateDeploymentOutput$inboundSchema)),
});
/** @internal */
export const Lambdas$outboundSchema = z.object({
    id: z.string(),
    createdAt: z.number().optional(),
    readyState: CreateDeploymentReadyState$outboundSchema.optional(),
    entrypoint: z.nullable(z.string()).optional(),
    readyStateAt: z.number().optional(),
    output: z.array(z.lazy(() => CreateDeploymentOutput$outboundSchema)),
});
export function lambdasToJSON(lambdas) {
    return JSON.stringify(Lambdas$outboundSchema.parse(lambdas));
}
export function lambdasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Lambdas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Lambdas' from JSON`);
}
/** @internal */
export const CreateDeploymentStatus$inboundSchema = z.nativeEnum(CreateDeploymentStatus);
/** @internal */
export const CreateDeploymentStatus$outboundSchema = CreateDeploymentStatus$inboundSchema;
/** @internal */
export const CreateDeploymentTeam$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatar: z.string().optional(),
});
/** @internal */
export const CreateDeploymentTeam$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    slug: z.string(),
    avatar: z.string().optional(),
});
export function createDeploymentTeamToJSON(createDeploymentTeam) {
    return JSON.stringify(CreateDeploymentTeam$outboundSchema.parse(createDeploymentTeam));
}
export function createDeploymentTeamFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentTeam$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentTeam' from JSON`);
}
/** @internal */
export const CustomEnvironment2$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomEnvironment2$outboundSchema = z.object({
    id: z.string(),
});
export function customEnvironment2ToJSON(customEnvironment2) {
    return JSON.stringify(CustomEnvironment2$outboundSchema.parse(customEnvironment2));
}
export function customEnvironment2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomEnvironment2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomEnvironment2' from JSON`);
}
/** @internal */
export const CustomEnvironmentType$inboundSchema = z.nativeEnum(CustomEnvironmentType);
/** @internal */
export const CustomEnvironmentType$outboundSchema = CustomEnvironmentType$inboundSchema;
/** @internal */
export const CreateDeploymentCustomEnvironmentType$inboundSchema = z.nativeEnum(CreateDeploymentCustomEnvironmentType);
/** @internal */
export const CreateDeploymentCustomEnvironmentType$outboundSchema = CreateDeploymentCustomEnvironmentType$inboundSchema;
/** @internal */
export const CustomEnvironmentBranchMatcher$inboundSchema = z.object({
    type: CreateDeploymentCustomEnvironmentType$inboundSchema,
    pattern: z.string(),
});
/** @internal */
export const CustomEnvironmentBranchMatcher$outboundSchema = z.object({
    type: CreateDeploymentCustomEnvironmentType$outboundSchema,
    pattern: z.string(),
});
export function customEnvironmentBranchMatcherToJSON(customEnvironmentBranchMatcher) {
    return JSON.stringify(CustomEnvironmentBranchMatcher$outboundSchema.parse(customEnvironmentBranchMatcher));
}
export function customEnvironmentBranchMatcherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomEnvironmentBranchMatcher$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomEnvironmentBranchMatcher' from JSON`);
}
/** @internal */
export const CustomEnvironmentVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const CustomEnvironmentVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
export function customEnvironmentVerificationToJSON(customEnvironmentVerification) {
    return JSON.stringify(CustomEnvironmentVerification$outboundSchema.parse(customEnvironmentVerification));
}
export function customEnvironmentVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomEnvironmentVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomEnvironmentVerification' from JSON`);
}
/** @internal */
export const CustomEnvironmentDomains$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => CustomEnvironmentVerification$inboundSchema)).optional(),
});
/** @internal */
export const CustomEnvironmentDomains$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => CustomEnvironmentVerification$outboundSchema)).optional(),
});
export function customEnvironmentDomainsToJSON(customEnvironmentDomains) {
    return JSON.stringify(CustomEnvironmentDomains$outboundSchema.parse(customEnvironmentDomains));
}
export function customEnvironmentDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomEnvironmentDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomEnvironmentDomains' from JSON`);
}
/** @internal */
export const CustomEnvironment1$inboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: CustomEnvironmentType$inboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => CustomEnvironmentBranchMatcher$inboundSchema)
        .optional(),
    domains: z.array(z.lazy(() => CustomEnvironmentDomains$inboundSchema))
        .optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const CustomEnvironment1$outboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: CustomEnvironmentType$outboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => CustomEnvironmentBranchMatcher$outboundSchema)
        .optional(),
    domains: z.array(z.lazy(() => CustomEnvironmentDomains$outboundSchema))
        .optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
export function customEnvironment1ToJSON(customEnvironment1) {
    return JSON.stringify(CustomEnvironment1$outboundSchema.parse(customEnvironment1));
}
export function customEnvironment1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomEnvironment1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomEnvironment1' from JSON`);
}
/** @internal */
export const CustomEnvironment$inboundSchema = z.union([
    z.lazy(() => CustomEnvironment1$inboundSchema),
    z.lazy(() => CustomEnvironment2$inboundSchema),
]);
/** @internal */
export const CustomEnvironment$outboundSchema = z.union([
    z.lazy(() => CustomEnvironment1$outboundSchema),
    z.lazy(() => CustomEnvironment2$outboundSchema),
]);
export function customEnvironmentToJSON(customEnvironment) {
    return JSON.stringify(CustomEnvironment$outboundSchema.parse(customEnvironment));
}
export function customEnvironmentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomEnvironment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomEnvironment' from JSON`);
}
/** @internal */
export const OomReport$inboundSchema = z
    .nativeEnum(OomReport);
/** @internal */
export const OomReport$outboundSchema = OomReport$inboundSchema;
/** @internal */
export const AliasWarning$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    link: z.string().optional(),
    action: z.string().optional(),
});
/** @internal */
export const AliasWarning$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    link: z.string().optional(),
    action: z.string().optional(),
});
export function aliasWarningToJSON(aliasWarning) {
    return JSON.stringify(AliasWarning$outboundSchema.parse(aliasWarning));
}
export function aliasWarningFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AliasWarning$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AliasWarning' from JSON`);
}
/** @internal */
export const ReadyState$inboundSchema = z
    .nativeEnum(ReadyState);
/** @internal */
export const ReadyState$outboundSchema = ReadyState$inboundSchema;
/** @internal */
export const CreateDeploymentType$inboundSchema = z.nativeEnum(CreateDeploymentType);
/** @internal */
export const CreateDeploymentType$outboundSchema = CreateDeploymentType$inboundSchema;
/** @internal */
export const AliasError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
/** @internal */
export const AliasError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
});
export function aliasErrorToJSON(aliasError) {
    return JSON.stringify(AliasError$outboundSchema.parse(aliasError));
}
export function aliasErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AliasError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AliasError' from JSON`);
}
/** @internal */
export const ChecksState$inboundSchema = z
    .nativeEnum(ChecksState);
/** @internal */
export const ChecksState$outboundSchema = ChecksState$inboundSchema;
/** @internal */
export const ChecksConclusion$inboundSchema = z.nativeEnum(ChecksConclusion);
/** @internal */
export const ChecksConclusion$outboundSchema = ChecksConclusion$inboundSchema;
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource15$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    owner: z.string().optional(),
    slug: z.string().optional(),
    workspaceUuid: z.string(),
    repoUuid: z.string(),
});
/** @internal */
export const CreateDeploymentGitSource15$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody15Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    owner: z.string().optional(),
    slug: z.string().optional(),
    workspaceUuid: z.string(),
    repoUuid: z.string(),
});
export function createDeploymentGitSource15ToJSON(createDeploymentGitSource15) {
    return JSON.stringify(CreateDeploymentGitSource15$outboundSchema.parse(createDeploymentGitSource15));
}
export function createDeploymentGitSource15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource15' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource14$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    projectId: z.number(),
});
/** @internal */
export const CreateDeploymentGitSource14$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody14Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    projectId: z.number(),
});
export function createDeploymentGitSource14ToJSON(createDeploymentGitSource14) {
    return JSON.stringify(CreateDeploymentGitSource14$outboundSchema.parse(createDeploymentGitSource14));
}
export function createDeploymentGitSource14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource14' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource13$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
/** @internal */
export const CreateDeploymentGitSource13$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody13Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
export function createDeploymentGitSource13ToJSON(createDeploymentGitSource13) {
    return JSON.stringify(CreateDeploymentGitSource13$outboundSchema.parse(createDeploymentGitSource13));
}
export function createDeploymentGitSource13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource13' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource12$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$inboundSchema,
    host: z.string(),
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
/** @internal */
export const CreateDeploymentGitSource12$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody12Type$outboundSchema,
    host: z.string(),
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
export function createDeploymentGitSource12ToJSON(createDeploymentGitSource12) {
    return JSON.stringify(CreateDeploymentGitSource12$outboundSchema.parse(createDeploymentGitSource12));
}
export function createDeploymentGitSource12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource12' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource11$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
/** @internal */
export const CreateDeploymentGitSource11$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody11Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    repoId: z.number(),
    org: z.string().optional(),
    repo: z.string().optional(),
});
export function createDeploymentGitSource11ToJSON(createDeploymentGitSource11) {
    return JSON.stringify(CreateDeploymentGitSource11$outboundSchema.parse(createDeploymentGitSource11));
}
export function createDeploymentGitSource11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource11' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource10$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$inboundSchema,
    ref: z.string(),
    sha: z.string(),
    gitUrl: z.string(),
});
/** @internal */
export const CreateDeploymentGitSource10$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody10Type$outboundSchema,
    ref: z.string(),
    sha: z.string(),
    gitUrl: z.string(),
});
export function createDeploymentGitSource10ToJSON(createDeploymentGitSource10) {
    return JSON.stringify(CreateDeploymentGitSource10$outboundSchema.parse(createDeploymentGitSource10));
}
export function createDeploymentGitSource10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource10' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource9$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$inboundSchema,
    owner: z.string(),
    slug: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource9$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody9Type$outboundSchema,
    owner: z.string(),
    slug: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource9ToJSON(createDeploymentGitSource9) {
    return JSON.stringify(CreateDeploymentGitSource9$outboundSchema.parse(createDeploymentGitSource9));
}
export function createDeploymentGitSource9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource9' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource8$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$inboundSchema,
    workspaceUuid: z.string().optional(),
    repoUuid: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource8$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody8Type$outboundSchema,
    workspaceUuid: z.string().optional(),
    repoUuid: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource8ToJSON(createDeploymentGitSource8) {
    return JSON.stringify(CreateDeploymentGitSource8$outboundSchema.parse(createDeploymentGitSource8));
}
export function createDeploymentGitSource8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource8' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$inboundSchema;
/** @internal */
export const GitSourceProjectId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const GitSourceProjectId$outboundSchema = z.union([z.string(), z.number()]);
export function gitSourceProjectIdToJSON(gitSourceProjectId) {
    return JSON.stringify(GitSourceProjectId$outboundSchema.parse(gitSourceProjectId));
}
export function gitSourceProjectIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitSourceProjectId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitSourceProjectId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSource7$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$inboundSchema,
    projectId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource7$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody7Type$outboundSchema,
    projectId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource7ToJSON(createDeploymentGitSource7) {
    return JSON.stringify(CreateDeploymentGitSource7$outboundSchema.parse(createDeploymentGitSource7));
}
export function createDeploymentGitSource7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource7' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource6$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type$inboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource6$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody6Type$outboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource6ToJSON(createDeploymentGitSource6) {
    return JSON.stringify(CreateDeploymentGitSource6$outboundSchema.parse(createDeploymentGitSource6));
}
export function createDeploymentGitSource6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource6' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponseRepoId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponseRepoId$outboundSchema = z.union([z.string(), z.number()]);
export function createDeploymentGitSourceDeploymentsResponseRepoIdToJSON(createDeploymentGitSourceDeploymentsResponseRepoId) {
    return JSON.stringify(CreateDeploymentGitSourceDeploymentsResponseRepoId$outboundSchema.parse(createDeploymentGitSourceDeploymentsResponseRepoId));
}
export function createDeploymentGitSourceDeploymentsResponseRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSourceDeploymentsResponseRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSourceDeploymentsResponseRepoId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSource5$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type$inboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource5$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBody5Type$outboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource5ToJSON(createDeploymentGitSource5) {
    return JSON.stringify(CreateDeploymentGitSource5$outboundSchema.parse(createDeploymentGitSource5));
}
export function createDeploymentGitSource5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource5' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource4$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$inboundSchema,
    host: z.string(),
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource4$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONResponseBodyType$outboundSchema,
    host: z.string(),
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource4ToJSON(createDeploymentGitSource4) {
    return JSON.stringify(CreateDeploymentGitSource4$outboundSchema.parse(createDeploymentGitSource4));
}
export function createDeploymentGitSource4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource4' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRepoId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsRepoId$outboundSchema = z.union([z.string(), z.number()]);
export function createDeploymentGitSourceDeploymentsRepoIdToJSON(createDeploymentGitSourceDeploymentsRepoId) {
    return JSON.stringify(CreateDeploymentGitSourceDeploymentsRepoId$outboundSchema.parse(createDeploymentGitSourceDeploymentsRepoId));
}
export function createDeploymentGitSourceDeploymentsRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSourceDeploymentsRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSourceDeploymentsRepoId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSource3$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$inboundSchema,
    host: z.string(),
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource3$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200ApplicationJSONType$outboundSchema,
    host: z.string(),
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource3ToJSON(createDeploymentGitSource3) {
    return JSON.stringify(CreateDeploymentGitSource3$outboundSchema.parse(createDeploymentGitSource3));
}
export function createDeploymentGitSource3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource3' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200Type$inboundSchema = z.nativeEnum(CreateDeploymentGitSourceDeploymentsResponse200Type);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponse200Type$outboundSchema = CreateDeploymentGitSourceDeploymentsResponse200Type$inboundSchema;
/** @internal */
export const CreateDeploymentGitSource2$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200Type$inboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource2$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponse200Type$outboundSchema,
    org: z.string(),
    repo: z.string(),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource2ToJSON(createDeploymentGitSource2) {
    return JSON.stringify(CreateDeploymentGitSource2$outboundSchema.parse(createDeploymentGitSource2));
}
export function createDeploymentGitSource2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource2' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponseType$inboundSchema = z
    .nativeEnum(CreateDeploymentGitSourceDeploymentsResponseType);
/** @internal */
export const CreateDeploymentGitSourceDeploymentsResponseType$outboundSchema = CreateDeploymentGitSourceDeploymentsResponseType$inboundSchema;
/** @internal */
export const CreateDeploymentGitSourceRepoId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CreateDeploymentGitSourceRepoId$outboundSchema = z.union([z.string(), z.number()]);
export function createDeploymentGitSourceRepoIdToJSON(createDeploymentGitSourceRepoId) {
    return JSON.stringify(CreateDeploymentGitSourceRepoId$outboundSchema.parse(createDeploymentGitSourceRepoId));
}
export function createDeploymentGitSourceRepoIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSourceRepoId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSourceRepoId' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSource1$inboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponseType$inboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
/** @internal */
export const CreateDeploymentGitSource1$outboundSchema = z.object({
    type: CreateDeploymentGitSourceDeploymentsResponseType$outboundSchema,
    repoId: z.union([z.string(), z.number()]),
    ref: z.nullable(z.string()).optional(),
    sha: z.string().optional(),
    prId: z.nullable(z.number()).optional(),
});
export function createDeploymentGitSource1ToJSON(createDeploymentGitSource1) {
    return JSON.stringify(CreateDeploymentGitSource1$outboundSchema.parse(createDeploymentGitSource1));
}
export function createDeploymentGitSource1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource1' from JSON`);
}
/** @internal */
export const CreateDeploymentGitSource$inboundSchema = z.union([
    z.lazy(() => CreateDeploymentGitSource12$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource15$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource4$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource10$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource11$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource13$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource14$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource2$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource3$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource6$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource9$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource1$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource5$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource7$inboundSchema),
    z.lazy(() => CreateDeploymentGitSource8$inboundSchema),
]);
/** @internal */
export const CreateDeploymentGitSource$outboundSchema = z.union([
    z.lazy(() => CreateDeploymentGitSource12$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource15$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource4$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource10$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource11$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource13$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource14$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource2$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource3$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource6$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource9$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource1$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource5$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource7$outboundSchema),
    z.lazy(() => CreateDeploymentGitSource8$outboundSchema),
]);
export function createDeploymentGitSourceToJSON(createDeploymentGitSource) {
    return JSON.stringify(CreateDeploymentGitSource$outboundSchema.parse(createDeploymentGitSource));
}
export function createDeploymentGitSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentGitSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentGitSource' from JSON`);
}
/** @internal */
export const CreateDeploymentNodeVersion$inboundSchema = z.nativeEnum(CreateDeploymentNodeVersion);
/** @internal */
export const CreateDeploymentNodeVersion$outboundSchema = CreateDeploymentNodeVersion$inboundSchema;
/** @internal */
export const CreateDeploymentProject$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    framework: z.nullable(z.string()).optional(),
});
/** @internal */
export const CreateDeploymentProject$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    framework: z.nullable(z.string()).optional(),
});
export function createDeploymentProjectToJSON(createDeploymentProject) {
    return JSON.stringify(CreateDeploymentProject$outboundSchema.parse(createDeploymentProject));
}
export function createDeploymentProjectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentProject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentProject' from JSON`);
}
/** @internal */
export const ReadySubstate$inboundSchema = z.nativeEnum(ReadySubstate);
/** @internal */
export const ReadySubstate$outboundSchema = ReadySubstate$inboundSchema;
/** @internal */
export const CreateDeploymentSource$inboundSchema = z.nativeEnum(CreateDeploymentSource);
/** @internal */
export const CreateDeploymentSource$outboundSchema = CreateDeploymentSource$inboundSchema;
/** @internal */
export const CreateDeploymentTarget$inboundSchema = z.nativeEnum(CreateDeploymentTarget);
/** @internal */
export const CreateDeploymentTarget$outboundSchema = CreateDeploymentTarget$inboundSchema;
/** @internal */
export const OidcTokenClaims$inboundSchema = z.object({
    iss: z.string(),
    sub: z.string(),
    scope: z.string(),
    aud: z.string(),
    owner: z.string(),
    owner_id: z.string(),
    project: z.string(),
    project_id: z.string(),
    environment: z.string(),
    plan: z.string(),
}).transform((v) => {
    return remap$(v, {
        "owner_id": "ownerId",
        "project_id": "projectId",
    });
});
/** @internal */
export const OidcTokenClaims$outboundSchema = z.object({
    iss: z.string(),
    sub: z.string(),
    scope: z.string(),
    aud: z.string(),
    owner: z.string(),
    ownerId: z.string(),
    project: z.string(),
    projectId: z.string(),
    environment: z.string(),
    plan: z.string(),
}).transform((v) => {
    return remap$(v, {
        ownerId: "owner_id",
        projectId: "project_id",
    });
});
export function oidcTokenClaimsToJSON(oidcTokenClaims) {
    return JSON.stringify(OidcTokenClaims$outboundSchema.parse(oidcTokenClaims));
}
export function oidcTokenClaimsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OidcTokenClaims$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OidcTokenClaims' from JSON`);
}
/** @internal */
export const CreateDeploymentPlan$inboundSchema = z.nativeEnum(CreateDeploymentPlan);
/** @internal */
export const CreateDeploymentPlan$outboundSchema = CreateDeploymentPlan$inboundSchema;
/** @internal */
export const CreateDeploymentCrons$inboundSchema = z.object({
    schedule: z.string(),
    path: z.string(),
});
/** @internal */
export const CreateDeploymentCrons$outboundSchema = z.object({
    schedule: z.string(),
    path: z.string(),
});
export function createDeploymentCronsToJSON(createDeploymentCrons) {
    return JSON.stringify(CreateDeploymentCrons$outboundSchema.parse(createDeploymentCrons));
}
export function createDeploymentCronsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentCrons$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentCrons' from JSON`);
}
/** @internal */
export const Architecture$inboundSchema = z.nativeEnum(Architecture);
/** @internal */
export const Architecture$outboundSchema = Architecture$inboundSchema;
/** @internal */
export const CreateDeploymentDeploymentsType$inboundSchema = z.nativeEnum(CreateDeploymentDeploymentsType);
/** @internal */
export const CreateDeploymentDeploymentsType$outboundSchema = CreateDeploymentDeploymentsType$inboundSchema;
/** @internal */
export const ExperimentalTriggers$inboundSchema = z.object({
    type: CreateDeploymentDeploymentsType$inboundSchema,
    topic: z.string(),
    consumer: z.string(),
    maxDeliveries: z.number().optional(),
    retryAfterSeconds: z.number().optional(),
    initialDelaySeconds: z.number().optional(),
});
/** @internal */
export const ExperimentalTriggers$outboundSchema = z.object({
    type: CreateDeploymentDeploymentsType$outboundSchema,
    topic: z.string(),
    consumer: z.string(),
    maxDeliveries: z.number().optional(),
    retryAfterSeconds: z.number().optional(),
    initialDelaySeconds: z.number().optional(),
});
export function experimentalTriggersToJSON(experimentalTriggers) {
    return JSON.stringify(ExperimentalTriggers$outboundSchema.parse(experimentalTriggers));
}
export function experimentalTriggersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ExperimentalTriggers$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ExperimentalTriggers' from JSON`);
}
/** @internal */
export const Functions$inboundSchema = z.object({
    architecture: Architecture$inboundSchema.optional(),
    memory: z.number().optional(),
    maxDuration: z.number().optional(),
    runtime: z.string().optional(),
    includeFiles: z.string().optional(),
    excludeFiles: z.string().optional(),
    experimentalTriggers: z.array(z.lazy(() => ExperimentalTriggers$inboundSchema)).optional(),
    supportsCancellation: z.boolean().optional(),
});
/** @internal */
export const Functions$outboundSchema = z.object({
    architecture: Architecture$outboundSchema.optional(),
    memory: z.number().optional(),
    maxDuration: z.number().optional(),
    runtime: z.string().optional(),
    includeFiles: z.string().optional(),
    excludeFiles: z.string().optional(),
    experimentalTriggers: z.array(z.lazy(() => ExperimentalTriggers$outboundSchema)).optional(),
    supportsCancellation: z.boolean().optional(),
});
export function functionsToJSON(functions) {
    return JSON.stringify(Functions$outboundSchema.parse(functions));
}
export function functionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Functions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Functions' from JSON`);
}
/** @internal */
export const Routes3$inboundSchema = z.object({
    src: z.string(),
    continue: z.boolean(),
    middleware: z.number(),
});
/** @internal */
export const Routes3$outboundSchema = z.object({
    src: z.string(),
    continue: z.boolean(),
    middleware: z.number(),
});
export function routes3ToJSON(routes3) {
    return JSON.stringify(Routes3$outboundSchema.parse(routes3));
}
export function routes3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Routes3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Routes3' from JSON`);
}
/** @internal */
export const Handle$inboundSchema = z
    .nativeEnum(Handle);
/** @internal */
export const Handle$outboundSchema = Handle$inboundSchema;
/** @internal */
export const Routes2$inboundSchema = z.object({
    handle: Handle$inboundSchema,
    src: z.string().optional(),
    dest: z.string().optional(),
    status: z.number().optional(),
});
/** @internal */
export const Routes2$outboundSchema = z.object({
    handle: Handle$outboundSchema,
    src: z.string().optional(),
    dest: z.string().optional(),
    status: z.number().optional(),
});
export function routes2ToJSON(routes2) {
    return JSON.stringify(Routes2$outboundSchema.parse(routes2));
}
export function routes2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Routes2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Routes2' from JSON`);
}
/** @internal */
export const CreateDeploymentHasDeploymentsType$inboundSchema = z.nativeEnum(CreateDeploymentHasDeploymentsType);
/** @internal */
export const CreateDeploymentHasDeploymentsType$outboundSchema = CreateDeploymentHasDeploymentsType$inboundSchema;
/** @internal */
export const ValueEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const ValueEq$outboundSchema = z.union([z.string(), z.number()]);
export function valueEqToJSON(valueEq) {
    return JSON.stringify(ValueEq$outboundSchema.parse(valueEq));
}
export function valueEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ValueEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValueEq' from JSON`);
}
/** @internal */
export const CreateDeploymentValue2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CreateDeploymentValue2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function createDeploymentValue2ToJSON(createDeploymentValue2) {
    return JSON.stringify(CreateDeploymentValue2$outboundSchema.parse(createDeploymentValue2));
}
export function createDeploymentValue2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentValue2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentValue2' from JSON`);
}
/** @internal */
export const CreateDeploymentHasDeploymentsValue$inboundSchema = z.union([z.string(), z.lazy(() => CreateDeploymentValue2$inboundSchema)]);
/** @internal */
export const CreateDeploymentHasDeploymentsValue$outboundSchema = z.union([z.string(), z.lazy(() => CreateDeploymentValue2$outboundSchema)]);
export function createDeploymentHasDeploymentsValueToJSON(createDeploymentHasDeploymentsValue) {
    return JSON.stringify(CreateDeploymentHasDeploymentsValue$outboundSchema.parse(createDeploymentHasDeploymentsValue));
}
export function createDeploymentHasDeploymentsValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentHasDeploymentsValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentHasDeploymentsValue' from JSON`);
}
/** @internal */
export const CreateDeploymentHas2$inboundSchema = z.object({
    type: CreateDeploymentHasDeploymentsType$inboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValue2$inboundSchema),
    ]).optional(),
});
/** @internal */
export const CreateDeploymentHas2$outboundSchema = z.object({
    type: CreateDeploymentHasDeploymentsType$outboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValue2$outboundSchema),
    ]).optional(),
});
export function createDeploymentHas2ToJSON(createDeploymentHas2) {
    return JSON.stringify(CreateDeploymentHas2$outboundSchema.parse(createDeploymentHas2));
}
export function createDeploymentHas2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentHas2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentHas2' from JSON`);
}
/** @internal */
export const CreateDeploymentHasType$inboundSchema = z.nativeEnum(CreateDeploymentHasType);
/** @internal */
export const CreateDeploymentHasType$outboundSchema = CreateDeploymentHasType$inboundSchema;
/** @internal */
export const Eq$inboundSchema = z.union([
    z.string(),
    z.number(),
]);
/** @internal */
export const Eq$outboundSchema = z
    .union([z.string(), z.number()]);
export function eqToJSON(eq) {
    return JSON.stringify(Eq$outboundSchema.parse(eq));
}
export function eqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Eq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Eq' from JSON`);
}
/** @internal */
export const CreateDeploymentValueDeploymentsResponse2002$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CreateDeploymentValueDeploymentsResponse2002$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function createDeploymentValueDeploymentsResponse2002ToJSON(createDeploymentValueDeploymentsResponse2002) {
    return JSON.stringify(CreateDeploymentValueDeploymentsResponse2002$outboundSchema.parse(createDeploymentValueDeploymentsResponse2002));
}
export function createDeploymentValueDeploymentsResponse2002FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentValueDeploymentsResponse2002$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentValueDeploymentsResponse2002' from JSON`);
}
/** @internal */
export const CreateDeploymentHasValue$inboundSchema = z.union([
    z.string(),
    z.lazy(() => CreateDeploymentValueDeploymentsResponse2002$inboundSchema),
]);
/** @internal */
export const CreateDeploymentHasValue$outboundSchema = z.union([
    z.string(),
    z.lazy(() => CreateDeploymentValueDeploymentsResponse2002$outboundSchema),
]);
export function createDeploymentHasValueToJSON(createDeploymentHasValue) {
    return JSON.stringify(CreateDeploymentHasValue$outboundSchema.parse(createDeploymentHasValue));
}
export function createDeploymentHasValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentHasValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentHasValue' from JSON`);
}
/** @internal */
export const CreateDeploymentHas1$inboundSchema = z.object({
    type: CreateDeploymentHasType$inboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValueDeploymentsResponse2002$inboundSchema),
    ]),
});
/** @internal */
export const CreateDeploymentHas1$outboundSchema = z.object({
    type: CreateDeploymentHasType$outboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValueDeploymentsResponse2002$outboundSchema),
    ]),
});
export function createDeploymentHas1ToJSON(createDeploymentHas1) {
    return JSON.stringify(CreateDeploymentHas1$outboundSchema.parse(createDeploymentHas1));
}
export function createDeploymentHas1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentHas1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentHas1' from JSON`);
}
/** @internal */
export const RoutesHas$inboundSchema = z.union([
    z.lazy(() => CreateDeploymentHas1$inboundSchema),
    z.lazy(() => CreateDeploymentHas2$inboundSchema),
]);
/** @internal */
export const RoutesHas$outboundSchema = z.union([
    z.lazy(() => CreateDeploymentHas1$outboundSchema),
    z.lazy(() => CreateDeploymentHas2$outboundSchema),
]);
export function routesHasToJSON(routesHas) {
    return JSON.stringify(RoutesHas$outboundSchema.parse(routesHas));
}
export function routesHasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesHas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesHas' from JSON`);
}
/** @internal */
export const CreateDeploymentMissingType$inboundSchema = z.nativeEnum(CreateDeploymentMissingType);
/** @internal */
export const CreateDeploymentMissingType$outboundSchema = CreateDeploymentMissingType$inboundSchema;
/** @internal */
export const CreateDeploymentValueDeploymentsEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CreateDeploymentValueDeploymentsEq$outboundSchema = z.union([z.string(), z.number()]);
export function createDeploymentValueDeploymentsEqToJSON(createDeploymentValueDeploymentsEq) {
    return JSON.stringify(CreateDeploymentValueDeploymentsEq$outboundSchema.parse(createDeploymentValueDeploymentsEq));
}
export function createDeploymentValueDeploymentsEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentValueDeploymentsEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentValueDeploymentsEq' from JSON`);
}
/** @internal */
export const CreateDeploymentValueDeploymentsResponse2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CreateDeploymentValueDeploymentsResponse2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function createDeploymentValueDeploymentsResponse2ToJSON(createDeploymentValueDeploymentsResponse2) {
    return JSON.stringify(CreateDeploymentValueDeploymentsResponse2$outboundSchema.parse(createDeploymentValueDeploymentsResponse2));
}
export function createDeploymentValueDeploymentsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentValueDeploymentsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentValueDeploymentsResponse2' from JSON`);
}
/** @internal */
export const CreateDeploymentMissingValue$inboundSchema = z.union([
    z.string(),
    z.lazy(() => CreateDeploymentValueDeploymentsResponse2$inboundSchema),
]);
/** @internal */
export const CreateDeploymentMissingValue$outboundSchema = z.union([
    z.string(),
    z.lazy(() => CreateDeploymentValueDeploymentsResponse2$outboundSchema),
]);
export function createDeploymentMissingValueToJSON(createDeploymentMissingValue) {
    return JSON.stringify(CreateDeploymentMissingValue$outboundSchema.parse(createDeploymentMissingValue));
}
export function createDeploymentMissingValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentMissingValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentMissingValue' from JSON`);
}
/** @internal */
export const Missing2$inboundSchema = z.object({
    type: CreateDeploymentMissingType$inboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValueDeploymentsResponse2$inboundSchema),
    ]).optional(),
});
/** @internal */
export const Missing2$outboundSchema = z.object({
    type: CreateDeploymentMissingType$outboundSchema,
    key: z.string(),
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValueDeploymentsResponse2$outboundSchema),
    ]).optional(),
});
export function missing2ToJSON(missing2) {
    return JSON.stringify(Missing2$outboundSchema.parse(missing2));
}
export function missing2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Missing2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Missing2' from JSON`);
}
/** @internal */
export const MissingType$inboundSchema = z
    .nativeEnum(MissingType);
/** @internal */
export const MissingType$outboundSchema = MissingType$inboundSchema;
/** @internal */
export const CreateDeploymentValueEq$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const CreateDeploymentValueEq$outboundSchema = z.union([z.string(), z.number()]);
export function createDeploymentValueEqToJSON(createDeploymentValueEq) {
    return JSON.stringify(CreateDeploymentValueEq$outboundSchema.parse(createDeploymentValueEq));
}
export function createDeploymentValueEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentValueEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentValueEq' from JSON`);
}
/** @internal */
export const CreateDeploymentValueDeployments2$inboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const CreateDeploymentValueDeployments2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    re: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function createDeploymentValueDeployments2ToJSON(createDeploymentValueDeployments2) {
    return JSON.stringify(CreateDeploymentValueDeployments2$outboundSchema.parse(createDeploymentValueDeployments2));
}
export function createDeploymentValueDeployments2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentValueDeployments2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentValueDeployments2' from JSON`);
}
/** @internal */
export const MissingValue$inboundSchema = z.union([
    z.string(),
    z.lazy(() => CreateDeploymentValueDeployments2$inboundSchema),
]);
/** @internal */
export const MissingValue$outboundSchema = z.union([
    z.string(),
    z.lazy(() => CreateDeploymentValueDeployments2$outboundSchema),
]);
export function missingValueToJSON(missingValue) {
    return JSON.stringify(MissingValue$outboundSchema.parse(missingValue));
}
export function missingValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MissingValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MissingValue' from JSON`);
}
/** @internal */
export const Missing1$inboundSchema = z.object({
    type: MissingType$inboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValueDeployments2$inboundSchema),
    ]),
});
/** @internal */
export const Missing1$outboundSchema = z.object({
    type: MissingType$outboundSchema,
    value: z.union([
        z.string(),
        z.lazy(() => CreateDeploymentValueDeployments2$outboundSchema),
    ]),
});
export function missing1ToJSON(missing1) {
    return JSON.stringify(Missing1$outboundSchema.parse(missing1));
}
export function missing1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Missing1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Missing1' from JSON`);
}
/** @internal */
export const Missing$inboundSchema = z.union([
    z.lazy(() => Missing1$inboundSchema),
    z.lazy(() => Missing2$inboundSchema),
]);
/** @internal */
export const Missing$outboundSchema = z.union([
    z.lazy(() => Missing1$outboundSchema),
    z.lazy(() => Missing2$outboundSchema),
]);
export function missingToJSON(missing) {
    return JSON.stringify(Missing$outboundSchema.parse(missing));
}
export function missingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Missing$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Missing' from JSON`);
}
/** @internal */
export const RoutesAction$inboundSchema = z.nativeEnum(RoutesAction);
/** @internal */
export const RoutesAction$outboundSchema = RoutesAction$inboundSchema;
/** @internal */
export const RoutesMitigate$inboundSchema = z.object({
    action: RoutesAction$inboundSchema,
});
/** @internal */
export const RoutesMitigate$outboundSchema = z.object({
    action: RoutesAction$outboundSchema,
});
export function routesMitigateToJSON(routesMitigate) {
    return JSON.stringify(RoutesMitigate$outboundSchema.parse(routesMitigate));
}
export function routesMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesMitigate' from JSON`);
}
/** @internal */
export const RoutesType$inboundSchema = z
    .nativeEnum(RoutesType);
/** @internal */
export const RoutesType$outboundSchema = RoutesType$inboundSchema;
/** @internal */
export const CreateDeploymentRoutesOp$inboundSchema = z.nativeEnum(CreateDeploymentRoutesOp);
/** @internal */
export const CreateDeploymentRoutesOp$outboundSchema = CreateDeploymentRoutesOp$inboundSchema;
/** @internal */
export const KeyEq$inboundSchema = z
    .union([z.string(), z.number()]);
/** @internal */
export const KeyEq$outboundSchema = z.union([z.string(), z.number()]);
export function keyEqToJSON(keyEq) {
    return JSON.stringify(KeyEq$outboundSchema.parse(keyEq));
}
export function keyEqFromJSON(jsonString) {
    return safeParse(jsonString, (x) => KeyEq$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'KeyEq' from JSON`);
}
/** @internal */
export const Key2$inboundSchema = z
    .object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
/** @internal */
export const Key2$outboundSchema = z.object({
    eq: z.union([z.string(), z.number()]).optional(),
    neq: z.string().optional(),
    inc: z.array(z.string()).optional(),
    ninc: z.array(z.string()).optional(),
    pre: z.string().optional(),
    suf: z.string().optional(),
    gt: z.number().optional(),
    gte: z.number().optional(),
    lt: z.number().optional(),
    lte: z.number().optional(),
});
export function key2ToJSON(key2) {
    return JSON.stringify(Key2$outboundSchema.parse(key2));
}
export function key2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Key2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Key2' from JSON`);
}
/** @internal */
export const RoutesKey$inboundSchema = z.union([z.string(), z.lazy(() => Key2$inboundSchema)]);
/** @internal */
export const RoutesKey$outboundSchema = z.union([z.string(), z.lazy(() => Key2$outboundSchema)]);
export function routesKeyToJSON(routesKey) {
    return JSON.stringify(RoutesKey$outboundSchema.parse(routesKey));
}
export function routesKeyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesKey$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesKey' from JSON`);
}
/** @internal */
export const RoutesTarget$inboundSchema = z.object({
    key: z.union([z.string(), z.lazy(() => Key2$inboundSchema)]),
});
/** @internal */
export const RoutesTarget$outboundSchema = z.object({
    key: z.union([z.string(), z.lazy(() => Key2$outboundSchema)]),
});
export function routesTargetToJSON(routesTarget) {
    return JSON.stringify(RoutesTarget$outboundSchema.parse(routesTarget));
}
export function routesTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RoutesTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RoutesTarget' from JSON`);
}
/** @internal */
export const Args$inboundSchema = z
    .union([z.string(), z.array(z.string())]);
/** @internal */
export const Args$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function argsToJSON(args) {
    return JSON.stringify(Args$outboundSchema.parse(args));
}
export function argsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Args$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Args' from JSON`);
}
/** @internal */
export const CreateDeploymentRoutesTransforms$inboundSchema = z.object({
    type: RoutesType$inboundSchema,
    op: CreateDeploymentRoutesOp$inboundSchema,
    target: z.lazy(() => RoutesTarget$inboundSchema),
    args: z.union([z.string(), z.array(z.string())]).optional(),
});
/** @internal */
export const CreateDeploymentRoutesTransforms$outboundSchema = z.object({
    type: RoutesType$outboundSchema,
    op: CreateDeploymentRoutesOp$outboundSchema,
    target: z.lazy(() => RoutesTarget$outboundSchema),
    args: z.union([z.string(), z.array(z.string())]).optional(),
});
export function createDeploymentRoutesTransformsToJSON(createDeploymentRoutesTransforms) {
    return JSON.stringify(CreateDeploymentRoutesTransforms$outboundSchema.parse(createDeploymentRoutesTransforms));
}
export function createDeploymentRoutesTransformsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentRoutesTransforms$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentRoutesTransforms' from JSON`);
}
/** @internal */
export const Locale$inboundSchema = z
    .object({
    redirect: z.record(z.string()).optional(),
    cookie: z.string().optional(),
});
/** @internal */
export const Locale$outboundSchema = z.object({
    redirect: z.record(z.string()).optional(),
    cookie: z.string().optional(),
});
export function localeToJSON(locale) {
    return JSON.stringify(Locale$outboundSchema.parse(locale));
}
export function localeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Locale$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Locale' from JSON`);
}
/** @internal */
export const Routes1$inboundSchema = z.object({
    src: z.string(),
    dest: z.string().optional(),
    headers: z.record(z.string()).optional(),
    methods: z.array(z.string()).optional(),
    continue: z.boolean().optional(),
    override: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    check: z.boolean().optional(),
    important: z.boolean().optional(),
    status: z.number().optional(),
    has: z.array(z.union([
        z.lazy(() => CreateDeploymentHas1$inboundSchema),
        z.lazy(() => CreateDeploymentHas2$inboundSchema),
    ])).optional(),
    missing: z.array(z.union([
        z.lazy(() => Missing1$inboundSchema),
        z.lazy(() => Missing2$inboundSchema),
    ])).optional(),
    mitigate: z.lazy(() => RoutesMitigate$inboundSchema).optional(),
    transforms: z.array(z.lazy(() => CreateDeploymentRoutesTransforms$inboundSchema)).optional(),
    locale: z.lazy(() => Locale$inboundSchema).optional(),
    middlewarePath: z.string().optional(),
    middlewareRawSrc: z.array(z.string()).optional(),
    middleware: z.number().optional(),
});
/** @internal */
export const Routes1$outboundSchema = z.object({
    src: z.string(),
    dest: z.string().optional(),
    headers: z.record(z.string()).optional(),
    methods: z.array(z.string()).optional(),
    continue: z.boolean().optional(),
    override: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    check: z.boolean().optional(),
    important: z.boolean().optional(),
    status: z.number().optional(),
    has: z.array(z.union([
        z.lazy(() => CreateDeploymentHas1$outboundSchema),
        z.lazy(() => CreateDeploymentHas2$outboundSchema),
    ])).optional(),
    missing: z.array(z.union([
        z.lazy(() => Missing1$outboundSchema),
        z.lazy(() => Missing2$outboundSchema),
    ])).optional(),
    mitigate: z.lazy(() => RoutesMitigate$outboundSchema).optional(),
    transforms: z.array(z.lazy(() => CreateDeploymentRoutesTransforms$outboundSchema)).optional(),
    locale: z.lazy(() => Locale$outboundSchema).optional(),
    middlewarePath: z.string().optional(),
    middlewareRawSrc: z.array(z.string()).optional(),
    middleware: z.number().optional(),
});
export function routes1ToJSON(routes1) {
    return JSON.stringify(Routes1$outboundSchema.parse(routes1));
}
export function routes1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Routes1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Routes1' from JSON`);
}
/** @internal */
export const Routes$inboundSchema = z
    .union([
    z.lazy(() => Routes3$inboundSchema),
    z.lazy(() => Routes1$inboundSchema),
    z.lazy(() => Routes2$inboundSchema),
]);
/** @internal */
export const Routes$outboundSchema = z.union([
    z.lazy(() => Routes3$outboundSchema),
    z.lazy(() => Routes1$outboundSchema),
    z.lazy(() => Routes2$outboundSchema),
]);
export function routesToJSON(routes) {
    return JSON.stringify(Routes$outboundSchema.parse(routes));
}
export function routesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Routes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Routes' from JSON`);
}
/** @internal */
export const CreateDeploymentGitRepoDeploymentsType$inboundSchema = z.nativeEnum(CreateDeploymentGitRepoDeploymentsType);
/** @internal */
export const CreateDeploymentGitRepoDeploymentsType$outboundSchema = CreateDeploymentGitRepoDeploymentsType$inboundSchema;
/** @internal */
export const CreateDeploymentGitRepoOwnerType$inboundSchema = z.nativeEnum(CreateDeploymentGitRepoOwnerType);
/** @internal */
export const CreateDeploymentGitRepoOwnerType$outboundSchema = CreateDeploymentGitRepoOwnerType$inboundSchema;
/** @internal */
export const GitRepo3$inboundSchema = z.object({
    owner: z.string(),
    repoUuid: z.string(),
    slug: z.string(),
    type: CreateDeploymentGitRepoDeploymentsType$inboundSchema,
    workspaceUuid: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CreateDeploymentGitRepoOwnerType$inboundSchema,
});
/** @internal */
export const GitRepo3$outboundSchema = z.object({
    owner: z.string(),
    repoUuid: z.string(),
    slug: z.string(),
    type: CreateDeploymentGitRepoDeploymentsType$outboundSchema,
    workspaceUuid: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: CreateDeploymentGitRepoOwnerType$outboundSchema,
});
export function gitRepo3ToJSON(gitRepo3) {
    return JSON.stringify(GitRepo3$outboundSchema.parse(gitRepo3));
}
export function gitRepo3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitRepo3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitRepo3' from JSON`);
}
/** @internal */
export const CreateDeploymentGitRepoType$inboundSchema = z.nativeEnum(CreateDeploymentGitRepoType);
/** @internal */
export const CreateDeploymentGitRepoType$outboundSchema = CreateDeploymentGitRepoType$inboundSchema;
/** @internal */
export const GitRepoOwnerType$inboundSchema = z.nativeEnum(GitRepoOwnerType);
/** @internal */
export const GitRepoOwnerType$outboundSchema = GitRepoOwnerType$inboundSchema;
/** @internal */
export const GitRepo2$inboundSchema = z.object({
    org: z.string(),
    repo: z.string(),
    repoId: z.number(),
    type: CreateDeploymentGitRepoType$inboundSchema,
    repoOwnerId: z.number(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: GitRepoOwnerType$inboundSchema,
});
/** @internal */
export const GitRepo2$outboundSchema = z.object({
    org: z.string(),
    repo: z.string(),
    repoId: z.number(),
    type: CreateDeploymentGitRepoType$outboundSchema,
    repoOwnerId: z.number(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: GitRepoOwnerType$outboundSchema,
});
export function gitRepo2ToJSON(gitRepo2) {
    return JSON.stringify(GitRepo2$outboundSchema.parse(gitRepo2));
}
export function gitRepo2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitRepo2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitRepo2' from JSON`);
}
/** @internal */
export const GitRepoType$inboundSchema = z
    .nativeEnum(GitRepoType);
/** @internal */
export const GitRepoType$outboundSchema = GitRepoType$inboundSchema;
/** @internal */
export const OwnerType$inboundSchema = z
    .nativeEnum(OwnerType);
/** @internal */
export const OwnerType$outboundSchema = OwnerType$inboundSchema;
/** @internal */
export const GitRepo1$inboundSchema = z.object({
    namespace: z.string(),
    projectId: z.number(),
    type: GitRepoType$inboundSchema,
    url: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: OwnerType$inboundSchema,
});
/** @internal */
export const GitRepo1$outboundSchema = z.object({
    namespace: z.string(),
    projectId: z.number(),
    type: GitRepoType$outboundSchema,
    url: z.string(),
    path: z.string(),
    defaultBranch: z.string(),
    name: z.string(),
    private: z.boolean(),
    ownerType: OwnerType$outboundSchema,
});
export function gitRepo1ToJSON(gitRepo1) {
    return JSON.stringify(GitRepo1$outboundSchema.parse(gitRepo1));
}
export function gitRepo1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitRepo1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitRepo1' from JSON`);
}
/** @internal */
export const GitRepo$inboundSchema = z.union([
    z.lazy(() => GitRepo2$inboundSchema),
    z.lazy(() => GitRepo3$inboundSchema),
    z.lazy(() => GitRepo1$inboundSchema),
]);
/** @internal */
export const GitRepo$outboundSchema = z.union([
    z.lazy(() => GitRepo2$outboundSchema),
    z.lazy(() => GitRepo3$outboundSchema),
    z.lazy(() => GitRepo1$outboundSchema),
]);
export function gitRepoToJSON(gitRepo) {
    return JSON.stringify(GitRepo$outboundSchema.parse(gitRepo));
}
export function gitRepoFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitRepo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitRepo' from JSON`);
}
/** @internal */
export const Flags2$inboundSchema = z
    .object({});
/** @internal */
export const Flags2$outboundSchema = z.object({});
export function flags2ToJSON(flags2) {
    return JSON.stringify(Flags2$outboundSchema.parse(flags2));
}
export function flags2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Flags2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Flags2' from JSON`);
}
/** @internal */
export const Options$inboundSchema = z.object({
    value: z.nullable(FlagJSONValue$inboundSchema),
    label: z.string().optional(),
});
/** @internal */
export const Options$outboundSchema = z.object({
    value: z.nullable(FlagJSONValue$outboundSchema),
    label: z.string().optional(),
});
export function optionsToJSON(options) {
    return JSON.stringify(Options$outboundSchema.parse(options));
}
export function optionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Options$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Options' from JSON`);
}
/** @internal */
export const FlagsDefinitions$inboundSchema = z.object({
    options: z.array(z.lazy(() => Options$inboundSchema)).optional(),
    url: z.string().optional(),
    description: z.string().optional(),
});
/** @internal */
export const FlagsDefinitions$outboundSchema = z.object({
    options: z.array(z.lazy(() => Options$outboundSchema)).optional(),
    url: z.string().optional(),
    description: z.string().optional(),
});
export function flagsDefinitionsToJSON(flagsDefinitions) {
    return JSON.stringify(FlagsDefinitions$outboundSchema.parse(flagsDefinitions));
}
export function flagsDefinitionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FlagsDefinitions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FlagsDefinitions' from JSON`);
}
/** @internal */
export const Flags1$inboundSchema = z
    .object({
    definitions: z.record(z.lazy(() => FlagsDefinitions$inboundSchema)),
});
/** @internal */
export const Flags1$outboundSchema = z.object({
    definitions: z.record(z.lazy(() => FlagsDefinitions$outboundSchema)),
});
export function flags1ToJSON(flags1) {
    return JSON.stringify(Flags1$outboundSchema.parse(flags1));
}
export function flags1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Flags1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Flags1' from JSON`);
}
/** @internal */
export const Flags$inboundSchema = z
    .union([
    z.lazy(() => Flags1$inboundSchema),
    z.array(z.lazy(() => Flags2$inboundSchema)),
]);
/** @internal */
export const Flags$outboundSchema = z.union([
    z.lazy(() => Flags1$outboundSchema),
    z.array(z.lazy(() => Flags2$outboundSchema)),
]);
export function flagsToJSON(flags) {
    return JSON.stringify(Flags$outboundSchema.parse(flags));
}
export function flagsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Flags$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Flags' from JSON`);
}
/** @internal */
export const CreateDeploymentMicrofrontendsApplications$inboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    productionHost: z.string(),
    deploymentAlias: z.string().optional(),
    deploymentHost: z.string().optional(),
});
/** @internal */
export const CreateDeploymentMicrofrontendsApplications$outboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    productionHost: z.string(),
    deploymentAlias: z.string().optional(),
    deploymentHost: z.string().optional(),
});
export function createDeploymentMicrofrontendsApplicationsToJSON(createDeploymentMicrofrontendsApplications) {
    return JSON.stringify(CreateDeploymentMicrofrontendsApplications$outboundSchema.parse(createDeploymentMicrofrontendsApplications));
}
export function createDeploymentMicrofrontendsApplicationsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentMicrofrontendsApplications$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentMicrofrontendsApplications' from JSON`);
}
/** @internal */
export const MfeConfigUploadState$inboundSchema = z.nativeEnum(MfeConfigUploadState);
/** @internal */
export const MfeConfigUploadState$outboundSchema = MfeConfigUploadState$inboundSchema;
/** @internal */
export const CreateDeploymentMicrofrontends2$inboundSchema = z.object({
    isDefaultApp: z.boolean(),
    applications: z.record(z.lazy(() => CreateDeploymentMicrofrontendsApplications$inboundSchema)).optional(),
    mfeConfigUploadState: MfeConfigUploadState$inboundSchema.optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
/** @internal */
export const CreateDeploymentMicrofrontends2$outboundSchema = z.object({
    isDefaultApp: z.boolean(),
    applications: z.record(z.lazy(() => CreateDeploymentMicrofrontendsApplications$outboundSchema)).optional(),
    mfeConfigUploadState: MfeConfigUploadState$outboundSchema.optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
export function createDeploymentMicrofrontends2ToJSON(createDeploymentMicrofrontends2) {
    return JSON.stringify(CreateDeploymentMicrofrontends2$outboundSchema.parse(createDeploymentMicrofrontends2));
}
export function createDeploymentMicrofrontends2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentMicrofrontends2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentMicrofrontends2' from JSON`);
}
/** @internal */
export const CreateDeploymentMicrofrontends1$inboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
/** @internal */
export const CreateDeploymentMicrofrontends1$outboundSchema = z.object({
    isDefaultApp: z.boolean().optional(),
    defaultAppProjectName: z.string(),
    defaultRoute: z.string().optional(),
    groupIds: z.array(z.string()),
});
export function createDeploymentMicrofrontends1ToJSON(createDeploymentMicrofrontends1) {
    return JSON.stringify(CreateDeploymentMicrofrontends1$outboundSchema.parse(createDeploymentMicrofrontends1));
}
export function createDeploymentMicrofrontends1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentMicrofrontends1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentMicrofrontends1' from JSON`);
}
/** @internal */
export const CreateDeploymentMicrofrontends$inboundSchema = z.union([
    z.lazy(() => CreateDeploymentMicrofrontends2$inboundSchema),
    z.lazy(() => CreateDeploymentMicrofrontends1$inboundSchema),
]);
/** @internal */
export const CreateDeploymentMicrofrontends$outboundSchema = z.union([
    z.lazy(() => CreateDeploymentMicrofrontends2$outboundSchema),
    z.lazy(() => CreateDeploymentMicrofrontends1$outboundSchema),
]);
export function createDeploymentMicrofrontendsToJSON(createDeploymentMicrofrontends) {
    return JSON.stringify(CreateDeploymentMicrofrontends$outboundSchema.parse(createDeploymentMicrofrontends));
}
export function createDeploymentMicrofrontendsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentMicrofrontends$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentMicrofrontends' from JSON`);
}
/** @internal */
export const FunctionType$inboundSchema = z.nativeEnum(FunctionType);
/** @internal */
export const FunctionType$outboundSchema = FunctionType$inboundSchema;
/** @internal */
export const FunctionMemoryType$inboundSchema = z.nativeEnum(FunctionMemoryType);
/** @internal */
export const FunctionMemoryType$outboundSchema = FunctionMemoryType$inboundSchema;
/** @internal */
export const Config$inboundSchema = z
    .object({
    version: z.number().optional(),
    functionType: FunctionType$inboundSchema,
    functionMemoryType: FunctionMemoryType$inboundSchema,
    functionTimeout: z.nullable(z.number()),
    secureComputePrimaryRegion: z.nullable(z.string()),
    secureComputeFallbackRegion: z.nullable(z.string()),
    isUsingActiveCPU: z.boolean().optional(),
});
/** @internal */
export const Config$outboundSchema = z.object({
    version: z.number().optional(),
    functionType: FunctionType$outboundSchema,
    functionMemoryType: FunctionMemoryType$outboundSchema,
    functionTimeout: z.nullable(z.number()),
    secureComputePrimaryRegion: z.nullable(z.string()),
    secureComputeFallbackRegion: z.nullable(z.string()),
    isUsingActiveCPU: z.boolean().optional(),
});
export function configToJSON(config) {
    return JSON.stringify(Config$outboundSchema.parse(config));
}
export function configFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Config$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Config' from JSON`);
}
/** @internal */
export const CreateDeploymentState$inboundSchema = z.nativeEnum(CreateDeploymentState);
/** @internal */
export const CreateDeploymentState$outboundSchema = CreateDeploymentState$inboundSchema;
/** @internal */
export const DeploymentAlias$inboundSchema = z.object({
    state: CreateDeploymentState$inboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
});
/** @internal */
export const DeploymentAlias$outboundSchema = z.object({
    state: CreateDeploymentState$outboundSchema,
    startedAt: z.number(),
    completedAt: z.number().optional(),
});
export function deploymentAliasToJSON(deploymentAlias) {
    return JSON.stringify(DeploymentAlias$outboundSchema.parse(deploymentAlias));
}
export function deploymentAliasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeploymentAlias$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeploymentAlias' from JSON`);
}
/** @internal */
export const CreateDeploymentChecks$inboundSchema = z.object({
    "deployment-alias": z.lazy(() => DeploymentAlias$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "deployment-alias": "deploymentAlias",
    });
});
/** @internal */
export const CreateDeploymentChecks$outboundSchema = z.object({
    deploymentAlias: z.lazy(() => DeploymentAlias$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        deploymentAlias: "deployment-alias",
    });
});
export function createDeploymentChecksToJSON(createDeploymentChecks) {
    return JSON.stringify(CreateDeploymentChecks$outboundSchema.parse(createDeploymentChecks));
}
export function createDeploymentChecksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentChecks$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentChecks' from JSON`);
}
/** @internal */
export const CreateDeploymentResponseBody$inboundSchema = z.object({
    aliasAssignedAt: z.nullable(z.union([z.number(), z.boolean()])).optional(),
    alwaysRefuseToBuild: z.boolean().optional(),
    build: z.lazy(() => Build$inboundSchema),
    buildArtifactUrls: z.array(z.string()).optional(),
    builds: z.array(z.lazy(() => Builds$inboundSchema)).optional(),
    env: z.array(z.string()),
    inspectorUrl: z.nullable(z.string()),
    isInConcurrentBuildsQueue: z.boolean(),
    isInSystemBuildsQueue: z.boolean(),
    projectSettings: z.lazy(() => CreateDeploymentProjectSettings$inboundSchema),
    readyStateReason: z.string().optional(),
    integrations: z.lazy(() => Integrations$inboundSchema).optional(),
    images: z.lazy(() => Images$inboundSchema).optional(),
    alias: z.array(z.string()).optional(),
    aliasAssigned: z.boolean(),
    bootedAt: z.number(),
    buildingAt: z.number(),
    buildContainerFinishedAt: z.number().optional(),
    buildSkipped: z.boolean(),
    creator: z.lazy(() => Creator$inboundSchema),
    initReadyAt: z.number().optional(),
    isFirstBranchDeployment: z.boolean().optional(),
    lambdas: z.array(z.lazy(() => Lambdas$inboundSchema)).optional(),
    public: z.boolean(),
    ready: z.number().optional(),
    status: CreateDeploymentStatus$inboundSchema,
    team: z.lazy(() => CreateDeploymentTeam$inboundSchema).optional(),
    userAliases: z.array(z.string()).optional(),
    previewCommentsEnabled: z.boolean().optional(),
    ttyBuildLogs: z.boolean().optional(),
    customEnvironment: z.union([
        z.lazy(() => CustomEnvironment1$inboundSchema),
        z.lazy(() => CustomEnvironment2$inboundSchema),
    ]).optional(),
    oomReport: OomReport$inboundSchema.optional(),
    aliasWarning: z.nullable(z.lazy(() => AliasWarning$inboundSchema)).optional(),
    id: z.string(),
    createdAt: z.number(),
    readyState: ReadyState$inboundSchema,
    name: z.string(),
    type: CreateDeploymentType$inboundSchema,
    aliasError: z.nullable(z.lazy(() => AliasError$inboundSchema)).optional(),
    aliasFinal: z.nullable(z.string()).optional(),
    autoAssignCustomDomains: z.boolean().optional(),
    automaticAliases: z.array(z.string()).optional(),
    buildErrorAt: z.number().optional(),
    checksState: ChecksState$inboundSchema.optional(),
    checksConclusion: ChecksConclusion$inboundSchema.optional(),
    deletedAt: z.nullable(z.number()).optional(),
    defaultRoute: z.string().optional(),
    canceledAt: z.number().optional(),
    errorCode: z.string().optional(),
    errorLink: z.string().optional(),
    errorMessage: z.nullable(z.string()).optional(),
    errorStep: z.string().optional(),
    passiveRegions: z.array(z.string()).optional(),
    gitSource: z.union([
        z.lazy(() => CreateDeploymentGitSource12$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource15$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource4$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource10$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource11$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource13$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource14$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource2$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource3$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource6$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource9$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource1$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource5$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource7$inboundSchema),
        z.lazy(() => CreateDeploymentGitSource8$inboundSchema),
    ]).optional(),
    meta: z.record(z.string()),
    originCacheRegion: z.string().optional(),
    nodeVersion: CreateDeploymentNodeVersion$inboundSchema.optional(),
    project: z.lazy(() => CreateDeploymentProject$inboundSchema).optional(),
    readySubstate: ReadySubstate$inboundSchema.optional(),
    regions: z.array(z.string()),
    softDeletedByRetention: z.boolean().optional(),
    source: CreateDeploymentSource$inboundSchema.optional(),
    target: z.nullable(CreateDeploymentTarget$inboundSchema).optional(),
    undeletedAt: z.number().optional(),
    url: z.string(),
    version: z.number(),
    oidcTokenClaims: z.lazy(() => OidcTokenClaims$inboundSchema).optional(),
    projectId: z.string(),
    plan: CreateDeploymentPlan$inboundSchema,
    connectBuildsEnabled: z.boolean().optional(),
    connectConfigurationId: z.string().optional(),
    createdIn: z.string(),
    crons: z.array(z.lazy(() => CreateDeploymentCrons$inboundSchema)).optional(),
    functions: z.nullable(z.record(z.lazy(() => Functions$inboundSchema)))
        .optional(),
    monorepoManager: z.nullable(z.string()).optional(),
    ownerId: z.string(),
    passiveConnectConfigurationId: z.string().optional(),
    routes: z.nullable(z.array(z.union([
        z.lazy(() => Routes3$inboundSchema),
        z.lazy(() => Routes1$inboundSchema),
        z.lazy(() => Routes2$inboundSchema),
    ]))),
    gitRepo: z.nullable(z.union([
        z.lazy(() => GitRepo2$inboundSchema),
        z.lazy(() => GitRepo3$inboundSchema),
        z.lazy(() => GitRepo1$inboundSchema),
    ])).optional(),
    flags: z.union([
        z.lazy(() => Flags1$inboundSchema),
        z.array(z.lazy(() => Flags2$inboundSchema)),
    ]).optional(),
    microfrontends: z.union([
        z.lazy(() => CreateDeploymentMicrofrontends2$inboundSchema),
        z.lazy(() => CreateDeploymentMicrofrontends1$inboundSchema),
    ]).optional(),
    config: z.lazy(() => Config$inboundSchema).optional(),
    checks: z.lazy(() => CreateDeploymentChecks$inboundSchema).optional(),
});
/** @internal */
export const CreateDeploymentResponseBody$outboundSchema = z.object({
    aliasAssignedAt: z.nullable(z.union([z.number(), z.boolean()])).optional(),
    alwaysRefuseToBuild: z.boolean().optional(),
    build: z.lazy(() => Build$outboundSchema),
    buildArtifactUrls: z.array(z.string()).optional(),
    builds: z.array(z.lazy(() => Builds$outboundSchema)).optional(),
    env: z.array(z.string()),
    inspectorUrl: z.nullable(z.string()),
    isInConcurrentBuildsQueue: z.boolean(),
    isInSystemBuildsQueue: z.boolean(),
    projectSettings: z.lazy(() => CreateDeploymentProjectSettings$outboundSchema),
    readyStateReason: z.string().optional(),
    integrations: z.lazy(() => Integrations$outboundSchema).optional(),
    images: z.lazy(() => Images$outboundSchema).optional(),
    alias: z.array(z.string()).optional(),
    aliasAssigned: z.boolean(),
    bootedAt: z.number(),
    buildingAt: z.number(),
    buildContainerFinishedAt: z.number().optional(),
    buildSkipped: z.boolean(),
    creator: z.lazy(() => Creator$outboundSchema),
    initReadyAt: z.number().optional(),
    isFirstBranchDeployment: z.boolean().optional(),
    lambdas: z.array(z.lazy(() => Lambdas$outboundSchema)).optional(),
    public: z.boolean(),
    ready: z.number().optional(),
    status: CreateDeploymentStatus$outboundSchema,
    team: z.lazy(() => CreateDeploymentTeam$outboundSchema).optional(),
    userAliases: z.array(z.string()).optional(),
    previewCommentsEnabled: z.boolean().optional(),
    ttyBuildLogs: z.boolean().optional(),
    customEnvironment: z.union([
        z.lazy(() => CustomEnvironment1$outboundSchema),
        z.lazy(() => CustomEnvironment2$outboundSchema),
    ]).optional(),
    oomReport: OomReport$outboundSchema.optional(),
    aliasWarning: z.nullable(z.lazy(() => AliasWarning$outboundSchema))
        .optional(),
    id: z.string(),
    createdAt: z.number(),
    readyState: ReadyState$outboundSchema,
    name: z.string(),
    type: CreateDeploymentType$outboundSchema,
    aliasError: z.nullable(z.lazy(() => AliasError$outboundSchema)).optional(),
    aliasFinal: z.nullable(z.string()).optional(),
    autoAssignCustomDomains: z.boolean().optional(),
    automaticAliases: z.array(z.string()).optional(),
    buildErrorAt: z.number().optional(),
    checksState: ChecksState$outboundSchema.optional(),
    checksConclusion: ChecksConclusion$outboundSchema.optional(),
    deletedAt: z.nullable(z.number()).optional(),
    defaultRoute: z.string().optional(),
    canceledAt: z.number().optional(),
    errorCode: z.string().optional(),
    errorLink: z.string().optional(),
    errorMessage: z.nullable(z.string()).optional(),
    errorStep: z.string().optional(),
    passiveRegions: z.array(z.string()).optional(),
    gitSource: z.union([
        z.lazy(() => CreateDeploymentGitSource12$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource15$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource4$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource10$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource11$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource13$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource14$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource2$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource3$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource6$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource9$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource1$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource5$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource7$outboundSchema),
        z.lazy(() => CreateDeploymentGitSource8$outboundSchema),
    ]).optional(),
    meta: z.record(z.string()),
    originCacheRegion: z.string().optional(),
    nodeVersion: CreateDeploymentNodeVersion$outboundSchema.optional(),
    project: z.lazy(() => CreateDeploymentProject$outboundSchema).optional(),
    readySubstate: ReadySubstate$outboundSchema.optional(),
    regions: z.array(z.string()),
    softDeletedByRetention: z.boolean().optional(),
    source: CreateDeploymentSource$outboundSchema.optional(),
    target: z.nullable(CreateDeploymentTarget$outboundSchema).optional(),
    undeletedAt: z.number().optional(),
    url: z.string(),
    version: z.number(),
    oidcTokenClaims: z.lazy(() => OidcTokenClaims$outboundSchema).optional(),
    projectId: z.string(),
    plan: CreateDeploymentPlan$outboundSchema,
    connectBuildsEnabled: z.boolean().optional(),
    connectConfigurationId: z.string().optional(),
    createdIn: z.string(),
    crons: z.array(z.lazy(() => CreateDeploymentCrons$outboundSchema)).optional(),
    functions: z.nullable(z.record(z.lazy(() => Functions$outboundSchema)))
        .optional(),
    monorepoManager: z.nullable(z.string()).optional(),
    ownerId: z.string(),
    passiveConnectConfigurationId: z.string().optional(),
    routes: z.nullable(z.array(z.union([
        z.lazy(() => Routes3$outboundSchema),
        z.lazy(() => Routes1$outboundSchema),
        z.lazy(() => Routes2$outboundSchema),
    ]))),
    gitRepo: z.nullable(z.union([
        z.lazy(() => GitRepo2$outboundSchema),
        z.lazy(() => GitRepo3$outboundSchema),
        z.lazy(() => GitRepo1$outboundSchema),
    ])).optional(),
    flags: z.union([
        z.lazy(() => Flags1$outboundSchema),
        z.array(z.lazy(() => Flags2$outboundSchema)),
    ]).optional(),
    microfrontends: z.union([
        z.lazy(() => CreateDeploymentMicrofrontends2$outboundSchema),
        z.lazy(() => CreateDeploymentMicrofrontends1$outboundSchema),
    ]).optional(),
    config: z.lazy(() => Config$outboundSchema).optional(),
    checks: z.lazy(() => CreateDeploymentChecks$outboundSchema).optional(),
});
export function createDeploymentResponseBodyToJSON(createDeploymentResponseBody) {
    return JSON.stringify(CreateDeploymentResponseBody$outboundSchema.parse(createDeploymentResponseBody));
}
export function createDeploymentResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateDeploymentResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateDeploymentResponseBody' from JSON`);
}
