// @vercel/sdk/esm/models/createconfigurablelogdrainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The delivery log format
 */
export const DeliveryFormat = {
    Json: "json",
    Ndjson: "ndjson",
};
export const Sources = {
    Static: "static",
    Lambda: "lambda",
    Build: "build",
    Edge: "edge",
    External: "external",
    Firewall: "firewall",
};
export const Environments = {
    Preview: "preview",
    Production: "production",
};
/** @internal */
export const DeliveryFormat$inboundSchema = z.nativeEnum(DeliveryFormat);
/** @internal */
export const DeliveryFormat$outboundSchema = DeliveryFormat$inboundSchema;
/** @internal */
export const Sources$inboundSchema = z
    .nativeEnum(Sources);
/** @internal */
export const Sources$outboundSchema = Sources$inboundSchema;
/** @internal */
export const Environments$inboundSchema = z.nativeEnum(Environments);
/** @internal */
export const Environments$outboundSchema = Environments$inboundSchema;
/** @internal */
export const CreateConfigurableLogDrainRequestBody$inboundSchema = z.object({
    deliveryFormat: DeliveryFormat$inboundSchema,
    url: z.string(),
    headers: z.record(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    sources: z.array(Sources$inboundSchema),
    environments: z.array(Environments$inboundSchema).optional(),
    secret: z.string().optional(),
    samplingRate: z.number().optional(),
    name: z.string().optional(),
});
/** @internal */
export const CreateConfigurableLogDrainRequestBody$outboundSchema = z.object({
    deliveryFormat: DeliveryFormat$outboundSchema,
    url: z.string(),
    headers: z.record(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    sources: z.array(Sources$outboundSchema),
    environments: z.array(Environments$outboundSchema).optional(),
    secret: z.string().optional(),
    samplingRate: z.number().optional(),
    name: z.string().optional(),
});
export function createConfigurableLogDrainRequestBodyToJSON(createConfigurableLogDrainRequestBody) {
    return JSON.stringify(CreateConfigurableLogDrainRequestBody$outboundSchema.parse(createConfigurableLogDrainRequestBody));
}
export function createConfigurableLogDrainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateConfigurableLogDrainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateConfigurableLogDrainRequestBody' from JSON`);
}
/** @internal */
export const CreateConfigurableLogDrainRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => CreateConfigurableLogDrainRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const CreateConfigurableLogDrainRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => CreateConfigurableLogDrainRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function createConfigurableLogDrainRequestToJSON(createConfigurableLogDrainRequest) {
    return JSON.stringify(CreateConfigurableLogDrainRequest$outboundSchema.parse(createConfigurableLogDrainRequest));
}
export function createConfigurableLogDrainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateConfigurableLogDrainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateConfigurableLogDrainRequest' from JSON`);
}
/** @internal */
export const CreateConfigurableLogDrainResponseBody$inboundSchema = z.object({});
/** @internal */
export const CreateConfigurableLogDrainResponseBody$outboundSchema = z.object({});
export function createConfigurableLogDrainResponseBodyToJSON(createConfigurableLogDrainResponseBody) {
    return JSON.stringify(CreateConfigurableLogDrainResponseBody$outboundSchema.parse(createConfigurableLogDrainResponseBody));
}
export function createConfigurableLogDrainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateConfigurableLogDrainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateConfigurableLogDrainResponseBody' from JSON`);
}
