// @vercel/sdk/esm/models/createsharedenvvariableop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The type of environment variable
 */
export const CreateSharedEnvVariableRequestBodyEnvironmentType = {
    Encrypted: "encrypted",
    Sensitive: "sensitive",
};
export const CreateSharedEnvVariableRequestBodyTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of environment variable
 */
export const CreateSharedEnvVariableRequestBodyType = {
    Encrypted: "encrypted",
    Sensitive: "sensitive",
};
export const RequestBodyTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of this cosmos doc instance, if blank, assume secret.
 */
export const CreateSharedEnvVariableType = {
    Encrypted: "encrypted",
    Sensitive: "sensitive",
    System: "system",
    Plain: "plain",
};
/**
 * environments this env variable targets
 */
export const CreateSharedEnvVariableTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const Value2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateSharedEnvVariableTarget2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const CreateSharedEnvVariableTarget1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/** @internal */
export const RequestBodyEvs$inboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    comment: z.string().optional(),
});
/** @internal */
export const RequestBodyEvs$outboundSchema = z.object({
    key: z.string(),
    value: z.string(),
    comment: z.string().optional(),
});
export function requestBodyEvsToJSON(requestBodyEvs) {
    return JSON.stringify(RequestBodyEvs$outboundSchema.parse(requestBodyEvs));
}
export function requestBodyEvsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RequestBodyEvs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RequestBodyEvs' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableRequestBodyEnvironmentType$inboundSchema = z
    .nativeEnum(CreateSharedEnvVariableRequestBodyEnvironmentType);
/** @internal */
export const CreateSharedEnvVariableRequestBodyEnvironmentType$outboundSchema = CreateSharedEnvVariableRequestBodyEnvironmentType$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableRequestBodyTarget$inboundSchema = z
    .nativeEnum(CreateSharedEnvVariableRequestBodyTarget);
/** @internal */
export const CreateSharedEnvVariableRequestBodyTarget$outboundSchema = CreateSharedEnvVariableRequestBodyTarget$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableRequestBody2$inboundSchema = z.object({
    evs: z.array(z.lazy(() => RequestBodyEvs$inboundSchema)),
    type: CreateSharedEnvVariableRequestBodyEnvironmentType$inboundSchema
        .optional(),
    target: z.array(CreateSharedEnvVariableRequestBodyTarget$inboundSchema)
        .optional(),
    projectId: z.array(z.string()).optional(),
});
/** @internal */
export const CreateSharedEnvVariableRequestBody2$outboundSchema = z.object({
    evs: z.array(z.lazy(() => RequestBodyEvs$outboundSchema)),
    type: CreateSharedEnvVariableRequestBodyEnvironmentType$outboundSchema
        .optional(),
    target: z.array(CreateSharedEnvVariableRequestBodyTarget$outboundSchema)
        .optional(),
    projectId: z.array(z.string()).optional(),
});
export function createSharedEnvVariableRequestBody2ToJSON(createSharedEnvVariableRequestBody2) {
    return JSON.stringify(CreateSharedEnvVariableRequestBody2$outboundSchema.parse(createSharedEnvVariableRequestBody2));
}
export function createSharedEnvVariableRequestBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableRequestBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableRequestBody2' from JSON`);
}
/** @internal */
export const Evs$inboundSchema = z
    .object({
    key: z.string(),
    value: z.string(),
    comment: z.string().optional(),
});
/** @internal */
export const Evs$outboundSchema = z
    .object({
    key: z.string(),
    value: z.string(),
    comment: z.string().optional(),
});
export function evsToJSON(evs) {
    return JSON.stringify(Evs$outboundSchema.parse(evs));
}
export function evsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Evs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Evs' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableRequestBodyType$inboundSchema = z.nativeEnum(CreateSharedEnvVariableRequestBodyType);
/** @internal */
export const CreateSharedEnvVariableRequestBodyType$outboundSchema = CreateSharedEnvVariableRequestBodyType$inboundSchema;
/** @internal */
export const RequestBodyTarget$inboundSchema = z.nativeEnum(RequestBodyTarget);
/** @internal */
export const RequestBodyTarget$outboundSchema = RequestBodyTarget$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableRequestBody1$inboundSchema = z.object({
    evs: z.array(z.lazy(() => Evs$inboundSchema)),
    type: CreateSharedEnvVariableRequestBodyType$inboundSchema.optional(),
    target: z.array(RequestBodyTarget$inboundSchema),
    projectId: z.array(z.string()).optional(),
});
/** @internal */
export const CreateSharedEnvVariableRequestBody1$outboundSchema = z.object({
    evs: z.array(z.lazy(() => Evs$outboundSchema)),
    type: CreateSharedEnvVariableRequestBodyType$outboundSchema.optional(),
    target: z.array(RequestBodyTarget$outboundSchema),
    projectId: z.array(z.string()).optional(),
});
export function createSharedEnvVariableRequestBody1ToJSON(createSharedEnvVariableRequestBody1) {
    return JSON.stringify(CreateSharedEnvVariableRequestBody1$outboundSchema.parse(createSharedEnvVariableRequestBody1));
}
export function createSharedEnvVariableRequestBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableRequestBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableRequestBody1' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableRequestBody$inboundSchema = z.union([
    z.lazy(() => CreateSharedEnvVariableRequestBody1$inboundSchema),
    z.lazy(() => CreateSharedEnvVariableRequestBody2$inboundSchema),
]);
/** @internal */
export const CreateSharedEnvVariableRequestBody$outboundSchema = z.union([
    z.lazy(() => CreateSharedEnvVariableRequestBody1$outboundSchema),
    z.lazy(() => CreateSharedEnvVariableRequestBody2$outboundSchema),
]);
export function createSharedEnvVariableRequestBodyToJSON(createSharedEnvVariableRequestBody) {
    return JSON.stringify(CreateSharedEnvVariableRequestBody$outboundSchema.parse(createSharedEnvVariableRequestBody));
}
export function createSharedEnvVariableRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableRequestBody' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.union([
        z.lazy(() => CreateSharedEnvVariableRequestBody1$inboundSchema),
        z.lazy(() => CreateSharedEnvVariableRequestBody2$inboundSchema),
    ]).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const CreateSharedEnvVariableRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.union([
        z.lazy(() => CreateSharedEnvVariableRequestBody1$outboundSchema),
        z.lazy(() => CreateSharedEnvVariableRequestBody2$outboundSchema),
    ]).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function createSharedEnvVariableRequestToJSON(createSharedEnvVariableRequest) {
    return JSON.stringify(CreateSharedEnvVariableRequest$outboundSchema.parse(createSharedEnvVariableRequest));
}
export function createSharedEnvVariableRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableRequest' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableType$inboundSchema = z.nativeEnum(CreateSharedEnvVariableType);
/** @internal */
export const CreateSharedEnvVariableType$outboundSchema = CreateSharedEnvVariableType$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableTarget$inboundSchema = z.nativeEnum(CreateSharedEnvVariableTarget);
/** @internal */
export const CreateSharedEnvVariableTarget$outboundSchema = CreateSharedEnvVariableTarget$inboundSchema;
/** @internal */
export const Created$inboundSchema = z.object({
    created: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    key: z.string().optional(),
    ownerId: z.nullable(z.string()).optional(),
    id: z.string().optional(),
    createdBy: z.nullable(z.string()).optional(),
    deletedBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    deletedAt: z.number().optional(),
    updatedAt: z.number().optional(),
    value: z.string().optional(),
    projectId: z.array(z.string()).optional(),
    type: CreateSharedEnvVariableType$inboundSchema.optional(),
    target: z.array(CreateSharedEnvVariableTarget$inboundSchema).optional(),
    applyToAllCustomEnvironments: z.boolean().optional(),
    decrypted: z.boolean().optional(),
    comment: z.string().optional(),
    lastEditedByDisplayName: z.string().optional(),
});
/** @internal */
export const Created$outboundSchema = z.object({
    created: z.date().transform(v => v.toISOString()).optional(),
    key: z.string().optional(),
    ownerId: z.nullable(z.string()).optional(),
    id: z.string().optional(),
    createdBy: z.nullable(z.string()).optional(),
    deletedBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    deletedAt: z.number().optional(),
    updatedAt: z.number().optional(),
    value: z.string().optional(),
    projectId: z.array(z.string()).optional(),
    type: CreateSharedEnvVariableType$outboundSchema.optional(),
    target: z.array(CreateSharedEnvVariableTarget$outboundSchema).optional(),
    applyToAllCustomEnvironments: z.boolean().optional(),
    decrypted: z.boolean().optional(),
    comment: z.string().optional(),
    lastEditedByDisplayName: z.string().optional(),
});
export function createdToJSON(created) {
    return JSON.stringify(Created$outboundSchema.parse(created));
}
export function createdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Created$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Created' from JSON`);
}
/** @internal */
export const Value2$inboundSchema = z
    .nativeEnum(Value2);
/** @internal */
export const Value2$outboundSchema = Value2$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableValue$inboundSchema = z.union([z.string(), z.array(Value2$inboundSchema)]);
/** @internal */
export const CreateSharedEnvVariableValue$outboundSchema = z.union([z.string(), z.array(Value2$outboundSchema)]);
export function createSharedEnvVariableValueToJSON(createSharedEnvVariableValue) {
    return JSON.stringify(CreateSharedEnvVariableValue$outboundSchema.parse(createSharedEnvVariableValue));
}
export function createSharedEnvVariableValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableValue' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableTarget2$inboundSchema = z.nativeEnum(CreateSharedEnvVariableTarget2);
/** @internal */
export const CreateSharedEnvVariableTarget2$outboundSchema = CreateSharedEnvVariableTarget2$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableTarget1$inboundSchema = z.nativeEnum(CreateSharedEnvVariableTarget1);
/** @internal */
export const CreateSharedEnvVariableTarget1$outboundSchema = CreateSharedEnvVariableTarget1$inboundSchema;
/** @internal */
export const CreateSharedEnvVariableEnvironmentTarget$inboundSchema = z.union([
    z.array(CreateSharedEnvVariableTarget1$inboundSchema),
    CreateSharedEnvVariableTarget2$inboundSchema,
]);
/** @internal */
export const CreateSharedEnvVariableEnvironmentTarget$outboundSchema = z.union([
    z.array(CreateSharedEnvVariableTarget1$outboundSchema),
    CreateSharedEnvVariableTarget2$outboundSchema,
]);
export function createSharedEnvVariableEnvironmentTargetToJSON(createSharedEnvVariableEnvironmentTarget) {
    return JSON.stringify(CreateSharedEnvVariableEnvironmentTarget$outboundSchema.parse(createSharedEnvVariableEnvironmentTarget));
}
export function createSharedEnvVariableEnvironmentTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableEnvironmentTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableEnvironmentTarget' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    key: z.string().optional(),
    envVarId: z.string().optional(),
    envVarKey: z.string().optional(),
    action: z.string().optional(),
    link: z.string().optional(),
    value: z.union([z.string(), z.array(Value2$inboundSchema)]).optional(),
    gitBranch: z.string().optional(),
    target: z.union([
        z.array(CreateSharedEnvVariableTarget1$inboundSchema),
        CreateSharedEnvVariableTarget2$inboundSchema,
    ]).optional(),
    project: z.string().optional(),
});
/** @internal */
export const CreateSharedEnvVariableError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    key: z.string().optional(),
    envVarId: z.string().optional(),
    envVarKey: z.string().optional(),
    action: z.string().optional(),
    link: z.string().optional(),
    value: z.union([z.string(), z.array(Value2$outboundSchema)]).optional(),
    gitBranch: z.string().optional(),
    target: z.union([
        z.array(CreateSharedEnvVariableTarget1$outboundSchema),
        CreateSharedEnvVariableTarget2$outboundSchema,
    ]).optional(),
    project: z.string().optional(),
});
export function createSharedEnvVariableErrorToJSON(createSharedEnvVariableError) {
    return JSON.stringify(CreateSharedEnvVariableError$outboundSchema.parse(createSharedEnvVariableError));
}
export function createSharedEnvVariableErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableError' from JSON`);
}
/** @internal */
export const Failed$inboundSchema = z
    .object({
    error: z.lazy(() => CreateSharedEnvVariableError$inboundSchema),
});
/** @internal */
export const Failed$outboundSchema = z.object({
    error: z.lazy(() => CreateSharedEnvVariableError$outboundSchema),
});
export function failedToJSON(failed) {
    return JSON.stringify(Failed$outboundSchema.parse(failed));
}
export function failedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Failed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Failed' from JSON`);
}
/** @internal */
export const CreateSharedEnvVariableResponseBody$inboundSchema = z.object({
    created: z.array(z.lazy(() => Created$inboundSchema)),
    failed: z.array(z.lazy(() => Failed$inboundSchema)),
});
/** @internal */
export const CreateSharedEnvVariableResponseBody$outboundSchema = z.object({
    created: z.array(z.lazy(() => Created$outboundSchema)),
    failed: z.array(z.lazy(() => Failed$outboundSchema)),
});
export function createSharedEnvVariableResponseBodyToJSON(createSharedEnvVariableResponseBody) {
    return JSON.stringify(CreateSharedEnvVariableResponseBody$outboundSchema.parse(createSharedEnvVariableResponseBody));
}
export function createSharedEnvVariableResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateSharedEnvVariableResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateSharedEnvVariableResponseBody' from JSON`);
}
