// @vercel/sdk/esm/models/createlogdrainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The delivery log format
 */
export const CreateLogDrainDeliveryFormat = {
    Json: "json",
    Ndjson: "ndjson",
};
export const CreateLogDrainSources = {
    Static: "static",
    Lambda: "lambda",
    Build: "build",
    Edge: "edge",
    External: "external",
    Firewall: "firewall",
};
export const CreateLogDrainEnvironments = {
    Preview: "preview",
    Production: "production",
};
/**
 * The delivery log format
 */
export const CreateLogDrainLogDrainsDeliveryFormat = {
    Json: "json",
    Ndjson: "ndjson",
    Protobuf: "protobuf",
};
/**
 * The sources from which logs are currently being delivered to this log drain.
 */
export const CreateLogDrainLogDrainsSources = {
    Build: "build",
    Edge: "edge",
    Lambda: "lambda",
    Static: "static",
    External: "external",
    Firewall: "firewall",
    Redirect: "redirect",
};
/**
 * Whether the log drain was created by an integration or by a user
 */
export const CreateLogDrainCreatedFrom = {
    Integration: "integration",
    SelfServed: "self-served",
};
/**
 * The environment of log drain
 */
export const CreateLogDrainLogDrainsEnvironments = {
    Production: "production",
    Preview: "preview",
};
export const CreateLogDrainSourceLogDrainsKind = {
    Integration: "integration",
};
export const CreateLogDrainSourceKind = {
    SelfServed: "self-served",
};
/** @internal */
export const CreateLogDrainDeliveryFormat$inboundSchema = z.nativeEnum(CreateLogDrainDeliveryFormat);
/** @internal */
export const CreateLogDrainDeliveryFormat$outboundSchema = CreateLogDrainDeliveryFormat$inboundSchema;
/** @internal */
export const CreateLogDrainSources$inboundSchema = z.nativeEnum(CreateLogDrainSources);
/** @internal */
export const CreateLogDrainSources$outboundSchema = CreateLogDrainSources$inboundSchema;
/** @internal */
export const CreateLogDrainEnvironments$inboundSchema = z.nativeEnum(CreateLogDrainEnvironments);
/** @internal */
export const CreateLogDrainEnvironments$outboundSchema = CreateLogDrainEnvironments$inboundSchema;
/** @internal */
export const CreateLogDrainRequestBody$inboundSchema = z.object({
    name: z.string(),
    projectIds: z.array(z.string()).optional(),
    secret: z.string().optional(),
    deliveryFormat: CreateLogDrainDeliveryFormat$inboundSchema.optional(),
    url: z.string(),
    sources: z.array(CreateLogDrainSources$inboundSchema).optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(CreateLogDrainEnvironments$inboundSchema).optional(),
});
/** @internal */
export const CreateLogDrainRequestBody$outboundSchema = z.object({
    name: z.string(),
    projectIds: z.array(z.string()).optional(),
    secret: z.string().optional(),
    deliveryFormat: CreateLogDrainDeliveryFormat$outboundSchema.optional(),
    url: z.string(),
    sources: z.array(CreateLogDrainSources$outboundSchema).optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(CreateLogDrainEnvironments$outboundSchema).optional(),
});
export function createLogDrainRequestBodyToJSON(createLogDrainRequestBody) {
    return JSON.stringify(CreateLogDrainRequestBody$outboundSchema.parse(createLogDrainRequestBody));
}
export function createLogDrainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateLogDrainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateLogDrainRequestBody' from JSON`);
}
/** @internal */
export const CreateLogDrainRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => CreateLogDrainRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const CreateLogDrainRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => CreateLogDrainRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function createLogDrainRequestToJSON(createLogDrainRequest) {
    return JSON.stringify(CreateLogDrainRequest$outboundSchema.parse(createLogDrainRequest));
}
export function createLogDrainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateLogDrainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateLogDrainRequest' from JSON`);
}
/** @internal */
export const CreateLogDrainLogDrainsDeliveryFormat$inboundSchema = z.nativeEnum(CreateLogDrainLogDrainsDeliveryFormat);
/** @internal */
export const CreateLogDrainLogDrainsDeliveryFormat$outboundSchema = CreateLogDrainLogDrainsDeliveryFormat$inboundSchema;
/** @internal */
export const CreateLogDrainLogDrainsSources$inboundSchema = z.nativeEnum(CreateLogDrainLogDrainsSources);
/** @internal */
export const CreateLogDrainLogDrainsSources$outboundSchema = CreateLogDrainLogDrainsSources$inboundSchema;
/** @internal */
export const CreateLogDrainCreatedFrom$inboundSchema = z.nativeEnum(CreateLogDrainCreatedFrom);
/** @internal */
export const CreateLogDrainCreatedFrom$outboundSchema = CreateLogDrainCreatedFrom$inboundSchema;
/** @internal */
export const CreateLogDrainLogDrainsEnvironments$inboundSchema = z.nativeEnum(CreateLogDrainLogDrainsEnvironments);
/** @internal */
export const CreateLogDrainLogDrainsEnvironments$outboundSchema = CreateLogDrainLogDrainsEnvironments$inboundSchema;
/** @internal */
export const CreateLogDrainSourceLogDrainsKind$inboundSchema = z.nativeEnum(CreateLogDrainSourceLogDrainsKind);
/** @internal */
export const CreateLogDrainSourceLogDrainsKind$outboundSchema = CreateLogDrainSourceLogDrainsKind$inboundSchema;
/** @internal */
export const CreateLogDrainSource2$inboundSchema = z.object({
    kind: CreateLogDrainSourceLogDrainsKind$inboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const CreateLogDrainSource2$outboundSchema = z.object({
    kind: CreateLogDrainSourceLogDrainsKind$outboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
export function createLogDrainSource2ToJSON(createLogDrainSource2) {
    return JSON.stringify(CreateLogDrainSource2$outboundSchema.parse(createLogDrainSource2));
}
export function createLogDrainSource2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateLogDrainSource2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateLogDrainSource2' from JSON`);
}
/** @internal */
export const CreateLogDrainSourceKind$inboundSchema = z.nativeEnum(CreateLogDrainSourceKind);
/** @internal */
export const CreateLogDrainSourceKind$outboundSchema = CreateLogDrainSourceKind$inboundSchema;
/** @internal */
export const CreateLogDrainSource1$inboundSchema = z.object({
    kind: CreateLogDrainSourceKind$inboundSchema,
});
/** @internal */
export const CreateLogDrainSource1$outboundSchema = z.object({
    kind: CreateLogDrainSourceKind$outboundSchema,
});
export function createLogDrainSource1ToJSON(createLogDrainSource1) {
    return JSON.stringify(CreateLogDrainSource1$outboundSchema.parse(createLogDrainSource1));
}
export function createLogDrainSource1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateLogDrainSource1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateLogDrainSource1' from JSON`);
}
/** @internal */
export const CreateLogDrainSource$inboundSchema = z.union([
    z.lazy(() => CreateLogDrainSource2$inboundSchema),
    z.lazy(() => CreateLogDrainSource1$inboundSchema),
]);
/** @internal */
export const CreateLogDrainSource$outboundSchema = z.union([
    z.lazy(() => CreateLogDrainSource2$outboundSchema),
    z.lazy(() => CreateLogDrainSource1$outboundSchema),
]);
export function createLogDrainSourceToJSON(createLogDrainSource) {
    return JSON.stringify(CreateLogDrainSource$outboundSchema.parse(createLogDrainSource));
}
export function createLogDrainSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateLogDrainSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateLogDrainSource' from JSON`);
}
/** @internal */
export const CreateLogDrainResponseBody$inboundSchema = z.object({
    clientId: z.string().optional(),
    configurationId: z.string().optional(),
    createdAt: z.number(),
    id: z.string(),
    deliveryFormat: CreateLogDrainLogDrainsDeliveryFormat$inboundSchema
        .optional(),
    name: z.string(),
    ownerId: z.string(),
    projectId: z.nullable(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    url: z.string(),
    sources: z.array(CreateLogDrainLogDrainsSources$inboundSchema).optional(),
    createdFrom: CreateLogDrainCreatedFrom$inboundSchema.optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(CreateLogDrainLogDrainsEnvironments$inboundSchema)
        .optional(),
    branch: z.string().optional(),
    samplingRate: z.number().optional(),
    source: z.union([
        z.lazy(() => CreateLogDrainSource2$inboundSchema),
        z.lazy(() => CreateLogDrainSource1$inboundSchema),
    ]),
});
/** @internal */
export const CreateLogDrainResponseBody$outboundSchema = z.object({
    clientId: z.string().optional(),
    configurationId: z.string().optional(),
    createdAt: z.number(),
    id: z.string(),
    deliveryFormat: CreateLogDrainLogDrainsDeliveryFormat$outboundSchema
        .optional(),
    name: z.string(),
    ownerId: z.string(),
    projectId: z.nullable(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    url: z.string(),
    sources: z.array(CreateLogDrainLogDrainsSources$outboundSchema).optional(),
    createdFrom: CreateLogDrainCreatedFrom$outboundSchema.optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(CreateLogDrainLogDrainsEnvironments$outboundSchema)
        .optional(),
    branch: z.string().optional(),
    samplingRate: z.number().optional(),
    source: z.union([
        z.lazy(() => CreateLogDrainSource2$outboundSchema),
        z.lazy(() => CreateLogDrainSource1$outboundSchema),
    ]),
});
export function createLogDrainResponseBodyToJSON(createLogDrainResponseBody) {
    return JSON.stringify(CreateLogDrainResponseBody$outboundSchema.parse(createLogDrainResponseBody));
}
export function createLogDrainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CreateLogDrainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateLogDrainResponseBody' from JSON`);
}
