// @vercel/sdk/esm/models/deleteprojectclientcertop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteProjectClientCertRequest$inboundSchema = z.object({
    idOrName: z.string(),
    certId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const DeleteProjectClientCertRequest$outboundSchema = z.object({
    idOrName: z.string(),
    certId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function deleteProjectClientCertRequestToJSON(deleteProjectClientCertRequest) {
    return JSON.stringify(DeleteProjectClientCertRequest$outboundSchema.parse(deleteProjectClientCertRequest));
}
export function deleteProjectClientCertRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteProjectClientCertRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteProjectClientCertRequest' from JSON`);
}
/** @internal */
export const DeleteProjectClientCertResponseBody$inboundSchema = z.object({
    origin: z.string(),
    certId: z.string(),
});
/** @internal */
export const DeleteProjectClientCertResponseBody$outboundSchema = z.object({
    origin: z.string(),
    certId: z.string(),
});
export function deleteProjectClientCertResponseBodyToJSON(deleteProjectClientCertResponseBody) {
    return JSON.stringify(DeleteProjectClientCertResponseBody$outboundSchema.parse(deleteProjectClientCertResponseBody));
}
export function deleteProjectClientCertResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteProjectClientCertResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteProjectClientCertResponseBody' from JSON`);
}
