// @vercel/sdk/esm/models/editprojectenvop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const EditProjectEnvTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The type of environment variable
 */
export const EditProjectEnvType = {
    System: "system",
    Secret: "secret",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
};
export const EditProjectEnvTarget2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const EditProjectEnvResponseBodyType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type = {
    PostgresDatabase: "postgres-database",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type = {
    PostgresPassword: "postgres-password",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type = {
    PostgresHost: "postgres-host",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type = {
    PostgresUser: "postgres-user",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType = {
    PostgresUrl: "postgres-url",
};
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONType = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const EditProjectEnvContentHintProjectsResponse200Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const EditProjectEnvContentHintProjectsResponseType = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const EditProjectEnvContentHintProjectsType = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const EditProjectEnvContentHintType = {
    RedisUrl: "redis-url",
};
export const EditProjectEnvResponseBodyProjectsType = {
    FlagsSecret: "flags-secret",
};
/** @internal */
export const EditProjectEnvTarget$inboundSchema = z.nativeEnum(EditProjectEnvTarget);
/** @internal */
export const EditProjectEnvTarget$outboundSchema = EditProjectEnvTarget$inboundSchema;
/** @internal */
export const EditProjectEnvType$inboundSchema = z.nativeEnum(EditProjectEnvType);
/** @internal */
export const EditProjectEnvType$outboundSchema = EditProjectEnvType$inboundSchema;
/** @internal */
export const EditProjectEnvRequestBody$inboundSchema = z.object({
    key: z.string().optional(),
    target: z.array(EditProjectEnvTarget$inboundSchema).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    type: EditProjectEnvType$inboundSchema.optional(),
    value: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
    comment: z.string().optional(),
});
/** @internal */
export const EditProjectEnvRequestBody$outboundSchema = z.object({
    key: z.string().optional(),
    target: z.array(EditProjectEnvTarget$outboundSchema).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    type: EditProjectEnvType$outboundSchema.optional(),
    value: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
    comment: z.string().optional(),
});
export function editProjectEnvRequestBodyToJSON(editProjectEnvRequestBody) {
    return JSON.stringify(EditProjectEnvRequestBody$outboundSchema.parse(editProjectEnvRequestBody));
}
export function editProjectEnvRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvRequestBody' from JSON`);
}
/** @internal */
export const EditProjectEnvRequest$inboundSchema = z.object({
    idOrName: z.string(),
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => EditProjectEnvRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const EditProjectEnvRequest$outboundSchema = z.object({
    idOrName: z.string(),
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => EditProjectEnvRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function editProjectEnvRequestToJSON(editProjectEnvRequest) {
    return JSON.stringify(EditProjectEnvRequest$outboundSchema.parse(editProjectEnvRequest));
}
export function editProjectEnvRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvRequest' from JSON`);
}
/** @internal */
export const EditProjectEnvResponseBody2$inboundSchema = z.object({});
/** @internal */
export const EditProjectEnvResponseBody2$outboundSchema = z.object({});
export function editProjectEnvResponseBody2ToJSON(editProjectEnvResponseBody2) {
    return JSON.stringify(EditProjectEnvResponseBody2$outboundSchema.parse(editProjectEnvResponseBody2));
}
export function editProjectEnvResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvResponseBody2' from JSON`);
}
/** @internal */
export const EditProjectEnvTarget2$inboundSchema = z.nativeEnum(EditProjectEnvTarget2);
/** @internal */
export const EditProjectEnvTarget2$outboundSchema = EditProjectEnvTarget2$inboundSchema;
/** @internal */
export const EditProjectEnvResponseBodyTarget$inboundSchema = z.union([z.array(z.string()), EditProjectEnvTarget2$inboundSchema]);
/** @internal */
export const EditProjectEnvResponseBodyTarget$outboundSchema = z.union([z.array(z.string()), EditProjectEnvTarget2$outboundSchema]);
export function editProjectEnvResponseBodyTargetToJSON(editProjectEnvResponseBodyTarget) {
    return JSON.stringify(EditProjectEnvResponseBodyTarget$outboundSchema.parse(editProjectEnvResponseBodyTarget));
}
export function editProjectEnvResponseBodyTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvResponseBodyTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvResponseBodyTarget' from JSON`);
}
/** @internal */
export const EditProjectEnvResponseBodyType$inboundSchema = z.nativeEnum(EditProjectEnvResponseBodyType);
/** @internal */
export const EditProjectEnvResponseBodyType$outboundSchema = EditProjectEnvResponseBodyType$inboundSchema;
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint15$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint15$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$outboundSchema,
    projectId: z.string(),
});
export function editProjectEnvContentHint15ToJSON(editProjectEnvContentHint15) {
    return JSON.stringify(EditProjectEnvContentHint15$outboundSchema.parse(editProjectEnvContentHint15));
}
export function editProjectEnvContentHint15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint15' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint14$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint14$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function editProjectEnvContentHint14ToJSON(editProjectEnvContentHint14) {
    return JSON.stringify(EditProjectEnvContentHint14$outboundSchema.parse(editProjectEnvContentHint14));
}
export function editProjectEnvContentHint14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint14' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint13$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint13$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint13ToJSON(editProjectEnvContentHint13) {
    return JSON.stringify(EditProjectEnvContentHint13$outboundSchema.parse(editProjectEnvContentHint13));
}
export function editProjectEnvContentHint13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint13' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint12$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint12$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint12ToJSON(editProjectEnvContentHint12) {
    return JSON.stringify(EditProjectEnvContentHint12$outboundSchema.parse(editProjectEnvContentHint12));
}
export function editProjectEnvContentHint12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint12' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint11$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint11$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint11ToJSON(editProjectEnvContentHint11) {
    return JSON.stringify(EditProjectEnvContentHint11$outboundSchema.parse(editProjectEnvContentHint11));
}
export function editProjectEnvContentHint11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint11' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint10$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint10$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint10ToJSON(editProjectEnvContentHint10) {
    return JSON.stringify(EditProjectEnvContentHint10$outboundSchema.parse(editProjectEnvContentHint10));
}
export function editProjectEnvContentHint10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint10' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint9$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint9$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint9ToJSON(editProjectEnvContentHint9) {
    return JSON.stringify(EditProjectEnvContentHint9$outboundSchema.parse(editProjectEnvContentHint9));
}
export function editProjectEnvContentHint9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint9' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint8$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint8$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint8ToJSON(editProjectEnvContentHint8) {
    return JSON.stringify(EditProjectEnvContentHint8$outboundSchema.parse(editProjectEnvContentHint8));
}
export function editProjectEnvContentHint8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint8' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint7$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint7$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint7ToJSON(editProjectEnvContentHint7) {
    return JSON.stringify(EditProjectEnvContentHint7$outboundSchema.parse(editProjectEnvContentHint7));
}
export function editProjectEnvContentHint7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint7' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint6$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint6$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint6ToJSON(editProjectEnvContentHint6) {
    return JSON.stringify(EditProjectEnvContentHint6$outboundSchema.parse(editProjectEnvContentHint6));
}
export function editProjectEnvContentHint6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint6' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsResponse200ApplicationJSONType);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200ApplicationJSONType$outboundSchema = EditProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint5$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint5$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200ApplicationJSONType$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint5ToJSON(editProjectEnvContentHint5) {
    return JSON.stringify(EditProjectEnvContentHint5$outboundSchema.parse(editProjectEnvContentHint5));
}
export function editProjectEnvContentHint5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint5' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200Type$inboundSchema = z
    .nativeEnum(EditProjectEnvContentHintProjectsResponse200Type);
/** @internal */
export const EditProjectEnvContentHintProjectsResponse200Type$outboundSchema = EditProjectEnvContentHintProjectsResponse200Type$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint4$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint4$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponse200Type$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint4ToJSON(editProjectEnvContentHint4) {
    return JSON.stringify(EditProjectEnvContentHint4$outboundSchema.parse(editProjectEnvContentHint4));
}
export function editProjectEnvContentHint4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint4' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsResponseType$inboundSchema = z
    .nativeEnum(EditProjectEnvContentHintProjectsResponseType);
/** @internal */
export const EditProjectEnvContentHintProjectsResponseType$outboundSchema = EditProjectEnvContentHintProjectsResponseType$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint3$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponseType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint3$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsResponseType$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint3ToJSON(editProjectEnvContentHint3) {
    return JSON.stringify(EditProjectEnvContentHint3$outboundSchema.parse(editProjectEnvContentHint3));
}
export function editProjectEnvContentHint3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint3' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintProjectsType$inboundSchema = z.nativeEnum(EditProjectEnvContentHintProjectsType);
/** @internal */
export const EditProjectEnvContentHintProjectsType$outboundSchema = EditProjectEnvContentHintProjectsType$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint2$inboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint2$outboundSchema = z.object({
    type: EditProjectEnvContentHintProjectsType$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint2ToJSON(editProjectEnvContentHint2) {
    return JSON.stringify(EditProjectEnvContentHint2$outboundSchema.parse(editProjectEnvContentHint2));
}
export function editProjectEnvContentHint2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint2' from JSON`);
}
/** @internal */
export const EditProjectEnvContentHintType$inboundSchema = z.nativeEnum(EditProjectEnvContentHintType);
/** @internal */
export const EditProjectEnvContentHintType$outboundSchema = EditProjectEnvContentHintType$inboundSchema;
/** @internal */
export const EditProjectEnvContentHint1$inboundSchema = z.object({
    type: EditProjectEnvContentHintType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const EditProjectEnvContentHint1$outboundSchema = z.object({
    type: EditProjectEnvContentHintType$outboundSchema,
    storeId: z.string(),
});
export function editProjectEnvContentHint1ToJSON(editProjectEnvContentHint1) {
    return JSON.stringify(EditProjectEnvContentHint1$outboundSchema.parse(editProjectEnvContentHint1));
}
export function editProjectEnvContentHint1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvContentHint1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvContentHint1' from JSON`);
}
/** @internal */
export const EditProjectEnvResponseBodyContentHint$inboundSchema = z.union([
    z.lazy(() => EditProjectEnvContentHint14$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint1$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint2$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint3$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint4$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint5$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint6$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint7$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint8$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint9$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint10$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint11$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint12$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint13$inboundSchema),
    z.lazy(() => EditProjectEnvContentHint15$inboundSchema),
]);
/** @internal */
export const EditProjectEnvResponseBodyContentHint$outboundSchema = z.union([
    z.lazy(() => EditProjectEnvContentHint14$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint1$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint2$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint3$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint4$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint5$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint6$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint7$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint8$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint9$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint10$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint11$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint12$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint13$outboundSchema),
    z.lazy(() => EditProjectEnvContentHint15$outboundSchema),
]);
export function editProjectEnvResponseBodyContentHintToJSON(editProjectEnvResponseBodyContentHint) {
    return JSON.stringify(EditProjectEnvResponseBodyContentHint$outboundSchema.parse(editProjectEnvResponseBodyContentHint));
}
export function editProjectEnvResponseBodyContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvResponseBodyContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvResponseBodyContentHint' from JSON`);
}
/** @internal */
export const EditProjectEnvResponseBodyProjectsType$inboundSchema = z.nativeEnum(EditProjectEnvResponseBodyProjectsType);
/** @internal */
export const EditProjectEnvResponseBodyProjectsType$outboundSchema = EditProjectEnvResponseBodyProjectsType$inboundSchema;
/** @internal */
export const EditProjectEnvResponseBodyInternalContentHint$inboundSchema = z.object({
    type: EditProjectEnvResponseBodyProjectsType$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const EditProjectEnvResponseBodyInternalContentHint$outboundSchema = z.object({
    type: EditProjectEnvResponseBodyProjectsType$outboundSchema,
    encryptedValue: z.string(),
});
export function editProjectEnvResponseBodyInternalContentHintToJSON(editProjectEnvResponseBodyInternalContentHint) {
    return JSON.stringify(EditProjectEnvResponseBodyInternalContentHint$outboundSchema.parse(editProjectEnvResponseBodyInternalContentHint));
}
export function editProjectEnvResponseBodyInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvResponseBodyInternalContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvResponseBodyInternalContentHint' from JSON`);
}
/** @internal */
export const EditProjectEnvResponseBody1$inboundSchema = z.object({
    target: z.union([z.array(z.string()), EditProjectEnvTarget2$inboundSchema])
        .optional(),
    type: EditProjectEnvResponseBodyType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => EditProjectEnvContentHint14$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint1$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint2$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint3$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint4$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint5$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint6$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint7$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint8$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint9$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint10$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint11$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint12$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint13$inboundSchema),
        z.lazy(() => EditProjectEnvContentHint15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => EditProjectEnvResponseBodyInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const EditProjectEnvResponseBody1$outboundSchema = z.object({
    target: z.union([z.array(z.string()), EditProjectEnvTarget2$outboundSchema])
        .optional(),
    type: EditProjectEnvResponseBodyType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => EditProjectEnvContentHint14$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint1$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint2$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint3$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint4$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint5$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint6$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint7$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint8$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint9$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint10$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint11$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint12$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint13$outboundSchema),
        z.lazy(() => EditProjectEnvContentHint15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => EditProjectEnvResponseBodyInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function editProjectEnvResponseBody1ToJSON(editProjectEnvResponseBody1) {
    return JSON.stringify(EditProjectEnvResponseBody1$outboundSchema.parse(editProjectEnvResponseBody1));
}
export function editProjectEnvResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvResponseBody1' from JSON`);
}
/** @internal */
export const EditProjectEnvResponseBody$inboundSchema = z.union([
    z.lazy(() => EditProjectEnvResponseBody1$inboundSchema),
    z.lazy(() => EditProjectEnvResponseBody2$inboundSchema),
]);
/** @internal */
export const EditProjectEnvResponseBody$outboundSchema = z.union([
    z.lazy(() => EditProjectEnvResponseBody1$outboundSchema),
    z.lazy(() => EditProjectEnvResponseBody2$outboundSchema),
]);
export function editProjectEnvResponseBodyToJSON(editProjectEnvResponseBody) {
    return JSON.stringify(EditProjectEnvResponseBody$outboundSchema.parse(editProjectEnvResponseBody));
}
export function editProjectEnvResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditProjectEnvResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditProjectEnvResponseBody' from JSON`);
}
