// @vercel/sdk/esm/models/exchangessotokenop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * The grant type, when using x-www-form-urlencoded content type
 */
export const ExchangeSsoTokenGrantType = {
    AuthorizationCode: "authorization_code",
};
/** @internal */
export const ExchangeSsoTokenGrantType$inboundSchema = z.nativeEnum(ExchangeSsoTokenGrantType);
/** @internal */
export const ExchangeSsoTokenGrantType$outboundSchema = ExchangeSsoTokenGrantType$inboundSchema;
/** @internal */
export const ExchangeSsoTokenRequestBody$inboundSchema = z.object({
    code: z.string(),
    state: z.string().optional(),
    client_id: z.string(),
    client_secret: z.string(),
    redirect_uri: z.string().optional(),
    grant_type: ExchangeSsoTokenGrantType$inboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        "client_id": "clientId",
        "client_secret": "clientSecret",
        "redirect_uri": "redirectUri",
        "grant_type": "grantType",
    });
});
/** @internal */
export const ExchangeSsoTokenRequestBody$outboundSchema = z.object({
    code: z.string(),
    state: z.string().optional(),
    clientId: z.string(),
    clientSecret: z.string(),
    redirectUri: z.string().optional(),
    grantType: ExchangeSsoTokenGrantType$outboundSchema.optional(),
}).transform((v) => {
    return remap$(v, {
        clientId: "client_id",
        clientSecret: "client_secret",
        redirectUri: "redirect_uri",
        grantType: "grant_type",
    });
});
export function exchangeSsoTokenRequestBodyToJSON(exchangeSsoTokenRequestBody) {
    return JSON.stringify(ExchangeSsoTokenRequestBody$outboundSchema.parse(exchangeSsoTokenRequestBody));
}
export function exchangeSsoTokenRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ExchangeSsoTokenRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ExchangeSsoTokenRequestBody' from JSON`);
}
/** @internal */
export const ExchangeSsoTokenResponseBody$inboundSchema = z.object({
    id_token: z.string(),
    access_token: z.nullable(z.string()),
    token_type: z.nullable(z.string()),
    expires_in: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        "id_token": "idToken",
        "access_token": "accessToken",
        "token_type": "tokenType",
        "expires_in": "expiresIn",
    });
});
/** @internal */
export const ExchangeSsoTokenResponseBody$outboundSchema = z.object({
    idToken: z.string(),
    accessToken: z.nullable(z.string()),
    tokenType: z.nullable(z.string()),
    expiresIn: z.number().optional(),
}).transform((v) => {
    return remap$(v, {
        idToken: "id_token",
        accessToken: "access_token",
        tokenType: "token_type",
        expiresIn: "expires_in",
    });
});
export function exchangeSsoTokenResponseBodyToJSON(exchangeSsoTokenResponseBody) {
    return JSON.stringify(ExchangeSsoTokenResponseBody$outboundSchema.parse(exchangeSsoTokenResponseBody));
}
export function exchangeSsoTokenResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ExchangeSsoTokenResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ExchangeSsoTokenResponseBody' from JSON`);
}
