// @vercel/sdk/esm/models/getaccountinfoop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetAccountInfoRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
});
/** @internal */
export const GetAccountInfoRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
});
export function getAccountInfoRequestToJSON(getAccountInfoRequest) {
    return JSON.stringify(GetAccountInfoRequest$outboundSchema.parse(getAccountInfoRequest));
}
export function getAccountInfoRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAccountInfoRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAccountInfoRequest' from JSON`);
}
/** @internal */
export const Contact$inboundSchema = z.object({
    email: z.string(),
    name: z.string().optional(),
});
/** @internal */
export const Contact$outboundSchema = z.object({
    email: z.string(),
    name: z.string().optional(),
});
export function contactToJSON(contact) {
    return JSON.stringify(Contact$outboundSchema.parse(contact));
}
export function contactFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Contact$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Contact' from JSON`);
}
/** @internal */
export const GetAccountInfoResponseBody$inboundSchema = z.object({
    name: z.string().optional(),
    url: z.string(),
    contact: z.nullable(z.lazy(() => Contact$inboundSchema)),
});
/** @internal */
export const GetAccountInfoResponseBody$outboundSchema = z.object({
    name: z.string().optional(),
    url: z.string(),
    contact: z.nullable(z.lazy(() => Contact$outboundSchema)),
});
export function getAccountInfoResponseBodyToJSON(getAccountInfoResponseBody) {
    return JSON.stringify(GetAccountInfoResponseBody$outboundSchema.parse(getAccountInfoResponseBody));
}
export function getAccountInfoResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAccountInfoResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAccountInfoResponseBody' from JSON`);
}
