// @vercel/sdk/esm/models/getaliasop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const GetAliasProtectionBypassAliasesResponse200Scope = {
    EmailInvite: "email_invite",
};
export const GetAliasProtectionBypassAliasesResponseScope = {
    AliasProtectionOverride: "alias-protection-override",
};
export const ProtectionBypassAccess = {
    Requested: "requested",
    Granted: "granted",
};
export const GetAliasProtectionBypassAliasesScope = {
    User: "user",
};
export const GetAliasProtectionBypassScope = {
    ShareableLink: "shareable-link",
};
/** @internal */
export const GetAliasRequest$inboundSchema = z.object({
    from: z.number().optional(),
    idOrAlias: z.string(),
    projectId: z.string().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetAliasRequest$outboundSchema = z.object({
    from: z.number().optional(),
    idOrAlias: z.string(),
    projectId: z.string().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getAliasRequestToJSON(getAliasRequest) {
    return JSON.stringify(GetAliasRequest$outboundSchema.parse(getAliasRequest));
}
export function getAliasRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasRequest' from JSON`);
}
/** @internal */
export const GetAliasCreator$inboundSchema = z.object({
    uid: z.string(),
    email: z.string(),
    username: z.string(),
});
/** @internal */
export const GetAliasCreator$outboundSchema = z.object({
    uid: z.string(),
    email: z.string(),
    username: z.string(),
});
export function getAliasCreatorToJSON(getAliasCreator) {
    return JSON.stringify(GetAliasCreator$outboundSchema.parse(getAliasCreator));
}
export function getAliasCreatorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasCreator$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasCreator' from JSON`);
}
/** @internal */
export const GetAliasDeployment$inboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    meta: z.string().optional(),
});
/** @internal */
export const GetAliasDeployment$outboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    meta: z.string().optional(),
});
export function getAliasDeploymentToJSON(getAliasDeployment) {
    return JSON.stringify(GetAliasDeployment$outboundSchema.parse(getAliasDeployment));
}
export function getAliasDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasDeployment' from JSON`);
}
/** @internal */
export const GetAliasProtectionBypassAliasesResponse200Scope$inboundSchema = z
    .nativeEnum(GetAliasProtectionBypassAliasesResponse200Scope);
/** @internal */
export const GetAliasProtectionBypassAliasesResponse200Scope$outboundSchema = GetAliasProtectionBypassAliasesResponse200Scope$inboundSchema;
/** @internal */
export const ProtectionBypass4$inboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    scope: GetAliasProtectionBypassAliasesResponse200Scope$inboundSchema,
});
/** @internal */
export const ProtectionBypass4$outboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    scope: GetAliasProtectionBypassAliasesResponse200Scope$outboundSchema,
});
export function protectionBypass4ToJSON(protectionBypass4) {
    return JSON.stringify(ProtectionBypass4$outboundSchema.parse(protectionBypass4));
}
export function protectionBypass4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProtectionBypass4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProtectionBypass4' from JSON`);
}
/** @internal */
export const GetAliasProtectionBypassAliasesResponseScope$inboundSchema = z
    .nativeEnum(GetAliasProtectionBypassAliasesResponseScope);
/** @internal */
export const GetAliasProtectionBypassAliasesResponseScope$outboundSchema = GetAliasProtectionBypassAliasesResponseScope$inboundSchema;
/** @internal */
export const ProtectionBypass3$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: GetAliasProtectionBypassAliasesResponseScope$inboundSchema,
});
/** @internal */
export const ProtectionBypass3$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: GetAliasProtectionBypassAliasesResponseScope$outboundSchema,
});
export function protectionBypass3ToJSON(protectionBypass3) {
    return JSON.stringify(ProtectionBypass3$outboundSchema.parse(protectionBypass3));
}
export function protectionBypass3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProtectionBypass3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProtectionBypass3' from JSON`);
}
/** @internal */
export const ProtectionBypassAccess$inboundSchema = z.nativeEnum(ProtectionBypassAccess);
/** @internal */
export const ProtectionBypassAccess$outboundSchema = ProtectionBypassAccess$inboundSchema;
/** @internal */
export const GetAliasProtectionBypassAliasesScope$inboundSchema = z.nativeEnum(GetAliasProtectionBypassAliasesScope);
/** @internal */
export const GetAliasProtectionBypassAliasesScope$outboundSchema = GetAliasProtectionBypassAliasesScope$inboundSchema;
/** @internal */
export const GetAliasProtectionBypass2$inboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    access: ProtectionBypassAccess$inboundSchema,
    scope: GetAliasProtectionBypassAliasesScope$inboundSchema,
});
/** @internal */
export const GetAliasProtectionBypass2$outboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    access: ProtectionBypassAccess$outboundSchema,
    scope: GetAliasProtectionBypassAliasesScope$outboundSchema,
});
export function getAliasProtectionBypass2ToJSON(getAliasProtectionBypass2) {
    return JSON.stringify(GetAliasProtectionBypass2$outboundSchema.parse(getAliasProtectionBypass2));
}
export function getAliasProtectionBypass2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasProtectionBypass2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasProtectionBypass2' from JSON`);
}
/** @internal */
export const GetAliasProtectionBypassScope$inboundSchema = z.nativeEnum(GetAliasProtectionBypassScope);
/** @internal */
export const GetAliasProtectionBypassScope$outboundSchema = GetAliasProtectionBypassScope$inboundSchema;
/** @internal */
export const GetAliasProtectionBypass1$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: GetAliasProtectionBypassScope$inboundSchema,
    expires: z.number().optional(),
});
/** @internal */
export const GetAliasProtectionBypass1$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: GetAliasProtectionBypassScope$outboundSchema,
    expires: z.number().optional(),
});
export function getAliasProtectionBypass1ToJSON(getAliasProtectionBypass1) {
    return JSON.stringify(GetAliasProtectionBypass1$outboundSchema.parse(getAliasProtectionBypass1));
}
export function getAliasProtectionBypass1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasProtectionBypass1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasProtectionBypass1' from JSON`);
}
/** @internal */
export const GetAliasProtectionBypass$inboundSchema = z.union([
    z.lazy(() => GetAliasProtectionBypass2$inboundSchema),
    z.lazy(() => ProtectionBypass4$inboundSchema),
    z.lazy(() => GetAliasProtectionBypass1$inboundSchema),
    z.lazy(() => ProtectionBypass3$inboundSchema),
]);
/** @internal */
export const GetAliasProtectionBypass$outboundSchema = z.union([
    z.lazy(() => GetAliasProtectionBypass2$outboundSchema),
    z.lazy(() => ProtectionBypass4$outboundSchema),
    z.lazy(() => GetAliasProtectionBypass1$outboundSchema),
    z.lazy(() => ProtectionBypass3$outboundSchema),
]);
export function getAliasProtectionBypassToJSON(getAliasProtectionBypass) {
    return JSON.stringify(GetAliasProtectionBypass$outboundSchema.parse(getAliasProtectionBypass));
}
export function getAliasProtectionBypassFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasProtectionBypass$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasProtectionBypass' from JSON`);
}
/** @internal */
export const DefaultApp$inboundSchema = z.object({
    projectId: z.string(),
});
/** @internal */
export const DefaultApp$outboundSchema = z.object({
    projectId: z.string(),
});
export function defaultAppToJSON(defaultApp) {
    return JSON.stringify(DefaultApp$outboundSchema.parse(defaultApp));
}
export function defaultAppFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DefaultApp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DefaultApp' from JSON`);
}
/** @internal */
export const Applications3$inboundSchema = z.object({
    deploymentId: z.string().optional(),
    branchDeploymentId: z.string().optional(),
    fallbackDeploymentId: z.string().optional(),
    fallbackHost: z.string().optional(),
    branchAlias: z.string().optional(),
    projectId: z.string(),
});
/** @internal */
export const Applications3$outboundSchema = z.object({
    deploymentId: z.string().optional(),
    branchDeploymentId: z.string().optional(),
    fallbackDeploymentId: z.string().optional(),
    fallbackHost: z.string().optional(),
    branchAlias: z.string().optional(),
    projectId: z.string(),
});
export function applications3ToJSON(applications3) {
    return JSON.stringify(Applications3$outboundSchema.parse(applications3));
}
export function applications3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Applications3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Applications3' from JSON`);
}
/** @internal */
export const Applications2$inboundSchema = z.object({
    fallbackHost: z.string(),
    branchAlias: z.string(),
    projectId: z.string(),
});
/** @internal */
export const Applications2$outboundSchema = z.object({
    fallbackHost: z.string(),
    branchAlias: z.string(),
    projectId: z.string(),
});
export function applications2ToJSON(applications2) {
    return JSON.stringify(Applications2$outboundSchema.parse(applications2));
}
export function applications2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Applications2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Applications2' from JSON`);
}
/** @internal */
export const Applications1$inboundSchema = z.object({
    fallbackHost: z.string(),
    projectId: z.string(),
});
/** @internal */
export const Applications1$outboundSchema = z.object({
    fallbackHost: z.string(),
    projectId: z.string(),
});
export function applications1ToJSON(applications1) {
    return JSON.stringify(Applications1$outboundSchema.parse(applications1));
}
export function applications1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Applications1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Applications1' from JSON`);
}
/** @internal */
export const Applications$inboundSchema = z.union([
    z.array(z.lazy(() => Applications1$inboundSchema)),
    z.array(z.lazy(() => Applications2$inboundSchema)),
    z.array(z.lazy(() => Applications3$inboundSchema)),
]);
/** @internal */
export const Applications$outboundSchema = z.union([
    z.array(z.lazy(() => Applications1$outboundSchema)),
    z.array(z.lazy(() => Applications2$outboundSchema)),
    z.array(z.lazy(() => Applications3$outboundSchema)),
]);
export function applicationsToJSON(applications) {
    return JSON.stringify(Applications$outboundSchema.parse(applications));
}
export function applicationsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Applications$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Applications' from JSON`);
}
/** @internal */
export const GetAliasMicrofrontends$inboundSchema = z.object({
    defaultApp: z.lazy(() => DefaultApp$inboundSchema),
    applications: z.union([
        z.array(z.lazy(() => Applications1$inboundSchema)),
        z.array(z.lazy(() => Applications2$inboundSchema)),
        z.array(z.lazy(() => Applications3$inboundSchema)),
    ]),
});
/** @internal */
export const GetAliasMicrofrontends$outboundSchema = z.object({
    defaultApp: z.lazy(() => DefaultApp$outboundSchema),
    applications: z.union([
        z.array(z.lazy(() => Applications1$outboundSchema)),
        z.array(z.lazy(() => Applications2$outboundSchema)),
        z.array(z.lazy(() => Applications3$outboundSchema)),
    ]),
});
export function getAliasMicrofrontendsToJSON(getAliasMicrofrontends) {
    return JSON.stringify(GetAliasMicrofrontends$outboundSchema.parse(getAliasMicrofrontends));
}
export function getAliasMicrofrontendsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasMicrofrontends$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasMicrofrontends' from JSON`);
}
/** @internal */
export const GetAliasResponseBody$inboundSchema = z.object({
    alias: z.string(),
    created: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    createdAt: z.number().optional(),
    creator: z.lazy(() => GetAliasCreator$inboundSchema).optional(),
    deletedAt: z.number().optional(),
    deployment: z.lazy(() => GetAliasDeployment$inboundSchema).optional(),
    deploymentId: z.nullable(z.string()),
    projectId: z.nullable(z.string()),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    uid: z.string(),
    updatedAt: z.number().optional(),
    protectionBypass: z.record(z.union([
        z.lazy(() => GetAliasProtectionBypass2$inboundSchema),
        z.lazy(() => ProtectionBypass4$inboundSchema),
        z.lazy(() => GetAliasProtectionBypass1$inboundSchema),
        z.lazy(() => ProtectionBypass3$inboundSchema),
    ])).optional(),
    microfrontends: z.lazy(() => GetAliasMicrofrontends$inboundSchema).optional(),
});
/** @internal */
export const GetAliasResponseBody$outboundSchema = z.object({
    alias: z.string(),
    created: z.date().transform(v => v.toISOString()),
    createdAt: z.number().optional(),
    creator: z.lazy(() => GetAliasCreator$outboundSchema).optional(),
    deletedAt: z.number().optional(),
    deployment: z.lazy(() => GetAliasDeployment$outboundSchema).optional(),
    deploymentId: z.nullable(z.string()),
    projectId: z.nullable(z.string()),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    uid: z.string(),
    updatedAt: z.number().optional(),
    protectionBypass: z.record(z.union([
        z.lazy(() => GetAliasProtectionBypass2$outboundSchema),
        z.lazy(() => ProtectionBypass4$outboundSchema),
        z.lazy(() => GetAliasProtectionBypass1$outboundSchema),
        z.lazy(() => ProtectionBypass3$outboundSchema),
    ])).optional(),
    microfrontends: z.lazy(() => GetAliasMicrofrontends$outboundSchema)
        .optional(),
});
export function getAliasResponseBodyToJSON(getAliasResponseBody) {
    return JSON.stringify(GetAliasResponseBody$outboundSchema.parse(getAliasResponseBody));
}
export function getAliasResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetAliasResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAliasResponseBody' from JSON`);
}
