// @vercel/sdk/esm/models/getbillingplansop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const GetBillingPlansType = {
    Prepayment: "prepayment",
    Subscription: "subscription",
};
export const GetBillingPlansScope = {
    Installation: "installation",
    Resource: "resource",
};
/** @internal */
export const GetBillingPlansRequest$inboundSchema = z.object({
    integrationIdOrSlug: z.string(),
    productIdOrSlug: z.string(),
    metadata: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetBillingPlansRequest$outboundSchema = z.object({
    integrationIdOrSlug: z.string(),
    productIdOrSlug: z.string(),
    metadata: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getBillingPlansRequestToJSON(getBillingPlansRequest) {
    return JSON.stringify(GetBillingPlansRequest$outboundSchema.parse(getBillingPlansRequest));
}
export function getBillingPlansRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBillingPlansRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBillingPlansRequest' from JSON`);
}
/** @internal */
export const GetBillingPlansType$inboundSchema = z.nativeEnum(GetBillingPlansType);
/** @internal */
export const GetBillingPlansType$outboundSchema = GetBillingPlansType$inboundSchema;
/** @internal */
export const GetBillingPlansScope$inboundSchema = z.nativeEnum(GetBillingPlansScope);
/** @internal */
export const GetBillingPlansScope$outboundSchema = GetBillingPlansScope$inboundSchema;
/** @internal */
export const GetBillingPlansDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const GetBillingPlansDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
export function getBillingPlansDetailsToJSON(getBillingPlansDetails) {
    return JSON.stringify(GetBillingPlansDetails$outboundSchema.parse(getBillingPlansDetails));
}
export function getBillingPlansDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBillingPlansDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBillingPlansDetails' from JSON`);
}
/** @internal */
export const GetBillingPlansHighlightedDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const GetBillingPlansHighlightedDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
export function getBillingPlansHighlightedDetailsToJSON(getBillingPlansHighlightedDetails) {
    return JSON.stringify(GetBillingPlansHighlightedDetails$outboundSchema.parse(getBillingPlansHighlightedDetails));
}
export function getBillingPlansHighlightedDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBillingPlansHighlightedDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBillingPlansHighlightedDetails' from JSON`);
}
/** @internal */
export const Quote$inboundSchema = z
    .object({
    line: z.string(),
    amount: z.string(),
});
/** @internal */
export const Quote$outboundSchema = z.object({
    line: z.string(),
    amount: z.string(),
});
export function quoteToJSON(quote) {
    return JSON.stringify(Quote$outboundSchema.parse(quote));
}
export function quoteFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Quote$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Quote' from JSON`);
}
/** @internal */
export const Plans$inboundSchema = z
    .object({
    type: GetBillingPlansType$inboundSchema,
    description: z.string(),
    id: z.string(),
    name: z.string(),
    scope: GetBillingPlansScope$inboundSchema,
    paymentMethodRequired: z.boolean(),
    preauthorizationAmount: z.number().optional(),
    initialCharge: z.string().optional(),
    minimumAmount: z.string().optional(),
    maximumAmount: z.string().optional(),
    maximumAmountAutoPurchasePerPeriod: z.string().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => GetBillingPlansDetails$inboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => GetBillingPlansHighlightedDetails$inboundSchema)).optional(),
    quote: z.array(z.lazy(() => Quote$inboundSchema)).optional(),
    effectiveDate: z.string().optional(),
    disabled: z.boolean().optional(),
});
/** @internal */
export const Plans$outboundSchema = z.object({
    type: GetBillingPlansType$outboundSchema,
    description: z.string(),
    id: z.string(),
    name: z.string(),
    scope: GetBillingPlansScope$outboundSchema,
    paymentMethodRequired: z.boolean(),
    preauthorizationAmount: z.number().optional(),
    initialCharge: z.string().optional(),
    minimumAmount: z.string().optional(),
    maximumAmount: z.string().optional(),
    maximumAmountAutoPurchasePerPeriod: z.string().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => GetBillingPlansDetails$outboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => GetBillingPlansHighlightedDetails$outboundSchema)).optional(),
    quote: z.array(z.lazy(() => Quote$outboundSchema)).optional(),
    effectiveDate: z.string().optional(),
    disabled: z.boolean().optional(),
});
export function plansToJSON(plans) {
    return JSON.stringify(Plans$outboundSchema.parse(plans));
}
export function plansFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Plans$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Plans' from JSON`);
}
/** @internal */
export const GetBillingPlansResponseBody$inboundSchema = z.object({
    plans: z.array(z.lazy(() => Plans$inboundSchema)),
});
/** @internal */
export const GetBillingPlansResponseBody$outboundSchema = z.object({
    plans: z.array(z.lazy(() => Plans$outboundSchema)),
});
export function getBillingPlansResponseBodyToJSON(getBillingPlansResponseBody) {
    return JSON.stringify(GetBillingPlansResponseBody$outboundSchema.parse(getBillingPlansResponseBody));
}
export function getBillingPlansResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetBillingPlansResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBillingPlansResponseBody' from JSON`);
}
