// @vercel/sdk/esm/models/getcheckop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const GetCheckStatus = {
    Registered: "registered",
    Running: "running",
    Completed: "completed",
};
export const GetCheckConclusion = {
    Canceled: "canceled",
    Failed: "failed",
    Neutral: "neutral",
    Succeeded: "succeeded",
    Skipped: "skipped",
    Stale: "stale",
};
export const GetCheckSource = {
    WebVitals: "web-vitals",
};
export const GetCheckChecksSource = {
    WebVitals: "web-vitals",
};
export const GetCheckChecksResponseSource = {
    WebVitals: "web-vitals",
};
export const GetCheckChecksResponse200Source = {
    WebVitals: "web-vitals",
};
export const GetCheckChecksResponse200ApplicationJSONSource = {
    WebVitals: "web-vitals",
};
/** @internal */
export const GetCheckRequest$inboundSchema = z.object({
    deploymentId: z.string(),
    checkId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetCheckRequest$outboundSchema = z.object({
    deploymentId: z.string(),
    checkId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getCheckRequestToJSON(getCheckRequest) {
    return JSON.stringify(GetCheckRequest$outboundSchema.parse(getCheckRequest));
}
export function getCheckRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckRequest' from JSON`);
}
/** @internal */
export const GetCheckStatus$inboundSchema = z.nativeEnum(GetCheckStatus);
/** @internal */
export const GetCheckStatus$outboundSchema = GetCheckStatus$inboundSchema;
/** @internal */
export const GetCheckConclusion$inboundSchema = z.nativeEnum(GetCheckConclusion);
/** @internal */
export const GetCheckConclusion$outboundSchema = GetCheckConclusion$inboundSchema;
/** @internal */
export const GetCheckSource$inboundSchema = z.nativeEnum(GetCheckSource);
/** @internal */
export const GetCheckSource$outboundSchema = GetCheckSource$inboundSchema;
/** @internal */
export const GetCheckFCP$inboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckSource$inboundSchema,
});
/** @internal */
export const GetCheckFCP$outboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckSource$outboundSchema,
});
export function getCheckFCPToJSON(getCheckFCP) {
    return JSON.stringify(GetCheckFCP$outboundSchema.parse(getCheckFCP));
}
export function getCheckFCPFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckFCP$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckFCP' from JSON`);
}
/** @internal */
export const GetCheckChecksSource$inboundSchema = z.nativeEnum(GetCheckChecksSource);
/** @internal */
export const GetCheckChecksSource$outboundSchema = GetCheckChecksSource$inboundSchema;
/** @internal */
export const GetCheckLCP$inboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksSource$inboundSchema,
});
/** @internal */
export const GetCheckLCP$outboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksSource$outboundSchema,
});
export function getCheckLCPToJSON(getCheckLCP) {
    return JSON.stringify(GetCheckLCP$outboundSchema.parse(getCheckLCP));
}
export function getCheckLCPFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckLCP$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckLCP' from JSON`);
}
/** @internal */
export const GetCheckChecksResponseSource$inboundSchema = z.nativeEnum(GetCheckChecksResponseSource);
/** @internal */
export const GetCheckChecksResponseSource$outboundSchema = GetCheckChecksResponseSource$inboundSchema;
/** @internal */
export const GetCheckCLS$inboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksResponseSource$inboundSchema,
});
/** @internal */
export const GetCheckCLS$outboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksResponseSource$outboundSchema,
});
export function getCheckCLSToJSON(getCheckCLS) {
    return JSON.stringify(GetCheckCLS$outboundSchema.parse(getCheckCLS));
}
export function getCheckCLSFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckCLS$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckCLS' from JSON`);
}
/** @internal */
export const GetCheckChecksResponse200Source$inboundSchema = z.nativeEnum(GetCheckChecksResponse200Source);
/** @internal */
export const GetCheckChecksResponse200Source$outboundSchema = GetCheckChecksResponse200Source$inboundSchema;
/** @internal */
export const GetCheckTBT$inboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksResponse200Source$inboundSchema,
});
/** @internal */
export const GetCheckTBT$outboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksResponse200Source$outboundSchema,
});
export function getCheckTBTToJSON(getCheckTBT) {
    return JSON.stringify(GetCheckTBT$outboundSchema.parse(getCheckTBT));
}
export function getCheckTBTFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckTBT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckTBT' from JSON`);
}
/** @internal */
export const GetCheckChecksResponse200ApplicationJSONSource$inboundSchema = z
    .nativeEnum(GetCheckChecksResponse200ApplicationJSONSource);
/** @internal */
export const GetCheckChecksResponse200ApplicationJSONSource$outboundSchema = GetCheckChecksResponse200ApplicationJSONSource$inboundSchema;
/** @internal */
export const GetCheckVirtualExperienceScore$inboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksResponse200ApplicationJSONSource$inboundSchema,
});
/** @internal */
export const GetCheckVirtualExperienceScore$outboundSchema = z.object({
    value: z.nullable(z.number()),
    previousValue: z.number().optional(),
    source: GetCheckChecksResponse200ApplicationJSONSource$outboundSchema,
});
export function getCheckVirtualExperienceScoreToJSON(getCheckVirtualExperienceScore) {
    return JSON.stringify(GetCheckVirtualExperienceScore$outboundSchema.parse(getCheckVirtualExperienceScore));
}
export function getCheckVirtualExperienceScoreFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckVirtualExperienceScore$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckVirtualExperienceScore' from JSON`);
}
/** @internal */
export const GetCheckMetrics$inboundSchema = z.object({
    FCP: z.lazy(() => GetCheckFCP$inboundSchema),
    LCP: z.lazy(() => GetCheckLCP$inboundSchema),
    CLS: z.lazy(() => GetCheckCLS$inboundSchema),
    TBT: z.lazy(() => GetCheckTBT$inboundSchema),
    virtualExperienceScore: z.lazy(() => GetCheckVirtualExperienceScore$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "FCP": "fcp",
        "LCP": "lcp",
        "CLS": "cls",
        "TBT": "tbt",
    });
});
/** @internal */
export const GetCheckMetrics$outboundSchema = z.object({
    fcp: z.lazy(() => GetCheckFCP$outboundSchema),
    lcp: z.lazy(() => GetCheckLCP$outboundSchema),
    cls: z.lazy(() => GetCheckCLS$outboundSchema),
    tbt: z.lazy(() => GetCheckTBT$outboundSchema),
    virtualExperienceScore: z.lazy(() => GetCheckVirtualExperienceScore$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        fcp: "FCP",
        lcp: "LCP",
        cls: "CLS",
        tbt: "TBT",
    });
});
export function getCheckMetricsToJSON(getCheckMetrics) {
    return JSON.stringify(GetCheckMetrics$outboundSchema.parse(getCheckMetrics));
}
export function getCheckMetricsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckMetrics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckMetrics' from JSON`);
}
/** @internal */
export const GetCheckOutput$inboundSchema = z.object({
    metrics: z.lazy(() => GetCheckMetrics$inboundSchema).optional(),
});
/** @internal */
export const GetCheckOutput$outboundSchema = z.object({
    metrics: z.lazy(() => GetCheckMetrics$outboundSchema).optional(),
});
export function getCheckOutputToJSON(getCheckOutput) {
    return JSON.stringify(GetCheckOutput$outboundSchema.parse(getCheckOutput));
}
export function getCheckOutputFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckOutput$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckOutput' from JSON`);
}
/** @internal */
export const GetCheckResponseBody$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    path: z.string().optional(),
    status: GetCheckStatus$inboundSchema,
    conclusion: GetCheckConclusion$inboundSchema.optional(),
    blocking: z.boolean(),
    output: z.lazy(() => GetCheckOutput$inboundSchema).optional(),
    detailsUrl: z.string().optional(),
    integrationId: z.string(),
    deploymentId: z.string(),
    externalId: z.string().optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
    startedAt: z.number().optional(),
    completedAt: z.number().optional(),
    rerequestable: z.boolean().optional(),
});
/** @internal */
export const GetCheckResponseBody$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    path: z.string().optional(),
    status: GetCheckStatus$outboundSchema,
    conclusion: GetCheckConclusion$outboundSchema.optional(),
    blocking: z.boolean(),
    output: z.lazy(() => GetCheckOutput$outboundSchema).optional(),
    detailsUrl: z.string().optional(),
    integrationId: z.string(),
    deploymentId: z.string(),
    externalId: z.string().optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
    startedAt: z.number().optional(),
    completedAt: z.number().optional(),
    rerequestable: z.boolean().optional(),
});
export function getCheckResponseBodyToJSON(getCheckResponseBody) {
    return JSON.stringify(GetCheckResponseBody$outboundSchema.parse(getCheckResponseBody));
}
export function getCheckResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetCheckResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCheckResponseBody' from JSON`);
}
