// @vercel/sdk/esm/models/getconfigurationop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * Source defines where the configuration was installed from. It is used to analyze user engagement for integration installations in product metrics.
 */
export const GetConfigurationResponseBodyIntegrationsSource = {
    Marketplace: "marketplace",
    DeployButton: "deploy-button",
    External: "external",
    V0: "v0",
    ResourceClaims: "resource-claims",
};
export const GetConfigurationResponseBodyIntegrationsType = {
    IntegrationConfiguration: "integration-configuration",
};
export const GetConfigurationResponseBodyIntegrationsDisabledReason = {
    DisabledByOwner: "disabled-by-owner",
    FeatureNotAvailable: "feature-not-available",
    DisabledByAdmin: "disabled-by-admin",
    OriginalOwnerLeftTheTeam: "original-owner-left-the-team",
    AccountPlanDowngrade: "account-plan-downgrade",
    OriginalOwnerRoleDowngraded: "original-owner-role-downgraded",
};
/**
 * Defines the installation type. - 'external' integrations are installed via the existing integrations flow - 'marketplace' integrations are natively installed: - when accepting the TOS of a partner during the store creation process - if undefined, assume 'external'
 */
export const GetConfigurationResponseBodyIntegrationsInstallationType = {
    Marketplace: "marketplace",
    External: "external",
};
/**
 * A string representing the permission for projects. Possible values are `all` or `selected`.
 */
export const ProjectSelection = {
    Selected: "selected",
    All: "all",
};
export const GetConfigurationResponseBodyLevel = {
    Info: "info",
    Warn: "warn",
    Error: "error",
};
export const GetConfigurationTransferRequestKind = {
    TransferFromMarketplace: "transfer-from-marketplace",
};
export const TransferRequestKind = {
    TransferToMarketplace: "transfer-to-marketplace",
};
export const TransferRequestType = {
    Subscription: "subscription",
    Prepayment: "prepayment",
};
export const TransferRequestScope = {
    Installation: "installation",
    Resource: "resource",
};
export const GetConfigurationResponseBodyDisabledReason = {
    DisabledByOwner: "disabled-by-owner",
    FeatureNotAvailable: "feature-not-available",
    DisabledByAdmin: "disabled-by-admin",
    OriginalOwnerLeftTheTeam: "original-owner-left-the-team",
    AccountPlanDowngrade: "account-plan-downgrade",
    OriginalOwnerRoleDowngraded: "original-owner-role-downgraded",
};
/**
 * Source defines where the configuration was installed from. It is used to analyze user engagement for integration installations in product metrics.
 */
export const GetConfigurationResponseBodySource = {
    Marketplace: "marketplace",
    DeployButton: "deploy-button",
    External: "external",
    V0: "v0",
    ResourceClaims: "resource-claims",
};
/**
 * Defines the installation type. - 'external' integrations are installed via the existing integrations flow - 'marketplace' integrations are natively installed: - when accepting the TOS of a partner during the store creation process - if undefined, assume 'external'
 */
export const GetConfigurationResponseBodyInstallationType = {
    Marketplace: "marketplace",
    External: "external",
};
export const GetConfigurationResponseBodyType = {
    IntegrationConfiguration: "integration-configuration",
};
/** @internal */
export const GetConfigurationRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetConfigurationRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getConfigurationRequestToJSON(getConfigurationRequest) {
    return JSON.stringify(GetConfigurationRequest$outboundSchema.parse(getConfigurationRequest));
}
export function getConfigurationRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetConfigurationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetConfigurationRequest' from JSON`);
}
/** @internal */
export const GetConfigurationResponseBodyIntegrationsSource$inboundSchema = z
    .nativeEnum(GetConfigurationResponseBodyIntegrationsSource);
/** @internal */
export const GetConfigurationResponseBodyIntegrationsSource$outboundSchema = GetConfigurationResponseBodyIntegrationsSource$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodyIntegrationsType$inboundSchema = z
    .nativeEnum(GetConfigurationResponseBodyIntegrationsType);
/** @internal */
export const GetConfigurationResponseBodyIntegrationsType$outboundSchema = GetConfigurationResponseBodyIntegrationsType$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodyIntegrationsDisabledReason$inboundSchema = z.nativeEnum(GetConfigurationResponseBodyIntegrationsDisabledReason);
/** @internal */
export const GetConfigurationResponseBodyIntegrationsDisabledReason$outboundSchema = GetConfigurationResponseBodyIntegrationsDisabledReason$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodyIntegrationsInstallationType$inboundSchema = z.nativeEnum(GetConfigurationResponseBodyIntegrationsInstallationType);
/** @internal */
export const GetConfigurationResponseBodyIntegrationsInstallationType$outboundSchema = GetConfigurationResponseBodyIntegrationsInstallationType$inboundSchema;
/** @internal */
export const GetConfigurationResponseBody2$inboundSchema = z.object({
    completedAt: z.number().optional(),
    createdAt: z.number(),
    id: z.string(),
    integrationId: z.string(),
    ownerId: z.string(),
    projects: z.array(z.string()).optional(),
    source: GetConfigurationResponseBodyIntegrationsSource$inboundSchema
        .optional(),
    slug: z.string(),
    teamId: z.nullable(z.string()).optional(),
    type: GetConfigurationResponseBodyIntegrationsType$inboundSchema,
    updatedAt: z.number(),
    userId: z.string(),
    scopes: z.array(z.string()),
    disabledAt: z.number().optional(),
    deletedAt: z.nullable(z.number()).optional(),
    deleteRequestedAt: z.nullable(z.number()).optional(),
    disabledReason: GetConfigurationResponseBodyIntegrationsDisabledReason$inboundSchema
        .optional(),
    installationType: GetConfigurationResponseBodyIntegrationsInstallationType$inboundSchema
        .optional(),
});
/** @internal */
export const GetConfigurationResponseBody2$outboundSchema = z.object({
    completedAt: z.number().optional(),
    createdAt: z.number(),
    id: z.string(),
    integrationId: z.string(),
    ownerId: z.string(),
    projects: z.array(z.string()).optional(),
    source: GetConfigurationResponseBodyIntegrationsSource$outboundSchema
        .optional(),
    slug: z.string(),
    teamId: z.nullable(z.string()).optional(),
    type: GetConfigurationResponseBodyIntegrationsType$outboundSchema,
    updatedAt: z.number(),
    userId: z.string(),
    scopes: z.array(z.string()),
    disabledAt: z.number().optional(),
    deletedAt: z.nullable(z.number()).optional(),
    deleteRequestedAt: z.nullable(z.number()).optional(),
    disabledReason: GetConfigurationResponseBodyIntegrationsDisabledReason$outboundSchema
        .optional(),
    installationType: GetConfigurationResponseBodyIntegrationsInstallationType$outboundSchema
        .optional(),
});
export function getConfigurationResponseBody2ToJSON(getConfigurationResponseBody2) {
    return JSON.stringify(GetConfigurationResponseBody2$outboundSchema.parse(getConfigurationResponseBody2));
}
export function getConfigurationResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetConfigurationResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetConfigurationResponseBody2' from JSON`);
}
/** @internal */
export const ProjectSelection$inboundSchema = z.nativeEnum(ProjectSelection);
/** @internal */
export const ProjectSelection$outboundSchema = ProjectSelection$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodyLevel$inboundSchema = z.nativeEnum(GetConfigurationResponseBodyLevel);
/** @internal */
export const GetConfigurationResponseBodyLevel$outboundSchema = GetConfigurationResponseBodyLevel$inboundSchema;
/** @internal */
export const ResponseBodyNotification$inboundSchema = z.object({
    level: GetConfigurationResponseBodyLevel$inboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/** @internal */
export const ResponseBodyNotification$outboundSchema = z.object({
    level: GetConfigurationResponseBodyLevel$outboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
export function responseBodyNotificationToJSON(responseBodyNotification) {
    return JSON.stringify(ResponseBodyNotification$outboundSchema.parse(responseBodyNotification));
}
export function responseBodyNotificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyNotification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyNotification' from JSON`);
}
/** @internal */
export const GetConfigurationTransferRequestKind$inboundSchema = z.nativeEnum(GetConfigurationTransferRequestKind);
/** @internal */
export const GetConfigurationTransferRequestKind$outboundSchema = GetConfigurationTransferRequestKind$inboundSchema;
/** @internal */
export const TransferRequestRequester$inboundSchema = z.object({
    name: z.string(),
    email: z.string().optional(),
});
/** @internal */
export const TransferRequestRequester$outboundSchema = z.object({
    name: z.string(),
    email: z.string().optional(),
});
export function transferRequestRequesterToJSON(transferRequestRequester) {
    return JSON.stringify(TransferRequestRequester$outboundSchema.parse(transferRequestRequester));
}
export function transferRequestRequesterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferRequestRequester$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferRequestRequester' from JSON`);
}
/** @internal */
export const TransferRequest2$inboundSchema = z.object({
    kind: GetConfigurationTransferRequestKind$inboundSchema,
    requestId: z.string(),
    transferId: z.string(),
    requester: z.lazy(() => TransferRequestRequester$inboundSchema),
    createdAt: z.number(),
    expiresAt: z.number(),
    discardedAt: z.number().optional(),
    discardedBy: z.string().optional(),
    approvedAt: z.number().optional(),
    approvedBy: z.string().optional(),
    authorizationId: z.string().optional(),
});
/** @internal */
export const TransferRequest2$outboundSchema = z.object({
    kind: GetConfigurationTransferRequestKind$outboundSchema,
    requestId: z.string(),
    transferId: z.string(),
    requester: z.lazy(() => TransferRequestRequester$outboundSchema),
    createdAt: z.number(),
    expiresAt: z.number(),
    discardedAt: z.number().optional(),
    discardedBy: z.string().optional(),
    approvedAt: z.number().optional(),
    approvedBy: z.string().optional(),
    authorizationId: z.string().optional(),
});
export function transferRequest2ToJSON(transferRequest2) {
    return JSON.stringify(TransferRequest2$outboundSchema.parse(transferRequest2));
}
export function transferRequest2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferRequest2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferRequest2' from JSON`);
}
/** @internal */
export const TransferRequestKind$inboundSchema = z.nativeEnum(TransferRequestKind);
/** @internal */
export const TransferRequestKind$outboundSchema = TransferRequestKind$inboundSchema;
/** @internal */
export const TransferRequestType$inboundSchema = z.nativeEnum(TransferRequestType);
/** @internal */
export const TransferRequestType$outboundSchema = TransferRequestType$inboundSchema;
/** @internal */
export const TransferRequestScope$inboundSchema = z.nativeEnum(TransferRequestScope);
/** @internal */
export const TransferRequestScope$outboundSchema = TransferRequestScope$inboundSchema;
/** @internal */
export const TransferRequestBillingPlan$inboundSchema = z.object({
    id: z.string(),
    type: TransferRequestType$inboundSchema,
    scope: TransferRequestScope$inboundSchema.optional(),
    name: z.string(),
    description: z.string(),
    paymentMethodRequired: z.boolean().optional(),
    preauthorizationAmount: z.number().optional(),
});
/** @internal */
export const TransferRequestBillingPlan$outboundSchema = z.object({
    id: z.string(),
    type: TransferRequestType$outboundSchema,
    scope: TransferRequestScope$outboundSchema.optional(),
    name: z.string(),
    description: z.string(),
    paymentMethodRequired: z.boolean().optional(),
    preauthorizationAmount: z.number().optional(),
});
export function transferRequestBillingPlanToJSON(transferRequestBillingPlan) {
    return JSON.stringify(TransferRequestBillingPlan$outboundSchema.parse(transferRequestBillingPlan));
}
export function transferRequestBillingPlanFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferRequestBillingPlan$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferRequestBillingPlan' from JSON`);
}
/** @internal */
export const Requester$inboundSchema = z.object({
    name: z.string(),
    email: z.string().optional(),
});
/** @internal */
export const Requester$outboundSchema = z.object({
    name: z.string(),
    email: z.string().optional(),
});
export function requesterToJSON(requester) {
    return JSON.stringify(Requester$outboundSchema.parse(requester));
}
export function requesterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Requester$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Requester' from JSON`);
}
/** @internal */
export const TransferRequest1$inboundSchema = z.object({
    kind: TransferRequestKind$inboundSchema,
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => TransferRequestBillingPlan$inboundSchema)
        .optional(),
    requestId: z.string(),
    transferId: z.string(),
    requester: z.lazy(() => Requester$inboundSchema),
    createdAt: z.number(),
    expiresAt: z.number(),
    discardedAt: z.number().optional(),
    discardedBy: z.string().optional(),
    approvedAt: z.number().optional(),
    approvedBy: z.string().optional(),
    authorizationId: z.string().optional(),
});
/** @internal */
export const TransferRequest1$outboundSchema = z.object({
    kind: TransferRequestKind$outboundSchema,
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => TransferRequestBillingPlan$outboundSchema)
        .optional(),
    requestId: z.string(),
    transferId: z.string(),
    requester: z.lazy(() => Requester$outboundSchema),
    createdAt: z.number(),
    expiresAt: z.number(),
    discardedAt: z.number().optional(),
    discardedBy: z.string().optional(),
    approvedAt: z.number().optional(),
    approvedBy: z.string().optional(),
    authorizationId: z.string().optional(),
});
export function transferRequest1ToJSON(transferRequest1) {
    return JSON.stringify(TransferRequest1$outboundSchema.parse(transferRequest1));
}
export function transferRequest1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferRequest1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferRequest1' from JSON`);
}
/** @internal */
export const TransferRequest$inboundSchema = z.union([
    z.lazy(() => TransferRequest1$inboundSchema),
    z.lazy(() => TransferRequest2$inboundSchema),
]);
/** @internal */
export const TransferRequest$outboundSchema = z.union([
    z.lazy(() => TransferRequest1$outboundSchema),
    z.lazy(() => TransferRequest2$outboundSchema),
]);
export function transferRequestToJSON(transferRequest) {
    return JSON.stringify(TransferRequest$outboundSchema.parse(transferRequest));
}
export function transferRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => TransferRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransferRequest' from JSON`);
}
/** @internal */
export const GetConfigurationResponseBodyDisabledReason$inboundSchema = z
    .nativeEnum(GetConfigurationResponseBodyDisabledReason);
/** @internal */
export const GetConfigurationResponseBodyDisabledReason$outboundSchema = GetConfigurationResponseBodyDisabledReason$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodySource$inboundSchema = z.nativeEnum(GetConfigurationResponseBodySource);
/** @internal */
export const GetConfigurationResponseBodySource$outboundSchema = GetConfigurationResponseBodySource$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodyInstallationType$inboundSchema = z
    .nativeEnum(GetConfigurationResponseBodyInstallationType);
/** @internal */
export const GetConfigurationResponseBodyInstallationType$outboundSchema = GetConfigurationResponseBodyInstallationType$inboundSchema;
/** @internal */
export const GetConfigurationResponseBodyType$inboundSchema = z.nativeEnum(GetConfigurationResponseBodyType);
/** @internal */
export const GetConfigurationResponseBodyType$outboundSchema = GetConfigurationResponseBodyType$inboundSchema;
/** @internal */
export const GetConfigurationResponseBody1$inboundSchema = z.object({
    projectSelection: ProjectSelection$inboundSchema,
    notification: z.lazy(() => ResponseBodyNotification$inboundSchema),
    transferRequest: z.union([
        z.lazy(() => TransferRequest1$inboundSchema),
        z.lazy(() => TransferRequest2$inboundSchema),
    ]),
    projects: z.array(z.string()).optional(),
    createdAt: z.number(),
    completedAt: z.number().optional(),
    id: z.string(),
    integrationId: z.string(),
    ownerId: z.string(),
    slug: z.string(),
    teamId: z.nullable(z.string()).optional(),
    updatedAt: z.number(),
    userId: z.string(),
    scopes: z.array(z.string()),
    disabledAt: z.number().optional(),
    disabledReason: GetConfigurationResponseBodyDisabledReason$inboundSchema
        .optional(),
    source: GetConfigurationResponseBodySource$inboundSchema.optional(),
    canConfigureOpenTelemetry: z.boolean().optional(),
    installationType: GetConfigurationResponseBodyInstallationType$inboundSchema
        .optional(),
    deleteRequestedAt: z.nullable(z.number()).optional(),
    type: GetConfigurationResponseBodyType$inboundSchema,
    deletedAt: z.nullable(z.number()).optional(),
});
/** @internal */
export const GetConfigurationResponseBody1$outboundSchema = z.object({
    projectSelection: ProjectSelection$outboundSchema,
    notification: z.lazy(() => ResponseBodyNotification$outboundSchema),
    transferRequest: z.union([
        z.lazy(() => TransferRequest1$outboundSchema),
        z.lazy(() => TransferRequest2$outboundSchema),
    ]),
    projects: z.array(z.string()).optional(),
    createdAt: z.number(),
    completedAt: z.number().optional(),
    id: z.string(),
    integrationId: z.string(),
    ownerId: z.string(),
    slug: z.string(),
    teamId: z.nullable(z.string()).optional(),
    updatedAt: z.number(),
    userId: z.string(),
    scopes: z.array(z.string()),
    disabledAt: z.number().optional(),
    disabledReason: GetConfigurationResponseBodyDisabledReason$outboundSchema
        .optional(),
    source: GetConfigurationResponseBodySource$outboundSchema.optional(),
    canConfigureOpenTelemetry: z.boolean().optional(),
    installationType: GetConfigurationResponseBodyInstallationType$outboundSchema
        .optional(),
    deleteRequestedAt: z.nullable(z.number()).optional(),
    type: GetConfigurationResponseBodyType$outboundSchema,
    deletedAt: z.nullable(z.number()).optional(),
});
export function getConfigurationResponseBody1ToJSON(getConfigurationResponseBody1) {
    return JSON.stringify(GetConfigurationResponseBody1$outboundSchema.parse(getConfigurationResponseBody1));
}
export function getConfigurationResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetConfigurationResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetConfigurationResponseBody1' from JSON`);
}
/** @internal */
export const GetConfigurationResponseBody$inboundSchema = z.union([
    z.lazy(() => GetConfigurationResponseBody1$inboundSchema),
    z.lazy(() => GetConfigurationResponseBody2$inboundSchema),
]);
/** @internal */
export const GetConfigurationResponseBody$outboundSchema = z.union([
    z.lazy(() => GetConfigurationResponseBody1$outboundSchema),
    z.lazy(() => GetConfigurationResponseBody2$outboundSchema),
]);
export function getConfigurationResponseBodyToJSON(getConfigurationResponseBody) {
    return JSON.stringify(GetConfigurationResponseBody$outboundSchema.parse(getConfigurationResponseBody));
}
export function getConfigurationResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetConfigurationResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetConfigurationResponseBody' from JSON`);
}
