// @vercel/sdk/esm/models/getdeploymentfilecontentsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetDeploymentFileContentsRequest$inboundSchema = z.object({
    id: z.string(),
    fileId: z.string(),
    path: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetDeploymentFileContentsRequest$outboundSchema = z.object({
    id: z.string(),
    fileId: z.string(),
    path: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getDeploymentFileContentsRequestToJSON(getDeploymentFileContentsRequest) {
    return JSON.stringify(GetDeploymentFileContentsRequest$outboundSchema.parse(getDeploymentFileContentsRequest));
}
export function getDeploymentFileContentsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDeploymentFileContentsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDeploymentFileContentsRequest' from JSON`);
}
