// @vercel/sdk/esm/models/getdomaintransferop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * The domain's transfer policy (depends on TLD requirements). `charge-and-renew`: transfer will charge for renewal and will renew the existing domain's registration. `no-charge-no-change`: transfer will have no change to registration period and does not require charge. `no-change`: transfer charge is required, but no change in registration period. `new-term`: transfer charge is required and a new registry term is set based on the transfer date. `not-supported`: transfers are not supported for this domain or TLD. `null`: This TLD is not supported by Vercel's Registrar.
 */
export const ResponseBodyTransferPolicy = {
    ChargeAndRenew: "charge-and-renew",
    NoChargeNoChange: "no-charge-no-change",
    NoChange: "no-change",
    NewTerm: "new-term",
    NotSupported: "not-supported",
};
/**
 * The current state of an ongoing transfer. `pending_owner`: Awaiting approval by domain's admin contact (every transfer begins with this status). If approval is not given within five days, the transfer is cancelled. `pending_admin`: Waiting for approval by Vercel Registrar admin. `pending_registry`: Awaiting registry approval (the transfer completes after 7 days unless it is declined by the current registrar). `completed`: The transfer completed successfully. `cancelled`: The transfer was cancelled. `undef`: No transfer exists for this domain. `unknown`: This TLD is not supported by Vercel's Registrar.
 */
export const GetDomainTransferResponseBodyStatus = {
    Completed: "completed",
    Undef: "undef",
    PendingOwner: "pending_owner",
    PendingAdmin: "pending_admin",
    PendingRegistry: "pending_registry",
    Cancelled: "cancelled",
    Unknown: "unknown",
};
export const TransferPolicy = {
    ChargeAndRenew: "charge-and-renew",
};
/** @internal */
export const GetDomainTransferRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    domain: z.string(),
});
/** @internal */
export const GetDomainTransferRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    domain: z.string(),
});
export function getDomainTransferRequestToJSON(getDomainTransferRequest) {
    return JSON.stringify(GetDomainTransferRequest$outboundSchema.parse(getDomainTransferRequest));
}
export function getDomainTransferRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferRequest' from JSON`);
}
/** @internal */
export const ResponseBodyTransferPolicy$inboundSchema = z.nativeEnum(ResponseBodyTransferPolicy);
/** @internal */
export const ResponseBodyTransferPolicy$outboundSchema = ResponseBodyTransferPolicy$inboundSchema;
/** @internal */
export const GetDomainTransferResponseBodyStatus$inboundSchema = z.nativeEnum(GetDomainTransferResponseBodyStatus);
/** @internal */
export const GetDomainTransferResponseBodyStatus$outboundSchema = GetDomainTransferResponseBodyStatus$inboundSchema;
/** @internal */
export const GetDomainTransferResponseBody2$inboundSchema = z.object({
    transferable: z.boolean(),
    transferPolicy: z.nullable(ResponseBodyTransferPolicy$inboundSchema),
    reason: z.string(),
    status: GetDomainTransferResponseBodyStatus$inboundSchema,
});
/** @internal */
export const GetDomainTransferResponseBody2$outboundSchema = z.object({
    transferable: z.boolean(),
    transferPolicy: z.nullable(ResponseBodyTransferPolicy$outboundSchema),
    reason: z.string(),
    status: GetDomainTransferResponseBodyStatus$outboundSchema,
});
export function getDomainTransferResponseBody2ToJSON(getDomainTransferResponseBody2) {
    return JSON.stringify(GetDomainTransferResponseBody2$outboundSchema.parse(getDomainTransferResponseBody2));
}
export function getDomainTransferResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferResponseBody2' from JSON`);
}
/** @internal */
export const TransferPolicy$inboundSchema = z.nativeEnum(TransferPolicy);
/** @internal */
export const TransferPolicy$outboundSchema = TransferPolicy$inboundSchema;
/** @internal */
export const GetDomainTransferResponseBody1$inboundSchema = z.object({
    reason: z.string(),
    status: z.string(),
    transferable: z.boolean(),
    transferPolicy: TransferPolicy$inboundSchema,
});
/** @internal */
export const GetDomainTransferResponseBody1$outboundSchema = z.object({
    reason: z.string(),
    status: z.string(),
    transferable: z.boolean(),
    transferPolicy: TransferPolicy$outboundSchema,
});
export function getDomainTransferResponseBody1ToJSON(getDomainTransferResponseBody1) {
    return JSON.stringify(GetDomainTransferResponseBody1$outboundSchema.parse(getDomainTransferResponseBody1));
}
export function getDomainTransferResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferResponseBody1' from JSON`);
}
/** @internal */
export const GetDomainTransferResponseBody$inboundSchema = z.union([
    z.lazy(() => GetDomainTransferResponseBody1$inboundSchema),
    z.lazy(() => GetDomainTransferResponseBody2$inboundSchema),
]);
/** @internal */
export const GetDomainTransferResponseBody$outboundSchema = z.union([
    z.lazy(() => GetDomainTransferResponseBody1$outboundSchema),
    z.lazy(() => GetDomainTransferResponseBody2$outboundSchema),
]);
export function getDomainTransferResponseBodyToJSON(getDomainTransferResponseBody) {
    return JSON.stringify(GetDomainTransferResponseBody$outboundSchema.parse(getDomainTransferResponseBody));
}
export function getDomainTransferResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainTransferResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainTransferResponseBody' from JSON`);
}
