// @vercel/sdk/esm/models/getdomainpriceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { BadRequest$inboundSchema, BadRequest$outboundSchema, } from "./badrequest.js";
import { DomainTooShort$inboundSchema, DomainTooShort$outboundSchema, } from "./domaintooshort.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { TldNotSupported$inboundSchema, TldNotSupported$outboundSchema, } from "./tldnotsupported.js";
/** @internal */
export const GetDomainPriceRequest$inboundSchema = z.object({
    domain: z.string(),
    years: z.string().optional(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetDomainPriceRequest$outboundSchema = z.object({
    domain: z.string(),
    years: z.string().optional(),
    teamId: z.string().optional(),
});
export function getDomainPriceRequestToJSON(getDomainPriceRequest) {
    return JSON.stringify(GetDomainPriceRequest$outboundSchema.parse(getDomainPriceRequest));
}
export function getDomainPriceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainPriceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainPriceRequest' from JSON`);
}
/** @internal */
export const GetDomainPriceDomainsRegistrarResponseBody$inboundSchema = z.union([
    BadRequest$inboundSchema,
    DomainTooShort$inboundSchema,
    TldNotSupported$inboundSchema,
    HttpApiDecodeError$inboundSchema,
]);
/** @internal */
export const GetDomainPriceDomainsRegistrarResponseBody$outboundSchema = z.union([
    BadRequest$outboundSchema,
    DomainTooShort$outboundSchema,
    TldNotSupported$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
export function getDomainPriceDomainsRegistrarResponseBodyToJSON(getDomainPriceDomainsRegistrarResponseBody) {
    return JSON.stringify(GetDomainPriceDomainsRegistrarResponseBody$outboundSchema.parse(getDomainPriceDomainsRegistrarResponseBody));
}
export function getDomainPriceDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainPriceDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainPriceDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetDomainPriceResponseBody$inboundSchema = z.object({
    years: z.number(),
    purchasePrice: z.nullable(z.number()),
    renewalPrice: z.nullable(z.number()),
    transferPrice: z.nullable(z.number()),
});
/** @internal */
export const GetDomainPriceResponseBody$outboundSchema = z.object({
    years: z.number(),
    purchasePrice: z.nullable(z.number()),
    renewalPrice: z.nullable(z.number()),
    transferPrice: z.nullable(z.number()),
});
export function getDomainPriceResponseBodyToJSON(getDomainPriceResponseBody) {
    return JSON.stringify(GetDomainPriceResponseBody$outboundSchema.parse(getDomainPriceResponseBody));
}
export function getDomainPriceResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetDomainPriceResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDomainPriceResponseBody' from JSON`);
}
