// @vercel/sdk/esm/models/getedgeconfigschemaop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetEdgeConfigSchemaRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigSchemaRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getEdgeConfigSchemaRequestToJSON(getEdgeConfigSchemaRequest) {
    return JSON.stringify(GetEdgeConfigSchemaRequest$outboundSchema.parse(getEdgeConfigSchemaRequest));
}
export function getEdgeConfigSchemaRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigSchemaRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigSchemaRequest' from JSON`);
}
/** @internal */
export const GetEdgeConfigSchemaResponseBody$inboundSchema = z.object({});
/** @internal */
export const GetEdgeConfigSchemaResponseBody$outboundSchema = z.object({});
export function getEdgeConfigSchemaResponseBodyToJSON(getEdgeConfigSchemaResponseBody) {
    return JSON.stringify(GetEdgeConfigSchemaResponseBody$outboundSchema.parse(getEdgeConfigSchemaResponseBody));
}
export function getEdgeConfigSchemaResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigSchemaResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigSchemaResponseBody' from JSON`);
}
