// @vercel/sdk/esm/models/getedgeconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const GetEdgeConfigPurposeEdgeConfigType = {
    Experimentation: "experimentation",
};
export const GetEdgeConfigPurposeType = {
    Flags: "flags",
};
/** @internal */
export const GetEdgeConfigRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetEdgeConfigRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getEdgeConfigRequestToJSON(getEdgeConfigRequest) {
    return JSON.stringify(GetEdgeConfigRequest$outboundSchema.parse(getEdgeConfigRequest));
}
export function getEdgeConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigRequest' from JSON`);
}
/** @internal */
export const GetEdgeConfigTransfer$inboundSchema = z.object({
    fromAccountId: z.string(),
    startedAt: z.number(),
    doneAt: z.nullable(z.number()),
});
/** @internal */
export const GetEdgeConfigTransfer$outboundSchema = z.object({
    fromAccountId: z.string(),
    startedAt: z.number(),
    doneAt: z.nullable(z.number()),
});
export function getEdgeConfigTransferToJSON(getEdgeConfigTransfer) {
    return JSON.stringify(GetEdgeConfigTransfer$outboundSchema.parse(getEdgeConfigTransfer));
}
export function getEdgeConfigTransferFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigTransfer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigTransfer' from JSON`);
}
/** @internal */
export const GetEdgeConfigSchema$inboundSchema = z.object({});
/** @internal */
export const GetEdgeConfigSchema$outboundSchema = z.object({});
export function getEdgeConfigSchemaToJSON(getEdgeConfigSchema) {
    return JSON.stringify(GetEdgeConfigSchema$outboundSchema.parse(getEdgeConfigSchema));
}
export function getEdgeConfigSchemaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigSchema$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigSchema' from JSON`);
}
/** @internal */
export const GetEdgeConfigPurposeEdgeConfigType$inboundSchema = z.nativeEnum(GetEdgeConfigPurposeEdgeConfigType);
/** @internal */
export const GetEdgeConfigPurposeEdgeConfigType$outboundSchema = GetEdgeConfigPurposeEdgeConfigType$inboundSchema;
/** @internal */
export const GetEdgeConfigPurpose2$inboundSchema = z.object({
    type: GetEdgeConfigPurposeEdgeConfigType$inboundSchema,
    resourceId: z.string(),
});
/** @internal */
export const GetEdgeConfigPurpose2$outboundSchema = z.object({
    type: GetEdgeConfigPurposeEdgeConfigType$outboundSchema,
    resourceId: z.string(),
});
export function getEdgeConfigPurpose2ToJSON(getEdgeConfigPurpose2) {
    return JSON.stringify(GetEdgeConfigPurpose2$outboundSchema.parse(getEdgeConfigPurpose2));
}
export function getEdgeConfigPurpose2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigPurpose2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigPurpose2' from JSON`);
}
/** @internal */
export const GetEdgeConfigPurposeType$inboundSchema = z.nativeEnum(GetEdgeConfigPurposeType);
/** @internal */
export const GetEdgeConfigPurposeType$outboundSchema = GetEdgeConfigPurposeType$inboundSchema;
/** @internal */
export const GetEdgeConfigPurpose1$inboundSchema = z.object({
    type: GetEdgeConfigPurposeType$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const GetEdgeConfigPurpose1$outboundSchema = z.object({
    type: GetEdgeConfigPurposeType$outboundSchema,
    projectId: z.string(),
});
export function getEdgeConfigPurpose1ToJSON(getEdgeConfigPurpose1) {
    return JSON.stringify(GetEdgeConfigPurpose1$outboundSchema.parse(getEdgeConfigPurpose1));
}
export function getEdgeConfigPurpose1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigPurpose1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigPurpose1' from JSON`);
}
/** @internal */
export const GetEdgeConfigPurpose$inboundSchema = z.union([
    z.lazy(() => GetEdgeConfigPurpose1$inboundSchema),
    z.lazy(() => GetEdgeConfigPurpose2$inboundSchema),
]);
/** @internal */
export const GetEdgeConfigPurpose$outboundSchema = z.union([
    z.lazy(() => GetEdgeConfigPurpose1$outboundSchema),
    z.lazy(() => GetEdgeConfigPurpose2$outboundSchema),
]);
export function getEdgeConfigPurposeToJSON(getEdgeConfigPurpose) {
    return JSON.stringify(GetEdgeConfigPurpose$outboundSchema.parse(getEdgeConfigPurpose));
}
export function getEdgeConfigPurposeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigPurpose$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigPurpose' from JSON`);
}
/** @internal */
export const GetEdgeConfigResponseBody$inboundSchema = z.object({
    createdAt: z.number(),
    updatedAt: z.number(),
    deletedAt: z.nullable(z.number()).optional(),
    id: z.string(),
    slug: z.string(),
    ownerId: z.string(),
    digest: z.string(),
    transfer: z.lazy(() => GetEdgeConfigTransfer$inboundSchema).optional(),
    schema: z.lazy(() => GetEdgeConfigSchema$inboundSchema).optional(),
    purpose: z.union([
        z.lazy(() => GetEdgeConfigPurpose1$inboundSchema),
        z.lazy(() => GetEdgeConfigPurpose2$inboundSchema),
    ]).optional(),
    syncedToDynamoAt: z.number().optional(),
    sizeInBytes: z.number(),
    itemCount: z.number(),
});
/** @internal */
export const GetEdgeConfigResponseBody$outboundSchema = z.object({
    createdAt: z.number(),
    updatedAt: z.number(),
    deletedAt: z.nullable(z.number()).optional(),
    id: z.string(),
    slug: z.string(),
    ownerId: z.string(),
    digest: z.string(),
    transfer: z.lazy(() => GetEdgeConfigTransfer$outboundSchema).optional(),
    schema: z.lazy(() => GetEdgeConfigSchema$outboundSchema).optional(),
    purpose: z.union([
        z.lazy(() => GetEdgeConfigPurpose1$outboundSchema),
        z.lazy(() => GetEdgeConfigPurpose2$outboundSchema),
    ]).optional(),
    syncedToDynamoAt: z.number().optional(),
    sizeInBytes: z.number(),
    itemCount: z.number(),
});
export function getEdgeConfigResponseBodyToJSON(getEdgeConfigResponseBody) {
    return JSON.stringify(GetEdgeConfigResponseBody$outboundSchema.parse(getEdgeConfigResponseBody));
}
export function getEdgeConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetEdgeConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetEdgeConfigResponseBody' from JSON`);
}
