// @vercel/sdk/esm/models/getintegrationresourcesop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * The current status of the resource
 */
export const GetIntegrationResourcesStatus = {
    Ready: "ready",
    Pending: "pending",
    Onboarding: "onboarding",
    Suspended: "suspended",
    Resumed: "resumed",
    Uninstalled: "uninstalled",
    Error: "error",
};
export const GetIntegrationResourcesLevel = {
    Error: "error",
    Info: "info",
    Warn: "warn",
};
/** @internal */
export const GetIntegrationResourcesRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
});
/** @internal */
export const GetIntegrationResourcesRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
});
export function getIntegrationResourcesRequestToJSON(getIntegrationResourcesRequest) {
    return JSON.stringify(GetIntegrationResourcesRequest$outboundSchema.parse(getIntegrationResourcesRequest));
}
export function getIntegrationResourcesRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesRequest' from JSON`);
}
/** @internal */
export const GetIntegrationResourcesStatus$inboundSchema = z.nativeEnum(GetIntegrationResourcesStatus);
/** @internal */
export const GetIntegrationResourcesStatus$outboundSchema = GetIntegrationResourcesStatus$inboundSchema;
/** @internal */
export const GetIntegrationResourcesExperimentation$inboundSchema = z.object({
    edgeConfigSyncingEnabled: z.boolean().optional(),
    edgeConfigId: z.string().optional(),
    edgeConfigTokenId: z.string().optional(),
});
/** @internal */
export const GetIntegrationResourcesExperimentation$outboundSchema = z.object({
    edgeConfigSyncingEnabled: z.boolean().optional(),
    edgeConfigId: z.string().optional(),
    edgeConfigTokenId: z.string().optional(),
});
export function getIntegrationResourcesExperimentationToJSON(getIntegrationResourcesExperimentation) {
    return JSON.stringify(GetIntegrationResourcesExperimentation$outboundSchema.parse(getIntegrationResourcesExperimentation));
}
export function getIntegrationResourcesExperimentationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesExperimentation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesExperimentation' from JSON`);
}
/** @internal */
export const GetIntegrationResourcesProtocolSettings$inboundSchema = z.object({
    experimentation: z.lazy(() => GetIntegrationResourcesExperimentation$inboundSchema).optional(),
});
/** @internal */
export const GetIntegrationResourcesProtocolSettings$outboundSchema = z.object({
    experimentation: z.lazy(() => GetIntegrationResourcesExperimentation$outboundSchema).optional(),
});
export function getIntegrationResourcesProtocolSettingsToJSON(getIntegrationResourcesProtocolSettings) {
    return JSON.stringify(GetIntegrationResourcesProtocolSettings$outboundSchema.parse(getIntegrationResourcesProtocolSettings));
}
export function getIntegrationResourcesProtocolSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesProtocolSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesProtocolSettings' from JSON`);
}
/** @internal */
export const GetIntegrationResourcesLevel$inboundSchema = z.nativeEnum(GetIntegrationResourcesLevel);
/** @internal */
export const GetIntegrationResourcesLevel$outboundSchema = GetIntegrationResourcesLevel$inboundSchema;
/** @internal */
export const GetIntegrationResourcesNotification$inboundSchema = z.object({
    level: GetIntegrationResourcesLevel$inboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/** @internal */
export const GetIntegrationResourcesNotification$outboundSchema = z.object({
    level: GetIntegrationResourcesLevel$outboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
export function getIntegrationResourcesNotificationToJSON(getIntegrationResourcesNotification) {
    return JSON.stringify(GetIntegrationResourcesNotification$outboundSchema.parse(getIntegrationResourcesNotification));
}
export function getIntegrationResourcesNotificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesNotification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesNotification' from JSON`);
}
/** @internal */
export const GetIntegrationResourcesMetadata$inboundSchema = z.union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.string()),
    z.array(z.number()),
]);
/** @internal */
export const GetIntegrationResourcesMetadata$outboundSchema = z.union([
    z.string(),
    z.number(),
    z.boolean(),
    z.array(z.string()),
    z.array(z.number()),
]);
export function getIntegrationResourcesMetadataToJSON(getIntegrationResourcesMetadata) {
    return JSON.stringify(GetIntegrationResourcesMetadata$outboundSchema.parse(getIntegrationResourcesMetadata));
}
export function getIntegrationResourcesMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesMetadata' from JSON`);
}
/** @internal */
export const GetIntegrationResourcesResources$inboundSchema = z.object({
    partnerId: z.string(),
    internalId: z.string(),
    name: z.string(),
    status: GetIntegrationResourcesStatus$inboundSchema.optional(),
    productId: z.string(),
    protocolSettings: z.lazy(() => GetIntegrationResourcesProtocolSettings$inboundSchema).optional(),
    notification: z.lazy(() => GetIntegrationResourcesNotification$inboundSchema)
        .optional(),
    billingPlanId: z.string().optional(),
    metadata: z.record(z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.string()),
        z.array(z.number()),
    ])).optional(),
});
/** @internal */
export const GetIntegrationResourcesResources$outboundSchema = z.object({
    partnerId: z.string(),
    internalId: z.string(),
    name: z.string(),
    status: GetIntegrationResourcesStatus$outboundSchema.optional(),
    productId: z.string(),
    protocolSettings: z.lazy(() => GetIntegrationResourcesProtocolSettings$outboundSchema).optional(),
    notification: z.lazy(() => GetIntegrationResourcesNotification$outboundSchema)
        .optional(),
    billingPlanId: z.string().optional(),
    metadata: z.record(z.union([
        z.string(),
        z.number(),
        z.boolean(),
        z.array(z.string()),
        z.array(z.number()),
    ])).optional(),
});
export function getIntegrationResourcesResourcesToJSON(getIntegrationResourcesResources) {
    return JSON.stringify(GetIntegrationResourcesResources$outboundSchema.parse(getIntegrationResourcesResources));
}
export function getIntegrationResourcesResourcesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesResources$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesResources' from JSON`);
}
/** @internal */
export const GetIntegrationResourcesResponseBody$inboundSchema = z.object({
    resources: z.array(z.lazy(() => GetIntegrationResourcesResources$inboundSchema)),
});
/** @internal */
export const GetIntegrationResourcesResponseBody$outboundSchema = z.object({
    resources: z.array(z.lazy(() => GetIntegrationResourcesResources$outboundSchema)),
});
export function getIntegrationResourcesResponseBodyToJSON(getIntegrationResourcesResponseBody) {
    return JSON.stringify(GetIntegrationResourcesResponseBody$outboundSchema.parse(getIntegrationResourcesResponseBody));
}
export function getIntegrationResourcesResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationResourcesResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationResourcesResponseBody' from JSON`);
}
