// @vercel/sdk/esm/models/getintegrationlogdrainsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * The delivery log format
 */
export const GetIntegrationLogDrainsDeliveryFormat = {
    Json: "json",
    Ndjson: "ndjson",
    Protobuf: "protobuf",
};
/**
 * The sources from which logs are currently being delivered to this log drain.
 */
export const GetIntegrationLogDrainsSources = {
    Build: "build",
    Edge: "edge",
    Lambda: "lambda",
    Static: "static",
    External: "external",
    Firewall: "firewall",
    Redirect: "redirect",
};
/**
 * Whether the log drain was created by an integration or by a user
 */
export const CreatedFrom = {
    Integration: "integration",
    SelfServed: "self-served",
};
/**
 * The environment of log drain
 */
export const GetIntegrationLogDrainsEnvironments = {
    Production: "production",
    Preview: "preview",
};
export const SourceKind = {
    Integration: "integration",
};
export const Kind = {
    SelfServed: "self-served",
};
/** @internal */
export const GetIntegrationLogDrainsRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetIntegrationLogDrainsRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getIntegrationLogDrainsRequestToJSON(getIntegrationLogDrainsRequest) {
    return JSON.stringify(GetIntegrationLogDrainsRequest$outboundSchema.parse(getIntegrationLogDrainsRequest));
}
export function getIntegrationLogDrainsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsRequest' from JSON`);
}
/** @internal */
export const GetIntegrationLogDrainsDeliveryFormat$inboundSchema = z.nativeEnum(GetIntegrationLogDrainsDeliveryFormat);
/** @internal */
export const GetIntegrationLogDrainsDeliveryFormat$outboundSchema = GetIntegrationLogDrainsDeliveryFormat$inboundSchema;
/** @internal */
export const GetIntegrationLogDrainsSources$inboundSchema = z.nativeEnum(GetIntegrationLogDrainsSources);
/** @internal */
export const GetIntegrationLogDrainsSources$outboundSchema = GetIntegrationLogDrainsSources$inboundSchema;
/** @internal */
export const CreatedFrom$inboundSchema = z
    .nativeEnum(CreatedFrom);
/** @internal */
export const CreatedFrom$outboundSchema = CreatedFrom$inboundSchema;
/** @internal */
export const GetIntegrationLogDrainsEnvironments$inboundSchema = z.nativeEnum(GetIntegrationLogDrainsEnvironments);
/** @internal */
export const GetIntegrationLogDrainsEnvironments$outboundSchema = GetIntegrationLogDrainsEnvironments$inboundSchema;
/** @internal */
export const SourceKind$inboundSchema = z
    .nativeEnum(SourceKind);
/** @internal */
export const SourceKind$outboundSchema = SourceKind$inboundSchema;
/** @internal */
export const GetIntegrationLogDrainsSource2$inboundSchema = z.object({
    kind: SourceKind$inboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const GetIntegrationLogDrainsSource2$outboundSchema = z.object({
    kind: SourceKind$outboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
export function getIntegrationLogDrainsSource2ToJSON(getIntegrationLogDrainsSource2) {
    return JSON.stringify(GetIntegrationLogDrainsSource2$outboundSchema.parse(getIntegrationLogDrainsSource2));
}
export function getIntegrationLogDrainsSource2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsSource2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsSource2' from JSON`);
}
/** @internal */
export const Kind$inboundSchema = z.nativeEnum(Kind);
/** @internal */
export const Kind$outboundSchema = Kind$inboundSchema;
/** @internal */
export const GetIntegrationLogDrainsSource1$inboundSchema = z.object({
    kind: Kind$inboundSchema,
});
/** @internal */
export const GetIntegrationLogDrainsSource1$outboundSchema = z.object({
    kind: Kind$outboundSchema,
});
export function getIntegrationLogDrainsSource1ToJSON(getIntegrationLogDrainsSource1) {
    return JSON.stringify(GetIntegrationLogDrainsSource1$outboundSchema.parse(getIntegrationLogDrainsSource1));
}
export function getIntegrationLogDrainsSource1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsSource1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsSource1' from JSON`);
}
/** @internal */
export const GetIntegrationLogDrainsSource$inboundSchema = z.union([
    z.lazy(() => GetIntegrationLogDrainsSource2$inboundSchema),
    z.lazy(() => GetIntegrationLogDrainsSource1$inboundSchema),
]);
/** @internal */
export const GetIntegrationLogDrainsSource$outboundSchema = z.union([
    z.lazy(() => GetIntegrationLogDrainsSource2$outboundSchema),
    z.lazy(() => GetIntegrationLogDrainsSource1$outboundSchema),
]);
export function getIntegrationLogDrainsSourceToJSON(getIntegrationLogDrainsSource) {
    return JSON.stringify(GetIntegrationLogDrainsSource$outboundSchema.parse(getIntegrationLogDrainsSource));
}
export function getIntegrationLogDrainsSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsSource' from JSON`);
}
/** @internal */
export const GetIntegrationLogDrainsResponseBody$inboundSchema = z.object({
    clientId: z.string().optional(),
    configurationId: z.string().optional(),
    createdAt: z.number(),
    id: z.string(),
    deliveryFormat: GetIntegrationLogDrainsDeliveryFormat$inboundSchema
        .optional(),
    name: z.string(),
    ownerId: z.string(),
    projectId: z.nullable(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    url: z.string(),
    sources: z.array(GetIntegrationLogDrainsSources$inboundSchema).optional(),
    createdFrom: CreatedFrom$inboundSchema.optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(GetIntegrationLogDrainsEnvironments$inboundSchema)
        .optional(),
    branch: z.string().optional(),
    samplingRate: z.number().optional(),
    source: z.union([
        z.lazy(() => GetIntegrationLogDrainsSource2$inboundSchema),
        z.lazy(() => GetIntegrationLogDrainsSource1$inboundSchema),
    ]),
});
/** @internal */
export const GetIntegrationLogDrainsResponseBody$outboundSchema = z.object({
    clientId: z.string().optional(),
    configurationId: z.string().optional(),
    createdAt: z.number(),
    id: z.string(),
    deliveryFormat: GetIntegrationLogDrainsDeliveryFormat$outboundSchema
        .optional(),
    name: z.string(),
    ownerId: z.string(),
    projectId: z.nullable(z.string()).optional(),
    projectIds: z.array(z.string()).optional(),
    url: z.string(),
    sources: z.array(GetIntegrationLogDrainsSources$outboundSchema).optional(),
    createdFrom: CreatedFrom$outboundSchema.optional(),
    headers: z.record(z.string()).optional(),
    environments: z.array(GetIntegrationLogDrainsEnvironments$outboundSchema)
        .optional(),
    branch: z.string().optional(),
    samplingRate: z.number().optional(),
    source: z.union([
        z.lazy(() => GetIntegrationLogDrainsSource2$outboundSchema),
        z.lazy(() => GetIntegrationLogDrainsSource1$outboundSchema),
    ]),
});
export function getIntegrationLogDrainsResponseBodyToJSON(getIntegrationLogDrainsResponseBody) {
    return JSON.stringify(GetIntegrationLogDrainsResponseBody$outboundSchema.parse(getIntegrationLogDrainsResponseBody));
}
export function getIntegrationLogDrainsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetIntegrationLogDrainsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetIntegrationLogDrainsResponseBody' from JSON`);
}
