// @vercel/sdk/esm/models/getmemberop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * "The `ADMIN` role, by default, is provided to users capable of installing integrations, while the `USER` role can be granted to Vercel users with the Vercel `Billing` or Vercel `Viewer` role, which are considered to be Read-Only roles."
 */
export const GetMemberRole = {
    Admin: "ADMIN",
    User: "USER",
};
/** @internal */
export const GetMemberRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    memberId: z.string(),
});
/** @internal */
export const GetMemberRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    memberId: z.string(),
});
export function getMemberRequestToJSON(getMemberRequest) {
    return JSON.stringify(GetMemberRequest$outboundSchema.parse(getMemberRequest));
}
export function getMemberRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetMemberRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetMemberRequest' from JSON`);
}
/** @internal */
export const GetMemberRole$inboundSchema = z.nativeEnum(GetMemberRole);
/** @internal */
export const GetMemberRole$outboundSchema = GetMemberRole$inboundSchema;
/** @internal */
export const GetMemberResponseBody$inboundSchema = z.object({
    id: z.string(),
    role: GetMemberRole$inboundSchema,
});
/** @internal */
export const GetMemberResponseBody$outboundSchema = z.object({
    id: z.string(),
    role: GetMemberRole$outboundSchema,
});
export function getMemberResponseBodyToJSON(getMemberResponseBody) {
    return JSON.stringify(GetMemberResponseBody$outboundSchema.parse(getMemberResponseBody));
}
export function getMemberResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetMemberResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetMemberResponseBody' from JSON`);
}
