// @vercel/sdk/esm/models/getorderop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { Forbidden$inboundSchema, Forbidden$outboundSchema, } from "./forbidden.js";
import { NotAuthorizedForScope$inboundSchema, NotAuthorizedForScope$outboundSchema, } from "./notauthorizedforscope.js";
export const GetOrderDomainsPurchaseType = {
    Transfer: "transfer",
};
export const GetOrderDomainsDomainsRegistrarStatus = {
    Pending: "pending",
    Completed: "completed",
    Failed: "failed",
    Refunded: "refunded",
    RefundFailed: "refund-failed",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code = {
    ClaimsNoticeRequired: "claims-notice-required",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode = {
    ClientTransferProhibited: "client-transfer-prohibited",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode = {
    UnsupportedLanguageCode: "unsupported-language-code",
};
export const DomainsPurchaseType = {
    Renewal: "renewal",
};
export const GetOrderDomainsStatus = {
    Pending: "pending",
    Completed: "completed",
    Failed: "failed",
    Refunded: "refunded",
    RefundFailed: "refund-failed",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONCode = {
    ClaimsNoticeRequired: "claims-notice-required",
};
export const GetOrder1DomainsRegistrarResponse200Code = {
    ClientTransferProhibited: "client-transfer-prohibited",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code = {
    UnsupportedLanguageCode: "unsupported-language-code",
};
export const GetOrderDomainsDomainsRegistrarPurchaseType = {
    Purchase: "purchase",
};
export const DomainsStatus = {
    Pending: "pending",
    Completed: "completed",
    Failed: "failed",
    Refunded: "refunded",
    RefundFailed: "refund-failed",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code = {
    ClaimsNoticeRequired: "claims-notice-required",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode = {
    ClientTransferProhibited: "client-transfer-prohibited",
};
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code = {
    UnsupportedLanguageCode: "unsupported-language-code",
};
export const GetOrderStatus = {
    Draft: "draft",
    Purchasing: "purchasing",
    Completed: "completed",
    Failed: "failed",
};
export const GetOrder1DomainsRegistrarResponseCode = {
    UnexpectedError: "unexpected-error",
};
export const GetOrder1DomainsRegistrarCode = {
    PriceMismatch: "price-mismatch",
};
export const GetOrder1Code = {
    TldOutage: "tld-outage",
};
export const OneCode = {
    PaymentFailed: "payment-failed",
};
/** @internal */
export const GetOrderRequest$inboundSchema = z.object({
    orderId: z.string(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetOrderRequest$outboundSchema = z.object({
    orderId: z.string(),
    teamId: z.string().optional(),
});
export function getOrderRequestToJSON(getOrderRequest) {
    return JSON.stringify(GetOrderRequest$outboundSchema.parse(getOrderRequest));
}
export function getOrderRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderRequest' from JSON`);
}
/** @internal */
export const GetOrderDomainsRegistrarResponseBody$inboundSchema = z.union([NotAuthorizedForScope$inboundSchema, Forbidden$inboundSchema]);
/** @internal */
export const GetOrderDomainsRegistrarResponseBody$outboundSchema = z.union([NotAuthorizedForScope$outboundSchema, Forbidden$outboundSchema]);
export function getOrderDomainsRegistrarResponseBodyToJSON(getOrderDomainsRegistrarResponseBody) {
    return JSON.stringify(GetOrderDomainsRegistrarResponseBody$outboundSchema.parse(getOrderDomainsRegistrarResponseBody));
}
export function getOrderDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetOrderDomainsPurchaseType$inboundSchema = z.nativeEnum(GetOrderDomainsPurchaseType);
/** @internal */
export const GetOrderDomainsPurchaseType$outboundSchema = GetOrderDomainsPurchaseType$inboundSchema;
/** @internal */
export const GetOrderDomainsDomainsRegistrarStatus$inboundSchema = z.nativeEnum(GetOrderDomainsDomainsRegistrarStatus);
/** @internal */
export const GetOrderDomainsDomainsRegistrarStatus$outboundSchema = GetOrderDomainsDomainsRegistrarStatus$inboundSchema;
/** @internal */
export const GetOrderErrorDomainsRegistrar2$inboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
/** @internal */
export const GetOrderErrorDomainsRegistrar2$outboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
export function getOrderErrorDomainsRegistrar2ToJSON(getOrderErrorDomainsRegistrar2) {
    return JSON.stringify(GetOrderErrorDomainsRegistrar2$outboundSchema.parse(getOrderErrorDomainsRegistrar2));
}
export function getOrderErrorDomainsRegistrar2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderErrorDomainsRegistrar2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderErrorDomainsRegistrar2' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrar3$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code$inboundSchema,
});
/** @internal */
export const GetOrder1DomainsRegistrar3$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains3Code$outboundSchema,
});
export function getOrder1DomainsRegistrar3ToJSON(getOrder1DomainsRegistrar3) {
    return JSON.stringify(GetOrder1DomainsRegistrar3$outboundSchema.parse(getOrder1DomainsRegistrar3));
}
export function getOrder1DomainsRegistrar3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrar3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrar3' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrar2$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode$inboundSchema,
});
/** @internal */
export const GetOrder1DomainsRegistrar2$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomainsCode$outboundSchema,
});
export function getOrder1DomainsRegistrar2ToJSON(getOrder1DomainsRegistrar2) {
    return JSON.stringify(GetOrder1DomainsRegistrar2$outboundSchema.parse(getOrder1DomainsRegistrar2));
}
export function getOrder1DomainsRegistrar2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrar2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrar2' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarResponse200Details$inboundSchema = z.object({
    detectedLanguageCode: z.string(),
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse200Details$outboundSchema = z.object({
    detectedLanguageCode: z.string(),
});
export function getOrder1DomainsRegistrarResponse200DetailsToJSON(getOrder1DomainsRegistrarResponse200Details) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse200Details$outboundSchema.parse(getOrder1DomainsRegistrarResponse200Details));
}
export function getOrder1DomainsRegistrarResponse200DetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse200Details$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse200Details' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrar1$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode$inboundSchema,
    details: z.lazy(() => GetOrder1DomainsRegistrarResponse200Details$inboundSchema),
});
/** @internal */
export const GetOrder1DomainsRegistrar1$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyCode$outboundSchema,
    details: z.lazy(() => GetOrder1DomainsRegistrarResponse200Details$outboundSchema),
});
export function getOrder1DomainsRegistrar1ToJSON(getOrder1DomainsRegistrar1) {
    return JSON.stringify(GetOrder1DomainsRegistrar1$outboundSchema.parse(getOrder1DomainsRegistrar1));
}
export function getOrder1DomainsRegistrar1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrar1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrar1' from JSON`);
}
/** @internal */
export const GetOrderErrorDomainsRegistrar1$inboundSchema = z.union([
    z.lazy(() => GetOrder1DomainsRegistrar1$inboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrar2$inboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrar3$inboundSchema),
]);
/** @internal */
export const GetOrderErrorDomainsRegistrar1$outboundSchema = z.union([
    z.lazy(() => GetOrder1DomainsRegistrar1$outboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrar2$outboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrar3$outboundSchema),
]);
export function getOrderErrorDomainsRegistrar1ToJSON(getOrderErrorDomainsRegistrar1) {
    return JSON.stringify(GetOrderErrorDomainsRegistrar1$outboundSchema.parse(getOrderErrorDomainsRegistrar1));
}
export function getOrderErrorDomainsRegistrar1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderErrorDomainsRegistrar1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderErrorDomainsRegistrar1' from JSON`);
}
/** @internal */
export const GetOrderDomainsDomainsRegistrarError$inboundSchema = z.union([
    z.lazy(() => GetOrderErrorDomainsRegistrar2$inboundSchema),
    z.union([
        z.lazy(() => GetOrder1DomainsRegistrar1$inboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrar2$inboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrar3$inboundSchema),
    ]),
]);
/** @internal */
export const GetOrderDomainsDomainsRegistrarError$outboundSchema = z.union([
    z.lazy(() => GetOrderErrorDomainsRegistrar2$outboundSchema),
    z.union([
        z.lazy(() => GetOrder1DomainsRegistrar1$outboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrar2$outboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrar3$outboundSchema),
    ]),
]);
export function getOrderDomainsDomainsRegistrarErrorToJSON(getOrderDomainsDomainsRegistrarError) {
    return JSON.stringify(GetOrderDomainsDomainsRegistrarError$outboundSchema.parse(getOrderDomainsDomainsRegistrarError));
}
export function getOrderDomainsDomainsRegistrarErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderDomainsDomainsRegistrarError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderDomainsDomainsRegistrarError' from JSON`);
}
/** @internal */
export const Domains3$inboundSchema = z.object({
    purchaseType: GetOrderDomainsPurchaseType$inboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsDomainsRegistrarStatus$inboundSchema,
    price: z.number(),
    error: z.union([
        z.lazy(() => GetOrderErrorDomainsRegistrar2$inboundSchema),
        z.union([
            z.lazy(() => GetOrder1DomainsRegistrar1$inboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrar2$inboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrar3$inboundSchema),
        ]),
    ]).optional(),
});
/** @internal */
export const Domains3$outboundSchema = z.object({
    purchaseType: GetOrderDomainsPurchaseType$outboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsDomainsRegistrarStatus$outboundSchema,
    price: z.number(),
    error: z.union([
        z.lazy(() => GetOrderErrorDomainsRegistrar2$outboundSchema),
        z.union([
            z.lazy(() => GetOrder1DomainsRegistrar1$outboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrar2$outboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrar3$outboundSchema),
        ]),
    ]).optional(),
});
export function domains3ToJSON(domains3) {
    return JSON.stringify(Domains3$outboundSchema.parse(domains3));
}
export function domains3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domains3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domains3' from JSON`);
}
/** @internal */
export const DomainsPurchaseType$inboundSchema = z.nativeEnum(DomainsPurchaseType);
/** @internal */
export const DomainsPurchaseType$outboundSchema = DomainsPurchaseType$inboundSchema;
/** @internal */
export const GetOrderDomainsStatus$inboundSchema = z.nativeEnum(GetOrderDomainsStatus);
/** @internal */
export const GetOrderDomainsStatus$outboundSchema = GetOrderDomainsStatus$inboundSchema;
/** @internal */
export const GetOrderError2$inboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
/** @internal */
export const GetOrderError2$outboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
export function getOrderError2ToJSON(getOrderError2) {
    return JSON.stringify(GetOrderError2$outboundSchema.parse(getOrderError2));
}
export function getOrderError2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderError2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderError2' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONCode$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONCode);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONCode$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONCode$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarResponse2003$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONCode$inboundSchema,
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse2003$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONCode$outboundSchema,
});
export function getOrder1DomainsRegistrarResponse2003ToJSON(getOrder1DomainsRegistrarResponse2003) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse2003$outboundSchema.parse(getOrder1DomainsRegistrarResponse2003));
}
export function getOrder1DomainsRegistrarResponse2003FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse2003$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse2003' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200Code$inboundSchema = z
    .nativeEnum(GetOrder1DomainsRegistrarResponse200Code);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200Code$outboundSchema = GetOrder1DomainsRegistrarResponse200Code$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarResponse2002$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200Code$inboundSchema,
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse2002$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200Code$outboundSchema,
});
export function getOrder1DomainsRegistrarResponse2002ToJSON(getOrder1DomainsRegistrarResponse2002) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse2002$outboundSchema.parse(getOrder1DomainsRegistrarResponse2002));
}
export function getOrder1DomainsRegistrarResponse2002FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse2002$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse2002' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarResponseDetails$inboundSchema = z.object({
    detectedLanguageCode: z.string(),
});
/** @internal */
export const GetOrder1DomainsRegistrarResponseDetails$outboundSchema = z.object({
    detectedLanguageCode: z.string(),
});
export function getOrder1DomainsRegistrarResponseDetailsToJSON(getOrder1DomainsRegistrarResponseDetails) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponseDetails$outboundSchema.parse(getOrder1DomainsRegistrarResponseDetails));
}
export function getOrder1DomainsRegistrarResponseDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponseDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponseDetails' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse2001$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code$inboundSchema,
    details: z.lazy(() => GetOrder1DomainsRegistrarResponseDetails$inboundSchema),
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse2001$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains2Code$outboundSchema,
    details: z.lazy(() => GetOrder1DomainsRegistrarResponseDetails$outboundSchema),
});
export function getOrder1DomainsRegistrarResponse2001ToJSON(getOrder1DomainsRegistrarResponse2001) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse2001$outboundSchema.parse(getOrder1DomainsRegistrarResponse2001));
}
export function getOrder1DomainsRegistrarResponse2001FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse2001$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse2001' from JSON`);
}
/** @internal */
export const GetOrderError1$inboundSchema = z.union([
    z.lazy(() => GetOrder1DomainsRegistrarResponse2001$inboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse2002$inboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse2003$inboundSchema),
]);
/** @internal */
export const GetOrderError1$outboundSchema = z.union([
    z.lazy(() => GetOrder1DomainsRegistrarResponse2001$outboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse2002$outboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse2003$outboundSchema),
]);
export function getOrderError1ToJSON(getOrderError1) {
    return JSON.stringify(GetOrderError1$outboundSchema.parse(getOrderError1));
}
export function getOrderError1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderError1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderError1' from JSON`);
}
/** @internal */
export const GetOrderDomainsError$inboundSchema = z.union([
    z.lazy(() => GetOrderError2$inboundSchema),
    z.union([
        z.lazy(() => GetOrder1DomainsRegistrarResponse2001$inboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse2002$inboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse2003$inboundSchema),
    ]),
]);
/** @internal */
export const GetOrderDomainsError$outboundSchema = z.union([
    z.lazy(() => GetOrderError2$outboundSchema),
    z.union([
        z.lazy(() => GetOrder1DomainsRegistrarResponse2001$outboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse2002$outboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse2003$outboundSchema),
    ]),
]);
export function getOrderDomainsErrorToJSON(getOrderDomainsError) {
    return JSON.stringify(GetOrderDomainsError$outboundSchema.parse(getOrderDomainsError));
}
export function getOrderDomainsErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderDomainsError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderDomainsError' from JSON`);
}
/** @internal */
export const Domains2$inboundSchema = z.object({
    purchaseType: DomainsPurchaseType$inboundSchema,
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsStatus$inboundSchema,
    price: z.number(),
    error: z.union([
        z.lazy(() => GetOrderError2$inboundSchema),
        z.union([
            z.lazy(() => GetOrder1DomainsRegistrarResponse2001$inboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse2002$inboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse2003$inboundSchema),
        ]),
    ]).optional(),
});
/** @internal */
export const Domains2$outboundSchema = z.object({
    purchaseType: DomainsPurchaseType$outboundSchema,
    years: z.number(),
    domainName: z.string(),
    status: GetOrderDomainsStatus$outboundSchema,
    price: z.number(),
    error: z.union([
        z.lazy(() => GetOrderError2$outboundSchema),
        z.union([
            z.lazy(() => GetOrder1DomainsRegistrarResponse2001$outboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse2002$outboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse2003$outboundSchema),
        ]),
    ]).optional(),
});
export function domains2ToJSON(domains2) {
    return JSON.stringify(Domains2$outboundSchema.parse(domains2));
}
export function domains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domains2' from JSON`);
}
/** @internal */
export const GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema = z
    .nativeEnum(GetOrderDomainsDomainsRegistrarPurchaseType);
/** @internal */
export const GetOrderDomainsDomainsRegistrarPurchaseType$outboundSchema = GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema;
/** @internal */
export const DomainsStatus$inboundSchema = z.nativeEnum(DomainsStatus);
/** @internal */
export const DomainsStatus$outboundSchema = DomainsStatus$inboundSchema;
/** @internal */
export const GetOrderErrorDomainsRegistrarResponse2$inboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
/** @internal */
export const GetOrderErrorDomainsRegistrarResponse2$outboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
export function getOrderErrorDomainsRegistrarResponse2ToJSON(getOrderErrorDomainsRegistrarResponse2) {
    return JSON.stringify(GetOrderErrorDomainsRegistrarResponse2$outboundSchema.parse(getOrderErrorDomainsRegistrarResponse2));
}
export function getOrderErrorDomainsRegistrarResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderErrorDomainsRegistrarResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderErrorDomainsRegistrarResponse2' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarResponse3$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code$inboundSchema,
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse3$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Error3Code$outboundSchema,
});
export function getOrder1DomainsRegistrarResponse3ToJSON(getOrder1DomainsRegistrarResponse3) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse3$outboundSchema.parse(getOrder1DomainsRegistrarResponse3));
}
export function getOrder1DomainsRegistrarResponse3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse3' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarResponse2$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode$inboundSchema,
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse2$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1ErrorCode$outboundSchema,
});
export function getOrder1DomainsRegistrarResponse2ToJSON(getOrder1DomainsRegistrarResponse2) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse2$outboundSchema.parse(getOrder1DomainsRegistrarResponse2));
}
export function getOrder1DomainsRegistrarResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse2' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code);
/** @internal */
export const GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code$outboundSchema = GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code$inboundSchema;
/** @internal */
export const GetOrder1DomainsRegistrarDetails$inboundSchema = z.object({
    detectedLanguageCode: z.string(),
});
/** @internal */
export const GetOrder1DomainsRegistrarDetails$outboundSchema = z.object({
    detectedLanguageCode: z.string(),
});
export function getOrder1DomainsRegistrarDetailsToJSON(getOrder1DomainsRegistrarDetails) {
    return JSON.stringify(GetOrder1DomainsRegistrarDetails$outboundSchema.parse(getOrder1DomainsRegistrarDetails));
}
export function getOrder1DomainsRegistrarDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarDetails' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponse1$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code$inboundSchema,
    details: z.lazy(() => GetOrder1DomainsRegistrarDetails$inboundSchema),
});
/** @internal */
export const GetOrder1DomainsRegistrarResponse1$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponse200ApplicationJSONResponseBodyDomains1Code$outboundSchema,
    details: z.lazy(() => GetOrder1DomainsRegistrarDetails$outboundSchema),
});
export function getOrder1DomainsRegistrarResponse1ToJSON(getOrder1DomainsRegistrarResponse1) {
    return JSON.stringify(GetOrder1DomainsRegistrarResponse1$outboundSchema.parse(getOrder1DomainsRegistrarResponse1));
}
export function getOrder1DomainsRegistrarResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1DomainsRegistrarResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1DomainsRegistrarResponse1' from JSON`);
}
/** @internal */
export const GetOrderErrorDomainsRegistrarResponse1$inboundSchema = z.union([
    z.lazy(() => GetOrder1DomainsRegistrarResponse1$inboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse2$inboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse3$inboundSchema),
]);
/** @internal */
export const GetOrderErrorDomainsRegistrarResponse1$outboundSchema = z.union([
    z.lazy(() => GetOrder1DomainsRegistrarResponse1$outboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse2$outboundSchema),
    z.lazy(() => GetOrder1DomainsRegistrarResponse3$outboundSchema),
]);
export function getOrderErrorDomainsRegistrarResponse1ToJSON(getOrderErrorDomainsRegistrarResponse1) {
    return JSON.stringify(GetOrderErrorDomainsRegistrarResponse1$outboundSchema.parse(getOrderErrorDomainsRegistrarResponse1));
}
export function getOrderErrorDomainsRegistrarResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderErrorDomainsRegistrarResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderErrorDomainsRegistrarResponse1' from JSON`);
}
/** @internal */
export const DomainsError$inboundSchema = z.union([
    z.lazy(() => GetOrderErrorDomainsRegistrarResponse2$inboundSchema),
    z.union([
        z.lazy(() => GetOrder1DomainsRegistrarResponse1$inboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse2$inboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse3$inboundSchema),
    ]),
]);
/** @internal */
export const DomainsError$outboundSchema = z.union([
    z.lazy(() => GetOrderErrorDomainsRegistrarResponse2$outboundSchema),
    z.union([
        z.lazy(() => GetOrder1DomainsRegistrarResponse1$outboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse2$outboundSchema),
        z.lazy(() => GetOrder1DomainsRegistrarResponse3$outboundSchema),
    ]),
]);
export function domainsErrorToJSON(domainsError) {
    return JSON.stringify(DomainsError$outboundSchema.parse(domainsError));
}
export function domainsErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DomainsError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DomainsError' from JSON`);
}
/** @internal */
export const Domains1$inboundSchema = z.object({
    purchaseType: GetOrderDomainsDomainsRegistrarPurchaseType$inboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: DomainsStatus$inboundSchema,
    price: z.number(),
    error: z.union([
        z.lazy(() => GetOrderErrorDomainsRegistrarResponse2$inboundSchema),
        z.union([
            z.lazy(() => GetOrder1DomainsRegistrarResponse1$inboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse2$inboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse3$inboundSchema),
        ]),
    ]).optional(),
});
/** @internal */
export const Domains1$outboundSchema = z.object({
    purchaseType: GetOrderDomainsDomainsRegistrarPurchaseType$outboundSchema,
    autoRenew: z.boolean(),
    years: z.number(),
    domainName: z.string(),
    status: DomainsStatus$outboundSchema,
    price: z.number(),
    error: z.union([
        z.lazy(() => GetOrderErrorDomainsRegistrarResponse2$outboundSchema),
        z.union([
            z.lazy(() => GetOrder1DomainsRegistrarResponse1$outboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse2$outboundSchema),
            z.lazy(() => GetOrder1DomainsRegistrarResponse3$outboundSchema),
        ]),
    ]).optional(),
});
export function domains1ToJSON(domains1) {
    return JSON.stringify(Domains1$outboundSchema.parse(domains1));
}
export function domains1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domains1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domains1' from JSON`);
}
/** @internal */
export const GetOrderDomains$inboundSchema = z.union([
    z.lazy(() => Domains1$inboundSchema),
    z.lazy(() => Domains3$inboundSchema),
    z.lazy(() => Domains2$inboundSchema),
]);
/** @internal */
export const GetOrderDomains$outboundSchema = z.union([
    z.lazy(() => Domains1$outboundSchema),
    z.lazy(() => Domains3$outboundSchema),
    z.lazy(() => Domains2$outboundSchema),
]);
export function getOrderDomainsToJSON(getOrderDomains) {
    return JSON.stringify(GetOrderDomains$outboundSchema.parse(getOrderDomains));
}
export function getOrderDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderDomains' from JSON`);
}
/** @internal */
export const GetOrderStatus$inboundSchema = z.nativeEnum(GetOrderStatus);
/** @internal */
export const GetOrderStatus$outboundSchema = GetOrderStatus$inboundSchema;
/** @internal */
export const Error2$inboundSchema = z
    .object({
    code: z.string(),
    details: z.any().optional(),
});
/** @internal */
export const Error2$outboundSchema = z.object({
    code: z.string(),
    details: z.any().optional(),
});
export function error2ToJSON(error2) {
    return JSON.stringify(Error2$outboundSchema.parse(error2));
}
export function error2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Error2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Error2' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarResponseCode$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarResponseCode);
/** @internal */
export const GetOrder1DomainsRegistrarResponseCode$outboundSchema = GetOrder1DomainsRegistrarResponseCode$inboundSchema;
/** @internal */
export const One4$inboundSchema = z
    .object({
    code: GetOrder1DomainsRegistrarResponseCode$inboundSchema,
});
/** @internal */
export const One4$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarResponseCode$outboundSchema,
});
export function one4ToJSON(one4) {
    return JSON.stringify(One4$outboundSchema.parse(one4));
}
export function one4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => One4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'One4' from JSON`);
}
/** @internal */
export const GetOrder1DomainsRegistrarCode$inboundSchema = z.nativeEnum(GetOrder1DomainsRegistrarCode);
/** @internal */
export const GetOrder1DomainsRegistrarCode$outboundSchema = GetOrder1DomainsRegistrarCode$inboundSchema;
/** @internal */
export const OneDetails$inboundSchema = z.object({
    expectedPrice: z.number(),
    actualPrice: z.number().optional(),
});
/** @internal */
export const OneDetails$outboundSchema = z.object({
    expectedPrice: z.number(),
    actualPrice: z.number().optional(),
});
export function oneDetailsToJSON(oneDetails) {
    return JSON.stringify(OneDetails$outboundSchema.parse(oneDetails));
}
export function oneDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => OneDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OneDetails' from JSON`);
}
/** @internal */
export const GetOrder13$inboundSchema = z.object({
    code: GetOrder1DomainsRegistrarCode$inboundSchema,
    details: z.lazy(() => OneDetails$inboundSchema),
});
/** @internal */
export const GetOrder13$outboundSchema = z.object({
    code: GetOrder1DomainsRegistrarCode$outboundSchema,
    details: z.lazy(() => OneDetails$outboundSchema),
});
export function getOrder13ToJSON(getOrder13) {
    return JSON.stringify(GetOrder13$outboundSchema.parse(getOrder13));
}
export function getOrder13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder13' from JSON`);
}
/** @internal */
export const GetOrder1Code$inboundSchema = z.nativeEnum(GetOrder1Code);
/** @internal */
export const GetOrder1Code$outboundSchema = GetOrder1Code$inboundSchema;
/** @internal */
export const Tlds$inboundSchema = z
    .object({
    tldName: z.string(),
    endsAt: z.string(),
});
/** @internal */
export const Tlds$outboundSchema = z.object({
    tldName: z.string(),
    endsAt: z.string(),
});
export function tldsToJSON(tlds) {
    return JSON.stringify(Tlds$outboundSchema.parse(tlds));
}
export function tldsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Tlds$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Tlds' from JSON`);
}
/** @internal */
export const GetOrder1Details$inboundSchema = z.object({
    tlds: z.array(z.lazy(() => Tlds$inboundSchema)),
});
/** @internal */
export const GetOrder1Details$outboundSchema = z.object({
    tlds: z.array(z.lazy(() => Tlds$outboundSchema)),
});
export function getOrder1DetailsToJSON(getOrder1Details) {
    return JSON.stringify(GetOrder1Details$outboundSchema.parse(getOrder1Details));
}
export function getOrder1DetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder1Details$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder1Details' from JSON`);
}
/** @internal */
export const GetOrder12$inboundSchema = z.object({
    code: GetOrder1Code$inboundSchema,
    details: z.lazy(() => GetOrder1Details$inboundSchema),
});
/** @internal */
export const GetOrder12$outboundSchema = z.object({
    code: GetOrder1Code$outboundSchema,
    details: z.lazy(() => GetOrder1Details$outboundSchema),
});
export function getOrder12ToJSON(getOrder12) {
    return JSON.stringify(GetOrder12$outboundSchema.parse(getOrder12));
}
export function getOrder12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder12' from JSON`);
}
/** @internal */
export const OneCode$inboundSchema = z
    .nativeEnum(OneCode);
/** @internal */
export const OneCode$outboundSchema = OneCode$inboundSchema;
/** @internal */
export const GetOrder11$inboundSchema = z.object({
    code: OneCode$inboundSchema,
});
/** @internal */
export const GetOrder11$outboundSchema = z.object({
    code: OneCode$outboundSchema,
});
export function getOrder11ToJSON(getOrder11) {
    return JSON.stringify(GetOrder11$outboundSchema.parse(getOrder11));
}
export function getOrder11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrder11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrder11' from JSON`);
}
/** @internal */
export const Error1$inboundSchema = z
    .union([
    z.lazy(() => GetOrder12$inboundSchema),
    z.lazy(() => GetOrder13$inboundSchema),
    z.lazy(() => GetOrder11$inboundSchema),
    z.lazy(() => One4$inboundSchema),
]);
/** @internal */
export const Error1$outboundSchema = z.union([
    z.lazy(() => GetOrder12$outboundSchema),
    z.lazy(() => GetOrder13$outboundSchema),
    z.lazy(() => GetOrder11$outboundSchema),
    z.lazy(() => One4$outboundSchema),
]);
export function error1ToJSON(error1) {
    return JSON.stringify(Error1$outboundSchema.parse(error1));
}
export function error1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Error1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Error1' from JSON`);
}
/** @internal */
export const ErrorT$inboundSchema = z
    .union([
    z.lazy(() => Error2$inboundSchema),
    z.union([
        z.lazy(() => GetOrder12$inboundSchema),
        z.lazy(() => GetOrder13$inboundSchema),
        z.lazy(() => GetOrder11$inboundSchema),
        z.lazy(() => One4$inboundSchema),
    ]),
]);
/** @internal */
export const ErrorT$outboundSchema = z.union([
    z.lazy(() => Error2$outboundSchema),
    z.union([
        z.lazy(() => GetOrder12$outboundSchema),
        z.lazy(() => GetOrder13$outboundSchema),
        z.lazy(() => GetOrder11$outboundSchema),
        z.lazy(() => One4$outboundSchema),
    ]),
]);
export function errorToJSON(errorT) {
    return JSON.stringify(ErrorT$outboundSchema.parse(errorT));
}
export function errorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ErrorT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ErrorT' from JSON`);
}
/** @internal */
export const GetOrderResponseBody$inboundSchema = z.object({
    orderId: z.string(),
    domains: z.array(z.union([
        z.lazy(() => Domains1$inboundSchema),
        z.lazy(() => Domains3$inboundSchema),
        z.lazy(() => Domains2$inboundSchema),
    ])),
    status: GetOrderStatus$inboundSchema,
    error: z.union([
        z.lazy(() => Error2$inboundSchema),
        z.union([
            z.lazy(() => GetOrder12$inboundSchema),
            z.lazy(() => GetOrder13$inboundSchema),
            z.lazy(() => GetOrder11$inboundSchema),
            z.lazy(() => One4$inboundSchema),
        ]),
    ]).optional(),
});
/** @internal */
export const GetOrderResponseBody$outboundSchema = z.object({
    orderId: z.string(),
    domains: z.array(z.union([
        z.lazy(() => Domains1$outboundSchema),
        z.lazy(() => Domains3$outboundSchema),
        z.lazy(() => Domains2$outboundSchema),
    ])),
    status: GetOrderStatus$outboundSchema,
    error: z.union([
        z.lazy(() => Error2$outboundSchema),
        z.union([
            z.lazy(() => GetOrder12$outboundSchema),
            z.lazy(() => GetOrder13$outboundSchema),
            z.lazy(() => GetOrder11$outboundSchema),
            z.lazy(() => One4$outboundSchema),
        ]),
    ]).optional(),
});
export function getOrderResponseBodyToJSON(getOrderResponseBody) {
    return JSON.stringify(GetOrderResponseBody$outboundSchema.parse(getOrderResponseBody));
}
export function getOrderResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetOrderResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetOrderResponseBody' from JSON`);
}
