// @vercel/sdk/esm/models/getprojectclientcertsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetProjectClientCertsRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetProjectClientCertsRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getProjectClientCertsRequestToJSON(getProjectClientCertsRequest) {
    return JSON.stringify(GetProjectClientCertsRequest$outboundSchema.parse(getProjectClientCertsRequest));
}
export function getProjectClientCertsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectClientCertsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectClientCertsRequest' from JSON`);
}
/** @internal */
export const ClientCerts$inboundSchema = z.object({
    origin: z.string(),
    id: z.string(),
});
/** @internal */
export const ClientCerts$outboundSchema = z.object({
    origin: z.string(),
    id: z.string(),
});
export function clientCertsToJSON(clientCerts) {
    return JSON.stringify(ClientCerts$outboundSchema.parse(clientCerts));
}
export function clientCertsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ClientCerts$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ClientCerts' from JSON`);
}
/** @internal */
export const GetProjectClientCertsResponseBody$inboundSchema = z.object({
    clientCerts: z.array(z.lazy(() => ClientCerts$inboundSchema)),
});
/** @internal */
export const GetProjectClientCertsResponseBody$outboundSchema = z.object({
    clientCerts: z.array(z.lazy(() => ClientCerts$outboundSchema)),
});
export function getProjectClientCertsResponseBodyToJSON(getProjectClientCertsResponseBody) {
    return JSON.stringify(GetProjectClientCertsResponseBody$outboundSchema.parse(getProjectClientCertsResponseBody));
}
export function getProjectClientCertsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectClientCertsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectClientCertsResponseBody' from JSON`);
}
