// @vercel/sdk/esm/models/getprojectenvop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const GetProjectEnvTargetProjectsResponse2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const GetProjectEnvResponseBodyProjectsResponseType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type = {
    PostgresDatabase: "postgres-database",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type = {
    PostgresPassword: "postgres-password",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type = {
    PostgresHost: "postgres-host",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type = {
    PostgresUser: "postgres-user",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type = {
    PostgresUrl: "postgres-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type = {
    RedisUrl: "redis-url",
};
export const GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type = {
    FlagsSecret: "flags-secret",
};
export const GetProjectEnvTargetProjects2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const GetProjectEnvTargetProjects1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const GetProjectEnvResponseBodyProjectsType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type = {
    PostgresDatabase: "postgres-database",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type = {
    PostgresPassword: "postgres-password",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type = {
    PostgresHost: "postgres-host",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type = {
    PostgresUser: "postgres-user",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type = {
    PostgresUrl: "postgres-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type = {
    RedisUrl: "redis-url",
};
export const GetProjectEnvResponseBodyProjectsResponse200Type = {
    FlagsSecret: "flags-secret",
};
export const GetProjectEnvTarget2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const GetProjectEnvTarget1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const GetProjectEnvResponseBodyType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type = {
    PostgresDatabase: "postgres-database",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type = {
    PostgresPassword: "postgres-password",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type = {
    PostgresHost: "postgres-host",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type = {
    PostgresUser: "postgres-user",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType = {
    PostgresUrl: "postgres-url",
};
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONType = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const GetProjectEnvContentHintProjectsResponse200Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const GetProjectEnvContentHintProjectsResponseType = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const GetProjectEnvContentHintProjectsType = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const GetProjectEnvContentHintType = {
    RedisUrl: "redis-url",
};
export const GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType = {
    FlagsSecret: "flags-secret",
};
/** @internal */
export const GetProjectEnvRequest$inboundSchema = z.object({
    idOrName: z.string(),
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetProjectEnvRequest$outboundSchema = z.object({
    idOrName: z.string(),
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getProjectEnvRequestToJSON(getProjectEnvRequest) {
    return JSON.stringify(GetProjectEnvRequest$outboundSchema.parse(getProjectEnvRequest));
}
export function getProjectEnvRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvRequest' from JSON`);
}
/** @internal */
export const GetProjectEnvTargetProjectsResponse2$inboundSchema = z.nativeEnum(GetProjectEnvTargetProjectsResponse2);
/** @internal */
export const GetProjectEnvTargetProjectsResponse2$outboundSchema = GetProjectEnvTargetProjectsResponse2$inboundSchema;
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseTarget$inboundSchema = z.union([
    z.array(z.string()),
    GetProjectEnvTargetProjectsResponse2$inboundSchema,
]);
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseTarget$outboundSchema = z.union([
    z.array(z.string()),
    GetProjectEnvTargetProjectsResponse2$outboundSchema,
]);
export function getProjectEnvResponseBodyProjectsResponseTargetToJSON(getProjectEnvResponseBodyProjectsResponseTarget) {
    return JSON.stringify(GetProjectEnvResponseBodyProjectsResponseTarget$outboundSchema.parse(getProjectEnvResponseBodyProjectsResponseTarget));
}
export function getProjectEnvResponseBodyProjectsResponseTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyProjectsResponseTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyProjectsResponseTarget' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseType$inboundSchema = z
    .nativeEnum(GetProjectEnvResponseBodyProjectsResponseType);
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseType$outboundSchema = GetProjectEnvResponseBodyProjectsResponseType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse15$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse15$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$outboundSchema,
    projectId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse15ToJSON(getProjectEnvContentHintProjectsResponse15) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse15$outboundSchema.parse(getProjectEnvContentHintProjectsResponse15));
}
export function getProjectEnvContentHintProjectsResponse15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse15' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse14$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse14$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse14ToJSON(getProjectEnvContentHintProjectsResponse14) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse14$outboundSchema.parse(getProjectEnvContentHintProjectsResponse14));
}
export function getProjectEnvContentHintProjectsResponse14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse14' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse13$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse13$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse13ToJSON(getProjectEnvContentHintProjectsResponse13) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse13$outboundSchema.parse(getProjectEnvContentHintProjectsResponse13));
}
export function getProjectEnvContentHintProjectsResponse13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse13' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse12$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse12$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse12ToJSON(getProjectEnvContentHintProjectsResponse12) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse12$outboundSchema.parse(getProjectEnvContentHintProjectsResponse12));
}
export function getProjectEnvContentHintProjectsResponse12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse12' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse11$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse11$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse11ToJSON(getProjectEnvContentHintProjectsResponse11) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse11$outboundSchema.parse(getProjectEnvContentHintProjectsResponse11));
}
export function getProjectEnvContentHintProjectsResponse11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse11' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse10$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse10$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse10ToJSON(getProjectEnvContentHintProjectsResponse10) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse10$outboundSchema.parse(getProjectEnvContentHintProjectsResponse10));
}
export function getProjectEnvContentHintProjectsResponse10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse10' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse9$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse9$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse9ToJSON(getProjectEnvContentHintProjectsResponse9) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse9$outboundSchema.parse(getProjectEnvContentHintProjectsResponse9));
}
export function getProjectEnvContentHintProjectsResponse9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse9' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse8$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse8$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse8ToJSON(getProjectEnvContentHintProjectsResponse8) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse8$outboundSchema.parse(getProjectEnvContentHintProjectsResponse8));
}
export function getProjectEnvContentHintProjectsResponse8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse8' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse7$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse7$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse7ToJSON(getProjectEnvContentHintProjectsResponse7) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse7$outboundSchema.parse(getProjectEnvContentHintProjectsResponse7));
}
export function getProjectEnvContentHintProjectsResponse7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse7' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse6$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse6$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse6ToJSON(getProjectEnvContentHintProjectsResponse6) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse6$outboundSchema.parse(getProjectEnvContentHintProjectsResponse6));
}
export function getProjectEnvContentHintProjectsResponse6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse6' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse5$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse5$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse5ToJSON(getProjectEnvContentHintProjectsResponse5) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse5$outboundSchema.parse(getProjectEnvContentHintProjectsResponse5));
}
export function getProjectEnvContentHintProjectsResponse5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse5' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse4$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse4$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse4ToJSON(getProjectEnvContentHintProjectsResponse4) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse4$outboundSchema.parse(getProjectEnvContentHintProjectsResponse4));
}
export function getProjectEnvContentHintProjectsResponse4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse4' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse3$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse3$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse3ToJSON(getProjectEnvContentHintProjectsResponse3) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse3$outboundSchema.parse(getProjectEnvContentHintProjectsResponse3));
}
export function getProjectEnvContentHintProjectsResponse3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse3' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse2$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse2$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse2ToJSON(getProjectEnvContentHintProjectsResponse2) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse2$outboundSchema.parse(getProjectEnvContentHintProjectsResponse2));
}
export function getProjectEnvContentHintProjectsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse2' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse1$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjectsResponse1$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjectsResponse1ToJSON(getProjectEnvContentHintProjectsResponse1) {
    return JSON.stringify(GetProjectEnvContentHintProjectsResponse1$outboundSchema.parse(getProjectEnvContentHintProjectsResponse1));
}
export function getProjectEnvContentHintProjectsResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjectsResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjectsResponse1' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseContentHint$inboundSchema = z.union([
    z.lazy(() => GetProjectEnvContentHintProjectsResponse14$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse1$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse2$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse3$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse4$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse5$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse6$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse7$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse8$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse9$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse10$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse11$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse12$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse13$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse15$inboundSchema),
]);
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseContentHint$outboundSchema = z.union([
    z.lazy(() => GetProjectEnvContentHintProjectsResponse14$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse1$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse2$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse3$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse4$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse5$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse6$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse7$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse8$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse9$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse10$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse11$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse12$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse13$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjectsResponse15$outboundSchema),
]);
export function getProjectEnvResponseBodyProjectsResponseContentHintToJSON(getProjectEnvResponseBodyProjectsResponseContentHint) {
    return JSON.stringify(GetProjectEnvResponseBodyProjectsResponseContentHint$outboundSchema.parse(getProjectEnvResponseBodyProjectsResponseContentHint));
}
export function getProjectEnvResponseBodyProjectsResponseContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyProjectsResponseContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyProjectsResponseContentHint' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$inboundSchema = z.nativeEnum(GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type);
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$outboundSchema = GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$inboundSchema;
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseInternalContentHint$inboundSchema = z.object({
    type: GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponseInternalContentHint$outboundSchema = z.object({
    type: GetProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$outboundSchema,
    encryptedValue: z.string(),
});
export function getProjectEnvResponseBodyProjectsResponseInternalContentHintToJSON(getProjectEnvResponseBodyProjectsResponseInternalContentHint) {
    return JSON.stringify(GetProjectEnvResponseBodyProjectsResponseInternalContentHint$outboundSchema
        .parse(getProjectEnvResponseBodyProjectsResponseInternalContentHint));
}
export function getProjectEnvResponseBodyProjectsResponseInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyProjectsResponseInternalContentHint$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyProjectsResponseInternalContentHint' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBody3$inboundSchema = z.object({
    target: z.union([
        z.array(z.string()),
        GetProjectEnvTargetProjectsResponse2$inboundSchema,
    ]).optional(),
    type: GetProjectEnvResponseBodyProjectsResponseType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => GetProjectEnvContentHintProjectsResponse14$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse1$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse2$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse3$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse4$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse5$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse6$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse7$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse8$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse9$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse10$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse11$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse12$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse13$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => GetProjectEnvResponseBodyProjectsResponseInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const GetProjectEnvResponseBody3$outboundSchema = z.object({
    target: z.union([
        z.array(z.string()),
        GetProjectEnvTargetProjectsResponse2$outboundSchema,
    ]).optional(),
    type: GetProjectEnvResponseBodyProjectsResponseType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => GetProjectEnvContentHintProjectsResponse14$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse1$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse2$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse3$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse4$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse5$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse6$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse7$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse8$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse9$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse10$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse11$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse12$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse13$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjectsResponse15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => GetProjectEnvResponseBodyProjectsResponseInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function getProjectEnvResponseBody3ToJSON(getProjectEnvResponseBody3) {
    return JSON.stringify(GetProjectEnvResponseBody3$outboundSchema.parse(getProjectEnvResponseBody3));
}
export function getProjectEnvResponseBody3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBody3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBody3' from JSON`);
}
/** @internal */
export const GetProjectEnvTargetProjects2$inboundSchema = z.nativeEnum(GetProjectEnvTargetProjects2);
/** @internal */
export const GetProjectEnvTargetProjects2$outboundSchema = GetProjectEnvTargetProjects2$inboundSchema;
/** @internal */
export const GetProjectEnvTargetProjects1$inboundSchema = z.nativeEnum(GetProjectEnvTargetProjects1);
/** @internal */
export const GetProjectEnvTargetProjects1$outboundSchema = GetProjectEnvTargetProjects1$inboundSchema;
/** @internal */
export const GetProjectEnvResponseBodyProjectsTarget$inboundSchema = z.union([
    z.array(GetProjectEnvTargetProjects1$inboundSchema),
    GetProjectEnvTargetProjects2$inboundSchema,
]);
/** @internal */
export const GetProjectEnvResponseBodyProjectsTarget$outboundSchema = z.union([
    z.array(GetProjectEnvTargetProjects1$outboundSchema),
    GetProjectEnvTargetProjects2$outboundSchema,
]);
export function getProjectEnvResponseBodyProjectsTargetToJSON(getProjectEnvResponseBodyProjectsTarget) {
    return JSON.stringify(GetProjectEnvResponseBodyProjectsTarget$outboundSchema.parse(getProjectEnvResponseBodyProjectsTarget));
}
export function getProjectEnvResponseBodyProjectsTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyProjectsTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyProjectsTarget' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsType$inboundSchema = z.nativeEnum(GetProjectEnvResponseBodyProjectsType);
/** @internal */
export const GetProjectEnvResponseBodyProjectsType$outboundSchema = GetProjectEnvResponseBodyProjectsType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects15$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects15$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$outboundSchema,
    projectId: z.string(),
});
export function getProjectEnvContentHintProjects15ToJSON(getProjectEnvContentHintProjects15) {
    return JSON.stringify(GetProjectEnvContentHintProjects15$outboundSchema.parse(getProjectEnvContentHintProjects15));
}
export function getProjectEnvContentHintProjects15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects15' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects14$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects14$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function getProjectEnvContentHintProjects14ToJSON(getProjectEnvContentHintProjects14) {
    return JSON.stringify(GetProjectEnvContentHintProjects14$outboundSchema.parse(getProjectEnvContentHintProjects14));
}
export function getProjectEnvContentHintProjects14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects14' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects13$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects13$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects13ToJSON(getProjectEnvContentHintProjects13) {
    return JSON.stringify(GetProjectEnvContentHintProjects13$outboundSchema.parse(getProjectEnvContentHintProjects13));
}
export function getProjectEnvContentHintProjects13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects13' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects12$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects12$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects12ToJSON(getProjectEnvContentHintProjects12) {
    return JSON.stringify(GetProjectEnvContentHintProjects12$outboundSchema.parse(getProjectEnvContentHintProjects12));
}
export function getProjectEnvContentHintProjects12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects12' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects11$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects11$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects11ToJSON(getProjectEnvContentHintProjects11) {
    return JSON.stringify(GetProjectEnvContentHintProjects11$outboundSchema.parse(getProjectEnvContentHintProjects11));
}
export function getProjectEnvContentHintProjects11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects11' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects10$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects10$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects10ToJSON(getProjectEnvContentHintProjects10) {
    return JSON.stringify(GetProjectEnvContentHintProjects10$outboundSchema.parse(getProjectEnvContentHintProjects10));
}
export function getProjectEnvContentHintProjects10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects10' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects9$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects9$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects9ToJSON(getProjectEnvContentHintProjects9) {
    return JSON.stringify(GetProjectEnvContentHintProjects9$outboundSchema.parse(getProjectEnvContentHintProjects9));
}
export function getProjectEnvContentHintProjects9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects9' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects8$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects8$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects8ToJSON(getProjectEnvContentHintProjects8) {
    return JSON.stringify(GetProjectEnvContentHintProjects8$outboundSchema.parse(getProjectEnvContentHintProjects8));
}
export function getProjectEnvContentHintProjects8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects8' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects7$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects7$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects7ToJSON(getProjectEnvContentHintProjects7) {
    return JSON.stringify(GetProjectEnvContentHintProjects7$outboundSchema.parse(getProjectEnvContentHintProjects7));
}
export function getProjectEnvContentHintProjects7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects7' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects6$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects6$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects6ToJSON(getProjectEnvContentHintProjects6) {
    return JSON.stringify(GetProjectEnvContentHintProjects6$outboundSchema.parse(getProjectEnvContentHintProjects6));
}
export function getProjectEnvContentHintProjects6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects6' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects5$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects5$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects5ToJSON(getProjectEnvContentHintProjects5) {
    return JSON.stringify(GetProjectEnvContentHintProjects5$outboundSchema.parse(getProjectEnvContentHintProjects5));
}
export function getProjectEnvContentHintProjects5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects5' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects4$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects4$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects4ToJSON(getProjectEnvContentHintProjects4) {
    return JSON.stringify(GetProjectEnvContentHintProjects4$outboundSchema.parse(getProjectEnvContentHintProjects4));
}
export function getProjectEnvContentHintProjects4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects4' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects3$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects3$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects3ToJSON(getProjectEnvContentHintProjects3) {
    return JSON.stringify(GetProjectEnvContentHintProjects3$outboundSchema.parse(getProjectEnvContentHintProjects3));
}
export function getProjectEnvContentHintProjects3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects3' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects2$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects2$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects2ToJSON(getProjectEnvContentHintProjects2) {
    return JSON.stringify(GetProjectEnvContentHintProjects2$outboundSchema.parse(getProjectEnvContentHintProjects2));
}
export function getProjectEnvContentHintProjects2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects2' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjects1$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHintProjects1$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHintProjects1ToJSON(getProjectEnvContentHintProjects1) {
    return JSON.stringify(GetProjectEnvContentHintProjects1$outboundSchema.parse(getProjectEnvContentHintProjects1));
}
export function getProjectEnvContentHintProjects1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHintProjects1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHintProjects1' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsContentHint$inboundSchema = z.union([
    z.lazy(() => GetProjectEnvContentHintProjects14$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects1$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects2$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects3$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects4$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects5$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects6$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects7$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects8$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects9$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects10$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects11$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects12$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects13$inboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects15$inboundSchema),
]);
/** @internal */
export const GetProjectEnvResponseBodyProjectsContentHint$outboundSchema = z.union([
    z.lazy(() => GetProjectEnvContentHintProjects14$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects1$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects2$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects3$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects4$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects5$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects6$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects7$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects8$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects9$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects10$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects11$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects12$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects13$outboundSchema),
    z.lazy(() => GetProjectEnvContentHintProjects15$outboundSchema),
]);
export function getProjectEnvResponseBodyProjectsContentHintToJSON(getProjectEnvResponseBodyProjectsContentHint) {
    return JSON.stringify(GetProjectEnvResponseBodyProjectsContentHint$outboundSchema.parse(getProjectEnvResponseBodyProjectsContentHint));
}
export function getProjectEnvResponseBodyProjectsContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyProjectsContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyProjectsContentHint' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponse200Type$inboundSchema = z
    .nativeEnum(GetProjectEnvResponseBodyProjectsResponse200Type);
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponse200Type$outboundSchema = GetProjectEnvResponseBodyProjectsResponse200Type$inboundSchema;
/** @internal */
export const GetProjectEnvResponseBodyProjectsInternalContentHint$inboundSchema = z.object({
    type: GetProjectEnvResponseBodyProjectsResponse200Type$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const GetProjectEnvResponseBodyProjectsInternalContentHint$outboundSchema = z.object({
    type: GetProjectEnvResponseBodyProjectsResponse200Type$outboundSchema,
    encryptedValue: z.string(),
});
export function getProjectEnvResponseBodyProjectsInternalContentHintToJSON(getProjectEnvResponseBodyProjectsInternalContentHint) {
    return JSON.stringify(GetProjectEnvResponseBodyProjectsInternalContentHint$outboundSchema.parse(getProjectEnvResponseBodyProjectsInternalContentHint));
}
export function getProjectEnvResponseBodyProjectsInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyProjectsInternalContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyProjectsInternalContentHint' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBody2$inboundSchema = z.object({
    target: z.union([
        z.array(GetProjectEnvTargetProjects1$inboundSchema),
        GetProjectEnvTargetProjects2$inboundSchema,
    ]).optional(),
    type: GetProjectEnvResponseBodyProjectsType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => GetProjectEnvContentHintProjects14$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects1$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects2$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects3$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects4$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects5$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects6$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects7$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects8$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects9$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects10$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects11$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects12$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects13$inboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => GetProjectEnvResponseBodyProjectsInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const GetProjectEnvResponseBody2$outboundSchema = z.object({
    target: z.union([
        z.array(GetProjectEnvTargetProjects1$outboundSchema),
        GetProjectEnvTargetProjects2$outboundSchema,
    ]).optional(),
    type: GetProjectEnvResponseBodyProjectsType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => GetProjectEnvContentHintProjects14$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects1$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects2$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects3$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects4$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects5$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects6$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects7$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects8$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects9$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects10$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects11$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects12$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects13$outboundSchema),
        z.lazy(() => GetProjectEnvContentHintProjects15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => GetProjectEnvResponseBodyProjectsInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function getProjectEnvResponseBody2ToJSON(getProjectEnvResponseBody2) {
    return JSON.stringify(GetProjectEnvResponseBody2$outboundSchema.parse(getProjectEnvResponseBody2));
}
export function getProjectEnvResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBody2' from JSON`);
}
/** @internal */
export const GetProjectEnvTarget2$inboundSchema = z.nativeEnum(GetProjectEnvTarget2);
/** @internal */
export const GetProjectEnvTarget2$outboundSchema = GetProjectEnvTarget2$inboundSchema;
/** @internal */
export const GetProjectEnvTarget1$inboundSchema = z.nativeEnum(GetProjectEnvTarget1);
/** @internal */
export const GetProjectEnvTarget1$outboundSchema = GetProjectEnvTarget1$inboundSchema;
/** @internal */
export const GetProjectEnvResponseBodyTarget$inboundSchema = z.union([
    z.array(GetProjectEnvTarget1$inboundSchema),
    GetProjectEnvTarget2$inboundSchema,
]);
/** @internal */
export const GetProjectEnvResponseBodyTarget$outboundSchema = z.union([
    z.array(GetProjectEnvTarget1$outboundSchema),
    GetProjectEnvTarget2$outboundSchema,
]);
export function getProjectEnvResponseBodyTargetToJSON(getProjectEnvResponseBodyTarget) {
    return JSON.stringify(GetProjectEnvResponseBodyTarget$outboundSchema.parse(getProjectEnvResponseBodyTarget));
}
export function getProjectEnvResponseBodyTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyTarget' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyType$inboundSchema = z.nativeEnum(GetProjectEnvResponseBodyType);
/** @internal */
export const GetProjectEnvResponseBodyType$outboundSchema = GetProjectEnvResponseBodyType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint15$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint15$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$outboundSchema,
    projectId: z.string(),
});
export function getProjectEnvContentHint15ToJSON(getProjectEnvContentHint15) {
    return JSON.stringify(GetProjectEnvContentHint15$outboundSchema.parse(getProjectEnvContentHint15));
}
export function getProjectEnvContentHint15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint15' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint14$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint14$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function getProjectEnvContentHint14ToJSON(getProjectEnvContentHint14) {
    return JSON.stringify(GetProjectEnvContentHint14$outboundSchema.parse(getProjectEnvContentHint14));
}
export function getProjectEnvContentHint14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint14' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint13$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint13$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint13ToJSON(getProjectEnvContentHint13) {
    return JSON.stringify(GetProjectEnvContentHint13$outboundSchema.parse(getProjectEnvContentHint13));
}
export function getProjectEnvContentHint13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint13' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint12$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint12$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint12ToJSON(getProjectEnvContentHint12) {
    return JSON.stringify(GetProjectEnvContentHint12$outboundSchema.parse(getProjectEnvContentHint12));
}
export function getProjectEnvContentHint12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint12' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint11$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint11$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint11ToJSON(getProjectEnvContentHint11) {
    return JSON.stringify(GetProjectEnvContentHint11$outboundSchema.parse(getProjectEnvContentHint11));
}
export function getProjectEnvContentHint11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint11' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint10$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint10$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint10ToJSON(getProjectEnvContentHint10) {
    return JSON.stringify(GetProjectEnvContentHint10$outboundSchema.parse(getProjectEnvContentHint10));
}
export function getProjectEnvContentHint10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint10' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint9$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint9$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint9ToJSON(getProjectEnvContentHint9) {
    return JSON.stringify(GetProjectEnvContentHint9$outboundSchema.parse(getProjectEnvContentHint9));
}
export function getProjectEnvContentHint9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint9' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint8$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint8$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint8ToJSON(getProjectEnvContentHint8) {
    return JSON.stringify(GetProjectEnvContentHint8$outboundSchema.parse(getProjectEnvContentHint8));
}
export function getProjectEnvContentHint8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint8' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint7$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint7$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint7ToJSON(getProjectEnvContentHint7) {
    return JSON.stringify(GetProjectEnvContentHint7$outboundSchema.parse(getProjectEnvContentHint7));
}
export function getProjectEnvContentHint7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint7' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint6$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint6$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint6ToJSON(getProjectEnvContentHint6) {
    return JSON.stringify(GetProjectEnvContentHint6$outboundSchema.parse(getProjectEnvContentHint6));
}
export function getProjectEnvContentHint6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint6' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsResponse200ApplicationJSONType);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200ApplicationJSONType$outboundSchema = GetProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint5$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint5$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200ApplicationJSONType$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint5ToJSON(getProjectEnvContentHint5) {
    return JSON.stringify(GetProjectEnvContentHint5$outboundSchema.parse(getProjectEnvContentHint5));
}
export function getProjectEnvContentHint5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint5' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200Type$inboundSchema = z
    .nativeEnum(GetProjectEnvContentHintProjectsResponse200Type);
/** @internal */
export const GetProjectEnvContentHintProjectsResponse200Type$outboundSchema = GetProjectEnvContentHintProjectsResponse200Type$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint4$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint4$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponse200Type$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint4ToJSON(getProjectEnvContentHint4) {
    return JSON.stringify(GetProjectEnvContentHint4$outboundSchema.parse(getProjectEnvContentHint4));
}
export function getProjectEnvContentHint4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint4' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsResponseType$inboundSchema = z
    .nativeEnum(GetProjectEnvContentHintProjectsResponseType);
/** @internal */
export const GetProjectEnvContentHintProjectsResponseType$outboundSchema = GetProjectEnvContentHintProjectsResponseType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint3$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponseType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint3$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsResponseType$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint3ToJSON(getProjectEnvContentHint3) {
    return JSON.stringify(GetProjectEnvContentHint3$outboundSchema.parse(getProjectEnvContentHint3));
}
export function getProjectEnvContentHint3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint3' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintProjectsType$inboundSchema = z.nativeEnum(GetProjectEnvContentHintProjectsType);
/** @internal */
export const GetProjectEnvContentHintProjectsType$outboundSchema = GetProjectEnvContentHintProjectsType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint2$inboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint2$outboundSchema = z.object({
    type: GetProjectEnvContentHintProjectsType$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint2ToJSON(getProjectEnvContentHint2) {
    return JSON.stringify(GetProjectEnvContentHint2$outboundSchema.parse(getProjectEnvContentHint2));
}
export function getProjectEnvContentHint2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint2' from JSON`);
}
/** @internal */
export const GetProjectEnvContentHintType$inboundSchema = z.nativeEnum(GetProjectEnvContentHintType);
/** @internal */
export const GetProjectEnvContentHintType$outboundSchema = GetProjectEnvContentHintType$inboundSchema;
/** @internal */
export const GetProjectEnvContentHint1$inboundSchema = z.object({
    type: GetProjectEnvContentHintType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const GetProjectEnvContentHint1$outboundSchema = z.object({
    type: GetProjectEnvContentHintType$outboundSchema,
    storeId: z.string(),
});
export function getProjectEnvContentHint1ToJSON(getProjectEnvContentHint1) {
    return JSON.stringify(GetProjectEnvContentHint1$outboundSchema.parse(getProjectEnvContentHint1));
}
export function getProjectEnvContentHint1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvContentHint1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvContentHint1' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyContentHint$inboundSchema = z.union([
    z.lazy(() => GetProjectEnvContentHint14$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint1$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint2$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint3$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint4$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint5$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint6$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint7$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint8$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint9$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint10$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint11$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint12$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint13$inboundSchema),
    z.lazy(() => GetProjectEnvContentHint15$inboundSchema),
]);
/** @internal */
export const GetProjectEnvResponseBodyContentHint$outboundSchema = z.union([
    z.lazy(() => GetProjectEnvContentHint14$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint1$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint2$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint3$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint4$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint5$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint6$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint7$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint8$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint9$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint10$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint11$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint12$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint13$outboundSchema),
    z.lazy(() => GetProjectEnvContentHint15$outboundSchema),
]);
export function getProjectEnvResponseBodyContentHintToJSON(getProjectEnvResponseBodyContentHint) {
    return JSON.stringify(GetProjectEnvResponseBodyContentHint$outboundSchema.parse(getProjectEnvResponseBodyContentHint));
}
export function getProjectEnvResponseBodyContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyContentHint' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType);
/** @internal */
export const GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$outboundSchema = GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const GetProjectEnvResponseBodyInternalContentHint$inboundSchema = z.object({
    type: GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const GetProjectEnvResponseBodyInternalContentHint$outboundSchema = z.object({
    type: GetProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$outboundSchema,
    encryptedValue: z.string(),
});
export function getProjectEnvResponseBodyInternalContentHintToJSON(getProjectEnvResponseBodyInternalContentHint) {
    return JSON.stringify(GetProjectEnvResponseBodyInternalContentHint$outboundSchema.parse(getProjectEnvResponseBodyInternalContentHint));
}
export function getProjectEnvResponseBodyInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBodyInternalContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBodyInternalContentHint' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBody1$inboundSchema = z.object({
    decrypted: z.boolean(),
    target: z.union([
        z.array(GetProjectEnvTarget1$inboundSchema),
        GetProjectEnvTarget2$inboundSchema,
    ]).optional(),
    type: GetProjectEnvResponseBodyType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => GetProjectEnvContentHint14$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint1$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint2$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint3$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint4$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint5$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint6$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint7$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint8$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint9$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint10$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint11$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint12$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint13$inboundSchema),
        z.lazy(() => GetProjectEnvContentHint15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => GetProjectEnvResponseBodyInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const GetProjectEnvResponseBody1$outboundSchema = z.object({
    decrypted: z.boolean(),
    target: z.union([
        z.array(GetProjectEnvTarget1$outboundSchema),
        GetProjectEnvTarget2$outboundSchema,
    ]).optional(),
    type: GetProjectEnvResponseBodyType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => GetProjectEnvContentHint14$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint1$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint2$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint3$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint4$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint5$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint6$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint7$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint8$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint9$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint10$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint11$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint12$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint13$outboundSchema),
        z.lazy(() => GetProjectEnvContentHint15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => GetProjectEnvResponseBodyInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function getProjectEnvResponseBody1ToJSON(getProjectEnvResponseBody1) {
    return JSON.stringify(GetProjectEnvResponseBody1$outboundSchema.parse(getProjectEnvResponseBody1));
}
export function getProjectEnvResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBody1' from JSON`);
}
/** @internal */
export const GetProjectEnvResponseBody$inboundSchema = z.union([
    z.lazy(() => GetProjectEnvResponseBody1$inboundSchema),
    z.lazy(() => GetProjectEnvResponseBody2$inboundSchema),
    z.lazy(() => GetProjectEnvResponseBody3$inboundSchema),
]);
/** @internal */
export const GetProjectEnvResponseBody$outboundSchema = z.union([
    z.lazy(() => GetProjectEnvResponseBody1$outboundSchema),
    z.lazy(() => GetProjectEnvResponseBody2$outboundSchema),
    z.lazy(() => GetProjectEnvResponseBody3$outboundSchema),
]);
export function getProjectEnvResponseBodyToJSON(getProjectEnvResponseBody) {
    return JSON.stringify(GetProjectEnvResponseBody$outboundSchema.parse(getProjectEnvResponseBody));
}
export function getProjectEnvResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectEnvResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectEnvResponseBody' from JSON`);
}
