// @vercel/sdk/esm/models/getprojectdomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetProjectDomainRequest$inboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetProjectDomainRequest$outboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getProjectDomainRequestToJSON(getProjectDomainRequest) {
    return JSON.stringify(GetProjectDomainRequest$outboundSchema.parse(getProjectDomainRequest));
}
export function getProjectDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainRequest' from JSON`);
}
/** @internal */
export const Verification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const Verification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
export function verificationToJSON(verification) {
    return JSON.stringify(Verification$outboundSchema.parse(verification));
}
export function verificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Verification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Verification' from JSON`);
}
/** @internal */
export const GetProjectDomainResponseBody$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => Verification$inboundSchema)).optional(),
});
/** @internal */
export const GetProjectDomainResponseBody$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => Verification$outboundSchema)).optional(),
});
export function getProjectDomainResponseBodyToJSON(getProjectDomainResponseBody) {
    return JSON.stringify(GetProjectDomainResponseBody$outboundSchema.parse(getProjectDomainResponseBody));
}
export function getProjectDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectDomainResponseBody' from JSON`);
}
