// @vercel/sdk/esm/models/getrollingreleaseop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * Filter by rolling release state
 */
export const State = {
    Active: "ACTIVE",
    Complete: "COMPLETE",
    Aborted: "ABORTED",
};
/**
 * The current state of the rolling release
 */
export const GetRollingReleaseState = {
    Active: "ACTIVE",
    Complete: "COMPLETE",
    Aborted: "ABORTED",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const GetRollingReleaseTarget = {
    Staging: "staging",
    Production: "production",
};
/**
 * Where was the deployment created from
 */
export const GetRollingReleaseSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const GetRollingReleaseReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * If defined, either `staging` if a staging alias in the format `<project>.<team>.now.sh` was assigned upon creation, or `production` if the aliases from `alias` were assigned. `null` value indicates the "preview" deployment.
 */
export const GetRollingReleaseRollingReleaseTarget = {
    Staging: "staging",
    Production: "production",
};
/**
 * Where was the deployment created from
 */
export const GetRollingReleaseRollingReleaseSource = {
    ApiTriggerGitDeploy: "api-trigger-git-deploy",
    Cli: "cli",
    CloneRepo: "clone/repo",
    Git: "git",
    Import: "import",
    ImportRepo: "import/repo",
    Redeploy: "redeploy",
    V0Web: "v0-web",
};
/**
 * The state of the deployment depending on the process of deploying, or if it is ready or in an error state
 */
export const GetRollingReleaseRollingReleaseReadyState = {
    Building: "BUILDING",
    Error: "ERROR",
    Initializing: "INITIALIZING",
    Queued: "QUEUED",
    Ready: "READY",
    Canceled: "CANCELED",
};
/**
 * The advancement type of the rolling release
 */
export const AdvancementType = {
    Automatic: "automatic",
    ManualApproval: "manual-approval",
};
/** @internal */
export const State$inboundSchema = z.nativeEnum(State);
/** @internal */
export const State$outboundSchema = State$inboundSchema;
/** @internal */
export const GetRollingReleaseRequest$inboundSchema = z.object({
    idOrName: z.string(),
    state: State$inboundSchema.optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetRollingReleaseRequest$outboundSchema = z.object({
    idOrName: z.string(),
    state: State$outboundSchema.optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getRollingReleaseRequestToJSON(getRollingReleaseRequest) {
    return JSON.stringify(GetRollingReleaseRequest$outboundSchema.parse(getRollingReleaseRequest));
}
export function getRollingReleaseRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRollingReleaseRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRollingReleaseRequest' from JSON`);
}
/** @internal */
export const GetRollingReleaseState$inboundSchema = z.nativeEnum(GetRollingReleaseState);
/** @internal */
export const GetRollingReleaseState$outboundSchema = GetRollingReleaseState$inboundSchema;
/** @internal */
export const GetRollingReleaseTarget$inboundSchema = z.nativeEnum(GetRollingReleaseTarget);
/** @internal */
export const GetRollingReleaseTarget$outboundSchema = GetRollingReleaseTarget$inboundSchema;
/** @internal */
export const GetRollingReleaseSource$inboundSchema = z.nativeEnum(GetRollingReleaseSource);
/** @internal */
export const GetRollingReleaseSource$outboundSchema = GetRollingReleaseSource$inboundSchema;
/** @internal */
export const GetRollingReleaseReadyState$inboundSchema = z.nativeEnum(GetRollingReleaseReadyState);
/** @internal */
export const GetRollingReleaseReadyState$outboundSchema = GetRollingReleaseReadyState$inboundSchema;
/** @internal */
export const CurrentDeployment$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(GetRollingReleaseTarget$inboundSchema).optional(),
    source: GetRollingReleaseSource$inboundSchema.optional(),
    createdAt: z.number(),
    readyState: GetRollingReleaseReadyState$inboundSchema,
    readyStateAt: z.number().optional(),
});
/** @internal */
export const CurrentDeployment$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(GetRollingReleaseTarget$outboundSchema).optional(),
    source: GetRollingReleaseSource$outboundSchema.optional(),
    createdAt: z.number(),
    readyState: GetRollingReleaseReadyState$outboundSchema,
    readyStateAt: z.number().optional(),
});
export function currentDeploymentToJSON(currentDeployment) {
    return JSON.stringify(CurrentDeployment$outboundSchema.parse(currentDeployment));
}
export function currentDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CurrentDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CurrentDeployment' from JSON`);
}
/** @internal */
export const GetRollingReleaseRollingReleaseTarget$inboundSchema = z.nativeEnum(GetRollingReleaseRollingReleaseTarget);
/** @internal */
export const GetRollingReleaseRollingReleaseTarget$outboundSchema = GetRollingReleaseRollingReleaseTarget$inboundSchema;
/** @internal */
export const GetRollingReleaseRollingReleaseSource$inboundSchema = z.nativeEnum(GetRollingReleaseRollingReleaseSource);
/** @internal */
export const GetRollingReleaseRollingReleaseSource$outboundSchema = GetRollingReleaseRollingReleaseSource$inboundSchema;
/** @internal */
export const GetRollingReleaseRollingReleaseReadyState$inboundSchema = z
    .nativeEnum(GetRollingReleaseRollingReleaseReadyState);
/** @internal */
export const GetRollingReleaseRollingReleaseReadyState$outboundSchema = GetRollingReleaseRollingReleaseReadyState$inboundSchema;
/** @internal */
export const CanaryDeployment$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(GetRollingReleaseRollingReleaseTarget$inboundSchema)
        .optional(),
    source: GetRollingReleaseRollingReleaseSource$inboundSchema.optional(),
    createdAt: z.number(),
    readyState: GetRollingReleaseRollingReleaseReadyState$inboundSchema,
    readyStateAt: z.number().optional(),
});
/** @internal */
export const CanaryDeployment$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    url: z.string(),
    target: z.nullable(GetRollingReleaseRollingReleaseTarget$outboundSchema)
        .optional(),
    source: GetRollingReleaseRollingReleaseSource$outboundSchema.optional(),
    createdAt: z.number(),
    readyState: GetRollingReleaseRollingReleaseReadyState$outboundSchema,
    readyStateAt: z.number().optional(),
});
export function canaryDeploymentToJSON(canaryDeployment) {
    return JSON.stringify(CanaryDeployment$outboundSchema.parse(canaryDeployment));
}
export function canaryDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CanaryDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CanaryDeployment' from JSON`);
}
/** @internal */
export const AdvancementType$inboundSchema = z.nativeEnum(AdvancementType);
/** @internal */
export const AdvancementType$outboundSchema = AdvancementType$inboundSchema;
/** @internal */
export const GetRollingReleaseStages$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const GetRollingReleaseStages$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function getRollingReleaseStagesToJSON(getRollingReleaseStages) {
    return JSON.stringify(GetRollingReleaseStages$outboundSchema.parse(getRollingReleaseStages));
}
export function getRollingReleaseStagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRollingReleaseStages$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRollingReleaseStages' from JSON`);
}
/** @internal */
export const ActiveStage$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const ActiveStage$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function activeStageToJSON(activeStage) {
    return JSON.stringify(ActiveStage$outboundSchema.parse(activeStage));
}
export function activeStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ActiveStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ActiveStage' from JSON`);
}
/** @internal */
export const NextStage$inboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const NextStage$outboundSchema = z.object({
    index: z.number(),
    isFinalStage: z.boolean(),
    targetPercentage: z.number(),
    requireApproval: z.boolean(),
    duration: z.nullable(z.number()),
    linearShift: z.boolean().optional(),
});
export function nextStageToJSON(nextStage) {
    return JSON.stringify(NextStage$outboundSchema.parse(nextStage));
}
export function nextStageFromJSON(jsonString) {
    return safeParse(jsonString, (x) => NextStage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'NextStage' from JSON`);
}
/** @internal */
export const GetRollingReleaseRollingRelease$inboundSchema = z.object({
    state: GetRollingReleaseState$inboundSchema,
    currentDeployment: z.nullable(z.lazy(() => CurrentDeployment$inboundSchema)),
    canaryDeployment: z.nullable(z.lazy(() => CanaryDeployment$inboundSchema)),
    queuedDeploymentId: z.nullable(z.string()),
    advancementType: AdvancementType$inboundSchema,
    stages: z.array(z.lazy(() => GetRollingReleaseStages$inboundSchema)),
    activeStage: z.nullable(z.lazy(() => ActiveStage$inboundSchema)),
    nextStage: z.nullable(z.lazy(() => NextStage$inboundSchema)),
    startedAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const GetRollingReleaseRollingRelease$outboundSchema = z.object({
    state: GetRollingReleaseState$outboundSchema,
    currentDeployment: z.nullable(z.lazy(() => CurrentDeployment$outboundSchema)),
    canaryDeployment: z.nullable(z.lazy(() => CanaryDeployment$outboundSchema)),
    queuedDeploymentId: z.nullable(z.string()),
    advancementType: AdvancementType$outboundSchema,
    stages: z.array(z.lazy(() => GetRollingReleaseStages$outboundSchema)),
    activeStage: z.nullable(z.lazy(() => ActiveStage$outboundSchema)),
    nextStage: z.nullable(z.lazy(() => NextStage$outboundSchema)),
    startedAt: z.number(),
    updatedAt: z.number(),
});
export function getRollingReleaseRollingReleaseToJSON(getRollingReleaseRollingRelease) {
    return JSON.stringify(GetRollingReleaseRollingRelease$outboundSchema.parse(getRollingReleaseRollingRelease));
}
export function getRollingReleaseRollingReleaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRollingReleaseRollingRelease$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRollingReleaseRollingRelease' from JSON`);
}
/** @internal */
export const GetRollingReleaseResponseBody$inboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => GetRollingReleaseRollingRelease$inboundSchema)),
});
/** @internal */
export const GetRollingReleaseResponseBody$outboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => GetRollingReleaseRollingRelease$outboundSchema)),
});
export function getRollingReleaseResponseBodyToJSON(getRollingReleaseResponseBody) {
    return JSON.stringify(GetRollingReleaseResponseBody$outboundSchema.parse(getRollingReleaseResponseBody));
}
export function getRollingReleaseResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRollingReleaseResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRollingReleaseResponseBody' from JSON`);
}
