// @vercel/sdk/esm/models/gettldpriceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { HttpApiDecodeError$inboundSchema, HttpApiDecodeError$outboundSchema, } from "./httpapidecodeerror.js";
import { TldNotSupported$inboundSchema, TldNotSupported$outboundSchema, } from "./tldnotsupported.js";
/** @internal */
export const GetTldPriceRequest$inboundSchema = z.object({
    tld: z.string(),
    years: z.string().optional(),
    teamId: z.string().optional(),
});
/** @internal */
export const GetTldPriceRequest$outboundSchema = z.object({
    tld: z.string(),
    years: z.string().optional(),
    teamId: z.string().optional(),
});
export function getTldPriceRequestToJSON(getTldPriceRequest) {
    return JSON.stringify(GetTldPriceRequest$outboundSchema.parse(getTldPriceRequest));
}
export function getTldPriceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTldPriceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTldPriceRequest' from JSON`);
}
/** @internal */
export const GetTldPriceDomainsRegistrarResponseBody$inboundSchema = z.union([TldNotSupported$inboundSchema, HttpApiDecodeError$inboundSchema]);
/** @internal */
export const GetTldPriceDomainsRegistrarResponseBody$outboundSchema = z.union([
    TldNotSupported$outboundSchema,
    HttpApiDecodeError$outboundSchema,
]);
export function getTldPriceDomainsRegistrarResponseBodyToJSON(getTldPriceDomainsRegistrarResponseBody) {
    return JSON.stringify(GetTldPriceDomainsRegistrarResponseBody$outboundSchema.parse(getTldPriceDomainsRegistrarResponseBody));
}
export function getTldPriceDomainsRegistrarResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTldPriceDomainsRegistrarResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTldPriceDomainsRegistrarResponseBody' from JSON`);
}
/** @internal */
export const GetTldPriceResponseBody$inboundSchema = z.object({
    years: z.number(),
    purchasePrice: z.nullable(z.number()),
    renewalPrice: z.nullable(z.number()),
    transferPrice: z.nullable(z.number()),
});
/** @internal */
export const GetTldPriceResponseBody$outboundSchema = z.object({
    years: z.number(),
    purchasePrice: z.nullable(z.number()),
    renewalPrice: z.nullable(z.number()),
    transferPrice: z.nullable(z.number()),
});
export function getTldPriceResponseBodyToJSON(getTldPriceResponseBody) {
    return JSON.stringify(GetTldPriceResponseBody$outboundSchema.parse(getTldPriceResponseBody));
}
export function getTldPriceResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTldPriceResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTldPriceResponseBody' from JSON`);
}
