// @vercel/sdk/esm/models/getteammembersop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * Only return members with the specified team role.
 */
export const QueryParamRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * Role of this user in the team.
 */
export const GetTeamMembersRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const GetTeamMembersOrigin = {
    Teams: "teams",
    Link: "link",
    Mail: "mail",
    Import: "import",
    Github: "github",
    Gitlab: "gitlab",
    Bitbucket: "bitbucket",
    Saml: "saml",
    Dsync: "dsync",
    Feedback: "feedback",
    OrganizationTeams: "organization-teams",
};
export const GetTeamMembersTeamsResponseRole = {
    Admin: "ADMIN",
    ProjectDeveloper: "PROJECT_DEVELOPER",
    ProjectViewer: "PROJECT_VIEWER",
};
export const GetTeamMembersTeamsRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const GetTeamMembersTeamRoles = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const GetTeamMembersTeamPermissions = {
    IntegrationManager: "IntegrationManager",
    CreateProject: "CreateProject",
    FullProductionDeployment: "FullProductionDeployment",
    UsageViewer: "UsageViewer",
    EnvVariableManager: "EnvVariableManager",
    EnvironmentManager: "EnvironmentManager",
    V0Builder: "V0Builder",
    V0Chatter: "V0Chatter",
    V0Viewer: "V0Viewer",
};
export const GetTeamMembersTeamsProjects = {
    Admin: "ADMIN",
    ProjectDeveloper: "PROJECT_DEVELOPER",
    ProjectViewer: "PROJECT_VIEWER",
};
/** @internal */
export const QueryParamRole$inboundSchema = z.nativeEnum(QueryParamRole);
/** @internal */
export const QueryParamRole$outboundSchema = QueryParamRole$inboundSchema;
/** @internal */
export const GetTeamMembersRequest$inboundSchema = z.object({
    limit: z.number().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    search: z.string().optional(),
    role: QueryParamRole$inboundSchema.optional(),
    excludeProject: z.string().optional(),
    eligibleMembersForProjectId: z.string().optional(),
});
/** @internal */
export const GetTeamMembersRequest$outboundSchema = z.object({
    limit: z.number().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    search: z.string().optional(),
    role: QueryParamRole$outboundSchema.optional(),
    excludeProject: z.string().optional(),
    eligibleMembersForProjectId: z.string().optional(),
});
export function getTeamMembersRequestToJSON(getTeamMembersRequest) {
    return JSON.stringify(GetTeamMembersRequest$outboundSchema.parse(getTeamMembersRequest));
}
export function getTeamMembersRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersRequest' from JSON`);
}
/** @internal */
export const GetTeamMembersGithub$inboundSchema = z.object({
    login: z.string().optional(),
});
/** @internal */
export const GetTeamMembersGithub$outboundSchema = z.object({
    login: z.string().optional(),
});
export function getTeamMembersGithubToJSON(getTeamMembersGithub) {
    return JSON.stringify(GetTeamMembersGithub$outboundSchema.parse(getTeamMembersGithub));
}
export function getTeamMembersGithubFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersGithub$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersGithub' from JSON`);
}
/** @internal */
export const GetTeamMembersGitlab$inboundSchema = z.object({
    login: z.string().optional(),
});
/** @internal */
export const GetTeamMembersGitlab$outboundSchema = z.object({
    login: z.string().optional(),
});
export function getTeamMembersGitlabToJSON(getTeamMembersGitlab) {
    return JSON.stringify(GetTeamMembersGitlab$outboundSchema.parse(getTeamMembersGitlab));
}
export function getTeamMembersGitlabFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersGitlab$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersGitlab' from JSON`);
}
/** @internal */
export const GetTeamMembersBitbucket$inboundSchema = z.object({
    login: z.string().optional(),
});
/** @internal */
export const GetTeamMembersBitbucket$outboundSchema = z.object({
    login: z.string().optional(),
});
export function getTeamMembersBitbucketToJSON(getTeamMembersBitbucket) {
    return JSON.stringify(GetTeamMembersBitbucket$outboundSchema.parse(getTeamMembersBitbucket));
}
export function getTeamMembersBitbucketFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersBitbucket$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersBitbucket' from JSON`);
}
/** @internal */
export const GetTeamMembersRole$inboundSchema = z.nativeEnum(GetTeamMembersRole);
/** @internal */
export const GetTeamMembersRole$outboundSchema = GetTeamMembersRole$inboundSchema;
/** @internal */
export const GetTeamMembersOrigin$inboundSchema = z.nativeEnum(GetTeamMembersOrigin);
/** @internal */
export const GetTeamMembersOrigin$outboundSchema = GetTeamMembersOrigin$inboundSchema;
/** @internal */
export const GetTeamMembersGitUserId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const GetTeamMembersGitUserId$outboundSchema = z.union([z.string(), z.number()]);
export function getTeamMembersGitUserIdToJSON(getTeamMembersGitUserId) {
    return JSON.stringify(GetTeamMembersGitUserId$outboundSchema.parse(getTeamMembersGitUserId));
}
export function getTeamMembersGitUserIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersGitUserId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersGitUserId' from JSON`);
}
/** @internal */
export const GetTeamMembersJoinedFrom$inboundSchema = z.object({
    origin: GetTeamMembersOrigin$inboundSchema,
    commitId: z.string().optional(),
    repoId: z.string().optional(),
    repoPath: z.string().optional(),
    gitUserId: z.union([z.string(), z.number()]).optional(),
    gitUserLogin: z.string().optional(),
    ssoUserId: z.string().optional(),
    ssoConnectedAt: z.number().optional(),
    idpUserId: z.string().optional(),
    dsyncUserId: z.string().optional(),
    dsyncConnectedAt: z.number().optional(),
});
/** @internal */
export const GetTeamMembersJoinedFrom$outboundSchema = z.object({
    origin: GetTeamMembersOrigin$outboundSchema,
    commitId: z.string().optional(),
    repoId: z.string().optional(),
    repoPath: z.string().optional(),
    gitUserId: z.union([z.string(), z.number()]).optional(),
    gitUserLogin: z.string().optional(),
    ssoUserId: z.string().optional(),
    ssoConnectedAt: z.number().optional(),
    idpUserId: z.string().optional(),
    dsyncUserId: z.string().optional(),
    dsyncConnectedAt: z.number().optional(),
});
export function getTeamMembersJoinedFromToJSON(getTeamMembersJoinedFrom) {
    return JSON.stringify(GetTeamMembersJoinedFrom$outboundSchema.parse(getTeamMembersJoinedFrom));
}
export function getTeamMembersJoinedFromFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersJoinedFrom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersJoinedFrom' from JSON`);
}
/** @internal */
export const GetTeamMembersTeamsResponseRole$inboundSchema = z.nativeEnum(GetTeamMembersTeamsResponseRole);
/** @internal */
export const GetTeamMembersTeamsResponseRole$outboundSchema = GetTeamMembersTeamsResponseRole$inboundSchema;
/** @internal */
export const GetTeamMembersProjects$inboundSchema = z.object({
    name: z.string(),
    id: z.string(),
    role: GetTeamMembersTeamsResponseRole$inboundSchema.optional(),
});
/** @internal */
export const GetTeamMembersProjects$outboundSchema = z.object({
    name: z.string(),
    id: z.string(),
    role: GetTeamMembersTeamsResponseRole$outboundSchema.optional(),
});
export function getTeamMembersProjectsToJSON(getTeamMembersProjects) {
    return JSON.stringify(GetTeamMembersProjects$outboundSchema.parse(getTeamMembersProjects));
}
export function getTeamMembersProjectsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersProjects$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersProjects' from JSON`);
}
/** @internal */
export const GetTeamMembersMembers$inboundSchema = z.object({
    avatar: z.string().optional(),
    confirmed: z.boolean(),
    email: z.string(),
    github: z.lazy(() => GetTeamMembersGithub$inboundSchema).optional(),
    gitlab: z.lazy(() => GetTeamMembersGitlab$inboundSchema).optional(),
    bitbucket: z.lazy(() => GetTeamMembersBitbucket$inboundSchema).optional(),
    role: GetTeamMembersRole$inboundSchema,
    uid: z.string(),
    username: z.string(),
    name: z.string().optional(),
    createdAt: z.number(),
    accessRequestedAt: z.number().optional(),
    joinedFrom: z.lazy(() => GetTeamMembersJoinedFrom$inboundSchema).optional(),
    projects: z.array(z.lazy(() => GetTeamMembersProjects$inboundSchema))
        .optional(),
});
/** @internal */
export const GetTeamMembersMembers$outboundSchema = z.object({
    avatar: z.string().optional(),
    confirmed: z.boolean(),
    email: z.string(),
    github: z.lazy(() => GetTeamMembersGithub$outboundSchema).optional(),
    gitlab: z.lazy(() => GetTeamMembersGitlab$outboundSchema).optional(),
    bitbucket: z.lazy(() => GetTeamMembersBitbucket$outboundSchema).optional(),
    role: GetTeamMembersRole$outboundSchema,
    uid: z.string(),
    username: z.string(),
    name: z.string().optional(),
    createdAt: z.number(),
    accessRequestedAt: z.number().optional(),
    joinedFrom: z.lazy(() => GetTeamMembersJoinedFrom$outboundSchema).optional(),
    projects: z.array(z.lazy(() => GetTeamMembersProjects$outboundSchema))
        .optional(),
});
export function getTeamMembersMembersToJSON(getTeamMembersMembers) {
    return JSON.stringify(GetTeamMembersMembers$outboundSchema.parse(getTeamMembersMembers));
}
export function getTeamMembersMembersFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersMembers$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersMembers' from JSON`);
}
/** @internal */
export const GetTeamMembersTeamsRole$inboundSchema = z.nativeEnum(GetTeamMembersTeamsRole);
/** @internal */
export const GetTeamMembersTeamsRole$outboundSchema = GetTeamMembersTeamsRole$inboundSchema;
/** @internal */
export const GetTeamMembersTeamRoles$inboundSchema = z.nativeEnum(GetTeamMembersTeamRoles);
/** @internal */
export const GetTeamMembersTeamRoles$outboundSchema = GetTeamMembersTeamRoles$inboundSchema;
/** @internal */
export const GetTeamMembersTeamPermissions$inboundSchema = z.nativeEnum(GetTeamMembersTeamPermissions);
/** @internal */
export const GetTeamMembersTeamPermissions$outboundSchema = GetTeamMembersTeamPermissions$inboundSchema;
/** @internal */
export const GetTeamMembersTeamsProjects$inboundSchema = z.nativeEnum(GetTeamMembersTeamsProjects);
/** @internal */
export const GetTeamMembersTeamsProjects$outboundSchema = GetTeamMembersTeamsProjects$inboundSchema;
/** @internal */
export const EmailInviteCodes$inboundSchema = z.object({
    accessGroups: z.array(z.string()).optional(),
    id: z.string(),
    email: z.string().optional(),
    role: GetTeamMembersTeamsRole$inboundSchema.optional(),
    teamRoles: z.array(GetTeamMembersTeamRoles$inboundSchema).optional(),
    teamPermissions: z.array(GetTeamMembersTeamPermissions$inboundSchema)
        .optional(),
    isDSyncUser: z.boolean(),
    createdAt: z.number().optional(),
    expired: z.boolean().optional(),
    projects: z.record(GetTeamMembersTeamsProjects$inboundSchema).optional(),
    entitlements: z.array(z.string()).optional(),
});
/** @internal */
export const EmailInviteCodes$outboundSchema = z.object({
    accessGroups: z.array(z.string()).optional(),
    id: z.string(),
    email: z.string().optional(),
    role: GetTeamMembersTeamsRole$outboundSchema.optional(),
    teamRoles: z.array(GetTeamMembersTeamRoles$outboundSchema).optional(),
    teamPermissions: z.array(GetTeamMembersTeamPermissions$outboundSchema)
        .optional(),
    isDSyncUser: z.boolean(),
    createdAt: z.number().optional(),
    expired: z.boolean().optional(),
    projects: z.record(GetTeamMembersTeamsProjects$outboundSchema).optional(),
    entitlements: z.array(z.string()).optional(),
});
export function emailInviteCodesToJSON(emailInviteCodes) {
    return JSON.stringify(EmailInviteCodes$outboundSchema.parse(emailInviteCodes));
}
export function emailInviteCodesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EmailInviteCodes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EmailInviteCodes' from JSON`);
}
/** @internal */
export const GetTeamMembersPagination$inboundSchema = z.object({
    hasNext: z.boolean(),
    count: z.number(),
    next: z.nullable(z.number()),
    prev: z.nullable(z.number()),
});
/** @internal */
export const GetTeamMembersPagination$outboundSchema = z.object({
    hasNext: z.boolean(),
    count: z.number(),
    next: z.nullable(z.number()),
    prev: z.nullable(z.number()),
});
export function getTeamMembersPaginationToJSON(getTeamMembersPagination) {
    return JSON.stringify(GetTeamMembersPagination$outboundSchema.parse(getTeamMembersPagination));
}
export function getTeamMembersPaginationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersPagination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersPagination' from JSON`);
}
/** @internal */
export const GetTeamMembersResponseBody$inboundSchema = z.object({
    members: z.array(z.lazy(() => GetTeamMembersMembers$inboundSchema)),
    emailInviteCodes: z.array(z.lazy(() => EmailInviteCodes$inboundSchema))
        .optional(),
    pagination: z.lazy(() => GetTeamMembersPagination$inboundSchema),
});
/** @internal */
export const GetTeamMembersResponseBody$outboundSchema = z.object({
    members: z.array(z.lazy(() => GetTeamMembersMembers$outboundSchema)),
    emailInviteCodes: z.array(z.lazy(() => EmailInviteCodes$outboundSchema))
        .optional(),
    pagination: z.lazy(() => GetTeamMembersPagination$outboundSchema),
});
export function getTeamMembersResponseBodyToJSON(getTeamMembersResponseBody) {
    return JSON.stringify(GetTeamMembersResponseBody$outboundSchema.parse(getTeamMembersResponseBody));
}
export function getTeamMembersResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetTeamMembersResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTeamMembersResponseBody' from JSON`);
}
