// @vercel/sdk/esm/models/getv9projectsidornamecustomenvironmentsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * The type of environment (production, preview, or development)
 */
export const GetV9ProjectsIdOrNameCustomEnvironmentsType = {
    Preview: "preview",
    Production: "production",
    Development: "development",
};
/**
 * The type of matching to perform
 */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType = {
    EndsWith: "endsWith",
    StartsWith: "startsWith",
    Equals: "equals",
};
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsRequest$inboundSchema = z.object({
    idOrName: z.string(),
    gitBranch: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsRequest$outboundSchema = z.object({
    idOrName: z.string(),
    gitBranch: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getV9ProjectsIdOrNameCustomEnvironmentsRequestToJSON(getV9ProjectsIdOrNameCustomEnvironmentsRequest) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsRequest$outboundSchema.parse(getV9ProjectsIdOrNameCustomEnvironmentsRequest));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsRequest' from JSON`);
}
/** @internal */
export const AccountLimit$inboundSchema = z.object({
    total: z.number(),
});
/** @internal */
export const AccountLimit$outboundSchema = z.object({
    total: z.number(),
});
export function accountLimitToJSON(accountLimit) {
    return JSON.stringify(AccountLimit$outboundSchema.parse(accountLimit));
}
export function accountLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AccountLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AccountLimit' from JSON`);
}
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsType$inboundSchema = z
    .nativeEnum(GetV9ProjectsIdOrNameCustomEnvironmentsType);
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsType$outboundSchema = GetV9ProjectsIdOrNameCustomEnvironmentsType$inboundSchema;
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType$inboundSchema = z.nativeEnum(GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType);
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType$outboundSchema = GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType$inboundSchema;
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher$inboundSchema = z.object({
    type: GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType$inboundSchema,
    pattern: z.string(),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher$outboundSchema = z.object({
    type: GetV9ProjectsIdOrNameCustomEnvironmentsEnvironmentType$outboundSchema,
    pattern: z.string(),
});
export function getV9ProjectsIdOrNameCustomEnvironmentsBranchMatcherToJSON(getV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher$outboundSchema.parse(getV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsBranchMatcherFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher' from JSON`);
}
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
export function getV9ProjectsIdOrNameCustomEnvironmentsVerificationToJSON(getV9ProjectsIdOrNameCustomEnvironmentsVerification) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsVerification$outboundSchema.parse(getV9ProjectsIdOrNameCustomEnvironmentsVerification));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsVerification' from JSON`);
}
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsDomains$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsVerification$inboundSchema)).optional(),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsDomains$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsVerification$outboundSchema)).optional(),
});
export function getV9ProjectsIdOrNameCustomEnvironmentsDomainsToJSON(getV9ProjectsIdOrNameCustomEnvironmentsDomains) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsDomains$outboundSchema.parse(getV9ProjectsIdOrNameCustomEnvironmentsDomains));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsDomainsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsDomains$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsDomains' from JSON`);
}
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments$inboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: GetV9ProjectsIdOrNameCustomEnvironmentsType$inboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher$inboundSchema).optional(),
    domains: z.array(z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsDomains$inboundSchema)).optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments$outboundSchema = z.object({
    id: z.string(),
    slug: z.string(),
    type: GetV9ProjectsIdOrNameCustomEnvironmentsType$outboundSchema,
    description: z.string().optional(),
    branchMatcher: z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsBranchMatcher$outboundSchema).optional(),
    domains: z.array(z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsDomains$outboundSchema)).optional(),
    currentDeploymentAliases: z.array(z.string()).optional(),
    createdAt: z.number(),
    updatedAt: z.number(),
});
export function getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentsToJSON(getV9ProjectsIdOrNameCustomEnvironmentsEnvironments) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments$outboundSchema.parse(getV9ProjectsIdOrNameCustomEnvironmentsEnvironments));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsEnvironmentsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments' from JSON`);
}
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsResponseBody$inboundSchema = z.object({
    accountLimit: z.lazy(() => AccountLimit$inboundSchema),
    environments: z.array(z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments$inboundSchema)),
});
/** @internal */
export const GetV9ProjectsIdOrNameCustomEnvironmentsResponseBody$outboundSchema = z.object({
    accountLimit: z.lazy(() => AccountLimit$outboundSchema),
    environments: z.array(z.lazy(() => GetV9ProjectsIdOrNameCustomEnvironmentsEnvironments$outboundSchema)),
});
export function getV9ProjectsIdOrNameCustomEnvironmentsResponseBodyToJSON(getV9ProjectsIdOrNameCustomEnvironmentsResponseBody) {
    return JSON.stringify(GetV9ProjectsIdOrNameCustomEnvironmentsResponseBody$outboundSchema.parse(getV9ProjectsIdOrNameCustomEnvironmentsResponseBody));
}
export function getV9ProjectsIdOrNameCustomEnvironmentsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV9ProjectsIdOrNameCustomEnvironmentsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV9ProjectsIdOrNameCustomEnvironmentsResponseBody' from JSON`);
}
