// @vercel/sdk/esm/models/getv1securityfirewalleventsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetV1SecurityFirewallEventsRequest$inboundSchema = z.object({
    projectId: z.string(),
    startTimestamp: z.number().optional(),
    endTimestamp: z.number().optional(),
    hosts: z.string().optional(),
});
/** @internal */
export const GetV1SecurityFirewallEventsRequest$outboundSchema = z.object({
    projectId: z.string(),
    startTimestamp: z.number().optional(),
    endTimestamp: z.number().optional(),
    hosts: z.string().optional(),
});
export function getV1SecurityFirewallEventsRequestToJSON(getV1SecurityFirewallEventsRequest) {
    return JSON.stringify(GetV1SecurityFirewallEventsRequest$outboundSchema.parse(getV1SecurityFirewallEventsRequest));
}
export function getV1SecurityFirewallEventsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV1SecurityFirewallEventsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV1SecurityFirewallEventsRequest' from JSON`);
}
/** @internal */
export const Actions$inboundSchema = z.object({
    startTime: z.string(),
    endTime: z.string(),
    isActive: z.boolean(),
    action_type: z.string(),
    host: z.string(),
    public_ip: z.string(),
    count: z.number(),
}).transform((v) => {
    return remap$(v, {
        "action_type": "actionType",
        "public_ip": "publicIp",
    });
});
/** @internal */
export const Actions$outboundSchema = z.object({
    startTime: z.string(),
    endTime: z.string(),
    isActive: z.boolean(),
    actionType: z.string(),
    host: z.string(),
    publicIp: z.string(),
    count: z.number(),
}).transform((v) => {
    return remap$(v, {
        actionType: "action_type",
        publicIp: "public_ip",
    });
});
export function actionsToJSON(actions) {
    return JSON.stringify(Actions$outboundSchema.parse(actions));
}
export function actionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Actions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Actions' from JSON`);
}
/** @internal */
export const GetV1SecurityFirewallEventsResponseBody$inboundSchema = z.object({
    actions: z.array(z.lazy(() => Actions$inboundSchema)),
});
/** @internal */
export const GetV1SecurityFirewallEventsResponseBody$outboundSchema = z.object({
    actions: z.array(z.lazy(() => Actions$outboundSchema)),
});
export function getV1SecurityFirewallEventsResponseBodyToJSON(getV1SecurityFirewallEventsResponseBody) {
    return JSON.stringify(GetV1SecurityFirewallEventsResponseBody$outboundSchema.parse(getV1SecurityFirewallEventsResponseBody));
}
export function getV1SecurityFirewallEventsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetV1SecurityFirewallEventsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetV1SecurityFirewallEventsResponseBody' from JSON`);
}
