// @vercel/sdk/esm/models/importresourceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
export const Ownership = {
    Owned: "owned",
    Linked: "linked",
    Sandbox: "sandbox",
};
export const ImportResourceStatus = {
    Ready: "ready",
    Pending: "pending",
    Onboarding: "onboarding",
    Suspended: "suspended",
    Resumed: "resumed",
    Uninstalled: "uninstalled",
    Error: "error",
};
export const ImportResourceType = {
    Prepayment: "prepayment",
    Subscription: "subscription",
};
export const ImportResourceLevel = {
    Info: "info",
    Warn: "warn",
    Error: "error",
};
/** @internal */
export const Ownership$inboundSchema = z
    .nativeEnum(Ownership);
/** @internal */
export const Ownership$outboundSchema = Ownership$inboundSchema;
/** @internal */
export const ImportResourceStatus$inboundSchema = z.nativeEnum(ImportResourceStatus);
/** @internal */
export const ImportResourceStatus$outboundSchema = ImportResourceStatus$inboundSchema;
/** @internal */
export const ImportResourceType$inboundSchema = z.nativeEnum(ImportResourceType);
/** @internal */
export const ImportResourceType$outboundSchema = ImportResourceType$inboundSchema;
/** @internal */
export const ImportResourceDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const ImportResourceDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
export function importResourceDetailsToJSON(importResourceDetails) {
    return JSON.stringify(ImportResourceDetails$outboundSchema.parse(importResourceDetails));
}
export function importResourceDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceDetails' from JSON`);
}
/** @internal */
export const ImportResourceHighlightedDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const ImportResourceHighlightedDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
export function importResourceHighlightedDetailsToJSON(importResourceHighlightedDetails) {
    return JSON.stringify(ImportResourceHighlightedDetails$outboundSchema.parse(importResourceHighlightedDetails));
}
export function importResourceHighlightedDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceHighlightedDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceHighlightedDetails' from JSON`);
}
/** @internal */
export const ImportResourceBillingPlan$inboundSchema = collectExtraKeys$(z.object({
    id: z.string(),
    type: ImportResourceType$inboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => ImportResourceDetails$inboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => ImportResourceHighlightedDetails$inboundSchema)).optional(),
    effectiveDate: z.string().optional(),
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const ImportResourceBillingPlan$outboundSchema = z.object({
    id: z.string(),
    type: ImportResourceType$outboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => ImportResourceDetails$outboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => ImportResourceHighlightedDetails$outboundSchema)).optional(),
    effectiveDate: z.string().optional(),
    additionalProperties: z.record(z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function importResourceBillingPlanToJSON(importResourceBillingPlan) {
    return JSON.stringify(ImportResourceBillingPlan$outboundSchema.parse(importResourceBillingPlan));
}
export function importResourceBillingPlanFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceBillingPlan$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceBillingPlan' from JSON`);
}
/** @internal */
export const ImportResourceLevel$inboundSchema = z.nativeEnum(ImportResourceLevel);
/** @internal */
export const ImportResourceLevel$outboundSchema = ImportResourceLevel$inboundSchema;
/** @internal */
export const ImportResourceNotification$inboundSchema = z.object({
    level: ImportResourceLevel$inboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/** @internal */
export const ImportResourceNotification$outboundSchema = z.object({
    level: ImportResourceLevel$outboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
export function importResourceNotificationToJSON(importResourceNotification) {
    return JSON.stringify(ImportResourceNotification$outboundSchema.parse(importResourceNotification));
}
export function importResourceNotificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceNotification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceNotification' from JSON`);
}
/** @internal */
export const EnvironmentOverrides$inboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
/** @internal */
export const EnvironmentOverrides$outboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
export function environmentOverridesToJSON(environmentOverrides) {
    return JSON.stringify(EnvironmentOverrides$outboundSchema.parse(environmentOverrides));
}
export function environmentOverridesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EnvironmentOverrides$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EnvironmentOverrides' from JSON`);
}
/** @internal */
export const ImportResourceSecrets$inboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => EnvironmentOverrides$inboundSchema)
        .optional(),
});
/** @internal */
export const ImportResourceSecrets$outboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => EnvironmentOverrides$outboundSchema)
        .optional(),
});
export function importResourceSecretsToJSON(importResourceSecrets) {
    return JSON.stringify(ImportResourceSecrets$outboundSchema.parse(importResourceSecrets));
}
export function importResourceSecretsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceSecrets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceSecrets' from JSON`);
}
/** @internal */
export const ImportResourceRequestBody$inboundSchema = z.object({
    ownership: Ownership$inboundSchema.optional(),
    productId: z.string(),
    name: z.string(),
    status: ImportResourceStatus$inboundSchema,
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => ImportResourceBillingPlan$inboundSchema).optional(),
    notification: z.lazy(() => ImportResourceNotification$inboundSchema)
        .optional(),
    extras: z.record(z.any()).optional(),
    secrets: z.array(z.lazy(() => ImportResourceSecrets$inboundSchema))
        .optional(),
});
/** @internal */
export const ImportResourceRequestBody$outboundSchema = z.object({
    ownership: Ownership$outboundSchema.optional(),
    productId: z.string(),
    name: z.string(),
    status: ImportResourceStatus$outboundSchema,
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => ImportResourceBillingPlan$outboundSchema)
        .optional(),
    notification: z.lazy(() => ImportResourceNotification$outboundSchema)
        .optional(),
    extras: z.record(z.any()).optional(),
    secrets: z.array(z.lazy(() => ImportResourceSecrets$outboundSchema))
        .optional(),
});
export function importResourceRequestBodyToJSON(importResourceRequestBody) {
    return JSON.stringify(ImportResourceRequestBody$outboundSchema.parse(importResourceRequestBody));
}
export function importResourceRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceRequestBody' from JSON`);
}
/** @internal */
export const ImportResourceRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
    RequestBody: z.lazy(() => ImportResourceRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const ImportResourceRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
    requestBody: z.lazy(() => ImportResourceRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function importResourceRequestToJSON(importResourceRequest) {
    return JSON.stringify(ImportResourceRequest$outboundSchema.parse(importResourceRequest));
}
export function importResourceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceRequest' from JSON`);
}
/** @internal */
export const ImportResourceResponseBody$inboundSchema = z.object({
    name: z.string(),
});
/** @internal */
export const ImportResourceResponseBody$outboundSchema = z.object({
    name: z.string(),
});
export function importResourceResponseBodyToJSON(importResourceResponseBody) {
    return JSON.stringify(ImportResourceResponseBody$outboundSchema.parse(importResourceResponseBody));
}
export function importResourceResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ImportResourceResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ImportResourceResponseBody' from JSON`);
}
