// @vercel/sdk/esm/models/invalidatebytagsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const InvalidateByTagsTarget = {
    Production: "production",
    Preview: "preview",
};
/** @internal */
export const Tags$inboundSchema = z
    .union([z.array(z.string()), z.string()]);
/** @internal */
export const Tags$outboundSchema = z.union([z.array(z.string()), z.string()]);
export function tagsToJSON(tags) {
    return JSON.stringify(Tags$outboundSchema.parse(tags));
}
export function tagsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Tags$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Tags' from JSON`);
}
/** @internal */
export const InvalidateByTagsTarget$inboundSchema = z.nativeEnum(InvalidateByTagsTarget);
/** @internal */
export const InvalidateByTagsTarget$outboundSchema = InvalidateByTagsTarget$inboundSchema;
/** @internal */
export const InvalidateByTagsRequestBody$inboundSchema = z.object({
    tags: z.union([z.array(z.string()), z.string()]),
    target: InvalidateByTagsTarget$inboundSchema.optional(),
});
/** @internal */
export const InvalidateByTagsRequestBody$outboundSchema = z.object({
    tags: z.union([z.array(z.string()), z.string()]),
    target: InvalidateByTagsTarget$outboundSchema.optional(),
});
export function invalidateByTagsRequestBodyToJSON(invalidateByTagsRequestBody) {
    return JSON.stringify(InvalidateByTagsRequestBody$outboundSchema.parse(invalidateByTagsRequestBody));
}
export function invalidateByTagsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InvalidateByTagsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InvalidateByTagsRequestBody' from JSON`);
}
/** @internal */
export const InvalidateByTagsRequest$inboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => InvalidateByTagsRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const InvalidateByTagsRequest$outboundSchema = z.object({
    projectIdOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => InvalidateByTagsRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function invalidateByTagsRequestToJSON(invalidateByTagsRequest) {
    return JSON.stringify(InvalidateByTagsRequest$outboundSchema.parse(invalidateByTagsRequest));
}
export function invalidateByTagsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InvalidateByTagsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InvalidateByTagsRequest' from JSON`);
}
