// @vercel/sdk/esm/models/invitedteammember.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * The role used for the invitation
 */
export const Role = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * The team roles of the user
 */
export const TeamRoles = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * The team permissions of the user
 */
export const TeamPermissions = {
    IntegrationManager: "IntegrationManager",
    CreateProject: "CreateProject",
    FullProductionDeployment: "FullProductionDeployment",
    UsageViewer: "UsageViewer",
    EnvVariableManager: "EnvVariableManager",
    EnvironmentManager: "EnvironmentManager",
    V0Builder: "V0Builder",
    V0Chatter: "V0Chatter",
    V0Viewer: "V0Viewer",
};
/** @internal */
export const Role$inboundSchema = z.nativeEnum(Role);
/** @internal */
export const Role$outboundSchema = Role$inboundSchema;
/** @internal */
export const TeamRoles$inboundSchema = z
    .nativeEnum(TeamRoles);
/** @internal */
export const TeamRoles$outboundSchema = TeamRoles$inboundSchema;
/** @internal */
export const TeamPermissions$inboundSchema = z.nativeEnum(TeamPermissions);
/** @internal */
export const TeamPermissions$outboundSchema = TeamPermissions$inboundSchema;
/** @internal */
export const InvitedTeamMember$inboundSchema = z.object({
    uid: z.string(),
    username: z.string(),
    email: z.string(),
    role: Role$inboundSchema,
    teamRoles: z.array(TeamRoles$inboundSchema).optional(),
    teamPermissions: z.array(TeamPermissions$inboundSchema).optional(),
});
/** @internal */
export const InvitedTeamMember$outboundSchema = z.object({
    uid: z.string(),
    username: z.string(),
    email: z.string(),
    role: Role$outboundSchema,
    teamRoles: z.array(TeamRoles$outboundSchema).optional(),
    teamPermissions: z.array(TeamPermissions$outboundSchema).optional(),
});
export function invitedTeamMemberToJSON(invitedTeamMember) {
    return JSON.stringify(InvitedTeamMember$outboundSchema.parse(invitedTeamMember));
}
export function invitedTeamMemberFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InvitedTeamMember$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InvitedTeamMember' from JSON`);
}
