// @vercel/sdk/esm/models/inviteusertoteamop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/**
 * The role of the user to invite
 */
export const InviteUserToTeamRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
/**
 * Sets the project roles for the invited user
 */
export const InviteUserToTeamTeamsRole = {
    Admin: "ADMIN",
    ProjectViewer: "PROJECT_VIEWER",
    ProjectDeveloper: "PROJECT_DEVELOPER",
};
/** @internal */
export const InviteUserToTeamRole$inboundSchema = z.nativeEnum(InviteUserToTeamRole);
/** @internal */
export const InviteUserToTeamRole$outboundSchema = InviteUserToTeamRole$inboundSchema;
/** @internal */
export const InviteUserToTeamTeamsRole$inboundSchema = z.nativeEnum(InviteUserToTeamTeamsRole);
/** @internal */
export const InviteUserToTeamTeamsRole$outboundSchema = InviteUserToTeamTeamsRole$inboundSchema;
/** @internal */
export const InviteUserToTeamProjects$inboundSchema = z.object({
    projectId: z.string(),
    role: InviteUserToTeamTeamsRole$inboundSchema,
});
/** @internal */
export const InviteUserToTeamProjects$outboundSchema = z.object({
    projectId: z.string(),
    role: InviteUserToTeamTeamsRole$outboundSchema,
});
export function inviteUserToTeamProjectsToJSON(inviteUserToTeamProjects) {
    return JSON.stringify(InviteUserToTeamProjects$outboundSchema.parse(inviteUserToTeamProjects));
}
export function inviteUserToTeamProjectsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InviteUserToTeamProjects$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InviteUserToTeamProjects' from JSON`);
}
/** @internal */
export const InviteUserToTeamRequestBody$inboundSchema = z.object({
    email: z.string(),
    role: InviteUserToTeamRole$inboundSchema.default("VIEWER"),
    projects: z.array(z.lazy(() => InviteUserToTeamProjects$inboundSchema))
        .optional(),
});
/** @internal */
export const InviteUserToTeamRequestBody$outboundSchema = z.object({
    email: z.string(),
    role: InviteUserToTeamRole$outboundSchema.default("VIEWER"),
    projects: z.array(z.lazy(() => InviteUserToTeamProjects$outboundSchema))
        .optional(),
});
export function inviteUserToTeamRequestBodyToJSON(inviteUserToTeamRequestBody) {
    return JSON.stringify(InviteUserToTeamRequestBody$outboundSchema.parse(inviteUserToTeamRequestBody));
}
export function inviteUserToTeamRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => InviteUserToTeamRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InviteUserToTeamRequestBody' from JSON`);
}
