// @vercel/sdk/esm/models/listdeploymentaliasesop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const ListDeploymentAliasesProtectionBypassAliasesResponse200Scope = {
    EmailInvite: "email_invite",
};
export const ListDeploymentAliasesProtectionBypassAliasesResponseScope = {
    AliasProtectionOverride: "alias-protection-override",
};
export const ListDeploymentAliasesProtectionBypassAccess = {
    Requested: "requested",
    Granted: "granted",
};
export const ListDeploymentAliasesProtectionBypassAliasesScope = {
    User: "user",
};
export const ListDeploymentAliasesProtectionBypassScope = {
    ShareableLink: "shareable-link",
};
/** @internal */
export const ListDeploymentAliasesRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const ListDeploymentAliasesRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function listDeploymentAliasesRequestToJSON(listDeploymentAliasesRequest) {
    return JSON.stringify(ListDeploymentAliasesRequest$outboundSchema.parse(listDeploymentAliasesRequest));
}
export function listDeploymentAliasesRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesRequest' from JSON`);
}
/** @internal */
export const ListDeploymentAliasesProtectionBypassAliasesResponse200Scope$inboundSchema = z.nativeEnum(ListDeploymentAliasesProtectionBypassAliasesResponse200Scope);
/** @internal */
export const ListDeploymentAliasesProtectionBypassAliasesResponse200Scope$outboundSchema = ListDeploymentAliasesProtectionBypassAliasesResponse200Scope$inboundSchema;
/** @internal */
export const ListDeploymentAliasesProtectionBypass4$inboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    scope: ListDeploymentAliasesProtectionBypassAliasesResponse200Scope$inboundSchema,
});
/** @internal */
export const ListDeploymentAliasesProtectionBypass4$outboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    scope: ListDeploymentAliasesProtectionBypassAliasesResponse200Scope$outboundSchema,
});
export function listDeploymentAliasesProtectionBypass4ToJSON(listDeploymentAliasesProtectionBypass4) {
    return JSON.stringify(ListDeploymentAliasesProtectionBypass4$outboundSchema.parse(listDeploymentAliasesProtectionBypass4));
}
export function listDeploymentAliasesProtectionBypass4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesProtectionBypass4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesProtectionBypass4' from JSON`);
}
/** @internal */
export const ListDeploymentAliasesProtectionBypassAliasesResponseScope$inboundSchema = z.nativeEnum(ListDeploymentAliasesProtectionBypassAliasesResponseScope);
/** @internal */
export const ListDeploymentAliasesProtectionBypassAliasesResponseScope$outboundSchema = ListDeploymentAliasesProtectionBypassAliasesResponseScope$inboundSchema;
/** @internal */
export const ListDeploymentAliasesProtectionBypass3$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListDeploymentAliasesProtectionBypassAliasesResponseScope$inboundSchema,
});
/** @internal */
export const ListDeploymentAliasesProtectionBypass3$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListDeploymentAliasesProtectionBypassAliasesResponseScope$outboundSchema,
});
export function listDeploymentAliasesProtectionBypass3ToJSON(listDeploymentAliasesProtectionBypass3) {
    return JSON.stringify(ListDeploymentAliasesProtectionBypass3$outboundSchema.parse(listDeploymentAliasesProtectionBypass3));
}
export function listDeploymentAliasesProtectionBypass3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesProtectionBypass3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesProtectionBypass3' from JSON`);
}
/** @internal */
export const ListDeploymentAliasesProtectionBypassAccess$inboundSchema = z
    .nativeEnum(ListDeploymentAliasesProtectionBypassAccess);
/** @internal */
export const ListDeploymentAliasesProtectionBypassAccess$outboundSchema = ListDeploymentAliasesProtectionBypassAccess$inboundSchema;
/** @internal */
export const ListDeploymentAliasesProtectionBypassAliasesScope$inboundSchema = z
    .nativeEnum(ListDeploymentAliasesProtectionBypassAliasesScope);
/** @internal */
export const ListDeploymentAliasesProtectionBypassAliasesScope$outboundSchema = ListDeploymentAliasesProtectionBypassAliasesScope$inboundSchema;
/** @internal */
export const ListDeploymentAliasesProtectionBypass2$inboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    access: ListDeploymentAliasesProtectionBypassAccess$inboundSchema,
    scope: ListDeploymentAliasesProtectionBypassAliasesScope$inboundSchema,
});
/** @internal */
export const ListDeploymentAliasesProtectionBypass2$outboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    access: ListDeploymentAliasesProtectionBypassAccess$outboundSchema,
    scope: ListDeploymentAliasesProtectionBypassAliasesScope$outboundSchema,
});
export function listDeploymentAliasesProtectionBypass2ToJSON(listDeploymentAliasesProtectionBypass2) {
    return JSON.stringify(ListDeploymentAliasesProtectionBypass2$outboundSchema.parse(listDeploymentAliasesProtectionBypass2));
}
export function listDeploymentAliasesProtectionBypass2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesProtectionBypass2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesProtectionBypass2' from JSON`);
}
/** @internal */
export const ListDeploymentAliasesProtectionBypassScope$inboundSchema = z
    .nativeEnum(ListDeploymentAliasesProtectionBypassScope);
/** @internal */
export const ListDeploymentAliasesProtectionBypassScope$outboundSchema = ListDeploymentAliasesProtectionBypassScope$inboundSchema;
/** @internal */
export const ListDeploymentAliasesProtectionBypass1$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListDeploymentAliasesProtectionBypassScope$inboundSchema,
    expires: z.number().optional(),
});
/** @internal */
export const ListDeploymentAliasesProtectionBypass1$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListDeploymentAliasesProtectionBypassScope$outboundSchema,
    expires: z.number().optional(),
});
export function listDeploymentAliasesProtectionBypass1ToJSON(listDeploymentAliasesProtectionBypass1) {
    return JSON.stringify(ListDeploymentAliasesProtectionBypass1$outboundSchema.parse(listDeploymentAliasesProtectionBypass1));
}
export function listDeploymentAliasesProtectionBypass1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesProtectionBypass1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesProtectionBypass1' from JSON`);
}
/** @internal */
export const ListDeploymentAliasesProtectionBypass$inboundSchema = z.union([
    z.lazy(() => ListDeploymentAliasesProtectionBypass2$inboundSchema),
    z.lazy(() => ListDeploymentAliasesProtectionBypass4$inboundSchema),
    z.lazy(() => ListDeploymentAliasesProtectionBypass1$inboundSchema),
    z.lazy(() => ListDeploymentAliasesProtectionBypass3$inboundSchema),
]);
/** @internal */
export const ListDeploymentAliasesProtectionBypass$outboundSchema = z.union([
    z.lazy(() => ListDeploymentAliasesProtectionBypass2$outboundSchema),
    z.lazy(() => ListDeploymentAliasesProtectionBypass4$outboundSchema),
    z.lazy(() => ListDeploymentAliasesProtectionBypass1$outboundSchema),
    z.lazy(() => ListDeploymentAliasesProtectionBypass3$outboundSchema),
]);
export function listDeploymentAliasesProtectionBypassToJSON(listDeploymentAliasesProtectionBypass) {
    return JSON.stringify(ListDeploymentAliasesProtectionBypass$outboundSchema.parse(listDeploymentAliasesProtectionBypass));
}
export function listDeploymentAliasesProtectionBypassFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesProtectionBypass$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesProtectionBypass' from JSON`);
}
/** @internal */
export const Aliases$inboundSchema = z.object({
    uid: z.string(),
    alias: z.string(),
    created: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    redirect: z.nullable(z.string()).optional(),
    protectionBypass: z.record(z.union([
        z.lazy(() => ListDeploymentAliasesProtectionBypass2$inboundSchema),
        z.lazy(() => ListDeploymentAliasesProtectionBypass4$inboundSchema),
        z.lazy(() => ListDeploymentAliasesProtectionBypass1$inboundSchema),
        z.lazy(() => ListDeploymentAliasesProtectionBypass3$inboundSchema),
    ])).optional(),
});
/** @internal */
export const Aliases$outboundSchema = z.object({
    uid: z.string(),
    alias: z.string(),
    created: z.date().transform(v => v.toISOString()),
    redirect: z.nullable(z.string()).optional(),
    protectionBypass: z.record(z.union([
        z.lazy(() => ListDeploymentAliasesProtectionBypass2$outboundSchema),
        z.lazy(() => ListDeploymentAliasesProtectionBypass4$outboundSchema),
        z.lazy(() => ListDeploymentAliasesProtectionBypass1$outboundSchema),
        z.lazy(() => ListDeploymentAliasesProtectionBypass3$outboundSchema),
    ])).optional(),
});
export function aliasesToJSON(aliases) {
    return JSON.stringify(Aliases$outboundSchema.parse(aliases));
}
export function aliasesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Aliases$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Aliases' from JSON`);
}
/** @internal */
export const ListDeploymentAliasesResponseBody$inboundSchema = z.object({
    aliases: z.array(z.lazy(() => Aliases$inboundSchema)),
});
/** @internal */
export const ListDeploymentAliasesResponseBody$outboundSchema = z.object({
    aliases: z.array(z.lazy(() => Aliases$outboundSchema)),
});
export function listDeploymentAliasesResponseBodyToJSON(listDeploymentAliasesResponseBody) {
    return JSON.stringify(ListDeploymentAliasesResponseBody$outboundSchema.parse(listDeploymentAliasesResponseBody));
}
export function listDeploymentAliasesResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListDeploymentAliasesResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListDeploymentAliasesResponseBody' from JSON`);
}
