// @vercel/sdk/esm/models/listaliasesop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
export const ListAliasesProtectionBypassAliasesResponse200Scope = {
    EmailInvite: "email_invite",
};
export const ListAliasesProtectionBypassAliasesResponseScope = {
    AliasProtectionOverride: "alias-protection-override",
};
export const ListAliasesProtectionBypassAccess = {
    Requested: "requested",
    Granted: "granted",
};
export const ListAliasesProtectionBypassAliasesScope = {
    User: "user",
};
export const ListAliasesProtectionBypassScope = {
    ShareableLink: "shareable-link",
};
/** @internal */
export const Domain$inboundSchema = z
    .union([z.array(z.string()), z.string()]);
/** @internal */
export const Domain$outboundSchema = z.union([z.array(z.string()), z.string()]);
export function domainToJSON(domain) {
    return JSON.stringify(Domain$outboundSchema.parse(domain));
}
export function domainFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Domain$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Domain' from JSON`);
}
/** @internal */
export const ListAliasesRequest$inboundSchema = z.object({
    domain: z.union([z.array(z.string()), z.string()]).optional(),
    from: z.number().optional(),
    limit: z.number().optional(),
    projectId: z.string().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    rollbackDeploymentId: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const ListAliasesRequest$outboundSchema = z.object({
    domain: z.union([z.array(z.string()), z.string()]).optional(),
    from: z.number().optional(),
    limit: z.number().optional(),
    projectId: z.string().optional(),
    since: z.number().optional(),
    until: z.number().optional(),
    rollbackDeploymentId: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function listAliasesRequestToJSON(listAliasesRequest) {
    return JSON.stringify(ListAliasesRequest$outboundSchema.parse(listAliasesRequest));
}
export function listAliasesRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesRequest' from JSON`);
}
/** @internal */
export const ListAliasesCreator$inboundSchema = z.object({
    uid: z.string(),
    email: z.string(),
    username: z.string(),
});
/** @internal */
export const ListAliasesCreator$outboundSchema = z.object({
    uid: z.string(),
    email: z.string(),
    username: z.string(),
});
export function listAliasesCreatorToJSON(listAliasesCreator) {
    return JSON.stringify(ListAliasesCreator$outboundSchema.parse(listAliasesCreator));
}
export function listAliasesCreatorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesCreator$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesCreator' from JSON`);
}
/** @internal */
export const ListAliasesDeployment$inboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    meta: z.string().optional(),
});
/** @internal */
export const ListAliasesDeployment$outboundSchema = z.object({
    id: z.string(),
    url: z.string(),
    meta: z.string().optional(),
});
export function listAliasesDeploymentToJSON(listAliasesDeployment) {
    return JSON.stringify(ListAliasesDeployment$outboundSchema.parse(listAliasesDeployment));
}
export function listAliasesDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesDeployment' from JSON`);
}
/** @internal */
export const ListAliasesProtectionBypassAliasesResponse200Scope$inboundSchema = z
    .nativeEnum(ListAliasesProtectionBypassAliasesResponse200Scope);
/** @internal */
export const ListAliasesProtectionBypassAliasesResponse200Scope$outboundSchema = ListAliasesProtectionBypassAliasesResponse200Scope$inboundSchema;
/** @internal */
export const ListAliasesProtectionBypass4$inboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    scope: ListAliasesProtectionBypassAliasesResponse200Scope$inboundSchema,
});
/** @internal */
export const ListAliasesProtectionBypass4$outboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    scope: ListAliasesProtectionBypassAliasesResponse200Scope$outboundSchema,
});
export function listAliasesProtectionBypass4ToJSON(listAliasesProtectionBypass4) {
    return JSON.stringify(ListAliasesProtectionBypass4$outboundSchema.parse(listAliasesProtectionBypass4));
}
export function listAliasesProtectionBypass4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesProtectionBypass4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesProtectionBypass4' from JSON`);
}
/** @internal */
export const ListAliasesProtectionBypassAliasesResponseScope$inboundSchema = z
    .nativeEnum(ListAliasesProtectionBypassAliasesResponseScope);
/** @internal */
export const ListAliasesProtectionBypassAliasesResponseScope$outboundSchema = ListAliasesProtectionBypassAliasesResponseScope$inboundSchema;
/** @internal */
export const ListAliasesProtectionBypass3$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListAliasesProtectionBypassAliasesResponseScope$inboundSchema,
});
/** @internal */
export const ListAliasesProtectionBypass3$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListAliasesProtectionBypassAliasesResponseScope$outboundSchema,
});
export function listAliasesProtectionBypass3ToJSON(listAliasesProtectionBypass3) {
    return JSON.stringify(ListAliasesProtectionBypass3$outboundSchema.parse(listAliasesProtectionBypass3));
}
export function listAliasesProtectionBypass3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesProtectionBypass3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesProtectionBypass3' from JSON`);
}
/** @internal */
export const ListAliasesProtectionBypassAccess$inboundSchema = z.nativeEnum(ListAliasesProtectionBypassAccess);
/** @internal */
export const ListAliasesProtectionBypassAccess$outboundSchema = ListAliasesProtectionBypassAccess$inboundSchema;
/** @internal */
export const ListAliasesProtectionBypassAliasesScope$inboundSchema = z
    .nativeEnum(ListAliasesProtectionBypassAliasesScope);
/** @internal */
export const ListAliasesProtectionBypassAliasesScope$outboundSchema = ListAliasesProtectionBypassAliasesScope$inboundSchema;
/** @internal */
export const ListAliasesProtectionBypass2$inboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    access: ListAliasesProtectionBypassAccess$inboundSchema,
    scope: ListAliasesProtectionBypassAliasesScope$inboundSchema,
});
/** @internal */
export const ListAliasesProtectionBypass2$outboundSchema = z.object({
    createdAt: z.number(),
    lastUpdatedAt: z.number(),
    lastUpdatedBy: z.string(),
    access: ListAliasesProtectionBypassAccess$outboundSchema,
    scope: ListAliasesProtectionBypassAliasesScope$outboundSchema,
});
export function listAliasesProtectionBypass2ToJSON(listAliasesProtectionBypass2) {
    return JSON.stringify(ListAliasesProtectionBypass2$outboundSchema.parse(listAliasesProtectionBypass2));
}
export function listAliasesProtectionBypass2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesProtectionBypass2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesProtectionBypass2' from JSON`);
}
/** @internal */
export const ListAliasesProtectionBypassScope$inboundSchema = z.nativeEnum(ListAliasesProtectionBypassScope);
/** @internal */
export const ListAliasesProtectionBypassScope$outboundSchema = ListAliasesProtectionBypassScope$inboundSchema;
/** @internal */
export const ListAliasesProtectionBypass1$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListAliasesProtectionBypassScope$inboundSchema,
    expires: z.number().optional(),
});
/** @internal */
export const ListAliasesProtectionBypass1$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: ListAliasesProtectionBypassScope$outboundSchema,
    expires: z.number().optional(),
});
export function listAliasesProtectionBypass1ToJSON(listAliasesProtectionBypass1) {
    return JSON.stringify(ListAliasesProtectionBypass1$outboundSchema.parse(listAliasesProtectionBypass1));
}
export function listAliasesProtectionBypass1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesProtectionBypass1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesProtectionBypass1' from JSON`);
}
/** @internal */
export const ListAliasesProtectionBypass$inboundSchema = z.union([
    z.lazy(() => ListAliasesProtectionBypass2$inboundSchema),
    z.lazy(() => ListAliasesProtectionBypass4$inboundSchema),
    z.lazy(() => ListAliasesProtectionBypass1$inboundSchema),
    z.lazy(() => ListAliasesProtectionBypass3$inboundSchema),
]);
/** @internal */
export const ListAliasesProtectionBypass$outboundSchema = z.union([
    z.lazy(() => ListAliasesProtectionBypass2$outboundSchema),
    z.lazy(() => ListAliasesProtectionBypass4$outboundSchema),
    z.lazy(() => ListAliasesProtectionBypass1$outboundSchema),
    z.lazy(() => ListAliasesProtectionBypass3$outboundSchema),
]);
export function listAliasesProtectionBypassToJSON(listAliasesProtectionBypass) {
    return JSON.stringify(ListAliasesProtectionBypass$outboundSchema.parse(listAliasesProtectionBypass));
}
export function listAliasesProtectionBypassFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesProtectionBypass$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesProtectionBypass' from JSON`);
}
/** @internal */
export const ListAliasesDefaultApp$inboundSchema = z.object({
    projectId: z.string(),
});
/** @internal */
export const ListAliasesDefaultApp$outboundSchema = z.object({
    projectId: z.string(),
});
export function listAliasesDefaultAppToJSON(listAliasesDefaultApp) {
    return JSON.stringify(ListAliasesDefaultApp$outboundSchema.parse(listAliasesDefaultApp));
}
export function listAliasesDefaultAppFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesDefaultApp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesDefaultApp' from JSON`);
}
/** @internal */
export const ListAliasesApplications3$inboundSchema = z.object({
    deploymentId: z.string().optional(),
    branchDeploymentId: z.string().optional(),
    fallbackDeploymentId: z.string().optional(),
    fallbackHost: z.string().optional(),
    branchAlias: z.string().optional(),
    projectId: z.string(),
});
/** @internal */
export const ListAliasesApplications3$outboundSchema = z.object({
    deploymentId: z.string().optional(),
    branchDeploymentId: z.string().optional(),
    fallbackDeploymentId: z.string().optional(),
    fallbackHost: z.string().optional(),
    branchAlias: z.string().optional(),
    projectId: z.string(),
});
export function listAliasesApplications3ToJSON(listAliasesApplications3) {
    return JSON.stringify(ListAliasesApplications3$outboundSchema.parse(listAliasesApplications3));
}
export function listAliasesApplications3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesApplications3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesApplications3' from JSON`);
}
/** @internal */
export const ListAliasesApplications2$inboundSchema = z.object({
    fallbackHost: z.string(),
    branchAlias: z.string(),
    projectId: z.string(),
});
/** @internal */
export const ListAliasesApplications2$outboundSchema = z.object({
    fallbackHost: z.string(),
    branchAlias: z.string(),
    projectId: z.string(),
});
export function listAliasesApplications2ToJSON(listAliasesApplications2) {
    return JSON.stringify(ListAliasesApplications2$outboundSchema.parse(listAliasesApplications2));
}
export function listAliasesApplications2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesApplications2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesApplications2' from JSON`);
}
/** @internal */
export const ListAliasesApplications1$inboundSchema = z.object({
    fallbackHost: z.string(),
    projectId: z.string(),
});
/** @internal */
export const ListAliasesApplications1$outboundSchema = z.object({
    fallbackHost: z.string(),
    projectId: z.string(),
});
export function listAliasesApplications1ToJSON(listAliasesApplications1) {
    return JSON.stringify(ListAliasesApplications1$outboundSchema.parse(listAliasesApplications1));
}
export function listAliasesApplications1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesApplications1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesApplications1' from JSON`);
}
/** @internal */
export const ListAliasesApplications$inboundSchema = z.union([
    z.array(z.lazy(() => ListAliasesApplications1$inboundSchema)),
    z.array(z.lazy(() => ListAliasesApplications2$inboundSchema)),
    z.array(z.lazy(() => ListAliasesApplications3$inboundSchema)),
]);
/** @internal */
export const ListAliasesApplications$outboundSchema = z.union([
    z.array(z.lazy(() => ListAliasesApplications1$outboundSchema)),
    z.array(z.lazy(() => ListAliasesApplications2$outboundSchema)),
    z.array(z.lazy(() => ListAliasesApplications3$outboundSchema)),
]);
export function listAliasesApplicationsToJSON(listAliasesApplications) {
    return JSON.stringify(ListAliasesApplications$outboundSchema.parse(listAliasesApplications));
}
export function listAliasesApplicationsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesApplications$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesApplications' from JSON`);
}
/** @internal */
export const ListAliasesMicrofrontends$inboundSchema = z.object({
    defaultApp: z.lazy(() => ListAliasesDefaultApp$inboundSchema),
    applications: z.union([
        z.array(z.lazy(() => ListAliasesApplications1$inboundSchema)),
        z.array(z.lazy(() => ListAliasesApplications2$inboundSchema)),
        z.array(z.lazy(() => ListAliasesApplications3$inboundSchema)),
    ]),
});
/** @internal */
export const ListAliasesMicrofrontends$outboundSchema = z.object({
    defaultApp: z.lazy(() => ListAliasesDefaultApp$outboundSchema),
    applications: z.union([
        z.array(z.lazy(() => ListAliasesApplications1$outboundSchema)),
        z.array(z.lazy(() => ListAliasesApplications2$outboundSchema)),
        z.array(z.lazy(() => ListAliasesApplications3$outboundSchema)),
    ]),
});
export function listAliasesMicrofrontendsToJSON(listAliasesMicrofrontends) {
    return JSON.stringify(ListAliasesMicrofrontends$outboundSchema.parse(listAliasesMicrofrontends));
}
export function listAliasesMicrofrontendsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesMicrofrontends$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesMicrofrontends' from JSON`);
}
/** @internal */
export const ListAliasesAliases$inboundSchema = z.object({
    alias: z.string(),
    created: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    createdAt: z.number().optional(),
    creator: z.lazy(() => ListAliasesCreator$inboundSchema).optional(),
    deletedAt: z.nullable(z.number()).optional(),
    deployment: z.lazy(() => ListAliasesDeployment$inboundSchema).optional(),
    deploymentId: z.nullable(z.string()),
    projectId: z.nullable(z.string()),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    uid: z.string(),
    updatedAt: z.number().optional(),
    protectionBypass: z.record(z.union([
        z.lazy(() => ListAliasesProtectionBypass2$inboundSchema),
        z.lazy(() => ListAliasesProtectionBypass4$inboundSchema),
        z.lazy(() => ListAliasesProtectionBypass1$inboundSchema),
        z.lazy(() => ListAliasesProtectionBypass3$inboundSchema),
    ])).optional(),
    microfrontends: z.lazy(() => ListAliasesMicrofrontends$inboundSchema)
        .optional(),
});
/** @internal */
export const ListAliasesAliases$outboundSchema = z.object({
    alias: z.string(),
    created: z.date().transform(v => v.toISOString()),
    createdAt: z.number().optional(),
    creator: z.lazy(() => ListAliasesCreator$outboundSchema).optional(),
    deletedAt: z.nullable(z.number()).optional(),
    deployment: z.lazy(() => ListAliasesDeployment$outboundSchema).optional(),
    deploymentId: z.nullable(z.string()),
    projectId: z.nullable(z.string()),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    uid: z.string(),
    updatedAt: z.number().optional(),
    protectionBypass: z.record(z.union([
        z.lazy(() => ListAliasesProtectionBypass2$outboundSchema),
        z.lazy(() => ListAliasesProtectionBypass4$outboundSchema),
        z.lazy(() => ListAliasesProtectionBypass1$outboundSchema),
        z.lazy(() => ListAliasesProtectionBypass3$outboundSchema),
    ])).optional(),
    microfrontends: z.lazy(() => ListAliasesMicrofrontends$outboundSchema)
        .optional(),
});
export function listAliasesAliasesToJSON(listAliasesAliases) {
    return JSON.stringify(ListAliasesAliases$outboundSchema.parse(listAliasesAliases));
}
export function listAliasesAliasesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesAliases$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesAliases' from JSON`);
}
/** @internal */
export const ListAliasesResponseBody$inboundSchema = z.object({
    aliases: z.array(z.lazy(() => ListAliasesAliases$inboundSchema)),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListAliasesResponseBody$outboundSchema = z.object({
    aliases: z.array(z.lazy(() => ListAliasesAliases$outboundSchema)),
    pagination: Pagination$outboundSchema,
});
export function listAliasesResponseBodyToJSON(listAliasesResponseBody) {
    return JSON.stringify(ListAliasesResponseBody$outboundSchema.parse(listAliasesResponseBody));
}
export function listAliasesResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListAliasesResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAliasesResponseBody' from JSON`);
}
