// @vercel/sdk/esm/models/listsharedenvvariableop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
import { Pagination$inboundSchema, Pagination$outboundSchema, } from "./pagination.js";
/**
 * The type of this cosmos doc instance, if blank, assume secret.
 */
export const ListSharedEnvVariableType = {
    Encrypted: "encrypted",
    Sensitive: "sensitive",
    System: "system",
    Plain: "plain",
};
/**
 * environments this env variable targets
 */
export const ListSharedEnvVariableTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/** @internal */
export const ListSharedEnvVariableRequest$inboundSchema = z.object({
    search: z.string().optional(),
    projectId: z.string().optional(),
    ids: z.string().optional(),
    exclude_idsQueryParameter: z.string().optional(),
    "exclude-idsQueryParameter1": z.string().optional(),
    exclude_projectIdQueryParameter: z.string().optional(),
    "exclude-projectIdQueryParameter1": z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "exclude_idsQueryParameter": "excludeIdsQueryParameter",
        "exclude-idsQueryParameter1": "excludeIdsQueryParameter1",
        "exclude_projectIdQueryParameter": "excludeProjectIdQueryParameter",
        "exclude-projectIdQueryParameter1": "excludeProjectIdQueryParameter1",
    });
});
/** @internal */
export const ListSharedEnvVariableRequest$outboundSchema = z.object({
    search: z.string().optional(),
    projectId: z.string().optional(),
    ids: z.string().optional(),
    excludeIdsQueryParameter: z.string().optional(),
    excludeIdsQueryParameter1: z.string().optional(),
    excludeProjectIdQueryParameter: z.string().optional(),
    excludeProjectIdQueryParameter1: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        excludeIdsQueryParameter: "exclude_idsQueryParameter",
        excludeIdsQueryParameter1: "exclude-idsQueryParameter1",
        excludeProjectIdQueryParameter: "exclude_projectIdQueryParameter",
        excludeProjectIdQueryParameter1: "exclude-projectIdQueryParameter1",
    });
});
export function listSharedEnvVariableRequestToJSON(listSharedEnvVariableRequest) {
    return JSON.stringify(ListSharedEnvVariableRequest$outboundSchema.parse(listSharedEnvVariableRequest));
}
export function listSharedEnvVariableRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListSharedEnvVariableRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListSharedEnvVariableRequest' from JSON`);
}
/** @internal */
export const ListSharedEnvVariableType$inboundSchema = z.nativeEnum(ListSharedEnvVariableType);
/** @internal */
export const ListSharedEnvVariableType$outboundSchema = ListSharedEnvVariableType$inboundSchema;
/** @internal */
export const ListSharedEnvVariableTarget$inboundSchema = z.nativeEnum(ListSharedEnvVariableTarget);
/** @internal */
export const ListSharedEnvVariableTarget$outboundSchema = ListSharedEnvVariableTarget$inboundSchema;
/** @internal */
export const Data$inboundSchema = z
    .object({
    created: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    key: z.string().optional(),
    ownerId: z.nullable(z.string()).optional(),
    id: z.string().optional(),
    createdBy: z.nullable(z.string()).optional(),
    deletedBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    deletedAt: z.number().optional(),
    updatedAt: z.number().optional(),
    value: z.string().optional(),
    projectId: z.array(z.string()).optional(),
    type: ListSharedEnvVariableType$inboundSchema.optional(),
    target: z.array(ListSharedEnvVariableTarget$inboundSchema).optional(),
    applyToAllCustomEnvironments: z.boolean().optional(),
    decrypted: z.boolean().optional(),
    comment: z.string().optional(),
    lastEditedByDisplayName: z.string().optional(),
});
/** @internal */
export const Data$outboundSchema = z.object({
    created: z.date().transform(v => v.toISOString()).optional(),
    key: z.string().optional(),
    ownerId: z.nullable(z.string()).optional(),
    id: z.string().optional(),
    createdBy: z.nullable(z.string()).optional(),
    deletedBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    deletedAt: z.number().optional(),
    updatedAt: z.number().optional(),
    value: z.string().optional(),
    projectId: z.array(z.string()).optional(),
    type: ListSharedEnvVariableType$outboundSchema.optional(),
    target: z.array(ListSharedEnvVariableTarget$outboundSchema).optional(),
    applyToAllCustomEnvironments: z.boolean().optional(),
    decrypted: z.boolean().optional(),
    comment: z.string().optional(),
    lastEditedByDisplayName: z.string().optional(),
});
export function dataToJSON(data) {
    return JSON.stringify(Data$outboundSchema.parse(data));
}
export function dataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Data$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Data' from JSON`);
}
/** @internal */
export const ListSharedEnvVariableResponseBody$inboundSchema = z.object({
    data: z.array(z.lazy(() => Data$inboundSchema)),
    pagination: Pagination$inboundSchema,
});
/** @internal */
export const ListSharedEnvVariableResponseBody$outboundSchema = z.object({
    data: z.array(z.lazy(() => Data$outboundSchema)),
    pagination: Pagination$outboundSchema,
});
export function listSharedEnvVariableResponseBodyToJSON(listSharedEnvVariableResponseBody) {
    return JSON.stringify(ListSharedEnvVariableResponseBody$outboundSchema.parse(listSharedEnvVariableResponseBody));
}
export function listSharedEnvVariableResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ListSharedEnvVariableResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListSharedEnvVariableResponseBody' from JSON`);
}
