// @vercel/sdk/esm/models/readaccessgroupop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const ReadAccessGroupTeamPermissions = {
    IntegrationManager: "IntegrationManager",
    CreateProject: "CreateProject",
    FullProductionDeployment: "FullProductionDeployment",
    UsageViewer: "UsageViewer",
    EnvVariableManager: "EnvVariableManager",
    EnvironmentManager: "EnvironmentManager",
    V0Builder: "V0Builder",
    V0Chatter: "V0Chatter",
    V0Viewer: "V0Viewer",
};
export const ReadAccessGroupEntitlements = {
    V0: "v0",
};
/** @internal */
export const ReadAccessGroupRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const ReadAccessGroupRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function readAccessGroupRequestToJSON(readAccessGroupRequest) {
    return JSON.stringify(ReadAccessGroupRequest$outboundSchema.parse(readAccessGroupRequest));
}
export function readAccessGroupRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReadAccessGroupRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReadAccessGroupRequest' from JSON`);
}
/** @internal */
export const ReadAccessGroupTeamPermissions$inboundSchema = z.nativeEnum(ReadAccessGroupTeamPermissions);
/** @internal */
export const ReadAccessGroupTeamPermissions$outboundSchema = ReadAccessGroupTeamPermissions$inboundSchema;
/** @internal */
export const ReadAccessGroupEntitlements$inboundSchema = z.nativeEnum(ReadAccessGroupEntitlements);
/** @internal */
export const ReadAccessGroupEntitlements$outboundSchema = ReadAccessGroupEntitlements$inboundSchema;
/** @internal */
export const ReadAccessGroupResponseBody$inboundSchema = z.object({
    teamPermissions: z.array(ReadAccessGroupTeamPermissions$inboundSchema)
        .optional(),
    entitlements: z.array(ReadAccessGroupEntitlements$inboundSchema).optional(),
    isDsyncManaged: z.boolean(),
    name: z.string(),
    createdAt: z.string(),
    teamId: z.string(),
    updatedAt: z.string(),
    accessGroupId: z.string(),
    membersCount: z.number(),
    projectsCount: z.number(),
    teamRoles: z.array(z.string()).optional(),
});
/** @internal */
export const ReadAccessGroupResponseBody$outboundSchema = z.object({
    teamPermissions: z.array(ReadAccessGroupTeamPermissions$outboundSchema)
        .optional(),
    entitlements: z.array(ReadAccessGroupEntitlements$outboundSchema).optional(),
    isDsyncManaged: z.boolean(),
    name: z.string(),
    createdAt: z.string(),
    teamId: z.string(),
    updatedAt: z.string(),
    accessGroupId: z.string(),
    membersCount: z.number(),
    projectsCount: z.number(),
    teamRoles: z.array(z.string()).optional(),
});
export function readAccessGroupResponseBodyToJSON(readAccessGroupResponseBody) {
    return JSON.stringify(ReadAccessGroupResponseBody$outboundSchema.parse(readAccessGroupResponseBody));
}
export function readAccessGroupResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ReadAccessGroupResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ReadAccessGroupResponseBody' from JSON`);
}
