// @vercel/sdk/esm/models/removeprojectenvop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
export const RemoveProjectEnvTargetProjectsResponse2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const RemoveProjectEnvTargetProjectsResponse1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const RemoveProjectEnvResponseBodyProjectsResponseType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type = {
    PostgresDatabase: "postgres-database",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type = {
    PostgresPassword: "postgres-password",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type = {
    PostgresHost: "postgres-host",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type = {
    PostgresUser: "postgres-user",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type = {
    PostgresUrl: "postgres-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type = {
    RedisUrl: "redis-url",
};
export const RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type = {
    FlagsSecret: "flags-secret",
};
export const RemoveProjectEnvTargetProjects2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const RemoveProjectEnvTargetProjects1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const RemoveProjectEnvResponseBodyProjectsType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type = {
    PostgresDatabase: "postgres-database",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type = {
    PostgresPassword: "postgres-password",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type = {
    PostgresHost: "postgres-host",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type = {
    PostgresUser: "postgres-user",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type = {
    PostgresUrl: "postgres-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type = {
    RedisUrl: "redis-url",
};
export const RemoveProjectEnvResponseBodyProjectsResponse200Type = {
    FlagsSecret: "flags-secret",
};
export const RemoveProjectEnvTarget2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const RemoveProjectEnvTarget1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const RemoveProjectEnvResponseBodyType = {
    System: "system",
    Encrypted: "encrypted",
    Plain: "plain",
    Sensitive: "sensitive",
    Secret: "secret",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type = {
    FlagsConnectionString: "flags-connection-string",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type = {
    IntegrationStoreSecret: "integration-store-secret",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type = {
    PostgresUrlNoSsl: "postgres-url-no-ssl",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type = {
    PostgresDatabase: "postgres-database",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type = {
    PostgresPassword: "postgres-password",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type = {
    PostgresHost: "postgres-host",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type = {
    PostgresUser: "postgres-user",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type = {
    PostgresPrismaUrl: "postgres-prisma-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type = {
    PostgresUrlNonPooling: "postgres-url-non-pooling",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType = {
    PostgresUrl: "postgres-url",
};
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType = {
    BlobReadWriteToken: "blob-read-write-token",
};
export const RemoveProjectEnvContentHintProjectsResponse200Type = {
    RedisRestApiReadOnlyToken: "redis-rest-api-read-only-token",
};
export const RemoveProjectEnvContentHintProjectsResponseType = {
    RedisRestApiToken: "redis-rest-api-token",
};
export const RemoveProjectEnvContentHintProjectsType = {
    RedisRestApiUrl: "redis-rest-api-url",
};
export const RemoveProjectEnvContentHintType = {
    RedisUrl: "redis-url",
};
export const RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType = {
    FlagsSecret: "flags-secret",
};
/** @internal */
export const RemoveProjectEnvRequest$inboundSchema = z.object({
    idOrName: z.string(),
    id: z.string(),
    customEnvironmentId: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const RemoveProjectEnvRequest$outboundSchema = z.object({
    idOrName: z.string(),
    id: z.string(),
    customEnvironmentId: z.string().optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function removeProjectEnvRequestToJSON(removeProjectEnvRequest) {
    return JSON.stringify(RemoveProjectEnvRequest$outboundSchema.parse(removeProjectEnvRequest));
}
export function removeProjectEnvRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvRequest' from JSON`);
}
/** @internal */
export const RemoveProjectEnvTargetProjectsResponse2$inboundSchema = z
    .nativeEnum(RemoveProjectEnvTargetProjectsResponse2);
/** @internal */
export const RemoveProjectEnvTargetProjectsResponse2$outboundSchema = RemoveProjectEnvTargetProjectsResponse2$inboundSchema;
/** @internal */
export const RemoveProjectEnvTargetProjectsResponse1$inboundSchema = z
    .nativeEnum(RemoveProjectEnvTargetProjectsResponse1);
/** @internal */
export const RemoveProjectEnvTargetProjectsResponse1$outboundSchema = RemoveProjectEnvTargetProjectsResponse1$inboundSchema;
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseTarget$inboundSchema = z.union([
    z.array(RemoveProjectEnvTargetProjectsResponse1$inboundSchema),
    RemoveProjectEnvTargetProjectsResponse2$inboundSchema,
]);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseTarget$outboundSchema = z.union([
    z.array(RemoveProjectEnvTargetProjectsResponse1$outboundSchema),
    RemoveProjectEnvTargetProjectsResponse2$outboundSchema,
]);
export function removeProjectEnvResponseBodyProjectsResponseTargetToJSON(removeProjectEnvResponseBodyProjectsResponseTarget) {
    return JSON.stringify(RemoveProjectEnvResponseBodyProjectsResponseTarget$outboundSchema.parse(removeProjectEnvResponseBodyProjectsResponseTarget));
}
export function removeProjectEnvResponseBodyProjectsResponseTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyProjectsResponseTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyProjectsResponseTarget' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseType$inboundSchema = z
    .nativeEnum(RemoveProjectEnvResponseBodyProjectsResponseType);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseType$outboundSchema = RemoveProjectEnvResponseBodyProjectsResponseType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse15$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse15$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody315Type$outboundSchema,
    projectId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse15ToJSON(removeProjectEnvContentHintProjectsResponse15) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse15$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse15));
}
export function removeProjectEnvContentHintProjectsResponse15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse15' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse14$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse14$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody314Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse14ToJSON(removeProjectEnvContentHintProjectsResponse14) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse14$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse14));
}
export function removeProjectEnvContentHintProjectsResponse14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse14' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse13$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse13$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody313Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse13ToJSON(removeProjectEnvContentHintProjectsResponse13) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse13$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse13));
}
export function removeProjectEnvContentHintProjectsResponse13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse13' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse12$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse12$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody312Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse12ToJSON(removeProjectEnvContentHintProjectsResponse12) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse12$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse12));
}
export function removeProjectEnvContentHintProjectsResponse12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse12' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse11$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse11$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody311Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse11ToJSON(removeProjectEnvContentHintProjectsResponse11) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse11$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse11));
}
export function removeProjectEnvContentHintProjectsResponse11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse11' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse10$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse10$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody310Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse10ToJSON(removeProjectEnvContentHintProjectsResponse10) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse10$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse10));
}
export function removeProjectEnvContentHintProjectsResponse10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse10' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse9$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse9$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody39Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse9ToJSON(removeProjectEnvContentHintProjectsResponse9) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse9$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse9));
}
export function removeProjectEnvContentHintProjectsResponse9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse9' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse8$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse8$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody38Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse8ToJSON(removeProjectEnvContentHintProjectsResponse8) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse8$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse8));
}
export function removeProjectEnvContentHintProjectsResponse8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse8' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse7$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse7$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody37Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse7ToJSON(removeProjectEnvContentHintProjectsResponse7) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse7$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse7));
}
export function removeProjectEnvContentHintProjectsResponse7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse7' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse6$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse6$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody36Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse6ToJSON(removeProjectEnvContentHintProjectsResponse6) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse6$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse6));
}
export function removeProjectEnvContentHintProjectsResponse6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse6' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse5$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse5$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody35Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse5ToJSON(removeProjectEnvContentHintProjectsResponse5) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse5$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse5));
}
export function removeProjectEnvContentHintProjectsResponse5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse5' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse4$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse4$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody34Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse4ToJSON(removeProjectEnvContentHintProjectsResponse4) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse4$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse4));
}
export function removeProjectEnvContentHintProjectsResponse4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse4' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse3$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse3$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody33Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse3ToJSON(removeProjectEnvContentHintProjectsResponse3) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse3$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse3));
}
export function removeProjectEnvContentHintProjectsResponse3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse3' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse2$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse2$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody32Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse2ToJSON(removeProjectEnvContentHintProjectsResponse2) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse2$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse2));
}
export function removeProjectEnvContentHintProjectsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse2' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse1$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse1$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody3Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjectsResponse1ToJSON(removeProjectEnvContentHintProjectsResponse1) {
    return JSON.stringify(RemoveProjectEnvContentHintProjectsResponse1$outboundSchema.parse(removeProjectEnvContentHintProjectsResponse1));
}
export function removeProjectEnvContentHintProjectsResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjectsResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjectsResponse1' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseContentHint$inboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse14$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse1$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse2$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse3$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse4$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse5$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse6$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse7$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse8$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse9$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse10$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse11$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse12$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse13$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse15$inboundSchema),
]);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseContentHint$outboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse14$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse1$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse2$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse3$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse4$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse5$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse6$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse7$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse8$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse9$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse10$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse11$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse12$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse13$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjectsResponse15$outboundSchema),
]);
export function removeProjectEnvResponseBodyProjectsResponseContentHintToJSON(removeProjectEnvResponseBodyProjectsResponseContentHint) {
    return JSON.stringify(RemoveProjectEnvResponseBodyProjectsResponseContentHint$outboundSchema
        .parse(removeProjectEnvResponseBodyProjectsResponseContentHint));
}
export function removeProjectEnvResponseBodyProjectsResponseContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyProjectsResponseContentHint$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyProjectsResponseContentHint' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$inboundSchema = z.nativeEnum(RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$outboundSchema = RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint$inboundSchema = z.object({
    type: RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint$outboundSchema = z.object({
    type: RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJson3Type$outboundSchema,
    encryptedValue: z.string(),
});
export function removeProjectEnvResponseBodyProjectsResponseInternalContentHintToJSON(removeProjectEnvResponseBodyProjectsResponseInternalContentHint) {
    return JSON.stringify(RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint$outboundSchema
        .parse(removeProjectEnvResponseBodyProjectsResponseInternalContentHint));
}
export function removeProjectEnvResponseBodyProjectsResponseInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBody3$inboundSchema = z.object({
    target: z.union([
        z.array(RemoveProjectEnvTargetProjectsResponse1$inboundSchema),
        RemoveProjectEnvTargetProjectsResponse2$inboundSchema,
    ]).optional(),
    type: RemoveProjectEnvResponseBodyProjectsResponseType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse14$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse1$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse2$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse3$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse4$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse5$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse6$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse7$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse8$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse9$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse10$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse11$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse12$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse13$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const RemoveProjectEnvResponseBody3$outboundSchema = z.object({
    target: z.union([
        z.array(RemoveProjectEnvTargetProjectsResponse1$outboundSchema),
        RemoveProjectEnvTargetProjectsResponse2$outboundSchema,
    ]).optional(),
    type: RemoveProjectEnvResponseBodyProjectsResponseType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse14$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse1$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse2$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse3$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse4$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse5$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse6$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse7$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse8$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse9$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse10$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse11$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse12$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse13$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjectsResponse15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => RemoveProjectEnvResponseBodyProjectsResponseInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function removeProjectEnvResponseBody3ToJSON(removeProjectEnvResponseBody3) {
    return JSON.stringify(RemoveProjectEnvResponseBody3$outboundSchema.parse(removeProjectEnvResponseBody3));
}
export function removeProjectEnvResponseBody3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBody3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBody3' from JSON`);
}
/** @internal */
export const RemoveProjectEnvTargetProjects2$inboundSchema = z.nativeEnum(RemoveProjectEnvTargetProjects2);
/** @internal */
export const RemoveProjectEnvTargetProjects2$outboundSchema = RemoveProjectEnvTargetProjects2$inboundSchema;
/** @internal */
export const RemoveProjectEnvTargetProjects1$inboundSchema = z.nativeEnum(RemoveProjectEnvTargetProjects1);
/** @internal */
export const RemoveProjectEnvTargetProjects1$outboundSchema = RemoveProjectEnvTargetProjects1$inboundSchema;
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsTarget$inboundSchema = z.union([
    z.array(RemoveProjectEnvTargetProjects1$inboundSchema),
    RemoveProjectEnvTargetProjects2$inboundSchema,
]);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsTarget$outboundSchema = z.union([
    z.array(RemoveProjectEnvTargetProjects1$outboundSchema),
    RemoveProjectEnvTargetProjects2$outboundSchema,
]);
export function removeProjectEnvResponseBodyProjectsTargetToJSON(removeProjectEnvResponseBodyProjectsTarget) {
    return JSON.stringify(RemoveProjectEnvResponseBodyProjectsTarget$outboundSchema.parse(removeProjectEnvResponseBodyProjectsTarget));
}
export function removeProjectEnvResponseBodyProjectsTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyProjectsTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyProjectsTarget' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsType$inboundSchema = z
    .nativeEnum(RemoveProjectEnvResponseBodyProjectsType);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsType$outboundSchema = RemoveProjectEnvResponseBodyProjectsType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects15$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects15$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody215Type$outboundSchema,
    projectId: z.string(),
});
export function removeProjectEnvContentHintProjects15ToJSON(removeProjectEnvContentHintProjects15) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects15$outboundSchema.parse(removeProjectEnvContentHintProjects15));
}
export function removeProjectEnvContentHintProjects15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects15' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects14$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects14$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody214Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function removeProjectEnvContentHintProjects14ToJSON(removeProjectEnvContentHintProjects14) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects14$outboundSchema.parse(removeProjectEnvContentHintProjects14));
}
export function removeProjectEnvContentHintProjects14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects14' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects13$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects13$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody213Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects13ToJSON(removeProjectEnvContentHintProjects13) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects13$outboundSchema.parse(removeProjectEnvContentHintProjects13));
}
export function removeProjectEnvContentHintProjects13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects13' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects12$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects12$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody212Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects12ToJSON(removeProjectEnvContentHintProjects12) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects12$outboundSchema.parse(removeProjectEnvContentHintProjects12));
}
export function removeProjectEnvContentHintProjects12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects12' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects11$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects11$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody211Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects11ToJSON(removeProjectEnvContentHintProjects11) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects11$outboundSchema.parse(removeProjectEnvContentHintProjects11));
}
export function removeProjectEnvContentHintProjects11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects11' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects10$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects10$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody210Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects10ToJSON(removeProjectEnvContentHintProjects10) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects10$outboundSchema.parse(removeProjectEnvContentHintProjects10));
}
export function removeProjectEnvContentHintProjects10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects10' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects9$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects9$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody29Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects9ToJSON(removeProjectEnvContentHintProjects9) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects9$outboundSchema.parse(removeProjectEnvContentHintProjects9));
}
export function removeProjectEnvContentHintProjects9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects9' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects8$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects8$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody28Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects8ToJSON(removeProjectEnvContentHintProjects8) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects8$outboundSchema.parse(removeProjectEnvContentHintProjects8));
}
export function removeProjectEnvContentHintProjects8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects8' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects7$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects7$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody27Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects7ToJSON(removeProjectEnvContentHintProjects7) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects7$outboundSchema.parse(removeProjectEnvContentHintProjects7));
}
export function removeProjectEnvContentHintProjects7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects7' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects6$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects6$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody26Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects6ToJSON(removeProjectEnvContentHintProjects6) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects6$outboundSchema.parse(removeProjectEnvContentHintProjects6));
}
export function removeProjectEnvContentHintProjects6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects6' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects5$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects5$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody25Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects5ToJSON(removeProjectEnvContentHintProjects5) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects5$outboundSchema.parse(removeProjectEnvContentHintProjects5));
}
export function removeProjectEnvContentHintProjects5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects5' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects4$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects4$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody24Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects4ToJSON(removeProjectEnvContentHintProjects4) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects4$outboundSchema.parse(removeProjectEnvContentHintProjects4));
}
export function removeProjectEnvContentHintProjects4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects4' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects3$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects3$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody23Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects3ToJSON(removeProjectEnvContentHintProjects3) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects3$outboundSchema.parse(removeProjectEnvContentHintProjects3));
}
export function removeProjectEnvContentHintProjects3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects3' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects2$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects2$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody22Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects2ToJSON(removeProjectEnvContentHintProjects2) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects2$outboundSchema.parse(removeProjectEnvContentHintProjects2));
}
export function removeProjectEnvContentHintProjects2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects2' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjects1$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHintProjects1$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody2Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHintProjects1ToJSON(removeProjectEnvContentHintProjects1) {
    return JSON.stringify(RemoveProjectEnvContentHintProjects1$outboundSchema.parse(removeProjectEnvContentHintProjects1));
}
export function removeProjectEnvContentHintProjects1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHintProjects1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHintProjects1' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsContentHint$inboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvContentHintProjects14$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects1$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects2$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects3$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects4$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects5$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects6$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects7$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects8$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects9$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects10$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects11$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects12$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects13$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects15$inboundSchema),
]);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsContentHint$outboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvContentHintProjects14$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects1$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects2$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects3$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects4$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects5$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects6$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects7$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects8$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects9$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects10$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects11$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects12$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects13$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHintProjects15$outboundSchema),
]);
export function removeProjectEnvResponseBodyProjectsContentHintToJSON(removeProjectEnvResponseBodyProjectsContentHint) {
    return JSON.stringify(RemoveProjectEnvResponseBodyProjectsContentHint$outboundSchema.parse(removeProjectEnvResponseBodyProjectsContentHint));
}
export function removeProjectEnvResponseBodyProjectsContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyProjectsContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyProjectsContentHint' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponse200Type$inboundSchema = z.nativeEnum(RemoveProjectEnvResponseBodyProjectsResponse200Type);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponse200Type$outboundSchema = RemoveProjectEnvResponseBodyProjectsResponse200Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsInternalContentHint$inboundSchema = z.object({
    type: RemoveProjectEnvResponseBodyProjectsResponse200Type$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsInternalContentHint$outboundSchema = z.object({
    type: RemoveProjectEnvResponseBodyProjectsResponse200Type$outboundSchema,
    encryptedValue: z.string(),
});
export function removeProjectEnvResponseBodyProjectsInternalContentHintToJSON(removeProjectEnvResponseBodyProjectsInternalContentHint) {
    return JSON.stringify(RemoveProjectEnvResponseBodyProjectsInternalContentHint$outboundSchema
        .parse(removeProjectEnvResponseBodyProjectsInternalContentHint));
}
export function removeProjectEnvResponseBodyProjectsInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyProjectsInternalContentHint$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyProjectsInternalContentHint' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBody2$inboundSchema = z.object({
    system: z.boolean().optional(),
    target: z.union([
        z.array(RemoveProjectEnvTargetProjects1$inboundSchema),
        RemoveProjectEnvTargetProjects2$inboundSchema,
    ]).optional(),
    type: RemoveProjectEnvResponseBodyProjectsType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => RemoveProjectEnvContentHintProjects14$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects1$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects2$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects3$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects4$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects5$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects6$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects7$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects8$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects9$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects10$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects11$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects12$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects13$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => RemoveProjectEnvResponseBodyProjectsInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const RemoveProjectEnvResponseBody2$outboundSchema = z.object({
    system: z.boolean().optional(),
    target: z.union([
        z.array(RemoveProjectEnvTargetProjects1$outboundSchema),
        RemoveProjectEnvTargetProjects2$outboundSchema,
    ]).optional(),
    type: RemoveProjectEnvResponseBodyProjectsType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => RemoveProjectEnvContentHintProjects14$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects1$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects2$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects3$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects4$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects5$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects6$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects7$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects8$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects9$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects10$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects11$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects12$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects13$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHintProjects15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => RemoveProjectEnvResponseBodyProjectsInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function removeProjectEnvResponseBody2ToJSON(removeProjectEnvResponseBody2) {
    return JSON.stringify(RemoveProjectEnvResponseBody2$outboundSchema.parse(removeProjectEnvResponseBody2));
}
export function removeProjectEnvResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBody2' from JSON`);
}
/** @internal */
export const RemoveProjectEnvTarget2$inboundSchema = z.nativeEnum(RemoveProjectEnvTarget2);
/** @internal */
export const RemoveProjectEnvTarget2$outboundSchema = RemoveProjectEnvTarget2$inboundSchema;
/** @internal */
export const RemoveProjectEnvTarget1$inboundSchema = z.nativeEnum(RemoveProjectEnvTarget1);
/** @internal */
export const RemoveProjectEnvTarget1$outboundSchema = RemoveProjectEnvTarget1$inboundSchema;
/** @internal */
export const RemoveProjectEnvResponseBodyTarget$inboundSchema = z.union([
    z.array(RemoveProjectEnvTarget1$inboundSchema),
    RemoveProjectEnvTarget2$inboundSchema,
]);
/** @internal */
export const RemoveProjectEnvResponseBodyTarget$outboundSchema = z.union([
    z.array(RemoveProjectEnvTarget1$outboundSchema),
    RemoveProjectEnvTarget2$outboundSchema,
]);
export function removeProjectEnvResponseBodyTargetToJSON(removeProjectEnvResponseBodyTarget) {
    return JSON.stringify(RemoveProjectEnvResponseBodyTarget$outboundSchema.parse(removeProjectEnvResponseBodyTarget));
}
export function removeProjectEnvResponseBodyTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyTarget' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyType$inboundSchema = z.nativeEnum(RemoveProjectEnvResponseBodyType);
/** @internal */
export const RemoveProjectEnvResponseBodyType$outboundSchema = RemoveProjectEnvResponseBodyType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint15$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint15$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody115Type$outboundSchema,
    projectId: z.string(),
});
export function removeProjectEnvContentHint15ToJSON(removeProjectEnvContentHint15) {
    return JSON.stringify(RemoveProjectEnvContentHint15$outboundSchema.parse(removeProjectEnvContentHint15));
}
export function removeProjectEnvContentHint15FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint15$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint15' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint14$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$inboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint14$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody114Type$outboundSchema,
    storeId: z.string(),
    integrationId: z.string(),
    integrationProductId: z.string(),
    integrationConfigurationId: z.string(),
});
export function removeProjectEnvContentHint14ToJSON(removeProjectEnvContentHint14) {
    return JSON.stringify(RemoveProjectEnvContentHint14$outboundSchema.parse(removeProjectEnvContentHint14));
}
export function removeProjectEnvContentHint14FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint14$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint14' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint13$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint13$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody113Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint13ToJSON(removeProjectEnvContentHint13) {
    return JSON.stringify(RemoveProjectEnvContentHint13$outboundSchema.parse(removeProjectEnvContentHint13));
}
export function removeProjectEnvContentHint13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint13' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint12$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint12$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody112Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint12ToJSON(removeProjectEnvContentHint12) {
    return JSON.stringify(RemoveProjectEnvContentHint12$outboundSchema.parse(removeProjectEnvContentHint12));
}
export function removeProjectEnvContentHint12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint12' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint11$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint11$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody111Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint11ToJSON(removeProjectEnvContentHint11) {
    return JSON.stringify(RemoveProjectEnvContentHint11$outboundSchema.parse(removeProjectEnvContentHint11));
}
export function removeProjectEnvContentHint11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint11' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint10$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint10$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody110Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint10ToJSON(removeProjectEnvContentHint10) {
    return JSON.stringify(RemoveProjectEnvContentHint10$outboundSchema.parse(removeProjectEnvContentHint10));
}
export function removeProjectEnvContentHint10FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint10$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint10' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint9$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint9$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody19Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint9ToJSON(removeProjectEnvContentHint9) {
    return JSON.stringify(RemoveProjectEnvContentHint9$outboundSchema.parse(removeProjectEnvContentHint9));
}
export function removeProjectEnvContentHint9FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint9$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint9' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint8$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint8$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody18Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint8ToJSON(removeProjectEnvContentHint8) {
    return JSON.stringify(RemoveProjectEnvContentHint8$outboundSchema.parse(removeProjectEnvContentHint8));
}
export function removeProjectEnvContentHint8FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint8$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint8' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint7$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint7$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBody1Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint7ToJSON(removeProjectEnvContentHint7) {
    return JSON.stringify(RemoveProjectEnvContentHint7$outboundSchema.parse(removeProjectEnvContentHint7));
}
export function removeProjectEnvContentHint7FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint7$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint7' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint6$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint6$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONResponseBodyType$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint6ToJSON(removeProjectEnvContentHint6) {
    return JSON.stringify(RemoveProjectEnvContentHint6$outboundSchema.parse(removeProjectEnvContentHint6));
}
export function removeProjectEnvContentHint6FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint6$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint6' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint5$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint5$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200ApplicationJSONType$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint5ToJSON(removeProjectEnvContentHint5) {
    return JSON.stringify(RemoveProjectEnvContentHint5$outboundSchema.parse(removeProjectEnvContentHint5));
}
export function removeProjectEnvContentHint5FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint5$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint5' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200Type$inboundSchema = z
    .nativeEnum(RemoveProjectEnvContentHintProjectsResponse200Type);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponse200Type$outboundSchema = RemoveProjectEnvContentHintProjectsResponse200Type$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint4$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200Type$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint4$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponse200Type$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint4ToJSON(removeProjectEnvContentHint4) {
    return JSON.stringify(RemoveProjectEnvContentHint4$outboundSchema.parse(removeProjectEnvContentHint4));
}
export function removeProjectEnvContentHint4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint4' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponseType$inboundSchema = z
    .nativeEnum(RemoveProjectEnvContentHintProjectsResponseType);
/** @internal */
export const RemoveProjectEnvContentHintProjectsResponseType$outboundSchema = RemoveProjectEnvContentHintProjectsResponseType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint3$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponseType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint3$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsResponseType$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint3ToJSON(removeProjectEnvContentHint3) {
    return JSON.stringify(RemoveProjectEnvContentHint3$outboundSchema.parse(removeProjectEnvContentHint3));
}
export function removeProjectEnvContentHint3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint3' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintProjectsType$inboundSchema = z
    .nativeEnum(RemoveProjectEnvContentHintProjectsType);
/** @internal */
export const RemoveProjectEnvContentHintProjectsType$outboundSchema = RemoveProjectEnvContentHintProjectsType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint2$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint2$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintProjectsType$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint2ToJSON(removeProjectEnvContentHint2) {
    return JSON.stringify(RemoveProjectEnvContentHint2$outboundSchema.parse(removeProjectEnvContentHint2));
}
export function removeProjectEnvContentHint2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint2' from JSON`);
}
/** @internal */
export const RemoveProjectEnvContentHintType$inboundSchema = z.nativeEnum(RemoveProjectEnvContentHintType);
/** @internal */
export const RemoveProjectEnvContentHintType$outboundSchema = RemoveProjectEnvContentHintType$inboundSchema;
/** @internal */
export const RemoveProjectEnvContentHint1$inboundSchema = z.object({
    type: RemoveProjectEnvContentHintType$inboundSchema,
    storeId: z.string(),
});
/** @internal */
export const RemoveProjectEnvContentHint1$outboundSchema = z.object({
    type: RemoveProjectEnvContentHintType$outboundSchema,
    storeId: z.string(),
});
export function removeProjectEnvContentHint1ToJSON(removeProjectEnvContentHint1) {
    return JSON.stringify(RemoveProjectEnvContentHint1$outboundSchema.parse(removeProjectEnvContentHint1));
}
export function removeProjectEnvContentHint1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvContentHint1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvContentHint1' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyContentHint$inboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvContentHint14$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint1$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint2$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint3$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint4$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint5$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint6$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint7$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint8$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint9$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint10$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint11$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint12$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint13$inboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint15$inboundSchema),
]);
/** @internal */
export const RemoveProjectEnvResponseBodyContentHint$outboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvContentHint14$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint1$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint2$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint3$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint4$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint5$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint6$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint7$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint8$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint9$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint10$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint11$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint12$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint13$outboundSchema),
    z.lazy(() => RemoveProjectEnvContentHint15$outboundSchema),
]);
export function removeProjectEnvResponseBodyContentHintToJSON(removeProjectEnvResponseBodyContentHint) {
    return JSON.stringify(RemoveProjectEnvResponseBodyContentHint$outboundSchema.parse(removeProjectEnvResponseBodyContentHint));
}
export function removeProjectEnvResponseBodyContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyContentHint' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType);
/** @internal */
export const RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$outboundSchema = RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const RemoveProjectEnvResponseBodyInternalContentHint$inboundSchema = z.object({
    type: RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$inboundSchema,
    encryptedValue: z.string(),
});
/** @internal */
export const RemoveProjectEnvResponseBodyInternalContentHint$outboundSchema = z.object({
    type: RemoveProjectEnvResponseBodyProjectsResponse200ApplicationJSONType$outboundSchema,
    encryptedValue: z.string(),
});
export function removeProjectEnvResponseBodyInternalContentHintToJSON(removeProjectEnvResponseBodyInternalContentHint) {
    return JSON.stringify(RemoveProjectEnvResponseBodyInternalContentHint$outboundSchema.parse(removeProjectEnvResponseBodyInternalContentHint));
}
export function removeProjectEnvResponseBodyInternalContentHintFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBodyInternalContentHint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBodyInternalContentHint' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBody1$inboundSchema = z.object({
    target: z.union([
        z.array(RemoveProjectEnvTarget1$inboundSchema),
        RemoveProjectEnvTarget2$inboundSchema,
    ]).optional(),
    type: RemoveProjectEnvResponseBodyType$inboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => RemoveProjectEnvContentHint14$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint1$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint2$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint3$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint4$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint5$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint6$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint7$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint8$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint9$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint10$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint11$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint12$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint13$inboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint15$inboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => RemoveProjectEnvResponseBodyInternalContentHint$inboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
/** @internal */
export const RemoveProjectEnvResponseBody1$outboundSchema = z.object({
    target: z.union([
        z.array(RemoveProjectEnvTarget1$outboundSchema),
        RemoveProjectEnvTarget2$outboundSchema,
    ]).optional(),
    type: RemoveProjectEnvResponseBodyType$outboundSchema,
    sunsetSecretId: z.string().optional(),
    decrypted: z.boolean().optional(),
    value: z.string(),
    vsmValue: z.string().optional(),
    id: z.string().optional(),
    key: z.string(),
    configurationId: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    updatedAt: z.number().optional(),
    createdBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    gitBranch: z.string().optional(),
    edgeConfigId: z.nullable(z.string()).optional(),
    edgeConfigTokenId: z.nullable(z.string()).optional(),
    contentHint: z.nullable(z.union([
        z.lazy(() => RemoveProjectEnvContentHint14$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint1$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint2$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint3$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint4$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint5$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint6$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint7$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint8$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint9$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint10$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint11$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint12$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint13$outboundSchema),
        z.lazy(() => RemoveProjectEnvContentHint15$outboundSchema),
    ])).optional(),
    internalContentHint: z.nullable(z.lazy(() => RemoveProjectEnvResponseBodyInternalContentHint$outboundSchema)).optional(),
    comment: z.string().optional(),
    customEnvironmentIds: z.array(z.string()).optional(),
});
export function removeProjectEnvResponseBody1ToJSON(removeProjectEnvResponseBody1) {
    return JSON.stringify(RemoveProjectEnvResponseBody1$outboundSchema.parse(removeProjectEnvResponseBody1));
}
export function removeProjectEnvResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBody1' from JSON`);
}
/** @internal */
export const RemoveProjectEnvResponseBody$inboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvResponseBody2$inboundSchema),
    z.lazy(() => RemoveProjectEnvResponseBody3$inboundSchema),
    z.array(z.lazy(() => RemoveProjectEnvResponseBody1$inboundSchema)),
]);
/** @internal */
export const RemoveProjectEnvResponseBody$outboundSchema = z.union([
    z.lazy(() => RemoveProjectEnvResponseBody2$outboundSchema),
    z.lazy(() => RemoveProjectEnvResponseBody3$outboundSchema),
    z.array(z.lazy(() => RemoveProjectEnvResponseBody1$outboundSchema)),
]);
export function removeProjectEnvResponseBodyToJSON(removeProjectEnvResponseBody) {
    return JSON.stringify(RemoveProjectEnvResponseBody$outboundSchema.parse(removeProjectEnvResponseBody));
}
export function removeProjectEnvResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveProjectEnvResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveProjectEnvResponseBody' from JSON`);
}
