// @vercel/sdk/esm/models/removeteammemberop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const RemoveTeamMemberRequest$inboundSchema = z.object({
    uid: z.string(),
    newDefaultTeamId: z.string().optional(),
    teamId: z.string(),
});
/** @internal */
export const RemoveTeamMemberRequest$outboundSchema = z.object({
    uid: z.string(),
    newDefaultTeamId: z.string().optional(),
    teamId: z.string(),
});
export function removeTeamMemberRequestToJSON(removeTeamMemberRequest) {
    return JSON.stringify(RemoveTeamMemberRequest$outboundSchema.parse(removeTeamMemberRequest));
}
export function removeTeamMemberRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveTeamMemberRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveTeamMemberRequest' from JSON`);
}
/** @internal */
export const RemoveTeamMemberResponseBody$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const RemoveTeamMemberResponseBody$outboundSchema = z.object({
    id: z.string(),
});
export function removeTeamMemberResponseBodyToJSON(removeTeamMemberResponseBody) {
    return JSON.stringify(RemoveTeamMemberResponseBody$outboundSchema.parse(removeTeamMemberResponseBody));
}
export function removeTeamMemberResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoveTeamMemberResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoveTeamMemberResponseBody' from JSON`);
}
