// @vercel/sdk/esm/models/submitprepaymentbalancesop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const Balances$inboundSchema = z.object({
    resourceId: z.string().optional(),
    credit: z.string().optional(),
    nameLabel: z.string().optional(),
    currencyValueInCents: z.number(),
});
/** @internal */
export const Balances$outboundSchema = z.object({
    resourceId: z.string().optional(),
    credit: z.string().optional(),
    nameLabel: z.string().optional(),
    currencyValueInCents: z.number(),
});
export function balancesToJSON(balances) {
    return JSON.stringify(Balances$outboundSchema.parse(balances));
}
export function balancesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Balances$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Balances' from JSON`);
}
/** @internal */
export const SubmitPrepaymentBalancesRequestBody$inboundSchema = z.object({
    timestamp: z.string().datetime({ offset: true }).transform(v => new Date(v)),
    balances: z.array(z.lazy(() => Balances$inboundSchema)),
});
/** @internal */
export const SubmitPrepaymentBalancesRequestBody$outboundSchema = z.object({
    timestamp: z.date().transform(v => v.toISOString()),
    balances: z.array(z.lazy(() => Balances$outboundSchema)),
});
export function submitPrepaymentBalancesRequestBodyToJSON(submitPrepaymentBalancesRequestBody) {
    return JSON.stringify(SubmitPrepaymentBalancesRequestBody$outboundSchema.parse(submitPrepaymentBalancesRequestBody));
}
export function submitPrepaymentBalancesRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubmitPrepaymentBalancesRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubmitPrepaymentBalancesRequestBody' from JSON`);
}
/** @internal */
export const SubmitPrepaymentBalancesRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    RequestBody: z.lazy(() => SubmitPrepaymentBalancesRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const SubmitPrepaymentBalancesRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    requestBody: z.lazy(() => SubmitPrepaymentBalancesRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function submitPrepaymentBalancesRequestToJSON(submitPrepaymentBalancesRequest) {
    return JSON.stringify(SubmitPrepaymentBalancesRequest$outboundSchema.parse(submitPrepaymentBalancesRequest));
}
export function submitPrepaymentBalancesRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SubmitPrepaymentBalancesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SubmitPrepaymentBalancesRequest' from JSON`);
}
