// @vercel/sdk/esm/models/team.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
/**
 * The default redirect URI to use after successful SAML authentication.
 */
export const DefaultRedirectUri = {
    VercelCom: "vercel.com",
    V0Dev: "v0.dev",
    V0App: "v0.app",
};
export const Roles2 = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const TeamTeamRoles = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const TeamTeamPermissions = {
    IntegrationManager: "IntegrationManager",
    CreateProject: "CreateProject",
    FullProductionDeployment: "FullProductionDeployment",
    UsageViewer: "UsageViewer",
    EnvVariableManager: "EnvVariableManager",
    EnvironmentManager: "EnvironmentManager",
    V0Builder: "V0Builder",
    V0Chatter: "V0Chatter",
    V0Viewer: "V0Viewer",
};
/**
 * Whether toolbar is enabled on preview deployments
 */
export const EnablePreviewFeedback = {
    Default: "default",
    On: "on",
    Off: "off",
    OnForce: "on-force",
    OffForce: "off-force",
    DefaultForce: "default-force",
};
/**
 * Whether toolbar is enabled on production deployments
 */
export const EnableProductionFeedback = {
    Default: "default",
    On: "on",
    Off: "off",
    OnForce: "on-force",
    OffForce: "off-force",
    DefaultForce: "default-force",
};
/**
 * Sensitive environment variable policy for this team
 */
export const SensitiveEnvironmentVariablePolicy = {
    Default: "default",
    On: "on",
    Off: "off",
};
export const TeamRole = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const TeamMembershipTeamRoles = {
    Owner: "OWNER",
    Member: "MEMBER",
    Developer: "DEVELOPER",
    Security: "SECURITY",
    Billing: "BILLING",
    Viewer: "VIEWER",
    ViewerForPlus: "VIEWER_FOR_PLUS",
    Contributor: "CONTRIBUTOR",
};
export const TeamMembershipTeamPermissions = {
    IntegrationManager: "IntegrationManager",
    CreateProject: "CreateProject",
    FullProductionDeployment: "FullProductionDeployment",
    UsageViewer: "UsageViewer",
    EnvVariableManager: "EnvVariableManager",
    EnvironmentManager: "EnvironmentManager",
    V0Builder: "V0Builder",
    V0Chatter: "V0Chatter",
    V0Viewer: "V0Viewer",
};
export const Origin = {
    Link: "link",
    Saml: "saml",
    Mail: "mail",
    Import: "import",
    Teams: "teams",
    Github: "github",
    Gitlab: "gitlab",
    Bitbucket: "bitbucket",
    Dsync: "dsync",
    Feedback: "feedback",
    OrganizationTeams: "organization-teams",
};
/** @internal */
export const Connect$inboundSchema = z.object({
    enabled: z.boolean().optional(),
});
/** @internal */
export const Connect$outboundSchema = z.object({
    enabled: z.boolean().optional(),
});
export function connectToJSON(connect) {
    return JSON.stringify(Connect$outboundSchema.parse(connect));
}
export function connectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Connect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Connect' from JSON`);
}
/** @internal */
export const Connection$inboundSchema = z.object({
    type: z.string(),
    status: z.string(),
    state: z.string(),
    connectedAt: z.number(),
    lastReceivedWebhookEvent: z.number().optional(),
});
/** @internal */
export const Connection$outboundSchema = z.object({
    type: z.string(),
    status: z.string(),
    state: z.string(),
    connectedAt: z.number(),
    lastReceivedWebhookEvent: z.number().optional(),
});
export function connectionToJSON(connection) {
    return JSON.stringify(Connection$outboundSchema.parse(connection));
}
export function connectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Connection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Connection' from JSON`);
}
/** @internal */
export const Directory$inboundSchema = z.object({
    type: z.string(),
    state: z.string(),
    connectedAt: z.number(),
    lastReceivedWebhookEvent: z.number().optional(),
});
/** @internal */
export const Directory$outboundSchema = z.object({
    type: z.string(),
    state: z.string(),
    connectedAt: z.number(),
    lastReceivedWebhookEvent: z.number().optional(),
});
export function directoryToJSON(directory) {
    return JSON.stringify(Directory$outboundSchema.parse(directory));
}
export function directoryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Directory$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Directory' from JSON`);
}
/** @internal */
export const DefaultRedirectUri$inboundSchema = z.nativeEnum(DefaultRedirectUri);
/** @internal */
export const DefaultRedirectUri$outboundSchema = DefaultRedirectUri$inboundSchema;
/** @internal */
export const Roles2$inboundSchema = z
    .nativeEnum(Roles2);
/** @internal */
export const Roles2$outboundSchema = Roles2$inboundSchema;
/** @internal */
export const Roles1$inboundSchema = z
    .object({
    accessGroupId: z.string(),
});
/** @internal */
export const Roles1$outboundSchema = z.object({
    accessGroupId: z.string(),
});
export function roles1ToJSON(roles1) {
    return JSON.stringify(Roles1$outboundSchema.parse(roles1));
}
export function roles1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Roles1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Roles1' from JSON`);
}
/** @internal */
export const Roles$inboundSchema = z
    .union([z.lazy(() => Roles1$inboundSchema), Roles2$inboundSchema]);
/** @internal */
export const Roles$outboundSchema = z.union([z.lazy(() => Roles1$outboundSchema), Roles2$outboundSchema]);
export function rolesToJSON(roles) {
    return JSON.stringify(Roles$outboundSchema.parse(roles));
}
export function rolesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Roles$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Roles' from JSON`);
}
/** @internal */
export const Saml$inboundSchema = z
    .object({
    connection: z.lazy(() => Connection$inboundSchema).optional(),
    directory: z.lazy(() => Directory$inboundSchema).optional(),
    enforced: z.boolean(),
    defaultRedirectUri: DefaultRedirectUri$inboundSchema.optional(),
    roles: z.record(z.union([z.lazy(() => Roles1$inboundSchema), Roles2$inboundSchema])).optional(),
});
/** @internal */
export const Saml$outboundSchema = z.object({
    connection: z.lazy(() => Connection$outboundSchema).optional(),
    directory: z.lazy(() => Directory$outboundSchema).optional(),
    enforced: z.boolean(),
    defaultRedirectUri: DefaultRedirectUri$outboundSchema.optional(),
    roles: z.record(z.union([z.lazy(() => Roles1$outboundSchema), Roles2$outboundSchema])).optional(),
});
export function samlToJSON(saml) {
    return JSON.stringify(Saml$outboundSchema.parse(saml));
}
export function samlFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Saml$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Saml' from JSON`);
}
/** @internal */
export const TeamTeamRoles$inboundSchema = z.nativeEnum(TeamTeamRoles);
/** @internal */
export const TeamTeamRoles$outboundSchema = TeamTeamRoles$inboundSchema;
/** @internal */
export const TeamTeamPermissions$inboundSchema = z.nativeEnum(TeamTeamPermissions);
/** @internal */
export const TeamTeamPermissions$outboundSchema = TeamTeamPermissions$inboundSchema;
/** @internal */
export const DefaultRoles$inboundSchema = z.object({
    teamRoles: z.array(TeamTeamRoles$inboundSchema).optional(),
    teamPermissions: z.array(TeamTeamPermissions$inboundSchema).optional(),
});
/** @internal */
export const DefaultRoles$outboundSchema = z.object({
    teamRoles: z.array(TeamTeamRoles$outboundSchema).optional(),
    teamPermissions: z.array(TeamTeamPermissions$outboundSchema).optional(),
});
export function defaultRolesToJSON(defaultRoles) {
    return JSON.stringify(DefaultRoles$outboundSchema.parse(defaultRoles));
}
export function defaultRolesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DefaultRoles$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DefaultRoles' from JSON`);
}
/** @internal */
export const BuildEntitlements$inboundSchema = z.object({
    enhancedBuilds: z.boolean().optional(),
});
/** @internal */
export const BuildEntitlements$outboundSchema = z.object({
    enhancedBuilds: z.boolean().optional(),
});
export function buildEntitlementsToJSON(buildEntitlements) {
    return JSON.stringify(BuildEntitlements$outboundSchema.parse(buildEntitlements));
}
export function buildEntitlementsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BuildEntitlements$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BuildEntitlements' from JSON`);
}
/** @internal */
export const ResourceConfig$inboundSchema = z.object({
    concurrentBuilds: z.number().optional(),
    elasticConcurrencyEnabled: z.boolean().optional(),
    edgeConfigSize: z.number().optional(),
    edgeConfigs: z.number().optional(),
    kvDatabases: z.number().optional(),
    blobStores: z.number().optional(),
    postgresDatabases: z.number().optional(),
    buildEntitlements: z.lazy(() => BuildEntitlements$inboundSchema).optional(),
});
/** @internal */
export const ResourceConfig$outboundSchema = z.object({
    concurrentBuilds: z.number().optional(),
    elasticConcurrencyEnabled: z.boolean().optional(),
    edgeConfigSize: z.number().optional(),
    edgeConfigs: z.number().optional(),
    kvDatabases: z.number().optional(),
    blobStores: z.number().optional(),
    postgresDatabases: z.number().optional(),
    buildEntitlements: z.lazy(() => BuildEntitlements$outboundSchema).optional(),
});
export function resourceConfigToJSON(resourceConfig) {
    return JSON.stringify(ResourceConfig$outboundSchema.parse(resourceConfig));
}
export function resourceConfigFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResourceConfig$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResourceConfig' from JSON`);
}
/** @internal */
export const DisableHardAutoBlocks$inboundSchema = z.union([z.number(), z.boolean()]);
/** @internal */
export const DisableHardAutoBlocks$outboundSchema = z.union([z.number(), z.boolean()]);
export function disableHardAutoBlocksToJSON(disableHardAutoBlocks) {
    return JSON.stringify(DisableHardAutoBlocks$outboundSchema.parse(disableHardAutoBlocks));
}
export function disableHardAutoBlocksFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DisableHardAutoBlocks$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DisableHardAutoBlocks' from JSON`);
}
/** @internal */
export const RemoteCaching$inboundSchema = z.object({
    enabled: z.boolean().optional(),
});
/** @internal */
export const RemoteCaching$outboundSchema = z.object({
    enabled: z.boolean().optional(),
});
export function remoteCachingToJSON(remoteCaching) {
    return JSON.stringify(RemoteCaching$outboundSchema.parse(remoteCaching));
}
export function remoteCachingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RemoteCaching$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RemoteCaching' from JSON`);
}
/** @internal */
export const PasswordProtection$inboundSchema = z.object({
    deploymentType: z.string(),
});
/** @internal */
export const PasswordProtection$outboundSchema = z.object({
    deploymentType: z.string(),
});
export function passwordProtectionToJSON(passwordProtection) {
    return JSON.stringify(PasswordProtection$outboundSchema.parse(passwordProtection));
}
export function passwordProtectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PasswordProtection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PasswordProtection' from JSON`);
}
/** @internal */
export const SsoProtection$inboundSchema = z.object({
    deploymentType: z.string(),
});
/** @internal */
export const SsoProtection$outboundSchema = z.object({
    deploymentType: z.string(),
});
export function ssoProtectionToJSON(ssoProtection) {
    return JSON.stringify(SsoProtection$outboundSchema.parse(ssoProtection));
}
export function ssoProtectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SsoProtection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SsoProtection' from JSON`);
}
/** @internal */
export const DefaultDeploymentProtection$inboundSchema = z.object({
    passwordProtection: z.lazy(() => PasswordProtection$inboundSchema).optional(),
    ssoProtection: z.lazy(() => SsoProtection$inboundSchema).optional(),
});
/** @internal */
export const DefaultDeploymentProtection$outboundSchema = z.object({
    passwordProtection: z.lazy(() => PasswordProtection$outboundSchema)
        .optional(),
    ssoProtection: z.lazy(() => SsoProtection$outboundSchema).optional(),
});
export function defaultDeploymentProtectionToJSON(defaultDeploymentProtection) {
    return JSON.stringify(DefaultDeploymentProtection$outboundSchema.parse(defaultDeploymentProtection));
}
export function defaultDeploymentProtectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DefaultDeploymentProtection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DefaultDeploymentProtection' from JSON`);
}
/** @internal */
export const DefaultExpirationSettings$inboundSchema = z.object({
    expirationDays: z.number().optional(),
    expirationDaysProduction: z.number().optional(),
    expirationDaysCanceled: z.number().optional(),
    expirationDaysErrored: z.number().optional(),
    deploymentsToKeep: z.number().optional(),
});
/** @internal */
export const DefaultExpirationSettings$outboundSchema = z.object({
    expirationDays: z.number().optional(),
    expirationDaysProduction: z.number().optional(),
    expirationDaysCanceled: z.number().optional(),
    expirationDaysErrored: z.number().optional(),
    deploymentsToKeep: z.number().optional(),
});
export function defaultExpirationSettingsToJSON(defaultExpirationSettings) {
    return JSON.stringify(DefaultExpirationSettings$outboundSchema.parse(defaultExpirationSettings));
}
export function defaultExpirationSettingsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DefaultExpirationSettings$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DefaultExpirationSettings' from JSON`);
}
/** @internal */
export const EnablePreviewFeedback$inboundSchema = z.nativeEnum(EnablePreviewFeedback);
/** @internal */
export const EnablePreviewFeedback$outboundSchema = EnablePreviewFeedback$inboundSchema;
/** @internal */
export const EnableProductionFeedback$inboundSchema = z.nativeEnum(EnableProductionFeedback);
/** @internal */
export const EnableProductionFeedback$outboundSchema = EnableProductionFeedback$inboundSchema;
/** @internal */
export const SensitiveEnvironmentVariablePolicy$inboundSchema = z.nativeEnum(SensitiveEnvironmentVariablePolicy);
/** @internal */
export const SensitiveEnvironmentVariablePolicy$outboundSchema = SensitiveEnvironmentVariablePolicy$inboundSchema;
/** @internal */
export const IpBuckets$inboundSchema = z.object({
    bucket: z.string(),
    supportUntil: z.number().optional(),
});
/** @internal */
export const IpBuckets$outboundSchema = z.object({
    bucket: z.string(),
    supportUntil: z.number().optional(),
});
export function ipBucketsToJSON(ipBuckets) {
    return JSON.stringify(IpBuckets$outboundSchema.parse(ipBuckets));
}
export function ipBucketsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => IpBuckets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'IpBuckets' from JSON`);
}
/** @internal */
export const Entitlements$inboundSchema = z.object({
    entitlement: z.string(),
});
/** @internal */
export const Entitlements$outboundSchema = z.object({
    entitlement: z.string(),
});
export function entitlementsToJSON(entitlements) {
    return JSON.stringify(Entitlements$outboundSchema.parse(entitlements));
}
export function entitlementsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Entitlements$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Entitlements' from JSON`);
}
/** @internal */
export const TeamRole$inboundSchema = z
    .nativeEnum(TeamRole);
/** @internal */
export const TeamRole$outboundSchema = TeamRole$inboundSchema;
/** @internal */
export const TeamMembershipTeamRoles$inboundSchema = z.nativeEnum(TeamMembershipTeamRoles);
/** @internal */
export const TeamMembershipTeamRoles$outboundSchema = TeamMembershipTeamRoles$inboundSchema;
/** @internal */
export const TeamMembershipTeamPermissions$inboundSchema = z.nativeEnum(TeamMembershipTeamPermissions);
/** @internal */
export const TeamMembershipTeamPermissions$outboundSchema = TeamMembershipTeamPermissions$inboundSchema;
/** @internal */
export const Origin$inboundSchema = z
    .nativeEnum(Origin);
/** @internal */
export const Origin$outboundSchema = Origin$inboundSchema;
/** @internal */
export const GitUserId$inboundSchema = z.union([z.string(), z.number()]);
/** @internal */
export const GitUserId$outboundSchema = z.union([z.string(), z.number()]);
export function gitUserIdToJSON(gitUserId) {
    return JSON.stringify(GitUserId$outboundSchema.parse(gitUserId));
}
export function gitUserIdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GitUserId$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GitUserId' from JSON`);
}
/** @internal */
export const JoinedFrom$inboundSchema = z.object({
    origin: Origin$inboundSchema,
    commitId: z.string().optional(),
    repoId: z.string().optional(),
    repoPath: z.string().optional(),
    gitUserId: z.union([z.string(), z.number()]).optional(),
    gitUserLogin: z.string().optional(),
    ssoUserId: z.string().optional(),
    ssoConnectedAt: z.number().optional(),
    idpUserId: z.string().optional(),
    dsyncUserId: z.string().optional(),
    dsyncConnectedAt: z.number().optional(),
});
/** @internal */
export const JoinedFrom$outboundSchema = z.object({
    origin: Origin$outboundSchema,
    commitId: z.string().optional(),
    repoId: z.string().optional(),
    repoPath: z.string().optional(),
    gitUserId: z.union([z.string(), z.number()]).optional(),
    gitUserLogin: z.string().optional(),
    ssoUserId: z.string().optional(),
    ssoConnectedAt: z.number().optional(),
    idpUserId: z.string().optional(),
    dsyncUserId: z.string().optional(),
    dsyncConnectedAt: z.number().optional(),
});
export function joinedFromToJSON(joinedFrom) {
    return JSON.stringify(JoinedFrom$outboundSchema.parse(joinedFrom));
}
export function joinedFromFromJSON(jsonString) {
    return safeParse(jsonString, (x) => JoinedFrom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'JoinedFrom' from JSON`);
}
/** @internal */
export const Membership$inboundSchema = z.object({
    uid: z.string().optional(),
    entitlements: z.array(z.lazy(() => Entitlements$inboundSchema)).optional(),
    teamId: z.string().optional(),
    confirmed: z.boolean(),
    accessRequestedAt: z.number().optional(),
    role: TeamRole$inboundSchema,
    teamRoles: z.array(TeamMembershipTeamRoles$inboundSchema).optional(),
    teamPermissions: z.array(TeamMembershipTeamPermissions$inboundSchema)
        .optional(),
    createdAt: z.number(),
    created: z.number(),
    joinedFrom: z.lazy(() => JoinedFrom$inboundSchema).optional(),
});
/** @internal */
export const Membership$outboundSchema = z.object({
    uid: z.string().optional(),
    entitlements: z.array(z.lazy(() => Entitlements$outboundSchema)).optional(),
    teamId: z.string().optional(),
    confirmed: z.boolean(),
    accessRequestedAt: z.number().optional(),
    role: TeamRole$outboundSchema,
    teamRoles: z.array(TeamMembershipTeamRoles$outboundSchema).optional(),
    teamPermissions: z.array(TeamMembershipTeamPermissions$outboundSchema)
        .optional(),
    createdAt: z.number(),
    created: z.number(),
    joinedFrom: z.lazy(() => JoinedFrom$outboundSchema).optional(),
});
export function membershipToJSON(membership) {
    return JSON.stringify(Membership$outboundSchema.parse(membership));
}
export function membershipFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Membership$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Membership' from JSON`);
}
/** @internal */
export const Team$inboundSchema = collectExtraKeys$(z.object({
    connect: z.lazy(() => Connect$inboundSchema).optional(),
    creatorId: z.string(),
    updatedAt: z.number(),
    emailDomain: z.nullable(z.string()).optional(),
    saml: z.lazy(() => Saml$inboundSchema).optional(),
    inviteCode: z.string().optional(),
    description: z.nullable(z.string()),
    defaultRoles: z.lazy(() => DefaultRoles$inboundSchema).optional(),
    stagingPrefix: z.string(),
    resourceConfig: z.lazy(() => ResourceConfig$inboundSchema).optional(),
    previewDeploymentSuffix: z.nullable(z.string()).optional(),
    platform: z.boolean().optional(),
    disableHardAutoBlocks: z.union([z.number(), z.boolean()]).optional(),
    remoteCaching: z.lazy(() => RemoteCaching$inboundSchema).optional(),
    defaultDeploymentProtection: z.lazy(() => DefaultDeploymentProtection$inboundSchema).optional(),
    defaultExpirationSettings: z.lazy(() => DefaultExpirationSettings$inboundSchema).optional(),
    enablePreviewFeedback: z.nullable(EnablePreviewFeedback$inboundSchema)
        .optional(),
    enableProductionFeedback: z.nullable(EnableProductionFeedback$inboundSchema).optional(),
    sensitiveEnvironmentVariablePolicy: z.nullable(SensitiveEnvironmentVariablePolicy$inboundSchema).optional(),
    hideIpAddresses: z.nullable(z.boolean()).optional(),
    hideIpAddressesInLogDrains: z.nullable(z.boolean()).optional(),
    ipBuckets: z.array(z.lazy(() => IpBuckets$inboundSchema)).optional(),
    id: z.string(),
    slug: z.string(),
    name: z.nullable(z.string()),
    avatar: z.nullable(z.string()),
    membership: z.lazy(() => Membership$inboundSchema),
    createdAt: z.number(),
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const Team$outboundSchema = z.object({
    connect: z.lazy(() => Connect$outboundSchema).optional(),
    creatorId: z.string(),
    updatedAt: z.number(),
    emailDomain: z.nullable(z.string()).optional(),
    saml: z.lazy(() => Saml$outboundSchema).optional(),
    inviteCode: z.string().optional(),
    description: z.nullable(z.string()),
    defaultRoles: z.lazy(() => DefaultRoles$outboundSchema).optional(),
    stagingPrefix: z.string(),
    resourceConfig: z.lazy(() => ResourceConfig$outboundSchema).optional(),
    previewDeploymentSuffix: z.nullable(z.string()).optional(),
    platform: z.boolean().optional(),
    disableHardAutoBlocks: z.union([z.number(), z.boolean()]).optional(),
    remoteCaching: z.lazy(() => RemoteCaching$outboundSchema).optional(),
    defaultDeploymentProtection: z.lazy(() => DefaultDeploymentProtection$outboundSchema).optional(),
    defaultExpirationSettings: z.lazy(() => DefaultExpirationSettings$outboundSchema).optional(),
    enablePreviewFeedback: z.nullable(EnablePreviewFeedback$outboundSchema)
        .optional(),
    enableProductionFeedback: z.nullable(EnableProductionFeedback$outboundSchema).optional(),
    sensitiveEnvironmentVariablePolicy: z.nullable(SensitiveEnvironmentVariablePolicy$outboundSchema).optional(),
    hideIpAddresses: z.nullable(z.boolean()).optional(),
    hideIpAddressesInLogDrains: z.nullable(z.boolean()).optional(),
    ipBuckets: z.array(z.lazy(() => IpBuckets$outboundSchema)).optional(),
    id: z.string(),
    slug: z.string(),
    name: z.nullable(z.string()),
    avatar: z.nullable(z.string()),
    membership: z.lazy(() => Membership$outboundSchema),
    createdAt: z.number(),
    additionalProperties: z.record(z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function teamToJSON(team) {
    return JSON.stringify(Team$outboundSchema.parse(team));
}
export function teamFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Team$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Team' from JSON`);
}
