// @vercel/sdk/esm/models/unlinksharedenvvariableop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const UnlinkSharedEnvVariableRequest$inboundSchema = z.object({
    id: z.string(),
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const UnlinkSharedEnvVariableRequest$outboundSchema = z.object({
    id: z.string(),
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function unlinkSharedEnvVariableRequestToJSON(unlinkSharedEnvVariableRequest) {
    return JSON.stringify(UnlinkSharedEnvVariableRequest$outboundSchema.parse(unlinkSharedEnvVariableRequest));
}
export function unlinkSharedEnvVariableRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnlinkSharedEnvVariableRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnlinkSharedEnvVariableRequest' from JSON`);
}
/** @internal */
export const UnlinkSharedEnvVariableResponseBody$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const UnlinkSharedEnvVariableResponseBody$outboundSchema = z.object({
    id: z.string(),
});
export function unlinkSharedEnvVariableResponseBodyToJSON(unlinkSharedEnvVariableResponseBody) {
    return JSON.stringify(UnlinkSharedEnvVariableResponseBody$outboundSchema.parse(unlinkSharedEnvVariableResponseBody));
}
export function unlinkSharedEnvVariableResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UnlinkSharedEnvVariableResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UnlinkSharedEnvVariableResponseBody' from JSON`);
}
