// @vercel/sdk/esm/models/updateattackchallengemodeop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const UpdateAttackChallengeModeRequestBody$inboundSchema = z.object({
    projectId: z.string(),
    attackModeEnabled: z.boolean(),
    attackModeActiveUntil: z.nullable(z.number()).optional(),
});
/** @internal */
export const UpdateAttackChallengeModeRequestBody$outboundSchema = z.object({
    projectId: z.string(),
    attackModeEnabled: z.boolean(),
    attackModeActiveUntil: z.nullable(z.number()).optional(),
});
export function updateAttackChallengeModeRequestBodyToJSON(updateAttackChallengeModeRequestBody) {
    return JSON.stringify(UpdateAttackChallengeModeRequestBody$outboundSchema.parse(updateAttackChallengeModeRequestBody));
}
export function updateAttackChallengeModeRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateAttackChallengeModeRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateAttackChallengeModeRequestBody' from JSON`);
}
/** @internal */
export const UpdateAttackChallengeModeRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateAttackChallengeModeRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateAttackChallengeModeRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateAttackChallengeModeRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateAttackChallengeModeRequestToJSON(updateAttackChallengeModeRequest) {
    return JSON.stringify(UpdateAttackChallengeModeRequest$outboundSchema.parse(updateAttackChallengeModeRequest));
}
export function updateAttackChallengeModeRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateAttackChallengeModeRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateAttackChallengeModeRequest' from JSON`);
}
/** @internal */
export const UpdateAttackChallengeModeResponseBody$inboundSchema = z.object({
    attackModeEnabled: z.boolean(),
    attackModeUpdatedAt: z.number(),
});
/** @internal */
export const UpdateAttackChallengeModeResponseBody$outboundSchema = z.object({
    attackModeEnabled: z.boolean(),
    attackModeUpdatedAt: z.number(),
});
export function updateAttackChallengeModeResponseBodyToJSON(updateAttackChallengeModeResponseBody) {
    return JSON.stringify(UpdateAttackChallengeModeResponseBody$outboundSchema.parse(updateAttackChallengeModeResponseBody));
}
export function updateAttackChallengeModeResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateAttackChallengeModeResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateAttackChallengeModeResponseBody' from JSON`);
}
