// @vercel/sdk/esm/models/updateresourceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { collectExtraKeys as collectExtraKeys$, safeParse, } from "../lib/schemas.js";
export const UpdateResourceOwnership = {
    Owned: "owned",
    Linked: "linked",
    Sandbox: "sandbox",
};
export const UpdateResourceStatus = {
    Ready: "ready",
    Pending: "pending",
    Onboarding: "onboarding",
    Suspended: "suspended",
    Resumed: "resumed",
    Uninstalled: "uninstalled",
    Error: "error",
};
export const UpdateResourceType = {
    Prepayment: "prepayment",
    Subscription: "subscription",
};
export const NotificationLevel = {
    Info: "info",
    Warn: "warn",
    Error: "error",
};
/** @internal */
export const UpdateResourceOwnership$inboundSchema = z.nativeEnum(UpdateResourceOwnership);
/** @internal */
export const UpdateResourceOwnership$outboundSchema = UpdateResourceOwnership$inboundSchema;
/** @internal */
export const UpdateResourceStatus$inboundSchema = z.nativeEnum(UpdateResourceStatus);
/** @internal */
export const UpdateResourceStatus$outboundSchema = UpdateResourceStatus$inboundSchema;
/** @internal */
export const UpdateResourceType$inboundSchema = z.nativeEnum(UpdateResourceType);
/** @internal */
export const UpdateResourceType$outboundSchema = UpdateResourceType$inboundSchema;
/** @internal */
export const UpdateResourceDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const UpdateResourceDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
export function updateResourceDetailsToJSON(updateResourceDetails) {
    return JSON.stringify(UpdateResourceDetails$outboundSchema.parse(updateResourceDetails));
}
export function updateResourceDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceDetails' from JSON`);
}
/** @internal */
export const UpdateResourceHighlightedDetails$inboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
/** @internal */
export const UpdateResourceHighlightedDetails$outboundSchema = z.object({
    label: z.string(),
    value: z.string().optional(),
});
export function updateResourceHighlightedDetailsToJSON(updateResourceHighlightedDetails) {
    return JSON.stringify(UpdateResourceHighlightedDetails$outboundSchema.parse(updateResourceHighlightedDetails));
}
export function updateResourceHighlightedDetailsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceHighlightedDetails$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceHighlightedDetails' from JSON`);
}
/** @internal */
export const UpdateResourceBillingPlan$inboundSchema = collectExtraKeys$(z.object({
    id: z.string(),
    type: UpdateResourceType$inboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => UpdateResourceDetails$inboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => UpdateResourceHighlightedDetails$inboundSchema)).optional(),
    effectiveDate: z.string().optional(),
}).catchall(z.any()), "additionalProperties", true);
/** @internal */
export const UpdateResourceBillingPlan$outboundSchema = z.object({
    id: z.string(),
    type: UpdateResourceType$outboundSchema,
    name: z.string(),
    description: z.string().optional(),
    paymentMethodRequired: z.boolean().optional(),
    cost: z.string().optional(),
    details: z.array(z.lazy(() => UpdateResourceDetails$outboundSchema))
        .optional(),
    highlightedDetails: z.array(z.lazy(() => UpdateResourceHighlightedDetails$outboundSchema)).optional(),
    effectiveDate: z.string().optional(),
    additionalProperties: z.record(z.any()).optional(),
}).transform((v) => {
    return {
        ...v.additionalProperties,
        ...remap$(v, {
            additionalProperties: null,
        }),
    };
});
export function updateResourceBillingPlanToJSON(updateResourceBillingPlan) {
    return JSON.stringify(UpdateResourceBillingPlan$outboundSchema.parse(updateResourceBillingPlan));
}
export function updateResourceBillingPlanFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceBillingPlan$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceBillingPlan' from JSON`);
}
/** @internal */
export const NotificationLevel$inboundSchema = z.nativeEnum(NotificationLevel);
/** @internal */
export const NotificationLevel$outboundSchema = NotificationLevel$inboundSchema;
/** @internal */
export const UpdateResourceNotification1$inboundSchema = z.object({
    level: NotificationLevel$inboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
/** @internal */
export const UpdateResourceNotification1$outboundSchema = z.object({
    level: NotificationLevel$outboundSchema,
    title: z.string(),
    message: z.string().optional(),
    href: z.string().optional(),
});
export function updateResourceNotification1ToJSON(updateResourceNotification1) {
    return JSON.stringify(UpdateResourceNotification1$outboundSchema.parse(updateResourceNotification1));
}
export function updateResourceNotification1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceNotification1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceNotification1' from JSON`);
}
/** @internal */
export const UpdateResourceNotification$inboundSchema = z.union([
    z.lazy(() => UpdateResourceNotification1$inboundSchema),
    z.string(),
]);
/** @internal */
export const UpdateResourceNotification$outboundSchema = z.union([
    z.lazy(() => UpdateResourceNotification1$outboundSchema),
    z.string(),
]);
export function updateResourceNotificationToJSON(updateResourceNotification) {
    return JSON.stringify(UpdateResourceNotification$outboundSchema.parse(updateResourceNotification));
}
export function updateResourceNotificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceNotification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceNotification' from JSON`);
}
/** @internal */
export const UpdateResourceSecretsMarketplaceEnvironmentOverrides$inboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
/** @internal */
export const UpdateResourceSecretsMarketplaceEnvironmentOverrides$outboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
export function updateResourceSecretsMarketplaceEnvironmentOverridesToJSON(updateResourceSecretsMarketplaceEnvironmentOverrides) {
    return JSON.stringify(UpdateResourceSecretsMarketplaceEnvironmentOverrides$outboundSchema.parse(updateResourceSecretsMarketplaceEnvironmentOverrides));
}
export function updateResourceSecretsMarketplaceEnvironmentOverridesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecretsMarketplaceEnvironmentOverrides$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecretsMarketplaceEnvironmentOverrides' from JSON`);
}
/** @internal */
export const SecretsSecrets$inboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => UpdateResourceSecretsMarketplaceEnvironmentOverrides$inboundSchema).optional(),
});
/** @internal */
export const SecretsSecrets$outboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => UpdateResourceSecretsMarketplaceEnvironmentOverrides$outboundSchema).optional(),
});
export function secretsSecretsToJSON(secretsSecrets) {
    return JSON.stringify(SecretsSecrets$outboundSchema.parse(secretsSecrets));
}
export function secretsSecretsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SecretsSecrets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SecretsSecrets' from JSON`);
}
/** @internal */
export const Secrets2$inboundSchema = z.object({
    secrets: z.array(z.lazy(() => SecretsSecrets$inboundSchema)),
    partial: z.boolean().optional(),
});
/** @internal */
export const Secrets2$outboundSchema = z.object({
    secrets: z.array(z.lazy(() => SecretsSecrets$outboundSchema)),
    partial: z.boolean().optional(),
});
export function secrets2ToJSON(secrets2) {
    return JSON.stringify(Secrets2$outboundSchema.parse(secrets2));
}
export function secrets2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Secrets2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Secrets2' from JSON`);
}
/** @internal */
export const SecretsEnvironmentOverrides$inboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
/** @internal */
export const SecretsEnvironmentOverrides$outboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
export function secretsEnvironmentOverridesToJSON(secretsEnvironmentOverrides) {
    return JSON.stringify(SecretsEnvironmentOverrides$outboundSchema.parse(secretsEnvironmentOverrides));
}
export function secretsEnvironmentOverridesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => SecretsEnvironmentOverrides$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'SecretsEnvironmentOverrides' from JSON`);
}
/** @internal */
export const Secrets1$inboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => SecretsEnvironmentOverrides$inboundSchema)
        .optional(),
});
/** @internal */
export const Secrets1$outboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => SecretsEnvironmentOverrides$outboundSchema)
        .optional(),
});
export function secrets1ToJSON(secrets1) {
    return JSON.stringify(Secrets1$outboundSchema.parse(secrets1));
}
export function secrets1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Secrets1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Secrets1' from JSON`);
}
/** @internal */
export const UpdateResourceSecrets$inboundSchema = z.union([
    z.lazy(() => Secrets2$inboundSchema),
    z.array(z.lazy(() => Secrets1$inboundSchema)),
]);
/** @internal */
export const UpdateResourceSecrets$outboundSchema = z.union([
    z.lazy(() => Secrets2$outboundSchema),
    z.array(z.lazy(() => Secrets1$outboundSchema)),
]);
export function updateResourceSecretsToJSON(updateResourceSecrets) {
    return JSON.stringify(UpdateResourceSecrets$outboundSchema.parse(updateResourceSecrets));
}
export function updateResourceSecretsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecrets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecrets' from JSON`);
}
/** @internal */
export const UpdateResourceRequestBody$inboundSchema = z.object({
    ownership: UpdateResourceOwnership$inboundSchema.optional(),
    name: z.string().optional(),
    status: UpdateResourceStatus$inboundSchema.optional(),
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => UpdateResourceBillingPlan$inboundSchema).optional(),
    notification: z.union([
        z.lazy(() => UpdateResourceNotification1$inboundSchema),
        z.string(),
    ]).optional(),
    extras: z.record(z.any()).optional(),
    secrets: z.union([
        z.lazy(() => Secrets2$inboundSchema),
        z.array(z.lazy(() => Secrets1$inboundSchema)),
    ]).optional(),
});
/** @internal */
export const UpdateResourceRequestBody$outboundSchema = z.object({
    ownership: UpdateResourceOwnership$outboundSchema.optional(),
    name: z.string().optional(),
    status: UpdateResourceStatus$outboundSchema.optional(),
    metadata: z.record(z.any()).optional(),
    billingPlan: z.lazy(() => UpdateResourceBillingPlan$outboundSchema)
        .optional(),
    notification: z.union([
        z.lazy(() => UpdateResourceNotification1$outboundSchema),
        z.string(),
    ]).optional(),
    extras: z.record(z.any()).optional(),
    secrets: z.union([
        z.lazy(() => Secrets2$outboundSchema),
        z.array(z.lazy(() => Secrets1$outboundSchema)),
    ]).optional(),
});
export function updateResourceRequestBodyToJSON(updateResourceRequestBody) {
    return JSON.stringify(UpdateResourceRequestBody$outboundSchema.parse(updateResourceRequestBody));
}
export function updateResourceRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceRequestBody' from JSON`);
}
/** @internal */
export const UpdateResourceRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
    RequestBody: z.lazy(() => UpdateResourceRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateResourceRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    resourceId: z.string(),
    requestBody: z.lazy(() => UpdateResourceRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateResourceRequestToJSON(updateResourceRequest) {
    return JSON.stringify(UpdateResourceRequest$outboundSchema.parse(updateResourceRequest));
}
export function updateResourceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceRequest' from JSON`);
}
/** @internal */
export const UpdateResourceResponseBody$inboundSchema = z.object({
    name: z.string(),
});
/** @internal */
export const UpdateResourceResponseBody$outboundSchema = z.object({
    name: z.string(),
});
export function updateResourceResponseBodyToJSON(updateResourceResponseBody) {
    return JSON.stringify(UpdateResourceResponseBody$outboundSchema.parse(updateResourceResponseBody));
}
export function updateResourceResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceResponseBody' from JSON`);
}
