// @vercel/sdk/esm/models/updatedrainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateDrainProjects = {
    Some: "some",
    All: "all",
};
export const UpdateDrainFilterSources = {
    Build: "build",
    Edge: "edge",
    Lambda: "lambda",
    Static: "static",
    External: "external",
    Firewall: "firewall",
    Redirect: "redirect",
};
export const UpdateDrainFilterEnvironments = {
    Production: "production",
    Preview: "preview",
};
export const UpdateDrainDeliveryDrainsEncoding = {
    Proto: "proto",
    Json: "json",
};
export const DeliveryCompression = {
    Gzip: "gzip",
    None: "none",
};
export const UpdateDrainDeliveryEncoding = {
    Json: "json",
    Ndjson: "ndjson",
};
/**
 * Environment to apply sampling to
 */
export const UpdateDrainEnv = {
    Production: "production",
    Preview: "preview",
};
export const UpdateDrainStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
};
export const UpdateDrainResponseBodyDrainsCreatedFrom = {
    SelfServed: "self-served",
    Integration: "integration",
};
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type = {
    Internal: "internal",
};
export const UpdateDrainDeliveryDrainsTarget = {
    VercelOtelTracesDb: "vercel-otel-traces-db",
};
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type = {
    Clickhouse: "clickhouse",
};
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType = {
    Otlphttp: "otlphttp",
};
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding = {
    Json: "json",
    Proto: "proto",
};
export const UpdateDrainSecretDrainsResponse200Kind = {
    IntegrationSecret: "INTEGRATION_SECRET",
};
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONType = {
    Http: "http",
};
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding = {
    Json: "json",
    Ndjson: "ndjson",
};
export const UpdateDrainDeliveryDrainsCompression = {
    Gzip: "gzip",
    None: "none",
};
export const UpdateDrainSecretDrainsResponseKind = {
    IntegrationSecret: "INTEGRATION_SECRET",
};
export const UpdateDrainResponseBodyDrainsType = {
    HeadSampling: "head_sampling",
};
export const UpdateDrainResponseBodyDrainsEnv = {
    Production: "production",
    Preview: "preview",
};
export const UpdateDrainResponseBodyDrainsStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
    Errored: "errored",
};
export const UpdateDrainResponseBodyDrainsDisabledReason = {
    DisabledByOwner: "disabled-by-owner",
    FeatureNotAvailable: "feature-not-available",
    AccountPlanDowngrade: "account-plan-downgrade",
    DisabledByAdmin: "disabled-by-admin",
};
export const UpdateDrainSourceDrainsResponse200Kind = {
    Integration: "integration",
};
export const UpdateDrainSourceDrainsResponseKind = {
    SelfServed: "self-served",
};
export const UpdateDrainFilterV2DrainsResponse200Version = {
    V2: "v2",
};
export const UpdateDrainFilterDrainsResponse200Type = {
    Odata: "odata",
};
export const UpdateDrainFilterDrainsResponseType = {
    Basic: "basic",
};
export const UpdateDrainFilterDrainsResponseSources = {
    Build: "build",
    Edge: "edge",
    Lambda: "lambda",
    Static: "static",
    External: "external",
    Firewall: "firewall",
    Redirect: "redirect",
};
export const UpdateDrainFilterDrainsResponseEnvironments = {
    Production: "production",
    Preview: "preview",
};
export const UpdateDrainFilterV2DrainsResponseVersion = {
    V1: "v1",
};
export const UpdateDrainResponseBodyFramework = {
    Blitzjs: "blitzjs",
    Nextjs: "nextjs",
    Gatsby: "gatsby",
    Remix: "remix",
    ReactRouter: "react-router",
    Astro: "astro",
    Hexo: "hexo",
    Eleventy: "eleventy",
    Docusaurus2: "docusaurus-2",
    Docusaurus: "docusaurus",
    Preact: "preact",
    Solidstart1: "solidstart-1",
    Solidstart: "solidstart",
    Dojo: "dojo",
    Ember: "ember",
    Vue: "vue",
    Scully: "scully",
    IonicAngular: "ionic-angular",
    Angular: "angular",
    Polymer: "polymer",
    Svelte: "svelte",
    Sveltekit: "sveltekit",
    Sveltekit1: "sveltekit-1",
    IonicReact: "ionic-react",
    CreateReactApp: "create-react-app",
    Gridsome: "gridsome",
    Umijs: "umijs",
    Sapper: "sapper",
    Saber: "saber",
    Stencil: "stencil",
    Nuxtjs: "nuxtjs",
    Redwoodjs: "redwoodjs",
    Hugo: "hugo",
    Jekyll: "jekyll",
    Brunch: "brunch",
    Middleman: "middleman",
    Zola: "zola",
    Hydrogen: "hydrogen",
    Vite: "vite",
    TanstackStart: "tanstack-start",
    Vitepress: "vitepress",
    Vuepress: "vuepress",
    Parcel: "parcel",
    Fastapi: "fastapi",
    Flask: "flask",
    Fasthtml: "fasthtml",
    SanityV3: "sanity-v3",
    Sanity: "sanity",
    Storybook: "storybook",
    Nitro: "nitro",
    Hono: "hono",
    Express: "express",
    H3: "h3",
    Nestjs: "nestjs",
    Elysia: "elysia",
    Fastify: "fastify",
    Xmcp: "xmcp",
};
export const UpdateDrainResponseBodyCreatedFrom = {
    SelfServed: "self-served",
    Integration: "integration",
};
export const UpdateDrainDeliveryDrainsResponse200Type = {
    Internal: "internal",
};
export const UpdateDrainDeliveryTarget = {
    VercelOtelTracesDb: "vercel-otel-traces-db",
};
export const UpdateDrainDeliveryDrainsResponseType = {
    Clickhouse: "clickhouse",
};
export const UpdateDrainDeliveryDrainsType = {
    Otlphttp: "otlphttp",
};
export const UpdateDrainDeliveryDrainsResponse200Encoding = {
    Json: "json",
    Proto: "proto",
};
export const UpdateDrainSecretDrainsKind = {
    IntegrationSecret: "INTEGRATION_SECRET",
};
export const UpdateDrainDeliveryType = {
    Http: "http",
};
export const UpdateDrainDeliveryDrainsResponseEncoding = {
    Json: "json",
    Ndjson: "ndjson",
};
export const UpdateDrainDeliveryCompression = {
    Gzip: "gzip",
    None: "none",
};
export const UpdateDrainSecretKind = {
    IntegrationSecret: "INTEGRATION_SECRET",
};
export const UpdateDrainResponseBodyType = {
    HeadSampling: "head_sampling",
};
export const UpdateDrainResponseBodyEnv = {
    Production: "production",
    Preview: "preview",
};
export const UpdateDrainResponseBodyStatus = {
    Enabled: "enabled",
    Disabled: "disabled",
    Errored: "errored",
};
export const UpdateDrainResponseBodyDisabledReason = {
    DisabledByOwner: "disabled-by-owner",
    FeatureNotAvailable: "feature-not-available",
    AccountPlanDowngrade: "account-plan-downgrade",
    DisabledByAdmin: "disabled-by-admin",
};
export const UpdateDrainSourceDrainsKind = {
    Integration: "integration",
};
export const UpdateDrainSourceKind = {
    SelfServed: "self-served",
};
export const UpdateDrainFilterV2DrainsVersion = {
    V2: "v2",
};
export const UpdateDrainFilterDrainsType = {
    Odata: "odata",
};
export const UpdateDrainFilterType = {
    Basic: "basic",
};
export const UpdateDrainFilterDrainsSources = {
    Build: "build",
    Edge: "edge",
    Lambda: "lambda",
    Static: "static",
    External: "external",
    Firewall: "firewall",
    Redirect: "redirect",
};
export const UpdateDrainFilterDrainsEnvironments = {
    Production: "production",
    Preview: "preview",
};
export const UpdateDrainFilterV2Version = {
    V1: "v1",
};
/** @internal */
export const UpdateDrainProjects$inboundSchema = z.nativeEnum(UpdateDrainProjects);
/** @internal */
export const UpdateDrainProjects$outboundSchema = UpdateDrainProjects$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrains2$inboundSchema = z.object({
    type: z.string(),
    text: z.string(),
});
/** @internal */
export const UpdateDrainFilterDrains2$outboundSchema = z.object({
    type: z.string(),
    text: z.string(),
});
export function updateDrainFilterDrains2ToJSON(updateDrainFilterDrains2) {
    return JSON.stringify(UpdateDrainFilterDrains2$outboundSchema.parse(updateDrainFilterDrains2));
}
export function updateDrainFilterDrains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrains2' from JSON`);
}
/** @internal */
export const FilterProject$inboundSchema = z.object({
    ids: z.array(z.string()).optional(),
});
/** @internal */
export const FilterProject$outboundSchema = z.object({
    ids: z.array(z.string()).optional(),
});
export function filterProjectToJSON(filterProject) {
    return JSON.stringify(FilterProject$outboundSchema.parse(filterProject));
}
export function filterProjectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilterProject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilterProject' from JSON`);
}
/** @internal */
export const UpdateDrainFilterSources$inboundSchema = z.nativeEnum(UpdateDrainFilterSources);
/** @internal */
export const UpdateDrainFilterSources$outboundSchema = UpdateDrainFilterSources$inboundSchema;
/** @internal */
export const FilterLog$inboundSchema = z.object({
    sources: z.array(UpdateDrainFilterSources$inboundSchema).optional(),
});
/** @internal */
export const FilterLog$outboundSchema = z.object({
    sources: z.array(UpdateDrainFilterSources$outboundSchema).optional(),
});
export function filterLogToJSON(filterLog) {
    return JSON.stringify(FilterLog$outboundSchema.parse(filterLog));
}
export function filterLogFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilterLog$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilterLog' from JSON`);
}
/** @internal */
export const UpdateDrainFilterEnvironments$inboundSchema = z.nativeEnum(UpdateDrainFilterEnvironments);
/** @internal */
export const UpdateDrainFilterEnvironments$outboundSchema = UpdateDrainFilterEnvironments$inboundSchema;
/** @internal */
export const FilterDeployment$inboundSchema = z.object({
    environments: z.array(UpdateDrainFilterEnvironments$inboundSchema).optional(),
});
/** @internal */
export const FilterDeployment$outboundSchema = z.object({
    environments: z.array(UpdateDrainFilterEnvironments$outboundSchema)
        .optional(),
});
export function filterDeploymentToJSON(filterDeployment) {
    return JSON.stringify(FilterDeployment$outboundSchema.parse(filterDeployment));
}
export function filterDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => FilterDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilterDeployment' from JSON`);
}
/** @internal */
export const UpdateDrainFilter1$inboundSchema = z.object({
    type: z.string(),
    project: z.lazy(() => FilterProject$inboundSchema).optional(),
    log: z.lazy(() => FilterLog$inboundSchema).optional(),
    deployment: z.lazy(() => FilterDeployment$inboundSchema).optional(),
});
/** @internal */
export const UpdateDrainFilter1$outboundSchema = z.object({
    type: z.string(),
    project: z.lazy(() => FilterProject$outboundSchema).optional(),
    log: z.lazy(() => FilterLog$outboundSchema).optional(),
    deployment: z.lazy(() => FilterDeployment$outboundSchema).optional(),
});
export function updateDrainFilter1ToJSON(updateDrainFilter1) {
    return JSON.stringify(UpdateDrainFilter1$outboundSchema.parse(updateDrainFilter1));
}
export function updateDrainFilter1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilter1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilter1' from JSON`);
}
/** @internal */
export const UpdateDrainFilterFilter$inboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterDrains2$inboundSchema),
    z.lazy(() => UpdateDrainFilter1$inboundSchema),
]);
/** @internal */
export const UpdateDrainFilterFilter$outboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterDrains2$outboundSchema),
    z.lazy(() => UpdateDrainFilter1$outboundSchema),
]);
export function updateDrainFilterFilterToJSON(updateDrainFilterFilter) {
    return JSON.stringify(UpdateDrainFilterFilter$outboundSchema.parse(updateDrainFilterFilter));
}
export function updateDrainFilterFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterFilter' from JSON`);
}
/** @internal */
export const UpdateDrainFilter2$inboundSchema = z.object({
    version: z.string(),
    filter: z.union([
        z.lazy(() => UpdateDrainFilterDrains2$inboundSchema),
        z.lazy(() => UpdateDrainFilter1$inboundSchema),
    ]),
});
/** @internal */
export const UpdateDrainFilter2$outboundSchema = z.object({
    version: z.string(),
    filter: z.union([
        z.lazy(() => UpdateDrainFilterDrains2$outboundSchema),
        z.lazy(() => UpdateDrainFilter1$outboundSchema),
    ]),
});
export function updateDrainFilter2ToJSON(updateDrainFilter2) {
    return JSON.stringify(UpdateDrainFilter2$outboundSchema.parse(updateDrainFilter2));
}
export function updateDrainFilter2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilter2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilter2' from JSON`);
}
/** @internal */
export const UpdateDrainFilter$inboundSchema = z.union([z.lazy(() => UpdateDrainFilter2$inboundSchema), z.string()]);
/** @internal */
export const UpdateDrainFilter$outboundSchema = z.union([z.lazy(() => UpdateDrainFilter2$outboundSchema), z.string()]);
export function updateDrainFilterToJSON(updateDrainFilter) {
    return JSON.stringify(UpdateDrainFilter$outboundSchema.parse(updateDrainFilter));
}
export function updateDrainFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilter' from JSON`);
}
/** @internal */
export const UpdateDrainSchemas$inboundSchema = z.object({
    version: z.string(),
});
/** @internal */
export const UpdateDrainSchemas$outboundSchema = z.object({
    version: z.string(),
});
export function updateDrainSchemasToJSON(updateDrainSchemas) {
    return JSON.stringify(UpdateDrainSchemas$outboundSchema.parse(updateDrainSchemas));
}
export function updateDrainSchemasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSchemas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSchemas' from JSON`);
}
/** @internal */
export const UpdateDrainEndpoint1$inboundSchema = z.object({
    traces: z.string(),
});
/** @internal */
export const UpdateDrainEndpoint1$outboundSchema = z.object({
    traces: z.string(),
});
export function updateDrainEndpoint1ToJSON(updateDrainEndpoint1) {
    return JSON.stringify(UpdateDrainEndpoint1$outboundSchema.parse(updateDrainEndpoint1));
}
export function updateDrainEndpoint1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainEndpoint1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainEndpoint1' from JSON`);
}
/** @internal */
export const DeliveryEndpoint$inboundSchema = z.lazy(() => UpdateDrainEndpoint1$inboundSchema);
/** @internal */
export const DeliveryEndpoint$outboundSchema = z.lazy(() => UpdateDrainEndpoint1$outboundSchema);
export function deliveryEndpointToJSON(deliveryEndpoint) {
    return JSON.stringify(DeliveryEndpoint$outboundSchema.parse(deliveryEndpoint));
}
export function deliveryEndpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeliveryEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeliveryEndpoint' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsEncoding$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsEncoding);
/** @internal */
export const UpdateDrainDeliveryDrainsEncoding$outboundSchema = UpdateDrainDeliveryDrainsEncoding$inboundSchema;
/** @internal */
export const UpdateDrainDelivery2$inboundSchema = z.object({
    type: z.string(),
    endpoint: z.lazy(() => UpdateDrainEndpoint1$inboundSchema),
    encoding: UpdateDrainDeliveryDrainsEncoding$inboundSchema,
    headers: z.record(z.string()),
    secret: z.string().optional(),
});
/** @internal */
export const UpdateDrainDelivery2$outboundSchema = z.object({
    type: z.string(),
    endpoint: z.lazy(() => UpdateDrainEndpoint1$outboundSchema),
    encoding: UpdateDrainDeliveryDrainsEncoding$outboundSchema,
    headers: z.record(z.string()),
    secret: z.string().optional(),
});
export function updateDrainDelivery2ToJSON(updateDrainDelivery2) {
    return JSON.stringify(UpdateDrainDelivery2$outboundSchema.parse(updateDrainDelivery2));
}
export function updateDrainDelivery2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDelivery2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDelivery2' from JSON`);
}
/** @internal */
export const DeliveryCompression$inboundSchema = z.nativeEnum(DeliveryCompression);
/** @internal */
export const DeliveryCompression$outboundSchema = DeliveryCompression$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryEncoding$inboundSchema = z.nativeEnum(UpdateDrainDeliveryEncoding);
/** @internal */
export const UpdateDrainDeliveryEncoding$outboundSchema = UpdateDrainDeliveryEncoding$inboundSchema;
/** @internal */
export const UpdateDrainDelivery1$inboundSchema = z.object({
    type: z.string(),
    endpoint: z.string(),
    compression: DeliveryCompression$inboundSchema.optional(),
    encoding: UpdateDrainDeliveryEncoding$inboundSchema,
    headers: z.record(z.string()),
    secret: z.string().optional(),
});
/** @internal */
export const UpdateDrainDelivery1$outboundSchema = z.object({
    type: z.string(),
    endpoint: z.string(),
    compression: DeliveryCompression$outboundSchema.optional(),
    encoding: UpdateDrainDeliveryEncoding$outboundSchema,
    headers: z.record(z.string()),
    secret: z.string().optional(),
});
export function updateDrainDelivery1ToJSON(updateDrainDelivery1) {
    return JSON.stringify(UpdateDrainDelivery1$outboundSchema.parse(updateDrainDelivery1));
}
export function updateDrainDelivery1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDelivery1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDelivery1' from JSON`);
}
/** @internal */
export const UpdateDrainDelivery$inboundSchema = z.union([
    z.lazy(() => UpdateDrainDelivery1$inboundSchema),
    z.lazy(() => UpdateDrainDelivery2$inboundSchema),
]);
/** @internal */
export const UpdateDrainDelivery$outboundSchema = z.union([
    z.lazy(() => UpdateDrainDelivery1$outboundSchema),
    z.lazy(() => UpdateDrainDelivery2$outboundSchema),
]);
export function updateDrainDeliveryToJSON(updateDrainDelivery) {
    return JSON.stringify(UpdateDrainDelivery$outboundSchema.parse(updateDrainDelivery));
}
export function updateDrainDeliveryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDelivery$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDelivery' from JSON`);
}
/** @internal */
export const UpdateDrainEnv$inboundSchema = z.nativeEnum(UpdateDrainEnv);
/** @internal */
export const UpdateDrainEnv$outboundSchema = UpdateDrainEnv$inboundSchema;
/** @internal */
export const UpdateDrainSampling$inboundSchema = z.object({
    type: z.string(),
    rate: z.number(),
    env: UpdateDrainEnv$inboundSchema.optional(),
    requestPath: z.string().optional(),
});
/** @internal */
export const UpdateDrainSampling$outboundSchema = z.object({
    type: z.string(),
    rate: z.number(),
    env: UpdateDrainEnv$outboundSchema.optional(),
    requestPath: z.string().optional(),
});
export function updateDrainSamplingToJSON(updateDrainSampling) {
    return JSON.stringify(UpdateDrainSampling$outboundSchema.parse(updateDrainSampling));
}
export function updateDrainSamplingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSampling$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSampling' from JSON`);
}
/** @internal */
export const UpdateDrainTransforms$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const UpdateDrainTransforms$outboundSchema = z.object({
    id: z.string(),
});
export function updateDrainTransformsToJSON(updateDrainTransforms) {
    return JSON.stringify(UpdateDrainTransforms$outboundSchema.parse(updateDrainTransforms));
}
export function updateDrainTransformsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainTransforms$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainTransforms' from JSON`);
}
/** @internal */
export const UpdateDrainStatus$inboundSchema = z.nativeEnum(UpdateDrainStatus);
/** @internal */
export const UpdateDrainStatus$outboundSchema = UpdateDrainStatus$inboundSchema;
/** @internal */
export const UpdateDrainSource2$inboundSchema = z.object({
    kind: z.string().default("self-served"),
});
/** @internal */
export const UpdateDrainSource2$outboundSchema = z.object({
    kind: z.string().default("self-served"),
});
export function updateDrainSource2ToJSON(updateDrainSource2) {
    return JSON.stringify(UpdateDrainSource2$outboundSchema.parse(updateDrainSource2));
}
export function updateDrainSource2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSource2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSource2' from JSON`);
}
/** @internal */
export const UpdateDrain13$inboundSchema = z.object({
    kind: z.string().default("integration"),
});
/** @internal */
export const UpdateDrain13$outboundSchema = z.object({
    kind: z.string().default("integration"),
});
export function updateDrain13ToJSON(updateDrain13) {
    return JSON.stringify(UpdateDrain13$outboundSchema.parse(updateDrain13));
}
export function updateDrain13FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrain13$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrain13' from JSON`);
}
/** @internal */
export const UpdateDrain12$inboundSchema = z.object({
    kind: z.string().default("integration"),
    resourceId: z.string(),
});
/** @internal */
export const UpdateDrain12$outboundSchema = z.object({
    kind: z.string().default("integration"),
    resourceId: z.string(),
});
export function updateDrain12ToJSON(updateDrain12) {
    return JSON.stringify(UpdateDrain12$outboundSchema.parse(updateDrain12));
}
export function updateDrain12FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrain12$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrain12' from JSON`);
}
/** @internal */
export const UpdateDrain11$inboundSchema = z.object({
    kind: z.string().default("integration"),
    externalResourceId: z.string(),
});
/** @internal */
export const UpdateDrain11$outboundSchema = z.object({
    kind: z.string().default("integration"),
    externalResourceId: z.string(),
});
export function updateDrain11ToJSON(updateDrain11) {
    return JSON.stringify(UpdateDrain11$outboundSchema.parse(updateDrain11));
}
export function updateDrain11FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrain11$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrain11' from JSON`);
}
/** @internal */
export const UpdateDrainSource1$inboundSchema = z.union([
    z.lazy(() => UpdateDrain11$inboundSchema),
    z.lazy(() => UpdateDrain12$inboundSchema),
    z.lazy(() => UpdateDrain13$inboundSchema),
]);
/** @internal */
export const UpdateDrainSource1$outboundSchema = z.union([
    z.lazy(() => UpdateDrain11$outboundSchema),
    z.lazy(() => UpdateDrain12$outboundSchema),
    z.lazy(() => UpdateDrain13$outboundSchema),
]);
export function updateDrainSource1ToJSON(updateDrainSource1) {
    return JSON.stringify(UpdateDrainSource1$outboundSchema.parse(updateDrainSource1));
}
export function updateDrainSource1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSource1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSource1' from JSON`);
}
/** @internal */
export const UpdateDrainSource$inboundSchema = z.union([
    z.union([
        z.lazy(() => UpdateDrain11$inboundSchema),
        z.lazy(() => UpdateDrain12$inboundSchema),
        z.lazy(() => UpdateDrain13$inboundSchema),
    ]),
    z.lazy(() => UpdateDrainSource2$inboundSchema),
]);
/** @internal */
export const UpdateDrainSource$outboundSchema = z.union([
    z.union([
        z.lazy(() => UpdateDrain11$outboundSchema),
        z.lazy(() => UpdateDrain12$outboundSchema),
        z.lazy(() => UpdateDrain13$outboundSchema),
    ]),
    z.lazy(() => UpdateDrainSource2$outboundSchema),
]);
export function updateDrainSourceToJSON(updateDrainSource) {
    return JSON.stringify(UpdateDrainSource$outboundSchema.parse(updateDrainSource));
}
export function updateDrainSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSource' from JSON`);
}
/** @internal */
export const UpdateDrainRequestBody$inboundSchema = z.object({
    name: z.string().optional(),
    projects: UpdateDrainProjects$inboundSchema.optional(),
    projectIds: z.nullable(z.array(z.string())).optional(),
    filter: z.nullable(z.union([z.lazy(() => UpdateDrainFilter2$inboundSchema), z.string()])).optional(),
    schemas: z.record(z.lazy(() => UpdateDrainSchemas$inboundSchema)).optional(),
    delivery: z.union([
        z.lazy(() => UpdateDrainDelivery1$inboundSchema),
        z.lazy(() => UpdateDrainDelivery2$inboundSchema),
    ]).optional(),
    sampling: z.nullable(z.array(z.lazy(() => UpdateDrainSampling$inboundSchema)))
        .optional(),
    transforms: z.nullable(z.array(z.lazy(() => UpdateDrainTransforms$inboundSchema))).optional(),
    status: UpdateDrainStatus$inboundSchema.optional(),
    source: z.union([
        z.union([
            z.lazy(() => UpdateDrain11$inboundSchema),
            z.lazy(() => UpdateDrain12$inboundSchema),
            z.lazy(() => UpdateDrain13$inboundSchema),
        ]),
        z.lazy(() => UpdateDrainSource2$inboundSchema),
    ]).optional(),
});
/** @internal */
export const UpdateDrainRequestBody$outboundSchema = z.object({
    name: z.string().optional(),
    projects: UpdateDrainProjects$outboundSchema.optional(),
    projectIds: z.nullable(z.array(z.string())).optional(),
    filter: z.nullable(z.union([z.lazy(() => UpdateDrainFilter2$outboundSchema), z.string()])).optional(),
    schemas: z.record(z.lazy(() => UpdateDrainSchemas$outboundSchema)).optional(),
    delivery: z.union([
        z.lazy(() => UpdateDrainDelivery1$outboundSchema),
        z.lazy(() => UpdateDrainDelivery2$outboundSchema),
    ]).optional(),
    sampling: z.nullable(z.array(z.lazy(() => UpdateDrainSampling$outboundSchema))).optional(),
    transforms: z.nullable(z.array(z.lazy(() => UpdateDrainTransforms$outboundSchema))).optional(),
    status: UpdateDrainStatus$outboundSchema.optional(),
    source: z.union([
        z.union([
            z.lazy(() => UpdateDrain11$outboundSchema),
            z.lazy(() => UpdateDrain12$outboundSchema),
            z.lazy(() => UpdateDrain13$outboundSchema),
        ]),
        z.lazy(() => UpdateDrainSource2$outboundSchema),
    ]).optional(),
});
export function updateDrainRequestBodyToJSON(updateDrainRequestBody) {
    return JSON.stringify(UpdateDrainRequestBody$outboundSchema.parse(updateDrainRequestBody));
}
export function updateDrainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainRequestBody' from JSON`);
}
/** @internal */
export const UpdateDrainRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateDrainRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateDrainRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateDrainRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateDrainRequestToJSON(updateDrainRequest) {
    return JSON.stringify(UpdateDrainRequest$outboundSchema.parse(updateDrainRequest));
}
export function updateDrainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainRequest' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsCreatedFrom$inboundSchema = z
    .nativeEnum(UpdateDrainResponseBodyDrainsCreatedFrom);
/** @internal */
export const UpdateDrainResponseBodyDrainsCreatedFrom$outboundSchema = UpdateDrainResponseBodyDrainsCreatedFrom$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyDrainsLog$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyDrainsLog$outboundSchema = z.object({});
export function updateDrainResponseBodyDrainsLogToJSON(updateDrainResponseBodyDrainsLog) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsLog$outboundSchema.parse(updateDrainResponseBodyDrainsLog));
}
export function updateDrainResponseBodyDrainsLogFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsLog$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsLog' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsTrace$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyDrainsTrace$outboundSchema = z.object({});
export function updateDrainResponseBodyDrainsTraceToJSON(updateDrainResponseBodyDrainsTrace) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsTrace$outboundSchema.parse(updateDrainResponseBodyDrainsTrace));
}
export function updateDrainResponseBodyDrainsTraceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsTrace$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsTrace' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsAnalytics$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyDrainsAnalytics$outboundSchema = z.object({});
export function updateDrainResponseBodyDrainsAnalyticsToJSON(updateDrainResponseBodyDrainsAnalytics) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsAnalytics$outboundSchema.parse(updateDrainResponseBodyDrainsAnalytics));
}
export function updateDrainResponseBodyDrainsAnalyticsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsAnalytics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsAnalytics' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsSpeedInsights$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyDrainsSpeedInsights$outboundSchema = z.object({});
export function updateDrainResponseBodyDrainsSpeedInsightsToJSON(updateDrainResponseBodyDrainsSpeedInsights) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsSpeedInsights$outboundSchema.parse(updateDrainResponseBodyDrainsSpeedInsights));
}
export function updateDrainResponseBodyDrainsSpeedInsightsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsSpeedInsights$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsSpeedInsights' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsSchemas$inboundSchema = z.object({
    log: z.lazy(() => UpdateDrainResponseBodyDrainsLog$inboundSchema).optional(),
    trace: z.lazy(() => UpdateDrainResponseBodyDrainsTrace$inboundSchema)
        .optional(),
    analytics: z.lazy(() => UpdateDrainResponseBodyDrainsAnalytics$inboundSchema)
        .optional(),
    speed_insights: z.lazy(() => UpdateDrainResponseBodyDrainsSpeedInsights$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "speed_insights": "speedInsights",
    });
});
/** @internal */
export const UpdateDrainResponseBodyDrainsSchemas$outboundSchema = z.object({
    log: z.lazy(() => UpdateDrainResponseBodyDrainsLog$outboundSchema).optional(),
    trace: z.lazy(() => UpdateDrainResponseBodyDrainsTrace$outboundSchema)
        .optional(),
    analytics: z.lazy(() => UpdateDrainResponseBodyDrainsAnalytics$outboundSchema)
        .optional(),
    speedInsights: z.lazy(() => UpdateDrainResponseBodyDrainsSpeedInsights$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        speedInsights: "speed_insights",
    });
});
export function updateDrainResponseBodyDrainsSchemasToJSON(updateDrainResponseBodyDrainsSchemas) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsSchemas$outboundSchema.parse(updateDrainResponseBodyDrainsSchemas));
}
export function updateDrainResponseBodyDrainsSchemasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsSchemas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsSchemas' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type$outboundSchema = UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrainsTarget$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsTarget);
/** @internal */
export const UpdateDrainDeliveryDrainsTarget$outboundSchema = UpdateDrainDeliveryDrainsTarget$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrains4$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type$inboundSchema,
    target: UpdateDrainDeliveryDrainsTarget$inboundSchema,
});
/** @internal */
export const UpdateDrainDeliveryDrains4$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody24Type$outboundSchema,
    target: UpdateDrainDeliveryDrainsTarget$outboundSchema,
});
export function updateDrainDeliveryDrains4ToJSON(updateDrainDeliveryDrains4) {
    return JSON.stringify(UpdateDrainDeliveryDrains4$outboundSchema.parse(updateDrainDeliveryDrains4));
}
export function updateDrainDeliveryDrains4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrains4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrains4' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type$outboundSchema = UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrains3$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type$inboundSchema,
    endpoint: z.string(),
    table: z.string(),
});
/** @internal */
export const UpdateDrainDeliveryDrains3$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBody2Type$outboundSchema,
    endpoint: z.string(),
    table: z.string(),
});
export function updateDrainDeliveryDrains3ToJSON(updateDrainDeliveryDrains3) {
    return JSON.stringify(UpdateDrainDeliveryDrains3$outboundSchema.parse(updateDrainDeliveryDrains3));
}
export function updateDrainDeliveryDrains3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrains3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrains3' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType$outboundSchema = UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrainsEndpoint$inboundSchema = z.object({
    traces: z.string(),
});
/** @internal */
export const UpdateDrainDeliveryDrainsEndpoint$outboundSchema = z.object({
    traces: z.string(),
});
export function updateDrainDeliveryDrainsEndpointToJSON(updateDrainDeliveryDrainsEndpoint) {
    return JSON.stringify(UpdateDrainDeliveryDrainsEndpoint$outboundSchema.parse(updateDrainDeliveryDrainsEndpoint));
}
export function updateDrainDeliveryDrainsEndpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrainsEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrainsEndpoint' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding$outboundSchema = UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding$inboundSchema;
/** @internal */
export const UpdateDrainSecretDrainsResponse200Kind$inboundSchema = z.nativeEnum(UpdateDrainSecretDrainsResponse200Kind);
/** @internal */
export const UpdateDrainSecretDrainsResponse200Kind$outboundSchema = UpdateDrainSecretDrainsResponse200Kind$inboundSchema;
/** @internal */
export const UpdateDrainSecretDrainsResponse2002$inboundSchema = z.object({
    kind: UpdateDrainSecretDrainsResponse200Kind$inboundSchema,
});
/** @internal */
export const UpdateDrainSecretDrainsResponse2002$outboundSchema = z.object({
    kind: UpdateDrainSecretDrainsResponse200Kind$outboundSchema,
});
export function updateDrainSecretDrainsResponse2002ToJSON(updateDrainSecretDrainsResponse2002) {
    return JSON.stringify(UpdateDrainSecretDrainsResponse2002$outboundSchema.parse(updateDrainSecretDrainsResponse2002));
}
export function updateDrainSecretDrainsResponse2002FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSecretDrainsResponse2002$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSecretDrainsResponse2002' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200Secret$inboundSchema = z.union([
    z.lazy(() => UpdateDrainSecretDrainsResponse2002$inboundSchema),
    z.string(),
]);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200Secret$outboundSchema = z.union([
    z.lazy(() => UpdateDrainSecretDrainsResponse2002$outboundSchema),
    z.string(),
]);
export function updateDrainDeliveryDrainsResponse200SecretToJSON(updateDrainDeliveryDrainsResponse200Secret) {
    return JSON.stringify(UpdateDrainDeliveryDrainsResponse200Secret$outboundSchema.parse(updateDrainDeliveryDrainsResponse200Secret));
}
export function updateDrainDeliveryDrainsResponse200SecretFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrainsResponse200Secret$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrainsResponse200Secret' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse2$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType$inboundSchema,
    endpoint: z.lazy(() => UpdateDrainDeliveryDrainsEndpoint$inboundSchema),
    encoding: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding$inboundSchema,
    headers: z.record(z.string()),
    secret: z.union([
        z.lazy(() => UpdateDrainSecretDrainsResponse2002$inboundSchema),
        z.string(),
    ]).optional(),
});
/** @internal */
export const UpdateDrainDeliveryDrainsResponse2$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyType$outboundSchema,
    endpoint: z.lazy(() => UpdateDrainDeliveryDrainsEndpoint$outboundSchema),
    encoding: UpdateDrainDeliveryDrainsResponse200ApplicationJSONResponseBodyEncoding$outboundSchema,
    headers: z.record(z.string()),
    secret: z.union([
        z.lazy(() => UpdateDrainSecretDrainsResponse2002$outboundSchema),
        z.string(),
    ]).optional(),
});
export function updateDrainDeliveryDrainsResponse2ToJSON(updateDrainDeliveryDrainsResponse2) {
    return JSON.stringify(UpdateDrainDeliveryDrainsResponse2$outboundSchema.parse(updateDrainDeliveryDrainsResponse2));
}
export function updateDrainDeliveryDrainsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrainsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrainsResponse2' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONType$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponse200ApplicationJSONType);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONType$outboundSchema = UpdateDrainDeliveryDrainsResponse200ApplicationJSONType$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding$outboundSchema = UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrainsCompression$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsCompression);
/** @internal */
export const UpdateDrainDeliveryDrainsCompression$outboundSchema = UpdateDrainDeliveryDrainsCompression$inboundSchema;
/** @internal */
export const UpdateDrainSecretDrainsResponseKind$inboundSchema = z.nativeEnum(UpdateDrainSecretDrainsResponseKind);
/** @internal */
export const UpdateDrainSecretDrainsResponseKind$outboundSchema = UpdateDrainSecretDrainsResponseKind$inboundSchema;
/** @internal */
export const UpdateDrainSecretDrainsResponse2$inboundSchema = z.object({
    kind: UpdateDrainSecretDrainsResponseKind$inboundSchema,
});
/** @internal */
export const UpdateDrainSecretDrainsResponse2$outboundSchema = z.object({
    kind: UpdateDrainSecretDrainsResponseKind$outboundSchema,
});
export function updateDrainSecretDrainsResponse2ToJSON(updateDrainSecretDrainsResponse2) {
    return JSON.stringify(UpdateDrainSecretDrainsResponse2$outboundSchema.parse(updateDrainSecretDrainsResponse2));
}
export function updateDrainSecretDrainsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSecretDrainsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSecretDrainsResponse2' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponseSecret$inboundSchema = z.union([
    z.lazy(() => UpdateDrainSecretDrainsResponse2$inboundSchema),
    z.string(),
]);
/** @internal */
export const UpdateDrainDeliveryDrainsResponseSecret$outboundSchema = z.union([
    z.lazy(() => UpdateDrainSecretDrainsResponse2$outboundSchema),
    z.string(),
]);
export function updateDrainDeliveryDrainsResponseSecretToJSON(updateDrainDeliveryDrainsResponseSecret) {
    return JSON.stringify(UpdateDrainDeliveryDrainsResponseSecret$outboundSchema.parse(updateDrainDeliveryDrainsResponseSecret));
}
export function updateDrainDeliveryDrainsResponseSecretFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrainsResponseSecret$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrainsResponseSecret' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse1$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONType$inboundSchema,
    endpoint: z.string(),
    encoding: UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding$inboundSchema,
    compression: UpdateDrainDeliveryDrainsCompression$inboundSchema.optional(),
    headers: z.record(z.string()),
    secret: z.union([
        z.lazy(() => UpdateDrainSecretDrainsResponse2$inboundSchema),
        z.string(),
    ]).optional(),
});
/** @internal */
export const UpdateDrainDeliveryDrainsResponse1$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200ApplicationJSONType$outboundSchema,
    endpoint: z.string(),
    encoding: UpdateDrainDeliveryDrainsResponse200ApplicationJSONEncoding$outboundSchema,
    compression: UpdateDrainDeliveryDrainsCompression$outboundSchema.optional(),
    headers: z.record(z.string()),
    secret: z.union([
        z.lazy(() => UpdateDrainSecretDrainsResponse2$outboundSchema),
        z.string(),
    ]).optional(),
});
export function updateDrainDeliveryDrainsResponse1ToJSON(updateDrainDeliveryDrainsResponse1) {
    return JSON.stringify(UpdateDrainDeliveryDrainsResponse1$outboundSchema.parse(updateDrainDeliveryDrainsResponse1));
}
export function updateDrainDeliveryDrainsResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrainsResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrainsResponse1' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsDelivery$inboundSchema = z.union([
    z.lazy(() => UpdateDrainDeliveryDrainsResponse1$inboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrainsResponse2$inboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrains3$inboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrains4$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBodyDrainsDelivery$outboundSchema = z.union([
    z.lazy(() => UpdateDrainDeliveryDrainsResponse1$outboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrainsResponse2$outboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrains3$outboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrains4$outboundSchema),
]);
export function updateDrainResponseBodyDrainsDeliveryToJSON(updateDrainResponseBodyDrainsDelivery) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsDelivery$outboundSchema.parse(updateDrainResponseBodyDrainsDelivery));
}
export function updateDrainResponseBodyDrainsDeliveryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsDelivery$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsDelivery' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsType$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyDrainsType);
/** @internal */
export const UpdateDrainResponseBodyDrainsType$outboundSchema = UpdateDrainResponseBodyDrainsType$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyDrainsEnv$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyDrainsEnv);
/** @internal */
export const UpdateDrainResponseBodyDrainsEnv$outboundSchema = UpdateDrainResponseBodyDrainsEnv$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyDrainsSampling$inboundSchema = z.object({
    type: UpdateDrainResponseBodyDrainsType$inboundSchema,
    rate: z.number(),
    env: UpdateDrainResponseBodyDrainsEnv$inboundSchema.optional(),
    requestPath: z.string().optional(),
});
/** @internal */
export const UpdateDrainResponseBodyDrainsSampling$outboundSchema = z.object({
    type: UpdateDrainResponseBodyDrainsType$outboundSchema,
    rate: z.number(),
    env: UpdateDrainResponseBodyDrainsEnv$outboundSchema.optional(),
    requestPath: z.string().optional(),
});
export function updateDrainResponseBodyDrainsSamplingToJSON(updateDrainResponseBodyDrainsSampling) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsSampling$outboundSchema.parse(updateDrainResponseBodyDrainsSampling));
}
export function updateDrainResponseBodyDrainsSamplingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsSampling$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsSampling' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsStatus$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyDrainsStatus);
/** @internal */
export const UpdateDrainResponseBodyDrainsStatus$outboundSchema = UpdateDrainResponseBodyDrainsStatus$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyDrainsDisabledReason$inboundSchema = z
    .nativeEnum(UpdateDrainResponseBodyDrainsDisabledReason);
/** @internal */
export const UpdateDrainResponseBodyDrainsDisabledReason$outboundSchema = UpdateDrainResponseBodyDrainsDisabledReason$inboundSchema;
/** @internal */
export const UpdateDrainSourceDrainsResponse200Kind$inboundSchema = z.nativeEnum(UpdateDrainSourceDrainsResponse200Kind);
/** @internal */
export const UpdateDrainSourceDrainsResponse200Kind$outboundSchema = UpdateDrainSourceDrainsResponse200Kind$inboundSchema;
/** @internal */
export const UpdateDrainSourceDrainsResponse2$inboundSchema = z.object({
    kind: UpdateDrainSourceDrainsResponse200Kind$inboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const UpdateDrainSourceDrainsResponse2$outboundSchema = z.object({
    kind: UpdateDrainSourceDrainsResponse200Kind$outboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
export function updateDrainSourceDrainsResponse2ToJSON(updateDrainSourceDrainsResponse2) {
    return JSON.stringify(UpdateDrainSourceDrainsResponse2$outboundSchema.parse(updateDrainSourceDrainsResponse2));
}
export function updateDrainSourceDrainsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSourceDrainsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSourceDrainsResponse2' from JSON`);
}
/** @internal */
export const UpdateDrainSourceDrainsResponseKind$inboundSchema = z.nativeEnum(UpdateDrainSourceDrainsResponseKind);
/** @internal */
export const UpdateDrainSourceDrainsResponseKind$outboundSchema = UpdateDrainSourceDrainsResponseKind$inboundSchema;
/** @internal */
export const UpdateDrainSourceDrainsResponse1$inboundSchema = z.object({
    kind: UpdateDrainSourceDrainsResponseKind$inboundSchema,
});
/** @internal */
export const UpdateDrainSourceDrainsResponse1$outboundSchema = z.object({
    kind: UpdateDrainSourceDrainsResponseKind$outboundSchema,
});
export function updateDrainSourceDrainsResponse1ToJSON(updateDrainSourceDrainsResponse1) {
    return JSON.stringify(UpdateDrainSourceDrainsResponse1$outboundSchema.parse(updateDrainSourceDrainsResponse1));
}
export function updateDrainSourceDrainsResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSourceDrainsResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSourceDrainsResponse1' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsSource$inboundSchema = z.union([
    z.lazy(() => UpdateDrainSourceDrainsResponse2$inboundSchema),
    z.lazy(() => UpdateDrainSourceDrainsResponse1$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBodyDrainsSource$outboundSchema = z.union([
    z.lazy(() => UpdateDrainSourceDrainsResponse2$outboundSchema),
    z.lazy(() => UpdateDrainSourceDrainsResponse1$outboundSchema),
]);
export function updateDrainResponseBodyDrainsSourceToJSON(updateDrainResponseBodyDrainsSource) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsSource$outboundSchema.parse(updateDrainResponseBodyDrainsSource));
}
export function updateDrainResponseBodyDrainsSourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsSource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsSource' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2DrainsResponse200Version$inboundSchema = z
    .nativeEnum(UpdateDrainFilterV2DrainsResponse200Version);
/** @internal */
export const UpdateDrainFilterV2DrainsResponse200Version$outboundSchema = UpdateDrainFilterV2DrainsResponse200Version$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsResponse200Type$inboundSchema = z.nativeEnum(UpdateDrainFilterDrainsResponse200Type);
/** @internal */
export const UpdateDrainFilterDrainsResponse200Type$outboundSchema = UpdateDrainFilterDrainsResponse200Type$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsResponse2002$inboundSchema = z.object({
    type: UpdateDrainFilterDrainsResponse200Type$inboundSchema,
    text: z.string(),
});
/** @internal */
export const UpdateDrainFilterDrainsResponse2002$outboundSchema = z.object({
    type: UpdateDrainFilterDrainsResponse200Type$outboundSchema,
    text: z.string(),
});
export function updateDrainFilterDrainsResponse2002ToJSON(updateDrainFilterDrainsResponse2002) {
    return JSON.stringify(UpdateDrainFilterDrainsResponse2002$outboundSchema.parse(updateDrainFilterDrainsResponse2002));
}
export function updateDrainFilterDrainsResponse2002FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrainsResponse2002$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrainsResponse2002' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrainsResponseType$inboundSchema = z.nativeEnum(UpdateDrainFilterDrainsResponseType);
/** @internal */
export const UpdateDrainFilterDrainsResponseType$outboundSchema = UpdateDrainFilterDrainsResponseType$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsProject$inboundSchema = z.object({
    ids: z.array(z.string()).optional(),
});
/** @internal */
export const UpdateDrainFilterDrainsProject$outboundSchema = z.object({
    ids: z.array(z.string()).optional(),
});
export function updateDrainFilterDrainsProjectToJSON(updateDrainFilterDrainsProject) {
    return JSON.stringify(UpdateDrainFilterDrainsProject$outboundSchema.parse(updateDrainFilterDrainsProject));
}
export function updateDrainFilterDrainsProjectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrainsProject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrainsProject' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrainsResponseSources$inboundSchema = z.nativeEnum(UpdateDrainFilterDrainsResponseSources);
/** @internal */
export const UpdateDrainFilterDrainsResponseSources$outboundSchema = UpdateDrainFilterDrainsResponseSources$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsLog$inboundSchema = z.object({
    sources: z.array(UpdateDrainFilterDrainsResponseSources$inboundSchema)
        .optional(),
    legacy_excludeCachedStaticAssetLogs: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        "legacy_excludeCachedStaticAssetLogs": "legacyExcludeCachedStaticAssetLogs",
    });
});
/** @internal */
export const UpdateDrainFilterDrainsLog$outboundSchema = z.object({
    sources: z.array(UpdateDrainFilterDrainsResponseSources$outboundSchema)
        .optional(),
    legacyExcludeCachedStaticAssetLogs: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        legacyExcludeCachedStaticAssetLogs: "legacy_excludeCachedStaticAssetLogs",
    });
});
export function updateDrainFilterDrainsLogToJSON(updateDrainFilterDrainsLog) {
    return JSON.stringify(UpdateDrainFilterDrainsLog$outboundSchema.parse(updateDrainFilterDrainsLog));
}
export function updateDrainFilterDrainsLogFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrainsLog$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrainsLog' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrainsResponseEnvironments$inboundSchema = z
    .nativeEnum(UpdateDrainFilterDrainsResponseEnvironments);
/** @internal */
export const UpdateDrainFilterDrainsResponseEnvironments$outboundSchema = UpdateDrainFilterDrainsResponseEnvironments$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsDeployment$inboundSchema = z.object({
    environments: z.array(UpdateDrainFilterDrainsResponseEnvironments$inboundSchema).optional(),
});
/** @internal */
export const UpdateDrainFilterDrainsDeployment$outboundSchema = z.object({
    environments: z.array(UpdateDrainFilterDrainsResponseEnvironments$outboundSchema).optional(),
});
export function updateDrainFilterDrainsDeploymentToJSON(updateDrainFilterDrainsDeployment) {
    return JSON.stringify(UpdateDrainFilterDrainsDeployment$outboundSchema.parse(updateDrainFilterDrainsDeployment));
}
export function updateDrainFilterDrainsDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrainsDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrainsDeployment' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrainsResponse1$inboundSchema = z.object({
    type: UpdateDrainFilterDrainsResponseType$inboundSchema,
    project: z.lazy(() => UpdateDrainFilterDrainsProject$inboundSchema)
        .optional(),
    log: z.lazy(() => UpdateDrainFilterDrainsLog$inboundSchema).optional(),
    deployment: z.lazy(() => UpdateDrainFilterDrainsDeployment$inboundSchema)
        .optional(),
});
/** @internal */
export const UpdateDrainFilterDrainsResponse1$outboundSchema = z.object({
    type: UpdateDrainFilterDrainsResponseType$outboundSchema,
    project: z.lazy(() => UpdateDrainFilterDrainsProject$outboundSchema)
        .optional(),
    log: z.lazy(() => UpdateDrainFilterDrainsLog$outboundSchema).optional(),
    deployment: z.lazy(() => UpdateDrainFilterDrainsDeployment$outboundSchema)
        .optional(),
});
export function updateDrainFilterDrainsResponse1ToJSON(updateDrainFilterDrainsResponse1) {
    return JSON.stringify(UpdateDrainFilterDrainsResponse1$outboundSchema.parse(updateDrainFilterDrainsResponse1));
}
export function updateDrainFilterDrainsResponse1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrainsResponse1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrainsResponse1' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2DrainsFilter$inboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterDrainsResponse2002$inboundSchema),
    z.lazy(() => UpdateDrainFilterDrainsResponse1$inboundSchema),
]);
/** @internal */
export const UpdateDrainFilterV2DrainsFilter$outboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterDrainsResponse2002$outboundSchema),
    z.lazy(() => UpdateDrainFilterDrainsResponse1$outboundSchema),
]);
export function updateDrainFilterV2DrainsFilterToJSON(updateDrainFilterV2DrainsFilter) {
    return JSON.stringify(UpdateDrainFilterV2DrainsFilter$outboundSchema.parse(updateDrainFilterV2DrainsFilter));
}
export function updateDrainFilterV2DrainsFilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterV2DrainsFilter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterV2DrainsFilter' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2Drains2$inboundSchema = z.object({
    version: UpdateDrainFilterV2DrainsResponse200Version$inboundSchema,
    filter: z.union([
        z.lazy(() => UpdateDrainFilterDrainsResponse2002$inboundSchema),
        z.lazy(() => UpdateDrainFilterDrainsResponse1$inboundSchema),
    ]),
});
/** @internal */
export const UpdateDrainFilterV2Drains2$outboundSchema = z.object({
    version: UpdateDrainFilterV2DrainsResponse200Version$outboundSchema,
    filter: z.union([
        z.lazy(() => UpdateDrainFilterDrainsResponse2002$outboundSchema),
        z.lazy(() => UpdateDrainFilterDrainsResponse1$outboundSchema),
    ]),
});
export function updateDrainFilterV2Drains2ToJSON(updateDrainFilterV2Drains2) {
    return JSON.stringify(UpdateDrainFilterV2Drains2$outboundSchema.parse(updateDrainFilterV2Drains2));
}
export function updateDrainFilterV2Drains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterV2Drains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterV2Drains2' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2DrainsResponseVersion$inboundSchema = z
    .nativeEnum(UpdateDrainFilterV2DrainsResponseVersion);
/** @internal */
export const UpdateDrainFilterV2DrainsResponseVersion$outboundSchema = UpdateDrainFilterV2DrainsResponseVersion$inboundSchema;
/** @internal */
export const UpdateDrainFilterV2Drains1$inboundSchema = z.object({
    version: UpdateDrainFilterV2DrainsResponseVersion$inboundSchema,
});
/** @internal */
export const UpdateDrainFilterV2Drains1$outboundSchema = z.object({
    version: UpdateDrainFilterV2DrainsResponseVersion$outboundSchema,
});
export function updateDrainFilterV2Drains1ToJSON(updateDrainFilterV2Drains1) {
    return JSON.stringify(UpdateDrainFilterV2Drains1$outboundSchema.parse(updateDrainFilterV2Drains1));
}
export function updateDrainFilterV2Drains1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterV2Drains1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterV2Drains1' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDrainsFilterV2$inboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterV2Drains2$inboundSchema),
    z.lazy(() => UpdateDrainFilterV2Drains1$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBodyDrainsFilterV2$outboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterV2Drains2$outboundSchema),
    z.lazy(() => UpdateDrainFilterV2Drains1$outboundSchema),
]);
export function updateDrainResponseBodyDrainsFilterV2ToJSON(updateDrainResponseBodyDrainsFilterV2) {
    return JSON.stringify(UpdateDrainResponseBodyDrainsFilterV2$outboundSchema.parse(updateDrainResponseBodyDrainsFilterV2));
}
export function updateDrainResponseBodyDrainsFilterV2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDrainsFilterV2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDrainsFilterV2' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyFramework$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyFramework);
/** @internal */
export const UpdateDrainResponseBodyFramework$outboundSchema = UpdateDrainResponseBodyFramework$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyProjectsMetadata$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    framework: z.nullable(UpdateDrainResponseBodyFramework$inboundSchema)
        .optional(),
    latestDeployment: z.string().optional(),
});
/** @internal */
export const UpdateDrainResponseBodyProjectsMetadata$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    framework: z.nullable(UpdateDrainResponseBodyFramework$outboundSchema)
        .optional(),
    latestDeployment: z.string().optional(),
});
export function updateDrainResponseBodyProjectsMetadataToJSON(updateDrainResponseBodyProjectsMetadata) {
    return JSON.stringify(UpdateDrainResponseBodyProjectsMetadata$outboundSchema.parse(updateDrainResponseBodyProjectsMetadata));
}
export function updateDrainResponseBodyProjectsMetadataFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyProjectsMetadata$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyProjectsMetadata' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBody2$inboundSchema = z.object({
    id: z.string(),
    ownerId: z.string(),
    name: z.string(),
    createdAt: z.number(),
    createdFrom: UpdateDrainResponseBodyDrainsCreatedFrom$inboundSchema
        .optional(),
    updatedAt: z.number(),
    projectIds: z.array(z.string()).optional(),
    schemas: z.lazy(() => UpdateDrainResponseBodyDrainsSchemas$inboundSchema)
        .optional(),
    delivery: z.union([
        z.lazy(() => UpdateDrainDeliveryDrainsResponse1$inboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrainsResponse2$inboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrains3$inboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrains4$inboundSchema),
    ]).optional(),
    sampling: z.array(z.lazy(() => UpdateDrainResponseBodyDrainsSampling$inboundSchema)).optional(),
    teamId: z.nullable(z.string()).optional(),
    status: UpdateDrainResponseBodyDrainsStatus$inboundSchema.optional(),
    disabledAt: z.number().optional(),
    disabledReason: UpdateDrainResponseBodyDrainsDisabledReason$inboundSchema
        .optional(),
    disabledBy: z.string().optional(),
    firstErrorTimestamp: z.number().optional(),
    configurationId: z.string().optional(),
    clientId: z.string().optional(),
    source: z.union([
        z.lazy(() => UpdateDrainSourceDrainsResponse2$inboundSchema),
        z.lazy(() => UpdateDrainSourceDrainsResponse1$inboundSchema),
    ]),
    filter: z.string().optional(),
    filterV2: z.union([
        z.lazy(() => UpdateDrainFilterV2Drains2$inboundSchema),
        z.lazy(() => UpdateDrainFilterV2Drains1$inboundSchema),
    ]).optional(),
    integrationIcon: z.string().optional(),
    integrationConfigurationUri: z.string().optional(),
    integrationWebsite: z.string().optional(),
    projectsMetadata: z.array(z.lazy(() => UpdateDrainResponseBodyProjectsMetadata$inboundSchema)).optional(),
});
/** @internal */
export const UpdateDrainResponseBody2$outboundSchema = z.object({
    id: z.string(),
    ownerId: z.string(),
    name: z.string(),
    createdAt: z.number(),
    createdFrom: UpdateDrainResponseBodyDrainsCreatedFrom$outboundSchema
        .optional(),
    updatedAt: z.number(),
    projectIds: z.array(z.string()).optional(),
    schemas: z.lazy(() => UpdateDrainResponseBodyDrainsSchemas$outboundSchema)
        .optional(),
    delivery: z.union([
        z.lazy(() => UpdateDrainDeliveryDrainsResponse1$outboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrainsResponse2$outboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrains3$outboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrains4$outboundSchema),
    ]).optional(),
    sampling: z.array(z.lazy(() => UpdateDrainResponseBodyDrainsSampling$outboundSchema)).optional(),
    teamId: z.nullable(z.string()).optional(),
    status: UpdateDrainResponseBodyDrainsStatus$outboundSchema.optional(),
    disabledAt: z.number().optional(),
    disabledReason: UpdateDrainResponseBodyDrainsDisabledReason$outboundSchema
        .optional(),
    disabledBy: z.string().optional(),
    firstErrorTimestamp: z.number().optional(),
    configurationId: z.string().optional(),
    clientId: z.string().optional(),
    source: z.union([
        z.lazy(() => UpdateDrainSourceDrainsResponse2$outboundSchema),
        z.lazy(() => UpdateDrainSourceDrainsResponse1$outboundSchema),
    ]),
    filter: z.string().optional(),
    filterV2: z.union([
        z.lazy(() => UpdateDrainFilterV2Drains2$outboundSchema),
        z.lazy(() => UpdateDrainFilterV2Drains1$outboundSchema),
    ]).optional(),
    integrationIcon: z.string().optional(),
    integrationConfigurationUri: z.string().optional(),
    integrationWebsite: z.string().optional(),
    projectsMetadata: z.array(z.lazy(() => UpdateDrainResponseBodyProjectsMetadata$outboundSchema)).optional(),
});
export function updateDrainResponseBody2ToJSON(updateDrainResponseBody2) {
    return JSON.stringify(UpdateDrainResponseBody2$outboundSchema.parse(updateDrainResponseBody2));
}
export function updateDrainResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBody2' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyCreatedFrom$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyCreatedFrom);
/** @internal */
export const UpdateDrainResponseBodyCreatedFrom$outboundSchema = UpdateDrainResponseBodyCreatedFrom$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyLog$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyLog$outboundSchema = z.object({});
export function updateDrainResponseBodyLogToJSON(updateDrainResponseBodyLog) {
    return JSON.stringify(UpdateDrainResponseBodyLog$outboundSchema.parse(updateDrainResponseBodyLog));
}
export function updateDrainResponseBodyLogFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyLog$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyLog' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyTrace$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyTrace$outboundSchema = z.object({});
export function updateDrainResponseBodyTraceToJSON(updateDrainResponseBodyTrace) {
    return JSON.stringify(UpdateDrainResponseBodyTrace$outboundSchema.parse(updateDrainResponseBodyTrace));
}
export function updateDrainResponseBodyTraceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyTrace$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyTrace' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyAnalytics$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodyAnalytics$outboundSchema = z.object({});
export function updateDrainResponseBodyAnalyticsToJSON(updateDrainResponseBodyAnalytics) {
    return JSON.stringify(UpdateDrainResponseBodyAnalytics$outboundSchema.parse(updateDrainResponseBodyAnalytics));
}
export function updateDrainResponseBodyAnalyticsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyAnalytics$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyAnalytics' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodySpeedInsights$inboundSchema = z.object({});
/** @internal */
export const UpdateDrainResponseBodySpeedInsights$outboundSchema = z.object({});
export function updateDrainResponseBodySpeedInsightsToJSON(updateDrainResponseBodySpeedInsights) {
    return JSON.stringify(UpdateDrainResponseBodySpeedInsights$outboundSchema.parse(updateDrainResponseBodySpeedInsights));
}
export function updateDrainResponseBodySpeedInsightsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodySpeedInsights$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodySpeedInsights' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodySchemas$inboundSchema = z.object({
    log: z.lazy(() => UpdateDrainResponseBodyLog$inboundSchema).optional(),
    trace: z.lazy(() => UpdateDrainResponseBodyTrace$inboundSchema).optional(),
    analytics: z.lazy(() => UpdateDrainResponseBodyAnalytics$inboundSchema)
        .optional(),
    speed_insights: z.lazy(() => UpdateDrainResponseBodySpeedInsights$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "speed_insights": "speedInsights",
    });
});
/** @internal */
export const UpdateDrainResponseBodySchemas$outboundSchema = z.object({
    log: z.lazy(() => UpdateDrainResponseBodyLog$outboundSchema).optional(),
    trace: z.lazy(() => UpdateDrainResponseBodyTrace$outboundSchema).optional(),
    analytics: z.lazy(() => UpdateDrainResponseBodyAnalytics$outboundSchema)
        .optional(),
    speedInsights: z.lazy(() => UpdateDrainResponseBodySpeedInsights$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        speedInsights: "speed_insights",
    });
});
export function updateDrainResponseBodySchemasToJSON(updateDrainResponseBodySchemas) {
    return JSON.stringify(UpdateDrainResponseBodySchemas$outboundSchema.parse(updateDrainResponseBodySchemas));
}
export function updateDrainResponseBodySchemasFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodySchemas$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodySchemas' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200Type$inboundSchema = z
    .nativeEnum(UpdateDrainDeliveryDrainsResponse200Type);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200Type$outboundSchema = UpdateDrainDeliveryDrainsResponse200Type$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryTarget$inboundSchema = z.nativeEnum(UpdateDrainDeliveryTarget);
/** @internal */
export const UpdateDrainDeliveryTarget$outboundSchema = UpdateDrainDeliveryTarget$inboundSchema;
/** @internal */
export const UpdateDrainDelivery4$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200Type$inboundSchema,
    target: UpdateDrainDeliveryTarget$inboundSchema,
});
/** @internal */
export const UpdateDrainDelivery4$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponse200Type$outboundSchema,
    target: UpdateDrainDeliveryTarget$outboundSchema,
});
export function updateDrainDelivery4ToJSON(updateDrainDelivery4) {
    return JSON.stringify(UpdateDrainDelivery4$outboundSchema.parse(updateDrainDelivery4));
}
export function updateDrainDelivery4FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDelivery4$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDelivery4' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponseType$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsResponseType);
/** @internal */
export const UpdateDrainDeliveryDrainsResponseType$outboundSchema = UpdateDrainDeliveryDrainsResponseType$inboundSchema;
/** @internal */
export const UpdateDrainDelivery3$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponseType$inboundSchema,
    endpoint: z.string(),
    table: z.string(),
});
/** @internal */
export const UpdateDrainDelivery3$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsResponseType$outboundSchema,
    endpoint: z.string(),
    table: z.string(),
});
export function updateDrainDelivery3ToJSON(updateDrainDelivery3) {
    return JSON.stringify(UpdateDrainDelivery3$outboundSchema.parse(updateDrainDelivery3));
}
export function updateDrainDelivery3FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDelivery3$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDelivery3' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsType$inboundSchema = z.nativeEnum(UpdateDrainDeliveryDrainsType);
/** @internal */
export const UpdateDrainDeliveryDrainsType$outboundSchema = UpdateDrainDeliveryDrainsType$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryEndpoint$inboundSchema = z.object({
    traces: z.string(),
});
/** @internal */
export const UpdateDrainDeliveryEndpoint$outboundSchema = z.object({
    traces: z.string(),
});
export function updateDrainDeliveryEndpointToJSON(updateDrainDeliveryEndpoint) {
    return JSON.stringify(UpdateDrainDeliveryEndpoint$outboundSchema.parse(updateDrainDeliveryEndpoint));
}
export function updateDrainDeliveryEndpointFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryEndpoint$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryEndpoint' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200Encoding$inboundSchema = z
    .nativeEnum(UpdateDrainDeliveryDrainsResponse200Encoding);
/** @internal */
export const UpdateDrainDeliveryDrainsResponse200Encoding$outboundSchema = UpdateDrainDeliveryDrainsResponse200Encoding$inboundSchema;
/** @internal */
export const UpdateDrainSecretDrainsKind$inboundSchema = z.nativeEnum(UpdateDrainSecretDrainsKind);
/** @internal */
export const UpdateDrainSecretDrainsKind$outboundSchema = UpdateDrainSecretDrainsKind$inboundSchema;
/** @internal */
export const UpdateDrainSecretDrains2$inboundSchema = z.object({
    kind: UpdateDrainSecretDrainsKind$inboundSchema,
});
/** @internal */
export const UpdateDrainSecretDrains2$outboundSchema = z.object({
    kind: UpdateDrainSecretDrainsKind$outboundSchema,
});
export function updateDrainSecretDrains2ToJSON(updateDrainSecretDrains2) {
    return JSON.stringify(UpdateDrainSecretDrains2$outboundSchema.parse(updateDrainSecretDrains2));
}
export function updateDrainSecretDrains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSecretDrains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSecretDrains2' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrainsSecret$inboundSchema = z.union([z.lazy(() => UpdateDrainSecretDrains2$inboundSchema), z.string()]);
/** @internal */
export const UpdateDrainDeliveryDrainsSecret$outboundSchema = z.union([
    z.lazy(() => UpdateDrainSecretDrains2$outboundSchema),
    z.string(),
]);
export function updateDrainDeliveryDrainsSecretToJSON(updateDrainDeliveryDrainsSecret) {
    return JSON.stringify(UpdateDrainDeliveryDrainsSecret$outboundSchema.parse(updateDrainDeliveryDrainsSecret));
}
export function updateDrainDeliveryDrainsSecretFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrainsSecret$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrainsSecret' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrains2$inboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsType$inboundSchema,
    endpoint: z.lazy(() => UpdateDrainDeliveryEndpoint$inboundSchema),
    encoding: UpdateDrainDeliveryDrainsResponse200Encoding$inboundSchema,
    headers: z.record(z.string()),
    secret: z.union([
        z.lazy(() => UpdateDrainSecretDrains2$inboundSchema),
        z.string(),
    ]).optional(),
});
/** @internal */
export const UpdateDrainDeliveryDrains2$outboundSchema = z.object({
    type: UpdateDrainDeliveryDrainsType$outboundSchema,
    endpoint: z.lazy(() => UpdateDrainDeliveryEndpoint$outboundSchema),
    encoding: UpdateDrainDeliveryDrainsResponse200Encoding$outboundSchema,
    headers: z.record(z.string()),
    secret: z.union([
        z.lazy(() => UpdateDrainSecretDrains2$outboundSchema),
        z.string(),
    ]).optional(),
});
export function updateDrainDeliveryDrains2ToJSON(updateDrainDeliveryDrains2) {
    return JSON.stringify(UpdateDrainDeliveryDrains2$outboundSchema.parse(updateDrainDeliveryDrains2));
}
export function updateDrainDeliveryDrains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrains2' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryType$inboundSchema = z.nativeEnum(UpdateDrainDeliveryType);
/** @internal */
export const UpdateDrainDeliveryType$outboundSchema = UpdateDrainDeliveryType$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryDrainsResponseEncoding$inboundSchema = z
    .nativeEnum(UpdateDrainDeliveryDrainsResponseEncoding);
/** @internal */
export const UpdateDrainDeliveryDrainsResponseEncoding$outboundSchema = UpdateDrainDeliveryDrainsResponseEncoding$inboundSchema;
/** @internal */
export const UpdateDrainDeliveryCompression$inboundSchema = z.nativeEnum(UpdateDrainDeliveryCompression);
/** @internal */
export const UpdateDrainDeliveryCompression$outboundSchema = UpdateDrainDeliveryCompression$inboundSchema;
/** @internal */
export const UpdateDrainSecretKind$inboundSchema = z.nativeEnum(UpdateDrainSecretKind);
/** @internal */
export const UpdateDrainSecretKind$outboundSchema = UpdateDrainSecretKind$inboundSchema;
/** @internal */
export const UpdateDrainSecret2$inboundSchema = z.object({
    kind: UpdateDrainSecretKind$inboundSchema,
});
/** @internal */
export const UpdateDrainSecret2$outboundSchema = z.object({
    kind: UpdateDrainSecretKind$outboundSchema,
});
export function updateDrainSecret2ToJSON(updateDrainSecret2) {
    return JSON.stringify(UpdateDrainSecret2$outboundSchema.parse(updateDrainSecret2));
}
export function updateDrainSecret2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSecret2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSecret2' from JSON`);
}
/** @internal */
export const UpdateDrainDeliverySecret$inboundSchema = z.union([z.lazy(() => UpdateDrainSecret2$inboundSchema), z.string()]);
/** @internal */
export const UpdateDrainDeliverySecret$outboundSchema = z.union([z.lazy(() => UpdateDrainSecret2$outboundSchema), z.string()]);
export function updateDrainDeliverySecretToJSON(updateDrainDeliverySecret) {
    return JSON.stringify(UpdateDrainDeliverySecret$outboundSchema.parse(updateDrainDeliverySecret));
}
export function updateDrainDeliverySecretFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliverySecret$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliverySecret' from JSON`);
}
/** @internal */
export const UpdateDrainDeliveryDrains1$inboundSchema = z.object({
    type: UpdateDrainDeliveryType$inboundSchema,
    endpoint: z.string(),
    encoding: UpdateDrainDeliveryDrainsResponseEncoding$inboundSchema,
    compression: UpdateDrainDeliveryCompression$inboundSchema.optional(),
    headers: z.record(z.string()),
    secret: z.union([z.lazy(() => UpdateDrainSecret2$inboundSchema), z.string()])
        .optional(),
});
/** @internal */
export const UpdateDrainDeliveryDrains1$outboundSchema = z.object({
    type: UpdateDrainDeliveryType$outboundSchema,
    endpoint: z.string(),
    encoding: UpdateDrainDeliveryDrainsResponseEncoding$outboundSchema,
    compression: UpdateDrainDeliveryCompression$outboundSchema.optional(),
    headers: z.record(z.string()),
    secret: z.union([z.lazy(() => UpdateDrainSecret2$outboundSchema), z.string()])
        .optional(),
});
export function updateDrainDeliveryDrains1ToJSON(updateDrainDeliveryDrains1) {
    return JSON.stringify(UpdateDrainDeliveryDrains1$outboundSchema.parse(updateDrainDeliveryDrains1));
}
export function updateDrainDeliveryDrains1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainDeliveryDrains1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainDeliveryDrains1' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyDelivery$inboundSchema = z.union([
    z.lazy(() => UpdateDrainDeliveryDrains1$inboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrains2$inboundSchema),
    z.lazy(() => UpdateDrainDelivery3$inboundSchema),
    z.lazy(() => UpdateDrainDelivery4$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBodyDelivery$outboundSchema = z.union([
    z.lazy(() => UpdateDrainDeliveryDrains1$outboundSchema),
    z.lazy(() => UpdateDrainDeliveryDrains2$outboundSchema),
    z.lazy(() => UpdateDrainDelivery3$outboundSchema),
    z.lazy(() => UpdateDrainDelivery4$outboundSchema),
]);
export function updateDrainResponseBodyDeliveryToJSON(updateDrainResponseBodyDelivery) {
    return JSON.stringify(UpdateDrainResponseBodyDelivery$outboundSchema.parse(updateDrainResponseBodyDelivery));
}
export function updateDrainResponseBodyDeliveryFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyDelivery$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyDelivery' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyType$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyType);
/** @internal */
export const UpdateDrainResponseBodyType$outboundSchema = UpdateDrainResponseBodyType$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyEnv$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyEnv);
/** @internal */
export const UpdateDrainResponseBodyEnv$outboundSchema = UpdateDrainResponseBodyEnv$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodySampling$inboundSchema = z.object({
    type: UpdateDrainResponseBodyType$inboundSchema,
    rate: z.number(),
    env: UpdateDrainResponseBodyEnv$inboundSchema.optional(),
    requestPath: z.string().optional(),
});
/** @internal */
export const UpdateDrainResponseBodySampling$outboundSchema = z.object({
    type: UpdateDrainResponseBodyType$outboundSchema,
    rate: z.number(),
    env: UpdateDrainResponseBodyEnv$outboundSchema.optional(),
    requestPath: z.string().optional(),
});
export function updateDrainResponseBodySamplingToJSON(updateDrainResponseBodySampling) {
    return JSON.stringify(UpdateDrainResponseBodySampling$outboundSchema.parse(updateDrainResponseBodySampling));
}
export function updateDrainResponseBodySamplingFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodySampling$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodySampling' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyStatus$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyStatus);
/** @internal */
export const UpdateDrainResponseBodyStatus$outboundSchema = UpdateDrainResponseBodyStatus$inboundSchema;
/** @internal */
export const UpdateDrainResponseBodyDisabledReason$inboundSchema = z.nativeEnum(UpdateDrainResponseBodyDisabledReason);
/** @internal */
export const UpdateDrainResponseBodyDisabledReason$outboundSchema = UpdateDrainResponseBodyDisabledReason$inboundSchema;
/** @internal */
export const UpdateDrainSourceDrainsKind$inboundSchema = z.nativeEnum(UpdateDrainSourceDrainsKind);
/** @internal */
export const UpdateDrainSourceDrainsKind$outboundSchema = UpdateDrainSourceDrainsKind$inboundSchema;
/** @internal */
export const UpdateDrainSourceDrains2$inboundSchema = z.object({
    kind: UpdateDrainSourceDrainsKind$inboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
/** @internal */
export const UpdateDrainSourceDrains2$outboundSchema = z.object({
    kind: UpdateDrainSourceDrainsKind$outboundSchema,
    resourceId: z.string().optional(),
    externalResourceId: z.string().optional(),
    integrationId: z.string(),
    integrationConfigurationId: z.string(),
});
export function updateDrainSourceDrains2ToJSON(updateDrainSourceDrains2) {
    return JSON.stringify(UpdateDrainSourceDrains2$outboundSchema.parse(updateDrainSourceDrains2));
}
export function updateDrainSourceDrains2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSourceDrains2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSourceDrains2' from JSON`);
}
/** @internal */
export const UpdateDrainSourceKind$inboundSchema = z.nativeEnum(UpdateDrainSourceKind);
/** @internal */
export const UpdateDrainSourceKind$outboundSchema = UpdateDrainSourceKind$inboundSchema;
/** @internal */
export const UpdateDrainSourceDrains1$inboundSchema = z.object({
    kind: UpdateDrainSourceKind$inboundSchema,
});
/** @internal */
export const UpdateDrainSourceDrains1$outboundSchema = z.object({
    kind: UpdateDrainSourceKind$outboundSchema,
});
export function updateDrainSourceDrains1ToJSON(updateDrainSourceDrains1) {
    return JSON.stringify(UpdateDrainSourceDrains1$outboundSchema.parse(updateDrainSourceDrains1));
}
export function updateDrainSourceDrains1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainSourceDrains1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainSourceDrains1' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodySource$inboundSchema = z.union([
    z.lazy(() => UpdateDrainSourceDrains2$inboundSchema),
    z.lazy(() => UpdateDrainSourceDrains1$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBodySource$outboundSchema = z.union([
    z.lazy(() => UpdateDrainSourceDrains2$outboundSchema),
    z.lazy(() => UpdateDrainSourceDrains1$outboundSchema),
]);
export function updateDrainResponseBodySourceToJSON(updateDrainResponseBodySource) {
    return JSON.stringify(UpdateDrainResponseBodySource$outboundSchema.parse(updateDrainResponseBodySource));
}
export function updateDrainResponseBodySourceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodySource$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodySource' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2DrainsVersion$inboundSchema = z.nativeEnum(UpdateDrainFilterV2DrainsVersion);
/** @internal */
export const UpdateDrainFilterV2DrainsVersion$outboundSchema = UpdateDrainFilterV2DrainsVersion$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsType$inboundSchema = z.nativeEnum(UpdateDrainFilterDrainsType);
/** @internal */
export const UpdateDrainFilterDrainsType$outboundSchema = UpdateDrainFilterDrainsType$inboundSchema;
/** @internal */
export const UpdateDrainFilterDrainsResponse2$inboundSchema = z.object({
    type: UpdateDrainFilterDrainsType$inboundSchema,
    text: z.string(),
});
/** @internal */
export const UpdateDrainFilterDrainsResponse2$outboundSchema = z.object({
    type: UpdateDrainFilterDrainsType$outboundSchema,
    text: z.string(),
});
export function updateDrainFilterDrainsResponse2ToJSON(updateDrainFilterDrainsResponse2) {
    return JSON.stringify(UpdateDrainFilterDrainsResponse2$outboundSchema.parse(updateDrainFilterDrainsResponse2));
}
export function updateDrainFilterDrainsResponse2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrainsResponse2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrainsResponse2' from JSON`);
}
/** @internal */
export const UpdateDrainFilterType$inboundSchema = z.nativeEnum(UpdateDrainFilterType);
/** @internal */
export const UpdateDrainFilterType$outboundSchema = UpdateDrainFilterType$inboundSchema;
/** @internal */
export const UpdateDrainFilterProject$inboundSchema = z.object({
    ids: z.array(z.string()).optional(),
});
/** @internal */
export const UpdateDrainFilterProject$outboundSchema = z.object({
    ids: z.array(z.string()).optional(),
});
export function updateDrainFilterProjectToJSON(updateDrainFilterProject) {
    return JSON.stringify(UpdateDrainFilterProject$outboundSchema.parse(updateDrainFilterProject));
}
export function updateDrainFilterProjectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterProject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterProject' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrainsSources$inboundSchema = z.nativeEnum(UpdateDrainFilterDrainsSources);
/** @internal */
export const UpdateDrainFilterDrainsSources$outboundSchema = UpdateDrainFilterDrainsSources$inboundSchema;
/** @internal */
export const UpdateDrainFilterLog$inboundSchema = z.object({
    sources: z.array(UpdateDrainFilterDrainsSources$inboundSchema).optional(),
    legacy_excludeCachedStaticAssetLogs: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        "legacy_excludeCachedStaticAssetLogs": "legacyExcludeCachedStaticAssetLogs",
    });
});
/** @internal */
export const UpdateDrainFilterLog$outboundSchema = z.object({
    sources: z.array(UpdateDrainFilterDrainsSources$outboundSchema).optional(),
    legacyExcludeCachedStaticAssetLogs: z.boolean().optional(),
}).transform((v) => {
    return remap$(v, {
        legacyExcludeCachedStaticAssetLogs: "legacy_excludeCachedStaticAssetLogs",
    });
});
export function updateDrainFilterLogToJSON(updateDrainFilterLog) {
    return JSON.stringify(UpdateDrainFilterLog$outboundSchema.parse(updateDrainFilterLog));
}
export function updateDrainFilterLogFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterLog$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterLog' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrainsEnvironments$inboundSchema = z.nativeEnum(UpdateDrainFilterDrainsEnvironments);
/** @internal */
export const UpdateDrainFilterDrainsEnvironments$outboundSchema = UpdateDrainFilterDrainsEnvironments$inboundSchema;
/** @internal */
export const UpdateDrainFilterDeployment$inboundSchema = z.object({
    environments: z.array(UpdateDrainFilterDrainsEnvironments$inboundSchema)
        .optional(),
});
/** @internal */
export const UpdateDrainFilterDeployment$outboundSchema = z.object({
    environments: z.array(UpdateDrainFilterDrainsEnvironments$outboundSchema)
        .optional(),
});
export function updateDrainFilterDeploymentToJSON(updateDrainFilterDeployment) {
    return JSON.stringify(UpdateDrainFilterDeployment$outboundSchema.parse(updateDrainFilterDeployment));
}
export function updateDrainFilterDeploymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDeployment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDeployment' from JSON`);
}
/** @internal */
export const UpdateDrainFilterDrains1$inboundSchema = z.object({
    type: UpdateDrainFilterType$inboundSchema,
    project: z.lazy(() => UpdateDrainFilterProject$inboundSchema).optional(),
    log: z.lazy(() => UpdateDrainFilterLog$inboundSchema).optional(),
    deployment: z.lazy(() => UpdateDrainFilterDeployment$inboundSchema)
        .optional(),
});
/** @internal */
export const UpdateDrainFilterDrains1$outboundSchema = z.object({
    type: UpdateDrainFilterType$outboundSchema,
    project: z.lazy(() => UpdateDrainFilterProject$outboundSchema).optional(),
    log: z.lazy(() => UpdateDrainFilterLog$outboundSchema).optional(),
    deployment: z.lazy(() => UpdateDrainFilterDeployment$outboundSchema)
        .optional(),
});
export function updateDrainFilterDrains1ToJSON(updateDrainFilterDrains1) {
    return JSON.stringify(UpdateDrainFilterDrains1$outboundSchema.parse(updateDrainFilterDrains1));
}
export function updateDrainFilterDrains1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterDrains1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterDrains1' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2Filter$inboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterDrainsResponse2$inboundSchema),
    z.lazy(() => UpdateDrainFilterDrains1$inboundSchema),
]);
/** @internal */
export const UpdateDrainFilterV2Filter$outboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterDrainsResponse2$outboundSchema),
    z.lazy(() => UpdateDrainFilterDrains1$outboundSchema),
]);
export function updateDrainFilterV2FilterToJSON(updateDrainFilterV2Filter) {
    return JSON.stringify(UpdateDrainFilterV2Filter$outboundSchema.parse(updateDrainFilterV2Filter));
}
export function updateDrainFilterV2FilterFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterV2Filter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterV2Filter' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV22$inboundSchema = z.object({
    version: UpdateDrainFilterV2DrainsVersion$inboundSchema,
    filter: z.union([
        z.lazy(() => UpdateDrainFilterDrainsResponse2$inboundSchema),
        z.lazy(() => UpdateDrainFilterDrains1$inboundSchema),
    ]),
});
/** @internal */
export const UpdateDrainFilterV22$outboundSchema = z.object({
    version: UpdateDrainFilterV2DrainsVersion$outboundSchema,
    filter: z.union([
        z.lazy(() => UpdateDrainFilterDrainsResponse2$outboundSchema),
        z.lazy(() => UpdateDrainFilterDrains1$outboundSchema),
    ]),
});
export function updateDrainFilterV22ToJSON(updateDrainFilterV22) {
    return JSON.stringify(UpdateDrainFilterV22$outboundSchema.parse(updateDrainFilterV22));
}
export function updateDrainFilterV22FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterV22$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterV22' from JSON`);
}
/** @internal */
export const UpdateDrainFilterV2Version$inboundSchema = z.nativeEnum(UpdateDrainFilterV2Version);
/** @internal */
export const UpdateDrainFilterV2Version$outboundSchema = UpdateDrainFilterV2Version$inboundSchema;
/** @internal */
export const UpdateDrainFilterV21$inboundSchema = z.object({
    version: UpdateDrainFilterV2Version$inboundSchema,
});
/** @internal */
export const UpdateDrainFilterV21$outboundSchema = z.object({
    version: UpdateDrainFilterV2Version$outboundSchema,
});
export function updateDrainFilterV21ToJSON(updateDrainFilterV21) {
    return JSON.stringify(UpdateDrainFilterV21$outboundSchema.parse(updateDrainFilterV21));
}
export function updateDrainFilterV21FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainFilterV21$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainFilterV21' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBodyFilterV2$inboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterV22$inboundSchema),
    z.lazy(() => UpdateDrainFilterV21$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBodyFilterV2$outboundSchema = z.union([
    z.lazy(() => UpdateDrainFilterV22$outboundSchema),
    z.lazy(() => UpdateDrainFilterV21$outboundSchema),
]);
export function updateDrainResponseBodyFilterV2ToJSON(updateDrainResponseBodyFilterV2) {
    return JSON.stringify(UpdateDrainResponseBodyFilterV2$outboundSchema.parse(updateDrainResponseBodyFilterV2));
}
export function updateDrainResponseBodyFilterV2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBodyFilterV2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBodyFilterV2' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBody1$inboundSchema = z.object({
    id: z.string(),
    ownerId: z.string(),
    name: z.string(),
    createdAt: z.number(),
    createdFrom: UpdateDrainResponseBodyCreatedFrom$inboundSchema.optional(),
    updatedAt: z.number(),
    projectIds: z.array(z.string()).optional(),
    schemas: z.lazy(() => UpdateDrainResponseBodySchemas$inboundSchema)
        .optional(),
    delivery: z.union([
        z.lazy(() => UpdateDrainDeliveryDrains1$inboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrains2$inboundSchema),
        z.lazy(() => UpdateDrainDelivery3$inboundSchema),
        z.lazy(() => UpdateDrainDelivery4$inboundSchema),
    ]).optional(),
    sampling: z.array(z.lazy(() => UpdateDrainResponseBodySampling$inboundSchema))
        .optional(),
    teamId: z.nullable(z.string()).optional(),
    status: UpdateDrainResponseBodyStatus$inboundSchema.optional(),
    disabledAt: z.number().optional(),
    disabledReason: UpdateDrainResponseBodyDisabledReason$inboundSchema
        .optional(),
    disabledBy: z.string().optional(),
    firstErrorTimestamp: z.number().optional(),
    configurationId: z.string().optional(),
    clientId: z.string().optional(),
    source: z.union([
        z.lazy(() => UpdateDrainSourceDrains2$inboundSchema),
        z.lazy(() => UpdateDrainSourceDrains1$inboundSchema),
    ]),
    filter: z.string().optional(),
    filterV2: z.union([
        z.lazy(() => UpdateDrainFilterV22$inboundSchema),
        z.lazy(() => UpdateDrainFilterV21$inboundSchema),
    ]).optional(),
});
/** @internal */
export const UpdateDrainResponseBody1$outboundSchema = z.object({
    id: z.string(),
    ownerId: z.string(),
    name: z.string(),
    createdAt: z.number(),
    createdFrom: UpdateDrainResponseBodyCreatedFrom$outboundSchema.optional(),
    updatedAt: z.number(),
    projectIds: z.array(z.string()).optional(),
    schemas: z.lazy(() => UpdateDrainResponseBodySchemas$outboundSchema)
        .optional(),
    delivery: z.union([
        z.lazy(() => UpdateDrainDeliveryDrains1$outboundSchema),
        z.lazy(() => UpdateDrainDeliveryDrains2$outboundSchema),
        z.lazy(() => UpdateDrainDelivery3$outboundSchema),
        z.lazy(() => UpdateDrainDelivery4$outboundSchema),
    ]).optional(),
    sampling: z.array(z.lazy(() => UpdateDrainResponseBodySampling$outboundSchema)).optional(),
    teamId: z.nullable(z.string()).optional(),
    status: UpdateDrainResponseBodyStatus$outboundSchema.optional(),
    disabledAt: z.number().optional(),
    disabledReason: UpdateDrainResponseBodyDisabledReason$outboundSchema
        .optional(),
    disabledBy: z.string().optional(),
    firstErrorTimestamp: z.number().optional(),
    configurationId: z.string().optional(),
    clientId: z.string().optional(),
    source: z.union([
        z.lazy(() => UpdateDrainSourceDrains2$outboundSchema),
        z.lazy(() => UpdateDrainSourceDrains1$outboundSchema),
    ]),
    filter: z.string().optional(),
    filterV2: z.union([
        z.lazy(() => UpdateDrainFilterV22$outboundSchema),
        z.lazy(() => UpdateDrainFilterV21$outboundSchema),
    ]).optional(),
});
export function updateDrainResponseBody1ToJSON(updateDrainResponseBody1) {
    return JSON.stringify(UpdateDrainResponseBody1$outboundSchema.parse(updateDrainResponseBody1));
}
export function updateDrainResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBody1' from JSON`);
}
/** @internal */
export const UpdateDrainResponseBody$inboundSchema = z.union([
    z.lazy(() => UpdateDrainResponseBody1$inboundSchema),
    z.lazy(() => UpdateDrainResponseBody2$inboundSchema),
]);
/** @internal */
export const UpdateDrainResponseBody$outboundSchema = z.union([
    z.lazy(() => UpdateDrainResponseBody1$outboundSchema),
    z.lazy(() => UpdateDrainResponseBody2$outboundSchema),
]);
export function updateDrainResponseBodyToJSON(updateDrainResponseBody) {
    return JSON.stringify(UpdateDrainResponseBody$outboundSchema.parse(updateDrainResponseBody));
}
export function updateDrainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateDrainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateDrainResponseBody' from JSON`);
}
