// @vercel/sdk/esm/models/updaterollingreleaseconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const UpdateRollingReleaseConfigRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const UpdateRollingReleaseConfigRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function updateRollingReleaseConfigRequestToJSON(updateRollingReleaseConfigRequest) {
    return JSON.stringify(UpdateRollingReleaseConfigRequest$outboundSchema.parse(updateRollingReleaseConfigRequest));
}
export function updateRollingReleaseConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateRollingReleaseConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateRollingReleaseConfigRequest' from JSON`);
}
/** @internal */
export const ResponseBodyStages$inboundSchema = z.object({
    targetPercentage: z.number(),
    requireApproval: z.boolean().optional(),
    duration: z.number().optional(),
    linearShift: z.boolean().optional(),
});
/** @internal */
export const ResponseBodyStages$outboundSchema = z.object({
    targetPercentage: z.number(),
    requireApproval: z.boolean().optional(),
    duration: z.number().optional(),
    linearShift: z.boolean().optional(),
});
export function responseBodyStagesToJSON(responseBodyStages) {
    return JSON.stringify(ResponseBodyStages$outboundSchema.parse(responseBodyStages));
}
export function responseBodyStagesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyStages$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyStages' from JSON`);
}
/** @internal */
export const ResponseBodyRollingRelease$inboundSchema = z.object({
    stages: z.nullable(z.array(z.lazy(() => ResponseBodyStages$inboundSchema)))
        .optional(),
});
/** @internal */
export const ResponseBodyRollingRelease$outboundSchema = z.object({
    stages: z.nullable(z.array(z.lazy(() => ResponseBodyStages$outboundSchema)))
        .optional(),
});
export function responseBodyRollingReleaseToJSON(responseBodyRollingRelease) {
    return JSON.stringify(ResponseBodyRollingRelease$outboundSchema.parse(responseBodyRollingRelease));
}
export function responseBodyRollingReleaseFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyRollingRelease$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyRollingRelease' from JSON`);
}
/** @internal */
export const UpdateRollingReleaseConfigResponseBody2$inboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => ResponseBodyRollingRelease$inboundSchema)),
});
/** @internal */
export const UpdateRollingReleaseConfigResponseBody2$outboundSchema = z.object({
    rollingRelease: z.nullable(z.lazy(() => ResponseBodyRollingRelease$outboundSchema)),
});
export function updateRollingReleaseConfigResponseBody2ToJSON(updateRollingReleaseConfigResponseBody2) {
    return JSON.stringify(UpdateRollingReleaseConfigResponseBody2$outboundSchema.parse(updateRollingReleaseConfigResponseBody2));
}
export function updateRollingReleaseConfigResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateRollingReleaseConfigResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateRollingReleaseConfigResponseBody2' from JSON`);
}
/** @internal */
export const UpdateRollingReleaseConfigResponseBody1$inboundSchema = z.object({
    rollingRelease: z.nullable(z.any()).optional(),
});
/** @internal */
export const UpdateRollingReleaseConfigResponseBody1$outboundSchema = z.object({
    rollingRelease: z.nullable(z.any()).optional(),
});
export function updateRollingReleaseConfigResponseBody1ToJSON(updateRollingReleaseConfigResponseBody1) {
    return JSON.stringify(UpdateRollingReleaseConfigResponseBody1$outboundSchema.parse(updateRollingReleaseConfigResponseBody1));
}
export function updateRollingReleaseConfigResponseBody1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateRollingReleaseConfigResponseBody1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateRollingReleaseConfigResponseBody1' from JSON`);
}
/** @internal */
export const UpdateRollingReleaseConfigResponseBody$inboundSchema = z.union([
    z.lazy(() => UpdateRollingReleaseConfigResponseBody1$inboundSchema),
    z.lazy(() => UpdateRollingReleaseConfigResponseBody2$inboundSchema),
]);
/** @internal */
export const UpdateRollingReleaseConfigResponseBody$outboundSchema = z.union([
    z.lazy(() => UpdateRollingReleaseConfigResponseBody1$outboundSchema),
    z.lazy(() => UpdateRollingReleaseConfigResponseBody2$outboundSchema),
]);
export function updateRollingReleaseConfigResponseBodyToJSON(updateRollingReleaseConfigResponseBody) {
    return JSON.stringify(UpdateRollingReleaseConfigResponseBody$outboundSchema.parse(updateRollingReleaseConfigResponseBody));
}
export function updateRollingReleaseConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateRollingReleaseConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateRollingReleaseConfigResponseBody' from JSON`);
}
