// @vercel/sdk/esm/models/updatesharedenvvariableop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateSharedEnvVariableTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/**
 * The new type of the Shared Environment Variable
 */
export const UpdateSharedEnvVariableType = {
    Encrypted: "encrypted",
    Sensitive: "sensitive",
};
/**
 * The type of this cosmos doc instance, if blank, assume secret.
 */
export const UpdateSharedEnvVariableEnvironmentType = {
    Encrypted: "encrypted",
    Sensitive: "sensitive",
    System: "system",
    Plain: "plain",
};
/**
 * environments this env variable targets
 */
export const UpdateSharedEnvVariableEnvironmentTarget = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const UpdateSharedEnvVariableValue2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const UpdateSharedEnvVariableTarget2 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
export const UpdateSharedEnvVariableTarget1 = {
    Production: "production",
    Preview: "preview",
    Development: "development",
};
/** @internal */
export const UpdateSharedEnvVariableTarget$inboundSchema = z.nativeEnum(UpdateSharedEnvVariableTarget);
/** @internal */
export const UpdateSharedEnvVariableTarget$outboundSchema = UpdateSharedEnvVariableTarget$inboundSchema;
/** @internal */
export const ProjectIdUpdates$inboundSchema = z.object({
    link: z.array(z.string()).optional(),
    unlink: z.array(z.string()).optional(),
});
/** @internal */
export const ProjectIdUpdates$outboundSchema = z.object({
    link: z.array(z.string()).optional(),
    unlink: z.array(z.string()).optional(),
});
export function projectIdUpdatesToJSON(projectIdUpdates) {
    return JSON.stringify(ProjectIdUpdates$outboundSchema.parse(projectIdUpdates));
}
export function projectIdUpdatesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ProjectIdUpdates$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ProjectIdUpdates' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableType$inboundSchema = z.nativeEnum(UpdateSharedEnvVariableType);
/** @internal */
export const UpdateSharedEnvVariableType$outboundSchema = UpdateSharedEnvVariableType$inboundSchema;
/** @internal */
export const Updates$inboundSchema = z.object({
    key: z.string().optional(),
    value: z.string().optional(),
    target: z.array(UpdateSharedEnvVariableTarget$inboundSchema).optional(),
    projectId: z.array(z.string()).optional(),
    projectIdUpdates: z.lazy(() => ProjectIdUpdates$inboundSchema).optional(),
    type: UpdateSharedEnvVariableType$inboundSchema.optional(),
    comment: z.string().optional(),
});
/** @internal */
export const Updates$outboundSchema = z.object({
    key: z.string().optional(),
    value: z.string().optional(),
    target: z.array(UpdateSharedEnvVariableTarget$outboundSchema).optional(),
    projectId: z.array(z.string()).optional(),
    projectIdUpdates: z.lazy(() => ProjectIdUpdates$outboundSchema).optional(),
    type: UpdateSharedEnvVariableType$outboundSchema.optional(),
    comment: z.string().optional(),
});
export function updatesToJSON(updates) {
    return JSON.stringify(Updates$outboundSchema.parse(updates));
}
export function updatesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Updates$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Updates' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableRequestBody$inboundSchema = z.object({
    updates: z.record(z.lazy(() => Updates$inboundSchema)),
});
/** @internal */
export const UpdateSharedEnvVariableRequestBody$outboundSchema = z.object({
    updates: z.record(z.lazy(() => Updates$outboundSchema)),
});
export function updateSharedEnvVariableRequestBodyToJSON(updateSharedEnvVariableRequestBody) {
    return JSON.stringify(UpdateSharedEnvVariableRequestBody$outboundSchema.parse(updateSharedEnvVariableRequestBody));
}
export function updateSharedEnvVariableRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableRequestBody' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateSharedEnvVariableRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateSharedEnvVariableRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateSharedEnvVariableRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateSharedEnvVariableRequestToJSON(updateSharedEnvVariableRequest) {
    return JSON.stringify(UpdateSharedEnvVariableRequest$outboundSchema.parse(updateSharedEnvVariableRequest));
}
export function updateSharedEnvVariableRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableRequest' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableEnvironmentType$inboundSchema = z.nativeEnum(UpdateSharedEnvVariableEnvironmentType);
/** @internal */
export const UpdateSharedEnvVariableEnvironmentType$outboundSchema = UpdateSharedEnvVariableEnvironmentType$inboundSchema;
/** @internal */
export const UpdateSharedEnvVariableEnvironmentTarget$inboundSchema = z
    .nativeEnum(UpdateSharedEnvVariableEnvironmentTarget);
/** @internal */
export const UpdateSharedEnvVariableEnvironmentTarget$outboundSchema = UpdateSharedEnvVariableEnvironmentTarget$inboundSchema;
/** @internal */
export const Updated$inboundSchema = z.object({
    created: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    key: z.string().optional(),
    ownerId: z.nullable(z.string()).optional(),
    id: z.string().optional(),
    createdBy: z.nullable(z.string()).optional(),
    deletedBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    deletedAt: z.number().optional(),
    updatedAt: z.number().optional(),
    value: z.string().optional(),
    projectId: z.array(z.string()).optional(),
    type: UpdateSharedEnvVariableEnvironmentType$inboundSchema.optional(),
    target: z.array(UpdateSharedEnvVariableEnvironmentTarget$inboundSchema)
        .optional(),
    applyToAllCustomEnvironments: z.boolean().optional(),
    decrypted: z.boolean().optional(),
    comment: z.string().optional(),
    lastEditedByDisplayName: z.string().optional(),
});
/** @internal */
export const Updated$outboundSchema = z.object({
    created: z.date().transform(v => v.toISOString()).optional(),
    key: z.string().optional(),
    ownerId: z.nullable(z.string()).optional(),
    id: z.string().optional(),
    createdBy: z.nullable(z.string()).optional(),
    deletedBy: z.nullable(z.string()).optional(),
    updatedBy: z.nullable(z.string()).optional(),
    createdAt: z.number().optional(),
    deletedAt: z.number().optional(),
    updatedAt: z.number().optional(),
    value: z.string().optional(),
    projectId: z.array(z.string()).optional(),
    type: UpdateSharedEnvVariableEnvironmentType$outboundSchema.optional(),
    target: z.array(UpdateSharedEnvVariableEnvironmentTarget$outboundSchema)
        .optional(),
    applyToAllCustomEnvironments: z.boolean().optional(),
    decrypted: z.boolean().optional(),
    comment: z.string().optional(),
    lastEditedByDisplayName: z.string().optional(),
});
export function updatedToJSON(updated) {
    return JSON.stringify(Updated$outboundSchema.parse(updated));
}
export function updatedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Updated$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Updated' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableValue2$inboundSchema = z.nativeEnum(UpdateSharedEnvVariableValue2);
/** @internal */
export const UpdateSharedEnvVariableValue2$outboundSchema = UpdateSharedEnvVariableValue2$inboundSchema;
/** @internal */
export const UpdateSharedEnvVariableValue$inboundSchema = z.union([z.string(), z.array(UpdateSharedEnvVariableValue2$inboundSchema)]);
/** @internal */
export const UpdateSharedEnvVariableValue$outboundSchema = z.union([
    z.string(),
    z.array(UpdateSharedEnvVariableValue2$outboundSchema),
]);
export function updateSharedEnvVariableValueToJSON(updateSharedEnvVariableValue) {
    return JSON.stringify(UpdateSharedEnvVariableValue$outboundSchema.parse(updateSharedEnvVariableValue));
}
export function updateSharedEnvVariableValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableValue' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableTarget2$inboundSchema = z.nativeEnum(UpdateSharedEnvVariableTarget2);
/** @internal */
export const UpdateSharedEnvVariableTarget2$outboundSchema = UpdateSharedEnvVariableTarget2$inboundSchema;
/** @internal */
export const UpdateSharedEnvVariableTarget1$inboundSchema = z.nativeEnum(UpdateSharedEnvVariableTarget1);
/** @internal */
export const UpdateSharedEnvVariableTarget1$outboundSchema = UpdateSharedEnvVariableTarget1$inboundSchema;
/** @internal */
export const UpdateSharedEnvVariableEnvironmentResponseTarget$inboundSchema = z.union([
    z.array(UpdateSharedEnvVariableTarget1$inboundSchema),
    UpdateSharedEnvVariableTarget2$inboundSchema,
]);
/** @internal */
export const UpdateSharedEnvVariableEnvironmentResponseTarget$outboundSchema = z.union([
    z.array(UpdateSharedEnvVariableTarget1$outboundSchema),
    UpdateSharedEnvVariableTarget2$outboundSchema,
]);
export function updateSharedEnvVariableEnvironmentResponseTargetToJSON(updateSharedEnvVariableEnvironmentResponseTarget) {
    return JSON.stringify(UpdateSharedEnvVariableEnvironmentResponseTarget$outboundSchema.parse(updateSharedEnvVariableEnvironmentResponseTarget));
}
export function updateSharedEnvVariableEnvironmentResponseTargetFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableEnvironmentResponseTarget$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableEnvironmentResponseTarget' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    key: z.string().optional(),
    envVarId: z.string().optional(),
    envVarKey: z.string().optional(),
    action: z.string().optional(),
    link: z.string().optional(),
    value: z.union([
        z.string(),
        z.array(UpdateSharedEnvVariableValue2$inboundSchema),
    ]).optional(),
    gitBranch: z.string().optional(),
    target: z.union([
        z.array(UpdateSharedEnvVariableTarget1$inboundSchema),
        UpdateSharedEnvVariableTarget2$inboundSchema,
    ]).optional(),
    project: z.string().optional(),
});
/** @internal */
export const UpdateSharedEnvVariableError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    key: z.string().optional(),
    envVarId: z.string().optional(),
    envVarKey: z.string().optional(),
    action: z.string().optional(),
    link: z.string().optional(),
    value: z.union([
        z.string(),
        z.array(UpdateSharedEnvVariableValue2$outboundSchema),
    ]).optional(),
    gitBranch: z.string().optional(),
    target: z.union([
        z.array(UpdateSharedEnvVariableTarget1$outboundSchema),
        UpdateSharedEnvVariableTarget2$outboundSchema,
    ]).optional(),
    project: z.string().optional(),
});
export function updateSharedEnvVariableErrorToJSON(updateSharedEnvVariableError) {
    return JSON.stringify(UpdateSharedEnvVariableError$outboundSchema.parse(updateSharedEnvVariableError));
}
export function updateSharedEnvVariableErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableError' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableFailed$inboundSchema = z.object({
    error: z.lazy(() => UpdateSharedEnvVariableError$inboundSchema),
});
/** @internal */
export const UpdateSharedEnvVariableFailed$outboundSchema = z.object({
    error: z.lazy(() => UpdateSharedEnvVariableError$outboundSchema),
});
export function updateSharedEnvVariableFailedToJSON(updateSharedEnvVariableFailed) {
    return JSON.stringify(UpdateSharedEnvVariableFailed$outboundSchema.parse(updateSharedEnvVariableFailed));
}
export function updateSharedEnvVariableFailedFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableFailed$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableFailed' from JSON`);
}
/** @internal */
export const UpdateSharedEnvVariableResponseBody$inboundSchema = z.object({
    updated: z.array(z.lazy(() => Updated$inboundSchema)),
    failed: z.array(z.lazy(() => UpdateSharedEnvVariableFailed$inboundSchema)),
});
/** @internal */
export const UpdateSharedEnvVariableResponseBody$outboundSchema = z.object({
    updated: z.array(z.lazy(() => Updated$outboundSchema)),
    failed: z.array(z.lazy(() => UpdateSharedEnvVariableFailed$outboundSchema)),
});
export function updateSharedEnvVariableResponseBodyToJSON(updateSharedEnvVariableResponseBody) {
    return JSON.stringify(UpdateSharedEnvVariableResponseBody$outboundSchema.parse(updateSharedEnvVariableResponseBody));
}
export function updateSharedEnvVariableResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateSharedEnvVariableResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateSharedEnvVariableResponseBody' from JSON`);
}
