// @vercel/sdk/esm/models/updateedgeconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateEdgeConfigPurposeEdgeConfigType = {
    Experimentation: "experimentation",
};
export const UpdateEdgeConfigPurposeType = {
    Flags: "flags",
};
/** @internal */
export const UpdateEdgeConfigRequestBody$inboundSchema = z.object({
    slug: z.string(),
});
/** @internal */
export const UpdateEdgeConfigRequestBody$outboundSchema = z.object({
    slug: z.string(),
});
export function updateEdgeConfigRequestBodyToJSON(updateEdgeConfigRequestBody) {
    return JSON.stringify(UpdateEdgeConfigRequestBody$outboundSchema.parse(updateEdgeConfigRequestBody));
}
export function updateEdgeConfigRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigRequestBody' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigRequest$inboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateEdgeConfigRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateEdgeConfigRequest$outboundSchema = z.object({
    edgeConfigId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateEdgeConfigRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateEdgeConfigRequestToJSON(updateEdgeConfigRequest) {
    return JSON.stringify(UpdateEdgeConfigRequest$outboundSchema.parse(updateEdgeConfigRequest));
}
export function updateEdgeConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigRequest' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigTransfer$inboundSchema = z.object({
    fromAccountId: z.string(),
    startedAt: z.number(),
    doneAt: z.nullable(z.number()),
});
/** @internal */
export const UpdateEdgeConfigTransfer$outboundSchema = z.object({
    fromAccountId: z.string(),
    startedAt: z.number(),
    doneAt: z.nullable(z.number()),
});
export function updateEdgeConfigTransferToJSON(updateEdgeConfigTransfer) {
    return JSON.stringify(UpdateEdgeConfigTransfer$outboundSchema.parse(updateEdgeConfigTransfer));
}
export function updateEdgeConfigTransferFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigTransfer$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigTransfer' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigSchema$inboundSchema = z.object({});
/** @internal */
export const UpdateEdgeConfigSchema$outboundSchema = z.object({});
export function updateEdgeConfigSchemaToJSON(updateEdgeConfigSchema) {
    return JSON.stringify(UpdateEdgeConfigSchema$outboundSchema.parse(updateEdgeConfigSchema));
}
export function updateEdgeConfigSchemaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigSchema$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigSchema' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigPurposeEdgeConfigType$inboundSchema = z.nativeEnum(UpdateEdgeConfigPurposeEdgeConfigType);
/** @internal */
export const UpdateEdgeConfigPurposeEdgeConfigType$outboundSchema = UpdateEdgeConfigPurposeEdgeConfigType$inboundSchema;
/** @internal */
export const UpdateEdgeConfigPurpose2$inboundSchema = z.object({
    type: UpdateEdgeConfigPurposeEdgeConfigType$inboundSchema,
    resourceId: z.string(),
});
/** @internal */
export const UpdateEdgeConfigPurpose2$outboundSchema = z.object({
    type: UpdateEdgeConfigPurposeEdgeConfigType$outboundSchema,
    resourceId: z.string(),
});
export function updateEdgeConfigPurpose2ToJSON(updateEdgeConfigPurpose2) {
    return JSON.stringify(UpdateEdgeConfigPurpose2$outboundSchema.parse(updateEdgeConfigPurpose2));
}
export function updateEdgeConfigPurpose2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigPurpose2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigPurpose2' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigPurposeType$inboundSchema = z.nativeEnum(UpdateEdgeConfigPurposeType);
/** @internal */
export const UpdateEdgeConfigPurposeType$outboundSchema = UpdateEdgeConfigPurposeType$inboundSchema;
/** @internal */
export const UpdateEdgeConfigPurpose1$inboundSchema = z.object({
    type: UpdateEdgeConfigPurposeType$inboundSchema,
    projectId: z.string(),
});
/** @internal */
export const UpdateEdgeConfigPurpose1$outboundSchema = z.object({
    type: UpdateEdgeConfigPurposeType$outboundSchema,
    projectId: z.string(),
});
export function updateEdgeConfigPurpose1ToJSON(updateEdgeConfigPurpose1) {
    return JSON.stringify(UpdateEdgeConfigPurpose1$outboundSchema.parse(updateEdgeConfigPurpose1));
}
export function updateEdgeConfigPurpose1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigPurpose1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigPurpose1' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigPurpose$inboundSchema = z.union([
    z.lazy(() => UpdateEdgeConfigPurpose1$inboundSchema),
    z.lazy(() => UpdateEdgeConfigPurpose2$inboundSchema),
]);
/** @internal */
export const UpdateEdgeConfigPurpose$outboundSchema = z.union([
    z.lazy(() => UpdateEdgeConfigPurpose1$outboundSchema),
    z.lazy(() => UpdateEdgeConfigPurpose2$outboundSchema),
]);
export function updateEdgeConfigPurposeToJSON(updateEdgeConfigPurpose) {
    return JSON.stringify(UpdateEdgeConfigPurpose$outboundSchema.parse(updateEdgeConfigPurpose));
}
export function updateEdgeConfigPurposeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigPurpose$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigPurpose' from JSON`);
}
/** @internal */
export const UpdateEdgeConfigResponseBody$inboundSchema = z.object({
    createdAt: z.number(),
    updatedAt: z.number(),
    deletedAt: z.nullable(z.number()).optional(),
    id: z.string(),
    slug: z.string(),
    ownerId: z.string(),
    digest: z.string(),
    transfer: z.lazy(() => UpdateEdgeConfigTransfer$inboundSchema).optional(),
    schema: z.lazy(() => UpdateEdgeConfigSchema$inboundSchema).optional(),
    purpose: z.union([
        z.lazy(() => UpdateEdgeConfigPurpose1$inboundSchema),
        z.lazy(() => UpdateEdgeConfigPurpose2$inboundSchema),
    ]).optional(),
    syncedToDynamoAt: z.number().optional(),
    sizeInBytes: z.number(),
    itemCount: z.number(),
});
/** @internal */
export const UpdateEdgeConfigResponseBody$outboundSchema = z.object({
    createdAt: z.number(),
    updatedAt: z.number(),
    deletedAt: z.nullable(z.number()).optional(),
    id: z.string(),
    slug: z.string(),
    ownerId: z.string(),
    digest: z.string(),
    transfer: z.lazy(() => UpdateEdgeConfigTransfer$outboundSchema).optional(),
    schema: z.lazy(() => UpdateEdgeConfigSchema$outboundSchema).optional(),
    purpose: z.union([
        z.lazy(() => UpdateEdgeConfigPurpose1$outboundSchema),
        z.lazy(() => UpdateEdgeConfigPurpose2$outboundSchema),
    ]).optional(),
    syncedToDynamoAt: z.number().optional(),
    sizeInBytes: z.number(),
    itemCount: z.number(),
});
export function updateEdgeConfigResponseBodyToJSON(updateEdgeConfigResponseBody) {
    return JSON.stringify(UpdateEdgeConfigResponseBody$outboundSchema.parse(updateEdgeConfigResponseBody));
}
export function updateEdgeConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateEdgeConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateEdgeConfigResponseBody' from JSON`);
}
