// @vercel/sdk/esm/models/updateresourcesecretsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const UpdateResourceSecretsEnvironmentOverrides$inboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
/** @internal */
export const UpdateResourceSecretsEnvironmentOverrides$outboundSchema = z.object({
    development: z.string().optional(),
    preview: z.string().optional(),
    production: z.string().optional(),
});
export function updateResourceSecretsEnvironmentOverridesToJSON(updateResourceSecretsEnvironmentOverrides) {
    return JSON.stringify(UpdateResourceSecretsEnvironmentOverrides$outboundSchema.parse(updateResourceSecretsEnvironmentOverrides));
}
export function updateResourceSecretsEnvironmentOverridesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecretsEnvironmentOverrides$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecretsEnvironmentOverrides' from JSON`);
}
/** @internal */
export const UpdateResourceSecretsSecrets$inboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => UpdateResourceSecretsEnvironmentOverrides$inboundSchema).optional(),
});
/** @internal */
export const UpdateResourceSecretsSecrets$outboundSchema = z.object({
    name: z.string(),
    value: z.string(),
    prefix: z.string().optional(),
    environmentOverrides: z.lazy(() => UpdateResourceSecretsEnvironmentOverrides$outboundSchema).optional(),
});
export function updateResourceSecretsSecretsToJSON(updateResourceSecretsSecrets) {
    return JSON.stringify(UpdateResourceSecretsSecrets$outboundSchema.parse(updateResourceSecretsSecrets));
}
export function updateResourceSecretsSecretsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecretsSecrets$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecretsSecrets' from JSON`);
}
/** @internal */
export const UpdateResourceSecretsRequestBody$inboundSchema = z.object({
    secrets: z.array(z.lazy(() => UpdateResourceSecretsSecrets$inboundSchema)),
    partial: z.boolean().optional(),
});
/** @internal */
export const UpdateResourceSecretsRequestBody$outboundSchema = z.object({
    secrets: z.array(z.lazy(() => UpdateResourceSecretsSecrets$outboundSchema)),
    partial: z.boolean().optional(),
});
export function updateResourceSecretsRequestBodyToJSON(updateResourceSecretsRequestBody) {
    return JSON.stringify(UpdateResourceSecretsRequestBody$outboundSchema.parse(updateResourceSecretsRequestBody));
}
export function updateResourceSecretsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecretsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecretsRequestBody' from JSON`);
}
/** @internal */
export const UpdateResourceSecretsRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    integrationProductIdOrSlug: z.string(),
    resourceId: z.string(),
    RequestBody: z.lazy(() => UpdateResourceSecretsRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateResourceSecretsRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    integrationProductIdOrSlug: z.string(),
    resourceId: z.string(),
    requestBody: z.lazy(() => UpdateResourceSecretsRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateResourceSecretsRequestToJSON(updateResourceSecretsRequest) {
    return JSON.stringify(UpdateResourceSecretsRequest$outboundSchema.parse(updateResourceSecretsRequest));
}
export function updateResourceSecretsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateResourceSecretsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateResourceSecretsRequest' from JSON`);
}
