// @vercel/sdk/esm/models/updateprojectprotectionbypassop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateProjectProtectionBypassProtectionBypassProjectsScope = {
    AutomationBypass: "automation-bypass",
};
export const UpdateProjectProtectionBypassProtectionBypassScope = {
    IntegrationAutomationBypass: "integration-automation-bypass",
};
/** @internal */
export const Revoke$inboundSchema = z
    .object({
    secret: z.string(),
    regenerate: z.boolean(),
});
/** @internal */
export const Revoke$outboundSchema = z.object({
    secret: z.string(),
    regenerate: z.boolean(),
});
export function revokeToJSON(revoke) {
    return JSON.stringify(Revoke$outboundSchema.parse(revoke));
}
export function revokeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Revoke$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Revoke' from JSON`);
}
/** @internal */
export const Generate$inboundSchema = z.object({
    secret: z.string().optional(),
});
/** @internal */
export const Generate$outboundSchema = z.object({
    secret: z.string().optional(),
});
export function generateToJSON(generate) {
    return JSON.stringify(Generate$outboundSchema.parse(generate));
}
export function generateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Generate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Generate' from JSON`);
}
/** @internal */
export const UpdateProjectProtectionBypassRequestBody$inboundSchema = z.object({
    revoke: z.lazy(() => Revoke$inboundSchema).optional(),
    generate: z.lazy(() => Generate$inboundSchema).optional(),
});
/** @internal */
export const UpdateProjectProtectionBypassRequestBody$outboundSchema = z.object({
    revoke: z.lazy(() => Revoke$outboundSchema).optional(),
    generate: z.lazy(() => Generate$outboundSchema).optional(),
});
export function updateProjectProtectionBypassRequestBodyToJSON(updateProjectProtectionBypassRequestBody) {
    return JSON.stringify(UpdateProjectProtectionBypassRequestBody$outboundSchema.parse(updateProjectProtectionBypassRequestBody));
}
export function updateProjectProtectionBypassRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateProjectProtectionBypassRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateProjectProtectionBypassRequestBody' from JSON`);
}
/** @internal */
export const UpdateProjectProtectionBypassRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateProjectProtectionBypassRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateProjectProtectionBypassRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateProjectProtectionBypassRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateProjectProtectionBypassRequestToJSON(updateProjectProtectionBypassRequest) {
    return JSON.stringify(UpdateProjectProtectionBypassRequest$outboundSchema.parse(updateProjectProtectionBypassRequest));
}
export function updateProjectProtectionBypassRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateProjectProtectionBypassRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateProjectProtectionBypassRequest' from JSON`);
}
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypassProjectsScope$inboundSchema = z.nativeEnum(UpdateProjectProtectionBypassProtectionBypassProjectsScope);
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypassProjectsScope$outboundSchema = UpdateProjectProtectionBypassProtectionBypassProjectsScope$inboundSchema;
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypass2$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: UpdateProjectProtectionBypassProtectionBypassProjectsScope$inboundSchema,
});
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypass2$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: UpdateProjectProtectionBypassProtectionBypassProjectsScope$outboundSchema,
});
export function updateProjectProtectionBypassProtectionBypass2ToJSON(updateProjectProtectionBypassProtectionBypass2) {
    return JSON.stringify(UpdateProjectProtectionBypassProtectionBypass2$outboundSchema.parse(updateProjectProtectionBypassProtectionBypass2));
}
export function updateProjectProtectionBypassProtectionBypass2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateProjectProtectionBypassProtectionBypass2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateProjectProtectionBypassProtectionBypass2' from JSON`);
}
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypassScope$inboundSchema = z
    .nativeEnum(UpdateProjectProtectionBypassProtectionBypassScope);
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypassScope$outboundSchema = UpdateProjectProtectionBypassProtectionBypassScope$inboundSchema;
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypass1$inboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: UpdateProjectProtectionBypassProtectionBypassScope$inboundSchema,
    integrationId: z.string(),
    configurationId: z.string(),
});
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypass1$outboundSchema = z.object({
    createdAt: z.number(),
    createdBy: z.string(),
    scope: UpdateProjectProtectionBypassProtectionBypassScope$outboundSchema,
    integrationId: z.string(),
    configurationId: z.string(),
});
export function updateProjectProtectionBypassProtectionBypass1ToJSON(updateProjectProtectionBypassProtectionBypass1) {
    return JSON.stringify(UpdateProjectProtectionBypassProtectionBypass1$outboundSchema.parse(updateProjectProtectionBypassProtectionBypass1));
}
export function updateProjectProtectionBypassProtectionBypass1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateProjectProtectionBypassProtectionBypass1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateProjectProtectionBypassProtectionBypass1' from JSON`);
}
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypass$inboundSchema = z.union([
    z.lazy(() => UpdateProjectProtectionBypassProtectionBypass1$inboundSchema),
    z.lazy(() => UpdateProjectProtectionBypassProtectionBypass2$inboundSchema),
]);
/** @internal */
export const UpdateProjectProtectionBypassProtectionBypass$outboundSchema = z.union([
    z.lazy(() => UpdateProjectProtectionBypassProtectionBypass1$outboundSchema),
    z.lazy(() => UpdateProjectProtectionBypassProtectionBypass2$outboundSchema),
]);
export function updateProjectProtectionBypassProtectionBypassToJSON(updateProjectProtectionBypassProtectionBypass) {
    return JSON.stringify(UpdateProjectProtectionBypassProtectionBypass$outboundSchema.parse(updateProjectProtectionBypassProtectionBypass));
}
export function updateProjectProtectionBypassProtectionBypassFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateProjectProtectionBypassProtectionBypass$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateProjectProtectionBypassProtectionBypass' from JSON`);
}
/** @internal */
export const UpdateProjectProtectionBypassResponseBody$inboundSchema = z.object({
    protectionBypass: z.record(z.union([
        z.lazy(() => UpdateProjectProtectionBypassProtectionBypass1$inboundSchema),
        z.lazy(() => UpdateProjectProtectionBypassProtectionBypass2$inboundSchema),
    ])).optional(),
});
/** @internal */
export const UpdateProjectProtectionBypassResponseBody$outboundSchema = z.object({
    protectionBypass: z.record(z.union([
        z.lazy(() => UpdateProjectProtectionBypassProtectionBypass1$outboundSchema),
        z.lazy(() => UpdateProjectProtectionBypassProtectionBypass2$outboundSchema),
    ])).optional(),
});
export function updateProjectProtectionBypassResponseBodyToJSON(updateProjectProtectionBypassResponseBody) {
    return JSON.stringify(UpdateProjectProtectionBypassResponseBody$outboundSchema.parse(updateProjectProtectionBypassResponseBody));
}
export function updateProjectProtectionBypassResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateProjectProtectionBypassResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateProjectProtectionBypassResponseBody' from JSON`);
}
