// @vercel/sdk/esm/models/uploadprojectclientcertop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const UploadProjectClientCertRequestBody$inboundSchema = z.object({
    cert: z.string(),
    key: z.string(),
    ca: z.string(),
    origin: z.string().optional(),
    skipValidation: z.boolean().optional(),
});
/** @internal */
export const UploadProjectClientCertRequestBody$outboundSchema = z.object({
    cert: z.string(),
    key: z.string(),
    ca: z.string(),
    origin: z.string().optional(),
    skipValidation: z.boolean().optional(),
});
export function uploadProjectClientCertRequestBodyToJSON(uploadProjectClientCertRequestBody) {
    return JSON.stringify(UploadProjectClientCertRequestBody$outboundSchema.parse(uploadProjectClientCertRequestBody));
}
export function uploadProjectClientCertRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UploadProjectClientCertRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UploadProjectClientCertRequestBody' from JSON`);
}
/** @internal */
export const UploadProjectClientCertRequest$inboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UploadProjectClientCertRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UploadProjectClientCertRequest$outboundSchema = z.object({
    idOrName: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UploadProjectClientCertRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function uploadProjectClientCertRequestToJSON(uploadProjectClientCertRequest) {
    return JSON.stringify(UploadProjectClientCertRequest$outboundSchema.parse(uploadProjectClientCertRequest));
}
export function uploadProjectClientCertRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UploadProjectClientCertRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UploadProjectClientCertRequest' from JSON`);
}
/** @internal */
export const UploadProjectClientCertResponseBody$inboundSchema = z.object({
    updated: z.boolean(),
    origin: z.string(),
    certId: z.string(),
});
/** @internal */
export const UploadProjectClientCertResponseBody$outboundSchema = z.object({
    updated: z.boolean(),
    origin: z.string(),
    certId: z.string(),
});
export function uploadProjectClientCertResponseBodyToJSON(uploadProjectClientCertResponseBody) {
    return JSON.stringify(UploadProjectClientCertResponseBody$outboundSchema.parse(uploadProjectClientCertResponseBody));
}
export function uploadProjectClientCertResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UploadProjectClientCertResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UploadProjectClientCertResponseBody' from JSON`);
}
