// @vercel/sdk/esm/models/verifyprojectdomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const VerifyProjectDomainRequest$inboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const VerifyProjectDomainRequest$outboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function verifyProjectDomainRequestToJSON(verifyProjectDomainRequest) {
    return JSON.stringify(VerifyProjectDomainRequest$outboundSchema.parse(verifyProjectDomainRequest));
}
export function verifyProjectDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => VerifyProjectDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VerifyProjectDomainRequest' from JSON`);
}
/** @internal */
export const VerifyProjectDomainResponseBody$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
});
/** @internal */
export const VerifyProjectDomainResponseBody$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
});
export function verifyProjectDomainResponseBodyToJSON(verifyProjectDomainResponseBody) {
    return JSON.stringify(VerifyProjectDomainResponseBody$outboundSchema.parse(verifyProjectDomainResponseBody));
}
export function verifyProjectDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => VerifyProjectDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VerifyProjectDomainResponseBody' from JSON`);
}
